{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": "true"
   },
   "source": [
    " # Table of Contents\n",
    "<div class=\"toc\" style=\"margin-top: 1em;\"><ul class=\"toc-item\" id=\"toc-level0\"><li><span><a href=\"#Select-SNPs-and-apply-recommended-hard-filters\" data-toc-modified-id=\"Select-SNPs-and-apply-recommended-hard-filters-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Select SNPs and apply recommended hard filters</a></span></li><li><span><a href=\"#Normalize-vcf\" data-toc-modified-id=\"Normalize-vcf-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Normalize vcf</a></span></li><li><span><a href=\"#Run-ANNOVAR\" data-toc-modified-id=\"Run-ANNOVAR-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Run ANNOVAR</a></span></li><li><span><a href=\"#Select-gnomAD-MAF-<-0.01\" data-toc-modified-id=\"Select-gnomAD-MAF-<-0.01-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Select gnomAD MAF &lt; 0.01</a></span></li><li><span><a href=\"#Select-AC->-1\" data-toc-modified-id=\"Select-AC->-1-5\"><span class=\"toc-item-num\">5&nbsp;&nbsp;</span>Select AC &gt; 1</a></span></li><li><span><a href=\"#Run-binomial-test\" data-toc-modified-id=\"Run-binomial-test-6\"><span class=\"toc-item-num\">6&nbsp;&nbsp;</span>Run binomial test</a></span></li><li><span><a href=\"#Filter-based-on-glioma-data\" data-toc-modified-id=\"Filter-based-on-glioma-data-7\"><span class=\"toc-item-num\">7&nbsp;&nbsp;</span>Filter based on glioma data</a></span></li></ul></div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pylab as py\n",
    "import scipy.stats as stats\n",
    "from scipy.stats import binom_test\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cromwell_path = '/mnt/chr10/Data/magda/Powroty/panel/cromwell-executions/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Select SNPs and apply recommended hard filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar CountVariants -V /mnt/chr10/Data/magda/Powroty/panel/cromwell-executions/JointGenotyping/604f468b-2e1f-42d4-8b02-c4b025130bba/call-FinalGatherVcf/execution/Lib1-6.without113.norm.vcf.gz\n",
      "12:00:57.002 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:00:57.181 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:00:57.182 INFO  CountVariants - The Genome Analysis Toolkit (GATK) v4.0.6.0\n",
      "12:00:57.182 INFO  CountVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:00:57.182 INFO  CountVariants - Executing as magda@lamina on Linux v4.4.0-31-generic amd64\n",
      "12:00:57.182 INFO  CountVariants - Java runtime: Java HotSpot(TM) 64-Bit Server VM v1.8.0_201-b09\n",
      "12:00:57.183 INFO  CountVariants - Start Date/Time: August 19, 2019 12:00:56 PM CEST\n",
      "12:00:57.183 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:00:57.183 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:00:57.183 INFO  CountVariants - HTSJDK Version: 2.16.0\n",
      "12:00:57.183 INFO  CountVariants - Picard Version: 2.18.7\n",
      "12:00:57.184 INFO  CountVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:00:57.184 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:00:57.184 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:00:57.184 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:00:57.184 INFO  CountVariants - Deflater: IntelDeflater\n",
      "12:00:57.184 INFO  CountVariants - Inflater: IntelInflater\n",
      "12:00:57.184 INFO  CountVariants - GCS max retries/reopens: 20\n",
      "12:00:57.184 INFO  CountVariants - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes\n",
      "12:00:57.184 INFO  CountVariants - Initializing engine\n",
      "12:00:57.809 INFO  FeatureManager - Using codec VCFCodec to read file file:///mnt/chr10/Data/magda/Powroty/panel/cromwell-executions/JointGenotyping/604f468b-2e1f-42d4-8b02-c4b025130bba/call-FinalGatherVcf/execution/Lib1-6.without113.norm.vcf.gz\n",
      "12:00:58.048 INFO  CountVariants - Done initializing engine\n",
      "12:00:58.048 INFO  ProgressMeter - Starting traversal\n",
      "12:00:58.049 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:01:02.546 INFO  ProgressMeter -       chrX:154437616              0.1                264271        3526748.2\n",
      "12:01:02.547 INFO  ProgressMeter - Traversal complete. Processed 264271 total variants in 0.1 minutes.\n",
      "12:01:02.547 INFO  CountVariants - Shutting down engine\n",
      "[August 19, 2019 12:01:02 PM CEST] org.broadinstitute.hellbender.tools.walkers.CountVariants done. Elapsed time: 0.09 minutes.\n",
      "Runtime.totalMemory()=3355967488\n",
      "Tool returned:\n",
      "264271\n"
     ]
    }
   ],
   "source": [
    "!/home/magda/gatk-4.0.6.0/gatk CountVariants -V \"$cromwell_path\"JointGenotyping/604f468b-2e1f-42d4-8b02-c4b025130bba/call-FinalGatherVcf/execution/Lib1-6.without113.norm.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "select_log = !/home/magda/gatk-4.0.6.0/gatk SelectVariants \\\n",
    "            -V \"$cromwell_path\"JointGenotyping/604f468b-2e1f-42d4-8b02-c4b025130bba/call-FinalGatherVcf/execution/Lib1-6.without113.norm.vcf.gz \\\n",
    "            --select-type-to-include SNP \\\n",
    "            -O out_files/Lib1-6.no113.SNPs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar CountVariants -V out_files/Lib1-6.no113.SNPs.vcf.gz\n",
      "12:05:53.502 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:05:53.675 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:05:53.675 INFO  CountVariants - The Genome Analysis Toolkit (GATK) v4.0.6.0\n",
      "12:05:53.675 INFO  CountVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:05:53.676 INFO  CountVariants - Executing as magda@lamina on Linux v4.4.0-31-generic amd64\n",
      "12:05:53.676 INFO  CountVariants - Java runtime: Java HotSpot(TM) 64-Bit Server VM v1.8.0_201-b09\n",
      "12:05:53.676 INFO  CountVariants - Start Date/Time: August 19, 2019 12:05:53 PM CEST\n",
      "12:05:53.676 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:05:53.676 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:05:53.677 INFO  CountVariants - HTSJDK Version: 2.16.0\n",
      "12:05:53.677 INFO  CountVariants - Picard Version: 2.18.7\n",
      "12:05:53.677 INFO  CountVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:05:53.677 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:05:53.677 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:05:53.677 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:05:53.678 INFO  CountVariants - Deflater: IntelDeflater\n",
      "12:05:53.678 INFO  CountVariants - Inflater: IntelInflater\n",
      "12:05:53.678 INFO  CountVariants - GCS max retries/reopens: 20\n",
      "12:05:53.678 INFO  CountVariants - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes\n",
      "12:05:53.678 INFO  CountVariants - Initializing engine\n",
      "12:05:54.256 INFO  FeatureManager - Using codec VCFCodec to read file file:///mnt/chr10/Data/magda/Powroty/experiments/working/2019-08-13-Recommended_hard_filters/out_files/Lib1-6.no113.SNPs.vcf.gz\n",
      "12:05:54.472 INFO  CountVariants - Done initializing engine\n",
      "12:05:54.472 INFO  ProgressMeter - Starting traversal\n",
      "12:05:54.473 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:06:01.162 INFO  ProgressMeter -       chrX:136066979              0.1                210469        1889026.2\n",
      "12:06:01.162 INFO  ProgressMeter - Traversal complete. Processed 210469 total variants in 0.1 minutes.\n",
      "12:06:01.163 INFO  CountVariants - Shutting down engine\n",
      "[August 19, 2019 12:06:01 PM CEST] org.broadinstitute.hellbender.tools.walkers.CountVariants done. Elapsed time: 0.13 minutes.\n",
      "Runtime.totalMemory()=2389180416\n",
      "Tool returned:\n",
      "210469\n"
     ]
    }
   ],
   "source": [
    "!/home/magda/gatk-4.0.6.0/gatk CountVariants -V out_files/Lib1-6.no113.SNPs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set\n",
    "filter_log = !/home/magda/gatk-4.0.6.0/gatk VariantFiltration \\\n",
    "    -R /mnt/chr11/Data/magda/gatk-ref/Homo_sapiens_assembly38.fasta \\\n",
    "    -V out_files/Lib1-6.no113.SNPs.vcf.gz \\\n",
    "    -O out_files/Lib1-6.no113.SNPs.hard_filtered.vcf.gz \\\n",
    "    --filter-expression \"QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0\" \\\n",
    "    --filter-name \"recommended_snp_filter\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "select_log = !/home/magda/gatk-4.0.6.0/gatk SelectVariants \\\n",
    "            -V out_files/Lib1-6.no113.SNPs.hard_filtered.vcf.gz \\\n",
    "            --exclude-filtered \\\n",
    "            -O out_files/Lib1-6.no113.SNPs.hard_filtered.selected.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.vcf.gz\n",
      "12:11:36.296 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:11:36.501 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:11:36.502 INFO  CountVariants - The Genome Analysis Toolkit (GATK) v4.0.6.0\n",
      "12:11:36.502 INFO  CountVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:11:36.502 INFO  CountVariants - Executing as magda@lamina on Linux v4.4.0-31-generic amd64\n",
      "12:11:36.503 INFO  CountVariants - Java runtime: Java HotSpot(TM) 64-Bit Server VM v1.8.0_201-b09\n",
      "12:11:36.503 INFO  CountVariants - Start Date/Time: August 19, 2019 12:11:36 PM CEST\n",
      "12:11:36.503 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:11:36.503 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:11:36.504 INFO  CountVariants - HTSJDK Version: 2.16.0\n",
      "12:11:36.504 INFO  CountVariants - Picard Version: 2.18.7\n",
      "12:11:36.504 INFO  CountVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:11:36.504 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:11:36.504 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:11:36.504 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:11:36.504 INFO  CountVariants - Deflater: IntelDeflater\n",
      "12:11:36.504 INFO  CountVariants - Inflater: IntelInflater\n",
      "12:11:36.505 INFO  CountVariants - GCS max retries/reopens: 20\n",
      "12:11:36.505 INFO  CountVariants - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes\n",
      "12:11:36.505 INFO  CountVariants - Initializing engine\n",
      "12:11:37.139 INFO  FeatureManager - Using codec VCFCodec to read file file:///mnt/chr10/Data/magda/Powroty/experiments/working/2019-08-13-Recommended_hard_filters/out_files/Lib1-6.no113.SNPs.hard_filtered.selected.vcf.gz\n",
      "12:11:37.380 INFO  CountVariants - Done initializing engine\n",
      "12:11:37.380 INFO  ProgressMeter - Starting traversal\n",
      "12:11:37.380 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:11:40.593 INFO  ProgressMeter -       chrX:102599481              0.1                185793        3471684.8\n",
      "12:11:40.594 INFO  ProgressMeter - Traversal complete. Processed 185793 total variants in 0.1 minutes.\n",
      "12:11:40.594 INFO  CountVariants - Shutting down engine\n",
      "[August 19, 2019 12:11:40 PM CEST] org.broadinstitute.hellbender.tools.walkers.CountVariants done. Elapsed time: 0.07 minutes.\n",
      "Runtime.totalMemory()=2419589120\n",
      "Tool returned:\n",
      "185793\n"
     ]
    }
   ],
   "source": [
    "!/home/magda/gatk-4.0.6.0/gatk CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalize vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lines total/modified/skipped:\t185793/0/0\r\n"
     ]
    }
   ],
   "source": [
    "!bcftools norm -m-both \\\n",
    "-f /mnt/chr11/Data/magda/gatk-ref/Homo_sapiens_assembly38.fasta \\\n",
    "-o out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.vcf \\\n",
    "out_files/Lib1-6.no113.SNPs.hard_filtered.selected.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl -includeinfo -allsample -withfreq -format vcf4 out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.vcf > out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.avinput>\n",
      "NOTICE: Finished reading 190544 lines from VCF file\n",
      "NOTICE: A total of 187122 locus in VCF file passed QC threshold, representing 186188 SNPs (120188 transitions and 66000 transversions) and 0 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 4282324 SNP genotypes (2764324 transitions and 1518000 transversions) and 0 indels/substitutions for 23 samples\n",
      "WARNING: 934 invalid alternative alleles found in input file\n",
      "\n",
      "NOTICE: Running with system command </home/magda/annovar/table_annovar.pl out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.avinput /home/magda/annovar/humandb/ -buildver hg38 -remove -protocol gnomad_genome,avsnp150 -operation f,f -nastring . -outfile out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=gnomad_genome\n",
      "NOTICE: Finished reading 8 column headers for '-dbtype gnomad_genome'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype gnomad_genome -buildver hg38 -outfile out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.avinput /home/magda/annovar/humandb/ -otherinfo>\n",
      "NOTICE: the --dbtype gnomad_genome is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Variants matching filtering criteria are written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_gnomad_genome_dropped, other variants are written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_gnomad_genome_filtered\n",
      "NOTICE: Processing next batch with 187122 unique variants in 187122 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28084439 and the number of bins to be scanned is 151548\n",
      "NOTICE: Scanning filter database /home/magda/annovar/humandb/hg38_gnomad_genome.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=avsnp150\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype avsnp150 -buildver hg38 -outfile out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.avinput /home/magda/annovar/humandb/>\n",
      "NOTICE: Variants matching filtering criteria are written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_avsnp150_dropped, other variants are written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_avsnp150_filtered\n",
      "NOTICE: Processing next batch with 187122 unique variants in 187122 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28304406 and the number of bins to be scanned is 151894\n",
      "NOTICE: Scanning filter database /home/magda/annovar/humandb/hg38_avsnp150.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.txt\n",
      "NOTICE: Reading from out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "!perl /home/magda/annovar/table_annovar.pl  \\\n",
    "out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.vcf  \\\n",
    "/home/magda/annovar/humandb/ -buildver hg38 -remove  \\\n",
    "-protocol gnomad_genome,avsnp150  \\\n",
    "-operation f,f -nastring .  \\\n",
    "-vcfinput  \\\n",
    "-out out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Select gnomAD MAF < 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations = ['gnomAD_genome_ALL',\n",
    "             'gnomAD_genome_AFR',\n",
    "             'gnomAD_genome_AMR',\n",
    "             'gnomAD_genome_ASJ',\n",
    "             'gnomAD_genome_EAS',\n",
    "             'gnomAD_genome_FIN',\n",
    "             'gnomAD_genome_NFE',\n",
    "             'gnomAD_genome_OTH']\n",
    "\n",
    "with open('out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.vcf', 'w') as o:\n",
    "    for line in open('out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.vcf').readlines():    \n",
    "        for el in annotations:\n",
    "            if el + '=.' in line:\n",
    "                line = line.replace(el + '=.', el + '=100.0')\n",
    "        o.write(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "  select_log = !/home/magda/gatk-4.0.6.0/gatk SelectVariants \\\n",
    "            -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.vcf \\\n",
    "            -select 'gnomAD_genome_ALL < 0.01' \\\n",
    "            -select 'gnomAD_genome_AFR < 0.01' \\\n",
    "            -select 'gnomAD_genome_AMR < 0.01' \\\n",
    "            -select 'gnomAD_genome_ASJ < 0.01' \\\n",
    "            -select 'gnomAD_genome_EAS < 0.01' \\\n",
    "            -select 'gnomAD_genome_FIN < 0.01' \\\n",
    "            -select 'gnomAD_genome_NFE < 0.01' \\\n",
    "            -select 'gnomAD_genome_OTH < 0.01' \\\n",
    "            -O out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "12:19:40.039 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:19:40.210 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:19:40.210 INFO  CountVariants - The Genome Analysis Toolkit (GATK) v4.0.6.0\n",
      "12:19:40.210 INFO  CountVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:19:40.211 INFO  CountVariants - Executing as magda@lamina on Linux v4.4.0-31-generic amd64\n",
      "12:19:40.211 INFO  CountVariants - Java runtime: Java HotSpot(TM) 64-Bit Server VM v1.8.0_201-b09\n",
      "12:19:40.211 INFO  CountVariants - Start Date/Time: August 19, 2019 12:19:40 PM CEST\n",
      "12:19:40.211 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:19:40.211 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:19:40.212 INFO  CountVariants - HTSJDK Version: 2.16.0\n",
      "12:19:40.212 INFO  CountVariants - Picard Version: 2.18.7\n",
      "12:19:40.212 INFO  CountVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:19:40.212 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:19:40.212 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:19:40.212 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:19:40.212 INFO  CountVariants - Deflater: IntelDeflater\n",
      "12:19:40.212 INFO  CountVariants - Inflater: IntelInflater\n",
      "12:19:40.213 INFO  CountVariants - GCS max retries/reopens: 20\n",
      "12:19:40.213 INFO  CountVariants - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes\n",
      "12:19:40.213 INFO  CountVariants - Initializing engine\n",
      "12:19:40.929 INFO  FeatureManager - Using codec VCFCodec to read file file:///mnt/chr10/Data/magda/Powroty/experiments/working/2019-08-13-Recommended_hard_filters/out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "12:19:41.082 INFO  CountVariants - Done initializing engine\n",
      "12:19:41.083 INFO  ProgressMeter - Starting traversal\n",
      "12:19:41.083 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:19:41.567 INFO  ProgressMeter -       chr22:45997399              0.0                 16268        2025062.2\n",
      "12:19:41.567 INFO  ProgressMeter - Traversal complete. Processed 16268 total variants in 0.0 minutes.\n",
      "12:19:41.567 INFO  CountVariants - Shutting down engine\n",
      "[August 19, 2019 12:19:41 PM CEST] org.broadinstitute.hellbender.tools.walkers.CountVariants done. Elapsed time: 0.03 minutes.\n",
      "Runtime.totalMemory()=2229796864\n",
      "Tool returned:\n",
      "16268\n"
     ]
    }
   ],
   "source": [
    "!/home/magda/gatk-4.0.6.0/gatk CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Select AC > 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "select_log = !/home/magda/gatk-4.0.6.0/gatk SelectVariants \\\n",
    "            -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.vcf \\\n",
    "            -select 'AC > 1' \\\n",
    "            -O out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.vcf\n",
      "12:21:23.397 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/magda/gatk-4.0.6.0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:21:23.596 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:21:23.596 INFO  CountVariants - The Genome Analysis Toolkit (GATK) v4.0.6.0\n",
      "12:21:23.596 INFO  CountVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:21:23.597 INFO  CountVariants - Executing as magda@lamina on Linux v4.4.0-31-generic amd64\n",
      "12:21:23.597 INFO  CountVariants - Java runtime: Java HotSpot(TM) 64-Bit Server VM v1.8.0_201-b09\n",
      "12:21:23.598 INFO  CountVariants - Start Date/Time: August 19, 2019 12:21:23 PM CEST\n",
      "12:21:23.598 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:21:23.598 INFO  CountVariants - ------------------------------------------------------------\n",
      "12:21:23.598 INFO  CountVariants - HTSJDK Version: 2.16.0\n",
      "12:21:23.598 INFO  CountVariants - Picard Version: 2.18.7\n",
      "12:21:23.598 INFO  CountVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:21:23.599 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:21:23.599 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:21:23.599 INFO  CountVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:21:23.599 INFO  CountVariants - Deflater: IntelDeflater\n",
      "12:21:23.599 INFO  CountVariants - Inflater: IntelInflater\n",
      "12:21:23.599 INFO  CountVariants - GCS max retries/reopens: 20\n",
      "12:21:23.599 INFO  CountVariants - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes\n",
      "12:21:23.599 INFO  CountVariants - Initializing engine\n",
      "12:21:24.250 INFO  FeatureManager - Using codec VCFCodec to read file file:///mnt/chr10/Data/magda/Powroty/experiments/working/2019-08-13-Recommended_hard_filters/out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.vcf\n",
      "12:21:24.427 INFO  CountVariants - Done initializing engine\n",
      "12:21:24.427 INFO  ProgressMeter - Starting traversal\n",
      "12:21:24.428 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:21:24.590 INFO  ProgressMeter -       chr17:38748304              0.0                  2531         949125.0\n",
      "12:21:24.591 INFO  ProgressMeter - Traversal complete. Processed 2531 total variants in 0.0 minutes.\n",
      "12:21:24.591 INFO  CountVariants - Shutting down engine\n",
      "[August 19, 2019 12:21:24 PM CEST] org.broadinstitute.hellbender.tools.walkers.CountVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=2080374784\n",
      "Tool returned:\n",
      "2531\n"
     ]
    }
   ],
   "source": [
    "!/home/magda/gatk-4.0.6.0/gatk CountVariants -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run binomial test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "totable_log = !/home/magda/gatk-4.0.6.0/gatk VariantsToTable \\\n",
    "    -V out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.vcf \\\n",
    "    -F CHROM -F POS -F ID -F REF -F ALT -F AC -F AF -F AN -F avsnp150 \\\n",
    "    -F gnomAD_genome_ALL -F gnomAD_genome_NFE \\\n",
    "    -O out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3</td>\n",
       "      <td>0.065</td>\n",
       "      <td>46</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.001700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1779742</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>0.003700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1808285</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1848381</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.045</td>\n",
       "      <td>44</td>\n",
       "      <td>rs914213962</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1859689</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.045</td>\n",
       "      <td>44</td>\n",
       "      <td>.</td>\n",
       "      <td>0.003400</td>\n",
       "      <td>0.002400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2072334</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46</td>\n",
       "      <td>rs565468909</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2643029</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs570074090</td>\n",
       "      <td>0.003900</td>\n",
       "      <td>0.007000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3425762</td>\n",
       "      <td>rs200643126</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs200643126</td>\n",
       "      <td>0.003200</td>\n",
       "      <td>0.004700</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS           ID REF ALT  AC     AF  AN     avsnp150  \\\n",
       "0  chr1   939561            .   G   A   2  0.043  46  rs888928964   \n",
       "1  chr1  1034902            .   A   T   3  0.065  46  rs751320169   \n",
       "2  chr1  1779742  rs143227614   G   A   2  0.043  46  rs143227614   \n",
       "3  chr1  1808285            .   G   A   2  0.091  22            .   \n",
       "4  chr1  1848381            .   G   A   2  0.045  44  rs914213962   \n",
       "5  chr1  1859689            .   T   G   2  0.045  44            .   \n",
       "6  chr1  2072334            .   C   G   4  0.087  46  rs565468909   \n",
       "7  chr1  2227533            .   C   T   2  0.043  46            .   \n",
       "8  chr1  2643029            .   T   C   2  0.043  46  rs570074090   \n",
       "9  chr1  3425762  rs200643126   G   C   2  0.043  46  rs200643126   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  \n",
       "0           0.000052           0.000000  \n",
       "1           0.001300           0.001700  \n",
       "2           0.002400           0.003700  \n",
       "3           0.000065           0.000067  \n",
       "4           0.000200           0.000200  \n",
       "5           0.003400           0.002400  \n",
       "6           0.004400           0.007800  \n",
       "7           0.000100           0.000300  \n",
       "8           0.003900           0.007000  \n",
       "9           0.003200           0.004700  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "How many variants have missing gnomAD_genome_NFE info? 0\n"
     ]
    }
   ],
   "source": [
    "print \"How many variants have missing gnomAD_genome_NFE info?\", len(df[df['gnomAD_genome_NFE']==100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binom(row):\n",
    "        x = row['AC']\n",
    "        n = row['AN']\n",
    "        try:\n",
    "            p = float(row['gnomAD_genome_NFE'])\n",
    "        except:\n",
    "            print \"error at:\", row['CHROM'], row['POS']\n",
    "        \n",
    "\n",
    "        return binom_test(x,n,p, alternative = 'greater')       \n",
    "            \n",
    "            \n",
    "df['pval']=df.apply(binom, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3</td>\n",
       "      <td>0.065</td>\n",
       "      <td>46</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.000071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1779742</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>0.003700</td>\n",
       "      <td>0.012719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1808285</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0.000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1848381</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.045</td>\n",
       "      <td>44</td>\n",
       "      <td>rs914213962</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000038</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS           ID REF ALT  AC     AF  AN     avsnp150  \\\n",
       "0  chr1   939561            .   G   A   2  0.043  46  rs888928964   \n",
       "1  chr1  1034902            .   A   T   3  0.065  46  rs751320169   \n",
       "2  chr1  1779742  rs143227614   G   A   2  0.043  46  rs143227614   \n",
       "3  chr1  1808285            .   G   A   2  0.091  22            .   \n",
       "4  chr1  1848381            .   G   A   2  0.045  44  rs914213962   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE      pval  \n",
       "0           0.000052           0.000000  0.000000  \n",
       "1           0.001300           0.001700  0.000071  \n",
       "2           0.002400           0.003700  0.012719  \n",
       "3           0.000065           0.000067  0.000001  \n",
       "4           0.000200           0.000200  0.000038  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "multipletests_correction = multipletests(df['pval'], alpha=0.01, \n",
    "              method='fdr_bh', is_sorted=False, returnsorted=False)\n",
    "df['B-H_reject_H0'] = multipletests_correction[0]\n",
    "df['corrected_p_value'] = multipletests_correction[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1736</th>\n",
       "      <td>chr14</td>\n",
       "      <td>64611478</td>\n",
       "      <td>rs138501516</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs138501516</td>\n",
       "      <td>0.0064</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.076205</td>\n",
       "      <td>False</td>\n",
       "      <td>0.076205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1660</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48976478</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs568959452</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.076205</td>\n",
       "      <td>False</td>\n",
       "      <td>0.076205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>chr11</td>\n",
       "      <td>117548865</td>\n",
       "      <td>rs138056897</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs138056897</td>\n",
       "      <td>0.0061</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.076205</td>\n",
       "      <td>False</td>\n",
       "      <td>0.076205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30909861</td>\n",
       "      <td>rs79374019</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs79374019</td>\n",
       "      <td>0.0068</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.076205</td>\n",
       "      <td>False</td>\n",
       "      <td>0.076205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2482</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71182704</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>rs758052869</td>\n",
       "      <td>0.0054</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.076205</td>\n",
       "      <td>False</td>\n",
       "      <td>0.076205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS           ID REF ALT  AC     AF  AN     avsnp150  \\\n",
       "1736  chr14   64611478  rs138501516   G   A   2  0.043  46  rs138501516   \n",
       "1660  chr13   48976478            .   G   A   2  0.043  46  rs568959452   \n",
       "1505  chr11  117548865  rs138056897   C   T   2  0.043  46  rs138056897   \n",
       "57     chr1   30909861   rs79374019   T   C   2  0.043  46   rs79374019   \n",
       "2482   chrX   71182704            .   T   C   2  0.043  46  rs758052869   \n",
       "\n",
       "      gnomAD_genome_ALL  gnomAD_genome_NFE      pval  B-H_reject_H0  \\\n",
       "1736             0.0064             0.0099  0.076205          False   \n",
       "1660             0.0062             0.0099  0.076205          False   \n",
       "1505             0.0061             0.0099  0.076205          False   \n",
       "57               0.0068             0.0099  0.076205          False   \n",
       "2482             0.0054             0.0099  0.076205          False   \n",
       "\n",
       "      corrected_p_value  \n",
       "1736           0.076205  \n",
       "1660           0.076205  \n",
       "1505           0.076205  \n",
       "57             0.076205  \n",
       "2482           0.076205  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by = \"pval\", ascending=False).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1575"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df[df['B-H_reject_H0']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2     1011\n",
       "3      297\n",
       "4      119\n",
       "5       47\n",
       "6       32\n",
       "7       17\n",
       "8       13\n",
       "10       9\n",
       "9        6\n",
       "12       5\n",
       "15       3\n",
       "13       2\n",
       "11       2\n",
       "14       2\n",
       "21       2\n",
       "24       2\n",
       "38       1\n",
       "16       1\n",
       "17       1\n",
       "18       1\n",
       "19       1\n",
       "43       1\n",
       "Name: AC, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['B-H_reject_H0']]['AC'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.FDR0.01.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter based on glioma data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Num variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19217</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19355</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20129</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20144</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20166</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS REF ALT  Num variant\n",
       "0  chr1  19217   G   C            5\n",
       "1  chr1  19355   G   A            5\n",
       "2  chr1  20129   C   T           13\n",
       "3  chr1  20144   G   A            7\n",
       "4  chr1  20166   A   G            8"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_for_filtering = pd.read_csv('../2019-05-07-Prepare_Symfonia_vcf_for_filtering/out_files/symfonia_variants_common.csv', sep='\\t')\n",
    "variants_for_filtering.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_p_value</th>\n",
       "      <th>Num variant</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46.0</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>46.0</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1779742</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46.0</td>\n",
       "      <td>rs143227614</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>0.003700</td>\n",
       "      <td>0.012719</td>\n",
       "      <td>False</td>\n",
       "      <td>0.017786</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1808285</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1848381</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>44.0</td>\n",
       "      <td>rs914213962</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS           ID REF ALT   AC     AF    AN     avsnp150  \\\n",
       "0  chr1   939561            .   G   A  2.0  0.043  46.0  rs888928964   \n",
       "1  chr1  1034902            .   A   T  3.0  0.065  46.0  rs751320169   \n",
       "2  chr1  1779742  rs143227614   G   A  2.0  0.043  46.0  rs143227614   \n",
       "3  chr1  1808285            .   G   A  2.0  0.091  22.0            .   \n",
       "4  chr1  1848381            .   G   A  2.0  0.045  44.0  rs914213962   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE      pval B-H_reject_H0  \\\n",
       "0           0.000052           0.000000  0.000000          True   \n",
       "1           0.001300           0.001700  0.000071          True   \n",
       "2           0.002400           0.003700  0.012719         False   \n",
       "3           0.000065           0.000067  0.000001          True   \n",
       "4           0.000200           0.000200  0.000038          True   \n",
       "\n",
       "   corrected_p_value  Num variant     _merge  \n",
       "0           0.000000          NaN  left_only  \n",
       "1           0.000235          NaN  left_only  \n",
       "2           0.017786          NaN  left_only  \n",
       "3           0.000005          NaN  left_only  \n",
       "4           0.000134          NaN  left_only  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_common_merged = pd.merge(df, variants_for_filtering, how = 'outer', on = ['CHROM', 'POS', 'REF', 'ALT'], indicator = True)\n",
    "df_common_merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_p_value</th>\n",
       "      <th>Num variant</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6101717</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46.0</td>\n",
       "      <td>rs867000634</td>\n",
       "      <td>0.0053</td>\n",
       "      <td>0.0058</td>\n",
       "      <td>1.520208e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>4.569652e-04</td>\n",
       "      <td>12.0</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6101720</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>1.308543e-09</td>\n",
       "      <td>True</td>\n",
       "      <td>9.462634e-09</td>\n",
       "      <td>10.0</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6101722</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>11.0</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6101725</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.068</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>3.543050e-07</td>\n",
       "      <td>True</td>\n",
       "      <td>2.024257e-06</td>\n",
       "      <td>11.0</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37082151</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>1.918805e-07</td>\n",
       "      <td>True</td>\n",
       "      <td>1.119008e-06</td>\n",
       "      <td>5.0</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS ID REF ALT   AC     AF    AN     avsnp150  \\\n",
       "11  chr1   6101717  .   A   C  4.0  0.087  46.0  rs867000634   \n",
       "12  chr1   6101720  .   A   C  4.0  0.087  46.0            .   \n",
       "13  chr1   6101722  .   T   C  4.0  0.091  44.0            .   \n",
       "14  chr1   6101725  .   T   C  3.0  0.068  44.0            .   \n",
       "69  chr1  37082151  .   G   A  4.0  0.091  44.0            .   \n",
       "\n",
       "    gnomAD_genome_ALL  gnomAD_genome_NFE          pval B-H_reject_H0  \\\n",
       "11             0.0053             0.0058  1.520208e-04          True   \n",
       "12             0.0012             0.0003  1.308543e-09          True   \n",
       "13             0.0010             0.0000  0.000000e+00          True   \n",
       "14             0.0007             0.0003  3.543050e-07          True   \n",
       "69             0.0008             0.0011  1.918805e-07          True   \n",
       "\n",
       "    corrected_p_value  Num variant _merge  \n",
       "11       4.569652e-04         12.0   both  \n",
       "12       9.462634e-09         10.0   both  \n",
       "13       0.000000e+00         11.0   both  \n",
       "14       2.024257e-06         11.0   both  \n",
       "69       1.119008e-06          5.0   both  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_common_merged[df_common_merged['_merge'] == 'both'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "166"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_common_merged[df_common_merged['_merge'] == 'both'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "ad_only = df_common_merged[df_common_merged['_merge'] == 'left_only']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1413"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ad_only[ad_only['B-H_reject_H0']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "ad_only[['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AC', 'AF', 'AN', \n",
    "         'avsnp150', 'gnomAD_genome_ALL', 'gnomAD_genome_NFE', \n",
    "         'pval', 'B-H_reject_H0', 'corrected_p_value']].to_csv('out_files/Lib1-6.no113.SNPs.hard_filtered.selected.normalized.hg38_multianno.nomissing.gnomad_below_0.01.ac_above_1.FDR0.01.below_0.2_glioma_samples.csv', \n",
    "                                                               sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": "block",
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
