{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": "true"
   },
   "source": [
    " # Table of Contents\n",
    "<div class=\"toc\" style=\"margin-top: 1em;\"><ul class=\"toc-item\" id=\"toc-level0\"><li><span><a href=\"#Read-data\" data-toc-modified-id=\"Read-data-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Read data</a></span></li><li><span><a href=\"#Add-PWMEnrich-info\" data-toc-modified-id=\"Add-PWMEnrich-info-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Add PWMEnrich info</a></span></li><li><span><a href=\"#Add-ENCODE-TFBS\" data-toc-modified-id=\"Add-ENCODE-TFBS-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Add ENCODE TFBS</a></span></li><li><span><a href=\"#Add-genomic-element-and-gene-info\" data-toc-modified-id=\"Add-genomic-element-and-gene-info-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Add genomic element and gene info</a></span><ul class=\"toc-item\"><li><span><a href=\"#Promoters\" data-toc-modified-id=\"Promoters-4.1\"><span class=\"toc-item-num\">4.1&nbsp;&nbsp;</span>Promoters</a></span></li><li><span><a href=\"#Enhancers\" data-toc-modified-id=\"Enhancers-4.2\"><span class=\"toc-item-num\">4.2&nbsp;&nbsp;</span>Enhancers</a></span><ul class=\"toc-item\"><li><span><a href=\"#Intronic\" data-toc-modified-id=\"Intronic-4.2.1\"><span class=\"toc-item-num\">4.2.1&nbsp;&nbsp;</span>Intronic</a></span></li><li><span><a href=\"#Intergenic\" data-toc-modified-id=\"Intergenic-4.2.2\"><span class=\"toc-item-num\">4.2.2&nbsp;&nbsp;</span>Intergenic</a></span><ul class=\"toc-item\"><li><span><a href=\"#Closest-gene\" data-toc-modified-id=\"Closest-gene-4.2.2.1\"><span class=\"toc-item-num\">4.2.2.1&nbsp;&nbsp;</span>Closest gene</a></span></li><li><span><a href=\"#Hi-C\" data-toc-modified-id=\"Hi-C-4.2.2.2\"><span class=\"toc-item-num\">4.2.2.2&nbsp;&nbsp;</span>Hi-C</a></span></li></ul></li></ul></li></ul></li><li><span><a href=\"#Add-POLGENOM-info\" data-toc-modified-id=\"Add-POLGENOM-info-5\"><span class=\"toc-item-num\">5&nbsp;&nbsp;</span>Add POLGENOM info</a></span><ul class=\"toc-item\"><li><span><a href=\"#Promoters\" data-toc-modified-id=\"Promoters-5.1\"><span class=\"toc-item-num\">5.1&nbsp;&nbsp;</span>Promoters</a></span></li><li><span><a href=\"#Enhancers\" data-toc-modified-id=\"Enhancers-5.2\"><span class=\"toc-item-num\">5.2&nbsp;&nbsp;</span>Enhancers</a></span></li></ul></li><li><span><a href=\"#Analyze-variants-with-PWMEnrich-hits\" data-toc-modified-id=\"Analyze-variants-with-PWMEnrich-hits-6\"><span class=\"toc-item-num\">6&nbsp;&nbsp;</span>Analyze variants with PWMEnrich hits</a></span><ul class=\"toc-item\"><li><span><a href=\"#Check-correlation-between-H3K27ac-in-enhancer-and-gene-expression\" data-toc-modified-id=\"Check-correlation-between-H3K27ac-in-enhancer-and-gene-expression-6.1\"><span class=\"toc-item-num\">6.1&nbsp;&nbsp;</span>Check correlation between H3K27ac in enhancer and gene expression</a></span><ul class=\"toc-item\"><li><span><a href=\"#Additional-checks-for-selected-SNPs\" data-toc-modified-id=\"Additional-checks-for-selected-SNPs-6.1.1\"><span class=\"toc-item-num\">6.1.1&nbsp;&nbsp;</span>Additional checks for selected SNPs</a></span></li></ul></li><li><span><a href=\"#Check-expression-of-target-genes-in-brain-and-lymphocytes\" data-toc-modified-id=\"Check-expression-of-target-genes-in-brain-and-lymphocytes-6.2\"><span class=\"toc-item-num\">6.2&nbsp;&nbsp;</span>Check expression of target genes in brain and lymphocytes</a></span></li><li><span><a href=\"#Check-expression-in-ADNI\" data-toc-modified-id=\"Check-expression-in-ADNI-6.3\"><span class=\"toc-item-num\">6.3&nbsp;&nbsp;</span>Check expression in ADNI</a></span></li><li><span><a href=\"#Add-previous-manual-annotations-for-enhancer-SNPs\" data-toc-modified-id=\"Add-previous-manual-annotations-for-enhancer-SNPs-6.4\"><span class=\"toc-item-num\">6.4&nbsp;&nbsp;</span>Add previous manual annotations for enhancer SNPs</a></span></li></ul></li></ul></div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import os\n",
    "import pybedtools as pbt\n",
    "import scipy.stats\n",
    "from pyliftover import LiftOver"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rsID, AF in my samples, MAF in gnomAD_NFE, genomic element, gene, PWMEnrich motif, ENCODE TFBS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT         VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1    939561   G   A  72,69;91,70  2.0  0.043  rs888928964   \n",
       "1  chr1   1034902   A   T  7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2  chr1   6180416   G   A    10,11;8,6  2.0  0.043  rs151212360   \n",
       "3  chr1  14924017   C   G    6,5;12,14  2.0  0.043  rs531269337   \n",
       "4  chr1  21653564   C   T  13,15;30,24  2.0  0.043  rs890141715   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \n",
       "0           0.000052             0.0000                     0  \n",
       "1           0.001300             0.0017                     3  \n",
       "2           0.000300             0.0005                     2  \n",
       "3           0.001300             0.0020                     0  \n",
       "4           0.000600             0.0013                     2  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters = pd.read_csv('out_files/promoters_filtered_mean-cov3_mean-vaf02_RNAsamples.csv', \n",
    "                        sep='\\t', usecols = [\"CHROM\", \"POS\", \"REF\", \"ALT\", \"AC\", \"AF\", \"avsnp150\", \"gnomAD_genome_ALL\", \"gnomAD_genome_NFE\", \"VAFs\", \"num_samples_with_rna\"])\n",
    "promoters.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>variant</th>\n",
       "      <th>start_position</th>\n",
       "      <th>pval_ref</th>\n",
       "      <th>pval_alt</th>\n",
       "      <th>-log10(pval_ref)</th>\n",
       "      <th>-log10(pval_alt)</th>\n",
       "      <th>-log10pvals_diff</th>\n",
       "      <th>-log10pvals_diff_abs</th>\n",
       "      <th>variant_short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr1:9823866G&gt;A::chr1:9823859-9823872(+)</td>\n",
       "      <td>17</td>\n",
       "      <td>0.036336</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>1.439664</td>\n",
       "      <td>4.445996</td>\n",
       "      <td>3.006332</td>\n",
       "      <td>3.006332</td>\n",
       "      <td>chr1:9823866G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr1:148522213A&gt;G::chr1:148522206-148522219(+)</td>\n",
       "      <td>6</td>\n",
       "      <td>0.083628</td>\n",
       "      <td>0.000845</td>\n",
       "      <td>1.077647</td>\n",
       "      <td>3.072947</td>\n",
       "      <td>1.995300</td>\n",
       "      <td>1.995300</td>\n",
       "      <td>chr1:148522213A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr3:50359509G&gt;A::chr3:50359502-50359515(+)</td>\n",
       "      <td>17</td>\n",
       "      <td>0.064109</td>\n",
       "      <td>0.000414</td>\n",
       "      <td>1.193078</td>\n",
       "      <td>3.382624</td>\n",
       "      <td>2.189547</td>\n",
       "      <td>2.189547</td>\n",
       "      <td>chr3:50359509G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr7:23492567G&gt;A::chr7:23492560-23492573(+)</td>\n",
       "      <td>17</td>\n",
       "      <td>0.066363</td>\n",
       "      <td>0.000628</td>\n",
       "      <td>1.178073</td>\n",
       "      <td>3.201848</td>\n",
       "      <td>2.023775</td>\n",
       "      <td>2.023775</td>\n",
       "      <td>chr7:23492567G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr10:132186178G&gt;A::chr10:132186171-132186184(+)</td>\n",
       "      <td>17</td>\n",
       "      <td>0.038616</td>\n",
       "      <td>0.000173</td>\n",
       "      <td>1.413231</td>\n",
       "      <td>3.761538</td>\n",
       "      <td>2.348306</td>\n",
       "      <td>2.348306</td>\n",
       "      <td>chr10:132186178G&gt;A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   motif                                           variant  \\\n",
       "0  CXXC1_HUMAN.H11MO.0.D          chr1:9823866G>A::chr1:9823859-9823872(+)   \n",
       "1  CXXC1_HUMAN.H11MO.0.D    chr1:148522213A>G::chr1:148522206-148522219(+)   \n",
       "2  CXXC1_HUMAN.H11MO.0.D       chr3:50359509G>A::chr3:50359502-50359515(+)   \n",
       "3  CXXC1_HUMAN.H11MO.0.D       chr7:23492567G>A::chr7:23492560-23492573(+)   \n",
       "4  CXXC1_HUMAN.H11MO.0.D  chr10:132186178G>A::chr10:132186171-132186184(+)   \n",
       "\n",
       "   start_position  pval_ref  pval_alt  -log10(pval_ref)  -log10(pval_alt)  \\\n",
       "0              17  0.036336  0.000036          1.439664          4.445996   \n",
       "1               6  0.083628  0.000845          1.077647          3.072947   \n",
       "2              17  0.064109  0.000414          1.193078          3.382624   \n",
       "3              17  0.066363  0.000628          1.178073          3.201848   \n",
       "4              17  0.038616  0.000173          1.413231          3.761538   \n",
       "\n",
       "   -log10pvals_diff  -log10pvals_diff_abs       variant_short  \n",
       "0          3.006332              3.006332     chr1:9823866G>A  \n",
       "1          1.995300              1.995300   chr1:148522213A>G  \n",
       "2          2.189547              2.189547    chr3:50359509G>A  \n",
       "3          2.023775              2.023775    chr7:23492567G>A  \n",
       "4          2.348306              2.348306  chr10:132186178G>A  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_pwmenrich = pd.read_csv(\"out_files/promoters_PWMEnrich_full_diff1.csv\", sep = '\\t')\n",
    "promoters_pwmenrich.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \n",
       "0           0.000100           0.000300                     2  \n",
       "1           0.000700           0.000700                     2  \n",
       "2           0.000032           0.000067                     1  \n",
       "3           0.000097           0.000067                     0  \n",
       "4           0.004400           0.007600                     2  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers = pd.read_csv('out_files/enhancers_filtered_mean-cov3_mean-vaf02_RNAsamples.csv', \n",
    "                        sep='\\t', usecols = [\"CHROM\", \"POS\", \"REF\", \"ALT\", \"AC\", \"AF\", \"avsnp150\", \"gnomAD_genome_ALL\", \"gnomAD_genome_NFE\", \"VAFs\", \"num_samples_with_rna\"])\n",
    "enhancers.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>motif</th>\n",
       "      <th>variant</th>\n",
       "      <th>start_position</th>\n",
       "      <th>pval_ref</th>\n",
       "      <th>pval_alt</th>\n",
       "      <th>-log10(pval_ref)</th>\n",
       "      <th>-log10(pval_alt)</th>\n",
       "      <th>-log10pvals_diff</th>\n",
       "      <th>-log10pvals_diff_abs</th>\n",
       "      <th>variant_short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr12:69438348G&gt;C::chr12:69438341-69438354(+)</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.072016</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>1.142574</td>\n",
       "      <td>3.151741</td>\n",
       "      <td>2.009167</td>\n",
       "      <td>2.009167</td>\n",
       "      <td>chr12:69438348G&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr13:35979734A&gt;C::chr13:35979727-35979740(+)</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.007038</td>\n",
       "      <td>0.000533</td>\n",
       "      <td>2.152527</td>\n",
       "      <td>3.273652</td>\n",
       "      <td>1.121125</td>\n",
       "      <td>1.121125</td>\n",
       "      <td>chr13:35979734A&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr15:51938615A&gt;G::chr15:51938608-51938621(+)</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.084388</td>\n",
       "      <td>0.000863</td>\n",
       "      <td>1.073722</td>\n",
       "      <td>3.063830</td>\n",
       "      <td>1.990109</td>\n",
       "      <td>1.990109</td>\n",
       "      <td>chr15:51938615A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chrX:131753299T&gt;A::chrX:131753292-131753305(+)</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.010910</td>\n",
       "      <td>0.000970</td>\n",
       "      <td>1.962191</td>\n",
       "      <td>3.013252</td>\n",
       "      <td>1.051061</td>\n",
       "      <td>1.051061</td>\n",
       "      <td>chrX:131753299T&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HMGA1_HUMAN.H11MO.0.D</td>\n",
       "      <td>chr1:94875371C&gt;T::chr1:94875364-94875377(+)</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.052682</td>\n",
       "      <td>0.000804</td>\n",
       "      <td>1.278337</td>\n",
       "      <td>3.094995</td>\n",
       "      <td>1.816659</td>\n",
       "      <td>1.816659</td>\n",
       "      <td>chr1:94875371C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   motif                                         variant  \\\n",
       "0  CXXC1_HUMAN.H11MO.0.D   chr12:69438348G>C::chr12:69438341-69438354(+)   \n",
       "1  CXXC1_HUMAN.H11MO.0.D   chr13:35979734A>C::chr13:35979727-35979740(+)   \n",
       "2  CXXC1_HUMAN.H11MO.0.D   chr15:51938615A>G::chr15:51938608-51938621(+)   \n",
       "3  CXXC1_HUMAN.H11MO.0.D  chrX:131753299T>A::chrX:131753292-131753305(+)   \n",
       "4  HMGA1_HUMAN.H11MO.0.D     chr1:94875371C>T::chr1:94875364-94875377(+)   \n",
       "\n",
       "   start_position  pval_ref  pval_alt  -log10(pval_ref)  -log10(pval_alt)  \\\n",
       "0            15.0  0.072016  0.000705          1.142574          3.151741   \n",
       "1             7.0  0.007038  0.000533          2.152527          3.273652   \n",
       "2             6.0  0.084388  0.000863          1.073722          3.063830   \n",
       "3            19.0  0.010910  0.000970          1.962191          3.013252   \n",
       "4             4.0  0.052682  0.000804          1.278337          3.094995   \n",
       "\n",
       "   -log10pvals_diff  -log10pvals_diff_abs      variant_short  \n",
       "0          2.009167              2.009167  chr12:69438348G>C  \n",
       "1          1.121125              1.121125  chr13:35979734A>C  \n",
       "2          1.990109              1.990109  chr15:51938615A>G  \n",
       "3          1.051061              1.051061  chrX:131753299T>A  \n",
       "4          1.816659              1.816659   chr1:94875371C>T  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_pwmenrich = pd.read_csv(\"out_files/enhancers_PWMEnrich_full_diff1.csv\", sep = '\\t')\n",
    "enhancers_pwmenrich.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add PWMEnrich info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_pwmenrich(df, pwm_hits_df):\n",
    "    df[\"PWMEnrich motifs\"] = df.apply(add_pwmenrich_column, args=(pwm_hits_df,), axis=1)  \n",
    "    df[\"PWMEnrich best -log10pval and diff\"] = df.apply(add_pwmenrich_scores, args=(pwm_hits_df,), axis=1)\n",
    "    return df\n",
    "\n",
    "def add_pwmenrich_column(row, pwm_df):\n",
    "    pwm_df_dedupl = pwm_df[[\"motif\", \"variant_short\"]].drop_duplicates()\n",
    "    \n",
    "    variant_short = row[\"CHROM\"] + ':' + str(row[\"POS\"]) + row[\"REF\"] + \">\" + row[\"ALT\"]\n",
    "    data_for_variant = pwm_df_dedupl[pwm_df_dedupl[\"variant_short\"] == variant_short]\n",
    "    if len(data_for_variant) > 0:\n",
    "        return \", \".join(data_for_variant[\"motif\"].values)\n",
    "    \n",
    "def add_pwmenrich_scores(row, pwm_df):    \n",
    "    variant_short = row[\"CHROM\"] + ':' + str(row[\"POS\"]) + row[\"REF\"] + \">\" + row[\"ALT\"]\n",
    "    data_for_variant = pwm_df[pwm_df[\"variant_short\"] == variant_short]\n",
    "    \n",
    "    if len(data_for_variant) > 0:\n",
    "        best_log_pval_ref = max(data_for_variant[\"-log10(pval_ref)\"])\n",
    "        best_log_pval_alt = max(data_for_variant[\"-log10(pval_alt)\"])\n",
    "        best_log_pval_diff_abs = max(data_for_variant[\"-log10pvals_diff_abs\"])\n",
    "        return \"%.2f;%.2f\" % (max([best_log_pval_ref, best_log_pval_alt]), best_log_pval_diff_abs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...</td>\n",
       "      <td>3.94;1.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>3</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.79;1.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...</td>\n",
       "      <td>10.00;6.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...</td>\n",
       "      <td>5.97;2.29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT         VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1    939561   G   A  72,69;91,70  2.0  0.043  rs888928964   \n",
       "1  chr1   1034902   A   T  7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2  chr1   6180416   G   A    10,11;8,6  2.0  0.043  rs151212360   \n",
       "3  chr1  14924017   C   G    6,5;12,14  2.0  0.043  rs531269337   \n",
       "4  chr1  21653564   C   T  13,15;30,24  2.0  0.043  rs890141715   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000052             0.0000                     0   \n",
       "1           0.001300             0.0017                     3   \n",
       "2           0.000300             0.0005                     2   \n",
       "3           0.001300             0.0020                     0   \n",
       "4           0.000600             0.0013                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...   \n",
       "1  UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "2  UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...   \n",
       "3  MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "4  ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \n",
       "0                          3.94;1.09  \n",
       "1                          4.79;1.83  \n",
       "2                         10.00;6.71  \n",
       "3                          5.20;2.79  \n",
       "4                          5.97;2.29  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs = add_pwmenrich(promoters, promoters_pwmenrich)\n",
    "promoters_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "454 450\n"
     ]
    }
   ],
   "source": [
    "print len(promoters), len(promoters_motifs[~promoters_motifs[\"PWMEnrich motifs\"].isnull()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000100           0.000300                     2   \n",
       "1           0.000700           0.000700                     2   \n",
       "2           0.000032           0.000067                     1   \n",
       "3           0.000097           0.000067                     0   \n",
       "4           0.004400           0.007600                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \n",
       "0                          4.41;1.32  \n",
       "1                          4.48;1.41  \n",
       "2                          4.16;1.45  \n",
       "3                          4.15;1.47  \n",
       "4                          4.75;1.76  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs = add_pwmenrich(enhancers, enhancers_pwmenrich)\n",
    "enhancers_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "351 344\n"
     ]
    }
   ],
   "source": [
    "print len(enhancers), len(enhancers_motifs[~enhancers_motifs[\"PWMEnrich motifs\"].isnull()])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add ENCODE TFBS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_encode(df):\n",
    "    #TFBS from ENCODE after liftover to hg38, copied from /mnr/chr3/People/Bartek/ENCODE-TFBS-CHIP/tfs\n",
    "    tfs_path = '../2019-07-03-Variants_in_ENCODE_TFBS/infiles/tfs/'    \n",
    "    df[\"ENCODE\"] = \"\"\n",
    "    variants_df_for_bed = df.copy()\n",
    "    variants_df_for_bed['POS-1'] = variants_df_for_bed['POS'] - 1\n",
    "    variants_df_for_bed['strand'] = '+'\n",
    "    variants_bedtool = pbt.BedTool.from_dataframe(variants_df_for_bed[['CHROM', 'POS-1', 'POS', 'strand', 'REF', 'ALT']])\n",
    "    for tf_file in os.listdir(tfs_path):\n",
    "        tf_name = tf_file.split('.bed')[0]\n",
    "        tf_bed = pbt.BedTool(os.path.join(tfs_path, tf_file))\n",
    "        intersection_bedtool = variants_bedtool.intersect(tf_bed, wa = True, wb=True) \n",
    "        \n",
    "        # if any variant intersect with this tf add info to its ENCODE column\n",
    "        if len(intersection_bedtool) > 0:\n",
    "            intersection_df = intersection_bedtool.to_dataframe(names = ['CHROM', 'POS', 'TF_'+tf_name],\n",
    "                                                                   usecols = [0,2,9])\n",
    "\n",
    "            df = pd.merge(df, intersection_df, how = \"left\", on = ['CHROM', 'POS']).fillna(\"\")   \n",
    "            df[\"ENCODE\"] = (df[\"ENCODE\"] + \", \" + df['TF_'+tf_name]).str.strip(\", \")\n",
    "            df = df.drop(labels = ['TF_'+tf_name], axis=1)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "enhancers_motifs_encode = add_encode(enhancers_motifs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000100           0.000300                     2   \n",
       "1           0.000700           0.000700                     2   \n",
       "2           0.000032           0.000067                     1   \n",
       "3           0.000097           0.000067                     0   \n",
       "4           0.004400           0.007600                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          4.41;1.32   \n",
       "1                          4.48;1.41   \n",
       "2                          4.16;1.45   \n",
       "3                          4.15;1.47   \n",
       "4                          4.75;1.76   \n",
       "\n",
       "                                              ENCODE  \n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...  \n",
       "1                                                     \n",
       "2                                                     \n",
       "3                                               CTCF  \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913970</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>18,8;24,14;15,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs537954556</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>HMX3_HUMAN.H11MO.0.D, CTCFL_HUMAN.H11MO.0.A, C...</td>\n",
       "      <td>4.54;1.46</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21625236</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>127,104;109,137</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs989937565</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0</td>\n",
       "      <td>NR2C1_HUMAN.H11MO.0.C, PPARD_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.46;1.19</td>\n",
       "      <td>ZNF143, REST, USF1, RCOR1, EP300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24705404</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>81,58;84,65</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs759876216</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>PBX3_HUMAN.H11MO.1.A, SOX9_HUMAN.H11MO.1.B, HX...</td>\n",
       "      <td>4.97;1.85</td>\n",
       "      <td>EGR1, TBP, SP2, NFYA, BACH1, FOS, TEAD4, SPI1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31811750</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>78,81;17,15</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs553603564</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>1</td>\n",
       "      <td>PAX3_HUMAN.H11MO.0.D, PAX8_HUMAN.H11MO.0.D, IR...</td>\n",
       "      <td>4.25;1.17</td>\n",
       "      <td>RAD21, SIN3A, MAX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43542202</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>36,35;8,8;23,28</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs561178520</td>\n",
       "      <td>0.004200</td>\n",
       "      <td>0.007700</td>\n",
       "      <td>2</td>\n",
       "      <td>MYF6_HUMAN.H11MO.0.C, SNAI1_HUMAN.H11MO.0.C, M...</td>\n",
       "      <td>4.09;1.96</td>\n",
       "      <td>HNF4G, RXRA, HNF4A, ZBTB7A, SP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65692298</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40,34;94,96</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs192789815</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>1</td>\n",
       "      <td>HXB7_HUMAN.H11MO.0.C, LHX6_HUMAN.H11MO.0.D, HX...</td>\n",
       "      <td>3.89;1.54</td>\n",
       "      <td>ZNF263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100189355</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>9,5;43,10;5,6;42,6;9,2;33,6</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.130</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>4</td>\n",
       "      <td>BARX2_HUMAN.H11MO.0.D, HXB13_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>5.21;2.61</td>\n",
       "      <td>ESR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>111515686</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>43,53;59,49</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs553576226</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>0</td>\n",
       "      <td>HIC2_HUMAN.H11MO.0.D, RELB_HUMAN.H11MO.0.C, ES...</td>\n",
       "      <td>4.33;1.41</td>\n",
       "      <td>FOS, POLR2A, HMGN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120844254</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0,229;293,160</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs587643657</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>1</td>\n",
       "      <td>CEBPD_HUMAN.H11MO.0.C, DBP_HUMAN.H11MO.0.B, CE...</td>\n",
       "      <td>3.33;1.49</td>\n",
       "      <td>PBX3, NR3C1, TBL1XR1, HSF1, PAX5, ATF3, REST, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>146038682</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>9,10;4,2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs115879572</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, KLF8_HUMAN.H11MO.0.C, N...</td>\n",
       "      <td>4.04;1.30</td>\n",
       "      <td>EGR1, TBP, TFAP2A, POLR2A, YY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>146039391</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>83,91;37,27</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs12091564</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>DLX5_HUMAN.H11MO.0.D, TYY2_HUMAN.H11MO.0.D</td>\n",
       "      <td>3.83;1.55</td>\n",
       "      <td>NR3C1, TBL1XR1, HSF1, ATF3, REST, ATF2, ESR1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>161727094</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>26,3;2,1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs28469291</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>ZBT7A_HUMAN.H11MO.0.A, ZIC1_HUMAN.H11MO.0.B, G...</td>\n",
       "      <td>5.17;2.83</td>\n",
       "      <td>CREB1, PAX5, REST, EGR1, MXI1, TBP, UBTF, BHLH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>164644793</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>120,105;64,86</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs544491415</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>PPARA_HUMAN.H11MO.1.B, PPARG_HUMAN.H11MO.1.A, ...</td>\n",
       "      <td>6.04;2.25</td>\n",
       "      <td>EP300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>167716310</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>27,4;19,7;9,3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs1039397715</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>3</td>\n",
       "      <td>IRF7_HUMAN.H11MO.0.C, GATA3_HUMAN.H11MO.0.A, S...</td>\n",
       "      <td>10.00;6.20</td>\n",
       "      <td>EGR1, RBBP5, UBTF, MAZ, POLR2A, SREBP1, CHD1, MAX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>201513397</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6,3;10,9;9,10</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs72739931</td>\n",
       "      <td>0.003200</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>1</td>\n",
       "      <td>FEZF1_HUMAN.H11MO.0.C, FOXO1_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>3.96;1.99</td>\n",
       "      <td>SRF, FOXM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>201552411</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>132,119;143,128;57,46</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs535033836</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>0.003500</td>\n",
       "      <td>1</td>\n",
       "      <td>ETV6_HUMAN.H11MO.0.D, NFAC2_HUMAN.H11MO.0.B, E...</td>\n",
       "      <td>4.37;1.79</td>\n",
       "      <td>POLR2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>201771524</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>17,13;48,7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.048</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>HMBX1_HUMAN.H11MO.0.D, HOMEZ_HUMAN.H11MO.0.D</td>\n",
       "      <td>3.59;1.26</td>\n",
       "      <td>FOS, STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>202085611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>43,50;57,36</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs573270783</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0</td>\n",
       "      <td>FOSB_HUMAN.H11MO.0.A, ATF1_HUMAN.H11MO.0.B, BA...</td>\n",
       "      <td>3.99;1.44</td>\n",
       "      <td>NR3C1, ESR1, MXI1, ARID3A, MAZ, NFIC, TFAP2C, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>204506594</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100,104;97,71</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs569615040</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN449_HUMAN.H11MO.0.C, NFYB_HUMAN.H11MO.0.A, N...</td>\n",
       "      <td>3.49;1.57</td>\n",
       "      <td>MTA3, ZNF143, BCL11A, TBL1XR1, HSF1, PAX5, ATF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>219556082</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>83,66;56,61</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs183526420</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0</td>\n",
       "      <td>E4F1_HUMAN.H11MO.0.D, EPAS1_HUMAN.H11MO.0.B, M...</td>\n",
       "      <td>4.51;1.50</td>\n",
       "      <td>FOSL2, FOS, TCF12, EP300, CEBPB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>223707099</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>21,5;9,3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000080</td>\n",
       "      <td>1</td>\n",
       "      <td>HSF1_HUMAN.H11MO.1.A, RUNX3_HUMAN.H11MO.0.A, G...</td>\n",
       "      <td>4.89;1.72</td>\n",
       "      <td>ZBTB33, GATA2, SIN3AK20, SMARCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr2</td>\n",
       "      <td>8684545</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>28,13;35,3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>2</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B, EGR3_HUMAN.H11MO.0.D, K...</td>\n",
       "      <td>6.26;1.69</td>\n",
       "      <td>FOXA1, TBP, UBTF, GATA3, ZBTB33, POLR2A, KDM5B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr2</td>\n",
       "      <td>10285627</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>90,103;113,86</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs769033826</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>0</td>\n",
       "      <td>ZFX_HUMAN.H11MO.0.A, NR5A2_HUMAN.H11MO.0.B, ZB...</td>\n",
       "      <td>10.00;6.24</td>\n",
       "      <td>USF2, NR3C1, ATF3, REST, MXI1, TBP, BACH1, MAF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr2</td>\n",
       "      <td>16070133</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>66,61;68,65</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs530153054</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0</td>\n",
       "      <td>TBX5_HUMAN.H11MO.0.D, NKX23_HUMAN.H11MO.0.D, S...</td>\n",
       "      <td>3.52;1.55</td>\n",
       "      <td>MAFK, MYBL2, POLR2A, MAFF, TAF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr2</td>\n",
       "      <td>22855701</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>95,85;87,103</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs774859729</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0</td>\n",
       "      <td>RUNX3_HUMAN.H11MO.0.A, TBP_HUMAN.H11MO.0.A, HX...</td>\n",
       "      <td>4.83;1.99</td>\n",
       "      <td>GATA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr2</td>\n",
       "      <td>22855850</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>32,34;55,60;40,38</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs548638610</td>\n",
       "      <td>0.003700</td>\n",
       "      <td>0.006700</td>\n",
       "      <td>2</td>\n",
       "      <td>NKX25_HUMAN.H11MO.0.B, NKX21_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>4.22;1.91</td>\n",
       "      <td>REST, GATA3, RAD21, CTCF, EP300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>chr2</td>\n",
       "      <td>28594772</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>52,35;40,45</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs960973165</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>2</td>\n",
       "      <td>PO5F1_HUMAN.H11MO.1.A, PO3F1_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>3.89;1.52</td>\n",
       "      <td>FOS, POLR2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>chr17</td>\n",
       "      <td>62820233</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>75,56;56,40;56,39</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs985067943</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXJ2_HUMAN.H11MO.0.C, P63_HUMAN.H11MO.1.A, FO...</td>\n",
       "      <td>3.32;1.36</td>\n",
       "      <td>ZBTB33, FOS, MAZ, TCF12, POLR2A, YY1, ZNF217, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>chr17</td>\n",
       "      <td>65079754</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>76,67;72,63</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs561611230</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0.002200</td>\n",
       "      <td>2</td>\n",
       "      <td>MAF_HUMAN.H11MO.1.B, MAFF_HUMAN.H11MO.1.B, ATO...</td>\n",
       "      <td>4.97;2.41</td>\n",
       "      <td>REST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>chr17</td>\n",
       "      <td>66374447</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>89,92;38,34</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs545058536</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>1</td>\n",
       "      <td>E2F5_HUMAN.H11MO.0.B, OLIG2_HUMAN.H11MO.1.B, Z...</td>\n",
       "      <td>3.92;1.48</td>\n",
       "      <td>REST, RCOR1, GATA2, CTCF, KAP1, SIN3AK20, GATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>chr17</td>\n",
       "      <td>72408934</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>16,14;12,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs563667690</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>0</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, GATA4_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>10.00;7.13</td>\n",
       "      <td>TFAP2A, TFAP2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>chr17</td>\n",
       "      <td>77367028</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>32,8;19,15</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs557169310</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>1</td>\n",
       "      <td>SOX15_HUMAN.H11MO.0.D, ZN341_HUMAN.H11MO.1.C, ...</td>\n",
       "      <td>4.52;1.60</td>\n",
       "      <td>USF2, SPI1, USF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>chr18</td>\n",
       "      <td>13578665</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>14,22;10,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs764346857</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>1</td>\n",
       "      <td>JUNB_HUMAN.H11MO.0.A, JUN_HUMAN.H11MO.0.A, MAF...</td>\n",
       "      <td>3.98;1.95</td>\n",
       "      <td>EBF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>chr19</td>\n",
       "      <td>2167742</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>9,2;7,2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...</td>\n",
       "      <td>10.00;7.20</td>\n",
       "      <td>POLR2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>chr19</td>\n",
       "      <td>4058396</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>28,27;80,53;44,38</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs565715379</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>1</td>\n",
       "      <td>HEN1_HUMAN.H11MO.0.C</td>\n",
       "      <td>3.32;1.03</td>\n",
       "      <td>ZNF143, EGR1, MXI1, UBTF, JUN, FOSL2, FOS, MAZ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>chr19</td>\n",
       "      <td>5037839</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>19,15;41,40</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs537370730</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>2</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C</td>\n",
       "      <td>3.59;1.10</td>\n",
       "      <td>GATA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>chr19</td>\n",
       "      <td>5586007</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>33,30;31,33</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs550202215</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>MYBB_HUMAN.H11MO.0.D, BSH_HUMAN.H11MO.0.D, HXC...</td>\n",
       "      <td>3.27;1.42</td>\n",
       "      <td>REST, EGR1, MAZ, RELA, E2F6, NR2F2, POLR2A, YY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6515973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>8,2;6,1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>.</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, FOXJ2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.34;1.78</td>\n",
       "      <td>MAZ, GATA2, JUND, MYC, GATA1, CEBPB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>chr19</td>\n",
       "      <td>7224268</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>39,15;59,13;50,14;53,12;57,11;43,9</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.130</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>3</td>\n",
       "      <td>SOX15_HUMAN.H11MO.0.D, SRY_HUMAN.H11MO.0.B, ST...</td>\n",
       "      <td>5.08;1.55</td>\n",
       "      <td>POU2F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>chr19</td>\n",
       "      <td>10888846</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>33,20;107,92</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs539661966</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>1</td>\n",
       "      <td>NR4A1_HUMAN.H11MO.0.A, ZN341_HUMAN.H11MO.1.C, ...</td>\n",
       "      <td>3.60;1.38</td>\n",
       "      <td>MXI1, RELA, POU2F2, SIN3AK20, SMC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>chr19</td>\n",
       "      <td>16341971</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>9,3;4,3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>1</td>\n",
       "      <td>FOXP1_HUMAN.H11MO.0.A, GATA4_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>5.13;1.88</td>\n",
       "      <td>MAZ, TCF3, STAT5A, ELF1, NFATC1, MEF2A, CEBPB,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>chr19</td>\n",
       "      <td>38666228</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>67,58;66,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs555684075</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>1</td>\n",
       "      <td>BCL6B_HUMAN.H11MO.0.D, ZNF18_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.21;1.45</td>\n",
       "      <td>REST, SRF, EGR1, ESR1, TBP, JUN, TEAD4, NANOG,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>chr19</td>\n",
       "      <td>43699794</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>14,11;109,102</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs563046331</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>1</td>\n",
       "      <td>SRBP1_HUMAN.H11MO.0.A, ZN784_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.19;1.31</td>\n",
       "      <td>RAD21, CTCF, SIN3A, SIN3AK20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>chr19</td>\n",
       "      <td>47600491</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>30,30;40,31;20,10</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs573875506</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.006600</td>\n",
       "      <td>1</td>\n",
       "      <td>HBP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, H...</td>\n",
       "      <td>4.18;1.56</td>\n",
       "      <td>CREB1, PAX5, FOXA1, REST, ATF2, MXI1, TBP, NFY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>chr20</td>\n",
       "      <td>34709316</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>14,18;38,36;11,7</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs199743434</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, HAND1_HUMAN.H11MO.1.D, ...</td>\n",
       "      <td>3.54;1.65</td>\n",
       "      <td>MTA3, CREB1, REST, ATF2, EGR1, MAZ, CCNT2, NFI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>chr20</td>\n",
       "      <td>46206927</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>78,74;35,47</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs370823375</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.002000</td>\n",
       "      <td>1</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C</td>\n",
       "      <td>3.84;1.08</td>\n",
       "      <td>TFAP2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>chr20</td>\n",
       "      <td>50573586</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>58,51;108,89</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs191855113</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>1</td>\n",
       "      <td>SOX15_HUMAN.H11MO.0.D, CEBPZ_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.45;1.63</td>\n",
       "      <td>ESR1, ZNF217, GATA2, CTCF, MYC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>chr21</td>\n",
       "      <td>31514521</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>3,2;21,5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.059</td>\n",
       "      <td>rs1052583672</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>0</td>\n",
       "      <td>HMGA1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, F...</td>\n",
       "      <td>10.00;7.53</td>\n",
       "      <td>KAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>chr21</td>\n",
       "      <td>45554469</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>rs569666409</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>SOX13_HUMAN.H11MO.0.D, ZIC1_HUMAN.H11MO.0.B, G...</td>\n",
       "      <td>4.86;1.42</td>\n",
       "      <td>CTCF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28034954</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91,102;99,84</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs984576954</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0</td>\n",
       "      <td>NKX61_HUMAN.H11MO.1.B, FOSB_HUMAN.H11MO.0.A, F...</td>\n",
       "      <td>4.05;1.64</td>\n",
       "      <td>REST, RAD21, POLR2A, YY1, USF1, MYC, EP300, GATA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>chr22</td>\n",
       "      <td>30872281</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>99,79;86,102;25,30</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs143554716</td>\n",
       "      <td>0.002900</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>2</td>\n",
       "      <td>DLX2_HUMAN.H11MO.0.D, SOX13_HUMAN.H11MO.0.D, O...</td>\n",
       "      <td>4.77;1.38</td>\n",
       "      <td>KAP1, GATA1, TCF7L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41412963</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>42,34;41,30</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs938357671</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL2_HUMAN.H11MO.0.D, NR5A2_HUMAN.H11MO.0.B, ...</td>\n",
       "      <td>3.57;1.41</td>\n",
       "      <td>E2F6, MAX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>chr22</td>\n",
       "      <td>49967521</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>53,58;27,34;11,17</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs142038940</td>\n",
       "      <td>0.004000</td>\n",
       "      <td>0.006300</td>\n",
       "      <td>2</td>\n",
       "      <td>ZFX_HUMAN.H11MO.1.A</td>\n",
       "      <td>4.53;1.22</td>\n",
       "      <td>POLR2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>chrX</td>\n",
       "      <td>47130033</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0,26</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>1</td>\n",
       "      <td>NKX61_HUMAN.H11MO.1.B, HSF1_HUMAN.H11MO.1.A, I...</td>\n",
       "      <td>10.00;6.51</td>\n",
       "      <td>MAFK, ZBTB7A, MAFF, TCF7L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>chrX</td>\n",
       "      <td>53072145</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs781871805</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002200</td>\n",
       "      <td>1</td>\n",
       "      <td>ZIC1_HUMAN.H11MO.0.B, SALL4_HUMAN.H11MO.0.B, E...</td>\n",
       "      <td>5.18;2.49</td>\n",
       "      <td>MXI1, TBP, POLR2A, KAP1, MYC, SMARCC1, SMARCB1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>chrX</td>\n",
       "      <td>129932525</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0,2;2,3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.075</td>\n",
       "      <td>rs1014404153</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0</td>\n",
       "      <td>GCM2_HUMAN.H11MO.0.D, NR2C2_HUMAN.H11MO.0.B, P...</td>\n",
       "      <td>5.59;1.78</td>\n",
       "      <td>RBBP5, MAZ, E2F6, PHF8, POLR2A, E2F1, CHD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>chrX</td>\n",
       "      <td>131753299</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0,27</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs111850126</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>1</td>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D, HXC11_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.27;1.85</td>\n",
       "      <td>ATF3, RAD21, ZBTB33, MAZ, YY1, CTCF, SP1, ETS1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>154 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                                VAFs   AC     AF  \\\n",
       "0     chr1    2227533   C   T                           4,4;13,11  2.0  0.043   \n",
       "3     chr1   16912416   G   A                       189,90;160,95  2.0  0.043   \n",
       "4     chr1   16913968   T   A                    18,9;24,13;17,11  3.0  0.065   \n",
       "5     chr1   16913970   A   G                    18,8;24,14;15,11  3.0  0.065   \n",
       "8     chr1   21625236   G   A                     127,104;109,137  2.0  0.043   \n",
       "9     chr1   24705404   T   A                         81,58;84,65  2.0  0.043   \n",
       "12    chr1   31811750   A   T                         78,81;17,15  2.0  0.043   \n",
       "15    chr1   43542202   G   A                     36,35;8,8;23,28  3.0  0.065   \n",
       "16    chr1   65692298   G   A                         40,34;94,96  2.0  0.043   \n",
       "19    chr1  100189355   A   T         9,5;43,10;5,6;42,6;9,2;33,6  6.0  0.130   \n",
       "20    chr1  111515686   G   C                         43,53;59,49  2.0  0.043   \n",
       "21    chr1  120844254   C   T                       0,229;293,160  3.0  0.065   \n",
       "22    chr1  146038682   G   A                            9,10;4,2  2.0  0.043   \n",
       "23    chr1  146039391   A   G                         83,91;37,27  2.0  0.043   \n",
       "25    chr1  161727094   G   C                            26,3;2,1  2.0  0.043   \n",
       "26    chr1  164644793   C   A                       120,105;64,86  2.0  0.043   \n",
       "28    chr1  167716310   C   T                       27,4;19,7;9,3  3.0  0.065   \n",
       "32    chr1  201513397   T   C                       6,3;10,9;9,10  3.0  0.065   \n",
       "33    chr1  201552411   C   T               132,119;143,128;57,46  3.0  0.065   \n",
       "34    chr1  201771524   T   C                          17,13;48,7  2.0  0.048   \n",
       "35    chr1  202085611   T   C                         43,50;57,36  2.0  0.043   \n",
       "37    chr1  204506594   C   T                       100,104;97,71  2.0  0.043   \n",
       "39    chr1  219556082   C   T                         83,66;56,61  2.0  0.043   \n",
       "40    chr1  223707099   T   A                            21,5;9,3  2.0  0.045   \n",
       "44    chr2    8684545   A   C                          28,13;35,3  2.0  0.043   \n",
       "45    chr2   10285627   C   T                       90,103;113,86  2.0  0.043   \n",
       "46    chr2   16070133   C   T                         66,61;68,65  2.0  0.043   \n",
       "47    chr2   22855701   C   T                        95,85;87,103  2.0  0.043   \n",
       "48    chr2   22855850   C   T                   32,34;55,60;40,38  3.0  0.065   \n",
       "50    chr2   28594772   G   A                         52,35;40,45  2.0  0.043   \n",
       "..     ...        ...  ..  ..                                 ...  ...    ...   \n",
       "300  chr17   62820233   A   G                   75,56;56,40;56,39  3.0  0.065   \n",
       "301  chr17   65079754   T   C                         76,67;72,63  2.0  0.043   \n",
       "303  chr17   66374447   C   T                         89,92;38,34  2.0  0.043   \n",
       "305  chr17   72408934   T   G                          16,14;12,6  2.0  0.043   \n",
       "307  chr17   77367028   G   C                          32,8;19,15  2.0  0.043   \n",
       "309  chr18   13578665   A   G                         14,22;10,11  2.0  0.043   \n",
       "312  chr19    2167742   T   C                             9,2;7,2  2.0  0.043   \n",
       "313  chr19    4058396   C   T                   28,27;80,53;44,38  3.0  0.065   \n",
       "314  chr19    5037839   C   A                         19,15;41,40  2.0  0.043   \n",
       "315  chr19    5586007   A   G                         33,30;31,33  2.0  0.043   \n",
       "316  chr19    6515973   G   A                             8,2;6,1  2.0  0.045   \n",
       "317  chr19    7224268   C   A  39,15;59,13;50,14;53,12;57,11;43,9  6.0  0.130   \n",
       "318  chr19   10888846   A   C                        33,20;107,92  2.0  0.043   \n",
       "320  chr19   16341971   G   A                             9,3;4,3  2.0  0.043   \n",
       "321  chr19   38666228   C   A                         67,58;66,70  2.0  0.043   \n",
       "322  chr19   43699794   G   A                       14,11;109,102  2.0  0.043   \n",
       "323  chr19   47600491   T   C                   30,30;40,31;20,10  3.0  0.065   \n",
       "328  chr20   34709316   G   C                    14,18;38,36;11,7  3.0  0.065   \n",
       "329  chr20   46206927   G   A                         78,74;35,47  2.0  0.043   \n",
       "330  chr20   50573586   G   A                        58,51;108,89  2.0  0.043   \n",
       "331  chr21   31514521   T   A                            3,2;21,5  2.0  0.059   \n",
       "333  chr21   45554469   C   A                                 0,6  2.0  0.045   \n",
       "334  chr22   28034954   T   C                        91,102;99,84  2.0  0.043   \n",
       "335  chr22   30872281   T   C                  99,79;86,102;25,30  3.0  0.065   \n",
       "336  chr22   41412963   C   T                         42,34;41,30  2.0  0.043   \n",
       "338  chr22   49967521   G   A                   53,58;27,34;11,17  3.0  0.065   \n",
       "342   chrX   47130033   A   G                                0,26  2.0  0.043   \n",
       "345   chrX   53072145   A   G                                0,32  2.0  0.043   \n",
       "348   chrX  129932525   G   A                             0,2;2,3  3.0  0.075   \n",
       "349   chrX  131753299   T   A                                0,27  2.0  0.043   \n",
       "\n",
       "         avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0               .           0.000100           0.000300                     2   \n",
       "3               .           0.000097           0.000067                     0   \n",
       "4     rs752881304           0.004400           0.007600                     2   \n",
       "5     rs537954556           0.004400           0.007600                     2   \n",
       "8     rs989937565           0.000065           0.000100                     0   \n",
       "9     rs759876216           0.000032           0.000000                     0   \n",
       "12    rs553603564           0.001700           0.002500                     1   \n",
       "15    rs561178520           0.004200           0.007700                     2   \n",
       "16    rs192789815           0.000500           0.000900                     1   \n",
       "19              .           0.000200           0.000100                     4   \n",
       "20    rs553576226           0.000700           0.000900                     0   \n",
       "21    rs587643657           0.000068           0.000073                     1   \n",
       "22    rs115879572           0.000032           0.000067                     1   \n",
       "23     rs12091564           0.000032           0.000000                     1   \n",
       "25     rs28469291           0.000100           0.000000                     1   \n",
       "26    rs544491415           0.000300           0.000300                     2   \n",
       "28   rs1039397715           0.000100           0.000077                     3   \n",
       "32     rs72739931           0.003200           0.004700                     1   \n",
       "33    rs535033836           0.002300           0.003500                     1   \n",
       "34              .           0.000043           0.000000                     0   \n",
       "35    rs573270783           0.000300           0.000300                     0   \n",
       "37    rs569615040           0.000500           0.000500                     0   \n",
       "39    rs183526420           0.001100           0.001500                     0   \n",
       "40              .           0.000400           0.000080                     1   \n",
       "44              .           0.000900           0.000500                     2   \n",
       "45    rs769033826           0.001200           0.002300                     0   \n",
       "46    rs530153054           0.000100           0.000300                     0   \n",
       "47    rs774859729           0.000100           0.000200                     0   \n",
       "48    rs548638610           0.003700           0.006700                     2   \n",
       "50    rs960973165           0.000700           0.001300                     2   \n",
       "..            ...                ...                ...                   ...   \n",
       "300   rs985067943           0.000300           0.000500                     3   \n",
       "301   rs561611230           0.001500           0.002200                     2   \n",
       "303   rs545058536           0.001300           0.002300                     1   \n",
       "305   rs563667690           0.000600           0.001000                     0   \n",
       "307   rs557169310           0.000400           0.000700                     1   \n",
       "309   rs764346857           0.000300           0.000600                     1   \n",
       "312             .           0.000036           0.000000                     2   \n",
       "313   rs565715379           0.000600           0.001100                     1   \n",
       "314   rs537370730           0.000500           0.000600                     2   \n",
       "315   rs550202215           0.000500           0.000700                     2   \n",
       "316             .           0.001200           0.000700                     2   \n",
       "317             .           0.000039           0.000079                     3   \n",
       "318   rs539661966           0.000500           0.000600                     1   \n",
       "320             .           0.000400           0.000600                     1   \n",
       "321   rs555684075           0.000400           0.000800                     1   \n",
       "322   rs563046331           0.000900           0.001800                     1   \n",
       "323   rs573875506           0.004700           0.006600                     1   \n",
       "328   rs199743434           0.000700           0.001300                     2   \n",
       "329   rs370823375           0.001100           0.002000                     1   \n",
       "330   rs191855113           0.002300           0.002300                     1   \n",
       "331  rs1052583672           0.001700           0.001800                     0   \n",
       "333   rs569666409           0.000000           0.000000                     1   \n",
       "334   rs984576954           0.000065           0.000100                     0   \n",
       "335   rs143554716           0.002900           0.004400                     2   \n",
       "336   rs938357671           0.000200           0.000500                     2   \n",
       "338   rs142038940           0.004000           0.006300                     2   \n",
       "342             .           0.000600           0.001000                     1   \n",
       "345   rs781871805           0.001300           0.002200                     1   \n",
       "348  rs1014404153           0.000700           0.001300                     0   \n",
       "349   rs111850126           0.000400           0.000700                     1   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "0    AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "3    MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4    TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "5    HMX3_HUMAN.H11MO.0.D, CTCFL_HUMAN.H11MO.0.A, C...   \n",
       "8    NR2C1_HUMAN.H11MO.0.C, PPARD_HUMAN.H11MO.0.D, ...   \n",
       "9    PBX3_HUMAN.H11MO.1.A, SOX9_HUMAN.H11MO.1.B, HX...   \n",
       "12   PAX3_HUMAN.H11MO.0.D, PAX8_HUMAN.H11MO.0.D, IR...   \n",
       "15   MYF6_HUMAN.H11MO.0.C, SNAI1_HUMAN.H11MO.0.C, M...   \n",
       "16   HXB7_HUMAN.H11MO.0.C, LHX6_HUMAN.H11MO.0.D, HX...   \n",
       "19   BARX2_HUMAN.H11MO.0.D, HXB13_HUMAN.H11MO.0.A, ...   \n",
       "20   HIC2_HUMAN.H11MO.0.D, RELB_HUMAN.H11MO.0.C, ES...   \n",
       "21   CEBPD_HUMAN.H11MO.0.C, DBP_HUMAN.H11MO.0.B, CE...   \n",
       "22   MECP2_HUMAN.H11MO.0.C, KLF8_HUMAN.H11MO.0.C, N...   \n",
       "23          DLX5_HUMAN.H11MO.0.D, TYY2_HUMAN.H11MO.0.D   \n",
       "25   ZBT7A_HUMAN.H11MO.0.A, ZIC1_HUMAN.H11MO.0.B, G...   \n",
       "26   PPARA_HUMAN.H11MO.1.B, PPARG_HUMAN.H11MO.1.A, ...   \n",
       "28   IRF7_HUMAN.H11MO.0.C, GATA3_HUMAN.H11MO.0.A, S...   \n",
       "32   FEZF1_HUMAN.H11MO.0.C, FOXO1_HUMAN.H11MO.0.A, ...   \n",
       "33   ETV6_HUMAN.H11MO.0.D, NFAC2_HUMAN.H11MO.0.B, E...   \n",
       "34        HMBX1_HUMAN.H11MO.0.D, HOMEZ_HUMAN.H11MO.0.D   \n",
       "35   FOSB_HUMAN.H11MO.0.A, ATF1_HUMAN.H11MO.0.B, BA...   \n",
       "37   ZN449_HUMAN.H11MO.0.C, NFYB_HUMAN.H11MO.0.A, N...   \n",
       "39   E4F1_HUMAN.H11MO.0.D, EPAS1_HUMAN.H11MO.0.B, M...   \n",
       "40   HSF1_HUMAN.H11MO.1.A, RUNX3_HUMAN.H11MO.0.A, G...   \n",
       "44   SALL4_HUMAN.H11MO.0.B, EGR3_HUMAN.H11MO.0.D, K...   \n",
       "45   ZFX_HUMAN.H11MO.0.A, NR5A2_HUMAN.H11MO.0.B, ZB...   \n",
       "46   TBX5_HUMAN.H11MO.0.D, NKX23_HUMAN.H11MO.0.D, S...   \n",
       "47   RUNX3_HUMAN.H11MO.0.A, TBP_HUMAN.H11MO.0.A, HX...   \n",
       "48   NKX25_HUMAN.H11MO.0.B, NKX21_HUMAN.H11MO.0.A, ...   \n",
       "50   PO5F1_HUMAN.H11MO.1.A, PO3F1_HUMAN.H11MO.0.C, ...   \n",
       "..                                                 ...   \n",
       "300  FOXJ2_HUMAN.H11MO.0.C, P63_HUMAN.H11MO.1.A, FO...   \n",
       "301  MAF_HUMAN.H11MO.1.B, MAFF_HUMAN.H11MO.1.B, ATO...   \n",
       "303  E2F5_HUMAN.H11MO.0.B, OLIG2_HUMAN.H11MO.1.B, Z...   \n",
       "305  UBIP1_HUMAN.H11MO.0.D, GATA4_HUMAN.H11MO.0.A, ...   \n",
       "307  SOX15_HUMAN.H11MO.0.D, ZN341_HUMAN.H11MO.1.C, ...   \n",
       "309  JUNB_HUMAN.H11MO.0.A, JUN_HUMAN.H11MO.0.A, MAF...   \n",
       "312  GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...   \n",
       "313                               HEN1_HUMAN.H11MO.0.C   \n",
       "314                              ZBT14_HUMAN.H11MO.0.C   \n",
       "315  MYBB_HUMAN.H11MO.0.D, BSH_HUMAN.H11MO.0.D, HXC...   \n",
       "316  UBIP1_HUMAN.H11MO.0.D, FOXJ2_HUMAN.H11MO.0.C, ...   \n",
       "317  SOX15_HUMAN.H11MO.0.D, SRY_HUMAN.H11MO.0.B, ST...   \n",
       "318  NR4A1_HUMAN.H11MO.0.A, ZN341_HUMAN.H11MO.1.C, ...   \n",
       "320  FOXP1_HUMAN.H11MO.0.A, GATA4_HUMAN.H11MO.0.A, ...   \n",
       "321  BCL6B_HUMAN.H11MO.0.D, ZNF18_HUMAN.H11MO.0.C, ...   \n",
       "322  SRBP1_HUMAN.H11MO.0.A, ZN784_HUMAN.H11MO.0.D, ...   \n",
       "323  HBP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, H...   \n",
       "328  UBIP1_HUMAN.H11MO.0.D, HAND1_HUMAN.H11MO.1.D, ...   \n",
       "329                              ZBT14_HUMAN.H11MO.0.C   \n",
       "330  SOX15_HUMAN.H11MO.0.D, CEBPZ_HUMAN.H11MO.0.D, ...   \n",
       "331  HMGA1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, F...   \n",
       "333  SOX13_HUMAN.H11MO.0.D, ZIC1_HUMAN.H11MO.0.B, G...   \n",
       "334  NKX61_HUMAN.H11MO.1.B, FOSB_HUMAN.H11MO.0.A, F...   \n",
       "335  DLX2_HUMAN.H11MO.0.D, SOX13_HUMAN.H11MO.0.D, O...   \n",
       "336  OVOL2_HUMAN.H11MO.0.D, NR5A2_HUMAN.H11MO.0.B, ...   \n",
       "338                                ZFX_HUMAN.H11MO.1.A   \n",
       "342  NKX61_HUMAN.H11MO.1.B, HSF1_HUMAN.H11MO.1.A, I...   \n",
       "345  ZIC1_HUMAN.H11MO.0.B, SALL4_HUMAN.H11MO.0.B, E...   \n",
       "348  GCM2_HUMAN.H11MO.0.D, NR2C2_HUMAN.H11MO.0.B, P...   \n",
       "349  CXXC1_HUMAN.H11MO.0.D, HXC11_HUMAN.H11MO.0.D, ...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "0                            4.41;1.32   \n",
       "3                            4.15;1.47   \n",
       "4                            4.75;1.76   \n",
       "5                            4.54;1.46   \n",
       "8                            4.46;1.19   \n",
       "9                            4.97;1.85   \n",
       "12                           4.25;1.17   \n",
       "15                           4.09;1.96   \n",
       "16                           3.89;1.54   \n",
       "19                           5.21;2.61   \n",
       "20                           4.33;1.41   \n",
       "21                           3.33;1.49   \n",
       "22                           4.04;1.30   \n",
       "23                           3.83;1.55   \n",
       "25                           5.17;2.83   \n",
       "26                           6.04;2.25   \n",
       "28                          10.00;6.20   \n",
       "32                           3.96;1.99   \n",
       "33                           4.37;1.79   \n",
       "34                           3.59;1.26   \n",
       "35                           3.99;1.44   \n",
       "37                           3.49;1.57   \n",
       "39                           4.51;1.50   \n",
       "40                           4.89;1.72   \n",
       "44                           6.26;1.69   \n",
       "45                          10.00;6.24   \n",
       "46                           3.52;1.55   \n",
       "47                           4.83;1.99   \n",
       "48                           4.22;1.91   \n",
       "50                           3.89;1.52   \n",
       "..                                 ...   \n",
       "300                          3.32;1.36   \n",
       "301                          4.97;2.41   \n",
       "303                          3.92;1.48   \n",
       "305                         10.00;7.13   \n",
       "307                          4.52;1.60   \n",
       "309                          3.98;1.95   \n",
       "312                         10.00;7.20   \n",
       "313                          3.32;1.03   \n",
       "314                          3.59;1.10   \n",
       "315                          3.27;1.42   \n",
       "316                          4.34;1.78   \n",
       "317                          5.08;1.55   \n",
       "318                          3.60;1.38   \n",
       "320                          5.13;1.88   \n",
       "321                          4.21;1.45   \n",
       "322                          4.19;1.31   \n",
       "323                          4.18;1.56   \n",
       "328                          3.54;1.65   \n",
       "329                          3.84;1.08   \n",
       "330                          4.45;1.63   \n",
       "331                         10.00;7.53   \n",
       "333                          4.86;1.42   \n",
       "334                          4.05;1.64   \n",
       "335                          4.77;1.38   \n",
       "336                          3.57;1.41   \n",
       "338                          4.53;1.22   \n",
       "342                         10.00;6.51   \n",
       "345                          5.18;2.49   \n",
       "348                          5.59;1.78   \n",
       "349                          3.27;1.85   \n",
       "\n",
       "                                                ENCODE  \n",
       "0    NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...  \n",
       "3                                                 CTCF  \n",
       "4    PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...  \n",
       "5    PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...  \n",
       "8                     ZNF143, REST, USF1, RCOR1, EP300  \n",
       "9    EGR1, TBP, SP2, NFYA, BACH1, FOS, TEAD4, SPI1,...  \n",
       "12                                   RAD21, SIN3A, MAX  \n",
       "15                     HNF4G, RXRA, HNF4A, ZBTB7A, SP1  \n",
       "16                                              ZNF263  \n",
       "19                                                ESR1  \n",
       "20                                  FOS, POLR2A, HMGN3  \n",
       "21   PBX3, NR3C1, TBL1XR1, HSF1, PAX5, ATF3, REST, ...  \n",
       "22                      EGR1, TBP, TFAP2A, POLR2A, YY1  \n",
       "23   NR3C1, TBL1XR1, HSF1, ATF3, REST, ATF2, ESR1, ...  \n",
       "25   CREB1, PAX5, REST, EGR1, MXI1, TBP, UBTF, BHLH...  \n",
       "26                                               EP300  \n",
       "28   EGR1, RBBP5, UBTF, MAZ, POLR2A, SREBP1, CHD1, MAX  \n",
       "32                                          SRF, FOXM1  \n",
       "33                                              POLR2A  \n",
       "34                                          FOS, STAT3  \n",
       "35   NR3C1, ESR1, MXI1, ARID3A, MAZ, NFIC, TFAP2C, ...  \n",
       "37   MTA3, ZNF143, BCL11A, TBL1XR1, HSF1, PAX5, ATF...  \n",
       "39                     FOSL2, FOS, TCF12, EP300, CEBPB  \n",
       "40                    ZBTB33, GATA2, SIN3AK20, SMARCC1  \n",
       "44   FOXA1, TBP, UBTF, GATA3, ZBTB33, POLR2A, KDM5B...  \n",
       "45   USF2, NR3C1, ATF3, REST, MXI1, TBP, BACH1, MAF...  \n",
       "46                     MAFK, MYBL2, POLR2A, MAFF, TAF1  \n",
       "47                                               GATA3  \n",
       "48                     REST, GATA3, RAD21, CTCF, EP300  \n",
       "50                                         FOS, POLR2A  \n",
       "..                                                 ...  \n",
       "300  ZBTB33, FOS, MAZ, TCF12, POLR2A, YY1, ZNF217, ...  \n",
       "301                                               REST  \n",
       "303  REST, RCOR1, GATA2, CTCF, KAP1, SIN3AK20, GATA...  \n",
       "305                                     TFAP2A, TFAP2C  \n",
       "307                                   USF2, SPI1, USF1  \n",
       "309                                               EBF1  \n",
       "312                                             POLR2A  \n",
       "313  ZNF143, EGR1, MXI1, UBTF, JUN, FOSL2, FOS, MAZ...  \n",
       "314                                              GATA2  \n",
       "315  REST, EGR1, MAZ, RELA, E2F6, NR2F2, POLR2A, YY...  \n",
       "316                MAZ, GATA2, JUND, MYC, GATA1, CEBPB  \n",
       "317                                             POU2F2  \n",
       "318                 MXI1, RELA, POU2F2, SIN3AK20, SMC3  \n",
       "320  MAZ, TCF3, STAT5A, ELF1, NFATC1, MEF2A, CEBPB,...  \n",
       "321  REST, SRF, EGR1, ESR1, TBP, JUN, TEAD4, NANOG,...  \n",
       "322                       RAD21, CTCF, SIN3A, SIN3AK20  \n",
       "323  CREB1, PAX5, FOXA1, REST, ATF2, MXI1, TBP, NFY...  \n",
       "328  MTA3, CREB1, REST, ATF2, EGR1, MAZ, CCNT2, NFI...  \n",
       "329                                             TFAP2C  \n",
       "330                     ESR1, ZNF217, GATA2, CTCF, MYC  \n",
       "331                                               KAP1  \n",
       "333                                               CTCF  \n",
       "334  REST, RAD21, POLR2A, YY1, USF1, MYC, EP300, GATA1  \n",
       "335                                KAP1, GATA1, TCF7L2  \n",
       "336                                          E2F6, MAX  \n",
       "338                                             POLR2A  \n",
       "342                         MAFK, ZBTB7A, MAFF, TCF7L2  \n",
       "345  MXI1, TBP, POLR2A, KAP1, MYC, SMARCC1, SMARCB1...  \n",
       "348         RBBP5, MAZ, E2F6, PHF8, POLR2A, E2F1, CHD1  \n",
       "349  ATF3, RAD21, ZBTB33, MAZ, YY1, CTCF, SP1, ETS1...  \n",
       "\n",
       "[154 rows x 14 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode[~(enhancers_motifs_encode[\"PWMEnrich motifs\"]==\"\") & ~(enhancers_motifs_encode[\"ENCODE\"]==\"\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...</td>\n",
       "      <td>3.94;1.09</td>\n",
       "      <td>EZH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>3</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.79;1.83</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...</td>\n",
       "      <td>10.00;6.71</td>\n",
       "      <td>ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "      <td>MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...</td>\n",
       "      <td>5.97;2.29</td>\n",
       "      <td>ZNF263, POLR2A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT         VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1    939561   G   A  72,69;91,70  2.0  0.043  rs888928964   \n",
       "1  chr1   1034902   A   T  7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2  chr1   6180416   G   A    10,11;8,6  2.0  0.043  rs151212360   \n",
       "3  chr1  14924017   C   G    6,5;12,14  2.0  0.043  rs531269337   \n",
       "4  chr1  21653564   C   T  13,15;30,24  2.0  0.043  rs890141715   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000052             0.0000                     0   \n",
       "1           0.001300             0.0017                     3   \n",
       "2           0.000300             0.0005                     2   \n",
       "3           0.001300             0.0020                     0   \n",
       "4           0.000600             0.0013                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...   \n",
       "1  UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "2  UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...   \n",
       "3  MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "4  ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          3.94;1.09   \n",
       "1                          4.79;1.83   \n",
       "2                         10.00;6.71   \n",
       "3                          5.20;2.79   \n",
       "4                          5.97;2.29   \n",
       "\n",
       "                                              ENCODE  \n",
       "0                                               EZH2  \n",
       "1                                                     \n",
       "2            ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1  \n",
       "3  MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...  \n",
       "4                                     ZNF263, POLR2A  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode = add_encode(promoters_motifs)\n",
    "promoters_motifs_encode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(454, 454)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoters_motifs_encode), len(promoters_motifs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add genomic element and gene info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Promoters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters_motifs_encode[\"genomic element\"] = \"promoter\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t28554\t29754\tENST00000473358\tENSG00000243485\tRP11-34P13.3\tactive promoter 20pts\n",
      "chr1\t29267\t30467\tENST00000469289\tENSG00000243485\tRP11-34P13.3\tactive promoter 20pts\n",
      "chr1\t29366\t30566\tENST00000607096\tENSG00000274890\tMIR1302-2\tactive promoter 20pts\n",
      "chr1\t29370\t30570\tENST00000488147\tENSG00000227232\tWASH7P\tactive promoter 20pts\n",
      "chr1\t199880\t201080\tENST00000611868\tENSG00000275135\tFO538757.1\tactive promoter 20pts\n",
      "chr1\t200122\t201322\tENST00000624735\tENSG00000279457\tFO538757.2\tactive promoter 20pts\n",
      "chr1\t777770\t778970\tENST00000434264\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      "chr1\t777782\t778982\tENST00000457084\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      "chr1\t777930\t779130\tENST00000589899\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      "chr1\t777937\t779137\tENST00000609830\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n"
     ]
    }
   ],
   "source": [
    "!head /home/magda/Symfonia/experiments/2017/2017-12-19-Collect_regions_for_panel/out_files/active_promoters_20pts_1kbup_200bpdown_not_merged.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t28554\t29754\tENST00000473358\tENSG00000243485\tRP11-34P13.3\tactive promoter 20pts\n",
      " chr1\t29267\t30467\tENST00000469289\tENSG00000243485\tRP11-34P13.3\tactive promoter 20pts\n",
      " chr1\t29366\t30566\tENST00000607096\tENSG00000274890\tMIR1302-2\tactive promoter 20pts\n",
      " chr1\t29370\t30570\tENST00000488147\tENSG00000227232\tWASH7P\tactive promoter 20pts\n",
      " chr1\t199880\t201080\tENST00000611868\tENSG00000275135\tFO538757.1\tactive promoter 20pts\n",
      " chr1\t200122\t201322\tENST00000624735\tENSG00000279457\tFO538757.2\tactive promoter 20pts\n",
      " chr1\t777770\t778970\tENST00000434264\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      " chr1\t777782\t778982\tENST00000457084\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      " chr1\t777930\t779130\tENST00000589899\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      " chr1\t777937\t779137\tENST00000609830\tENSG00000237491\tRP11-206L10.9\tactive promoter 20pts\n",
      " "
     ]
    }
   ],
   "source": [
    "promoters_info = pbt.BedTool(\"/home/magda/Symfonia/experiments/2017/2017-12-19-Collect_regions_for_panel/out_files/active_promoters_20pts_1kbup_200bpdown_not_merged.bed\")\n",
    "promoters_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>AGRN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>CHD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>KAZN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>RAP1GAP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS     Gene\n",
       "0  chr1    939561   SAMD11\n",
       "1  chr1   1034902     AGRN\n",
       "2  chr1   6180416     CHD5\n",
       "4  chr1  14924017     KAZN\n",
       "5  chr1  21653564  RAP1GAP"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode[\"POS-1\"] = promoters_motifs_encode[\"POS\"] - 1\n",
    "promoters_motifs_encode_bedtool = pbt.BedTool.from_dataframe(promoters_motifs_encode[[\"CHROM\", \"POS-1\", \"POS\"]])\n",
    "promoters_motifs_encode_intersection = promoters_motifs_encode_bedtool.intersect(promoters_info, wa=True, wb=True)\n",
    "promoters_motifs_encode_intersection_df = promoters_motifs_encode_intersection.to_dataframe(names = [\"CHROM\", \"POS\", \"Gene\"], usecols = [0, 2, 8]).drop_duplicates()\n",
    "promoters_motifs_encode_intersection_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "533"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoters_motifs_encode_intersection_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "454"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoters_motifs_encode_intersection_df.groupby([\"CHROM\", \"POS\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_gene(row, intersection_df):\n",
    "    genes = intersection_df[(intersection_df[\"CHROM\"] == row[\"CHROM\"]) & (intersection_df[\"POS\"] == row[\"POS\"])][\"Gene\"].values\n",
    "    try:\n",
    "        return \", \".join(genes)\n",
    "    except:\n",
    "        print row[\"CHROM\"], row[\"POS\"]\n",
    "        print genes\n",
    "                                                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...</td>\n",
       "      <td>3.94;1.09</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>3</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.79;1.83</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>AGRN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...</td>\n",
       "      <td>10.00;6.71</td>\n",
       "      <td>ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CHD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002000</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "      <td>MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>KAZN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>2</td>\n",
       "      <td>ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...</td>\n",
       "      <td>5.97;2.29</td>\n",
       "      <td>ZNF263, POLR2A</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RAP1GAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25268534</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>13,11;0,7;15,7</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs148017678</td>\n",
       "      <td>0.004100</td>\n",
       "      <td>0.006900</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.37;1.50</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RSRP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26889274</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>115,126;109,108</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs781750092</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0</td>\n",
       "      <td>IKZF1_HUMAN.H11MO.0.C, STA5A_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>10.00;6.89</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>GPN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26901253</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6,7;3,12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs556268801</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.001900</td>\n",
       "      <td>1</td>\n",
       "      <td>NFIA_HUMAN.H11MO.1.D, PRRX1_HUMAN.H11MO.0.D, H...</td>\n",
       "      <td>4.79;1.91</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>GPATCH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27490564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>65,44;52,41</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs530994873</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>2</td>\n",
       "      <td>NKX25_HUMAN.H11MO.0.B, E2F8_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>3.37;1.51</td>\n",
       "      <td>USF2, TBL1XR1, REST, SRF, ATF2, MXI1, TBP, JUN...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>WASF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28737136</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>41,33;38,43</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs199887916</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.001400</td>\n",
       "      <td>0</td>\n",
       "      <td>MYB_HUMAN.H11MO.0.A, NR1H4_HUMAN.H11MO.0.B</td>\n",
       "      <td>3.58;1.29</td>\n",
       "      <td>SAP30, TBP, RBBP5, UBTF, BCLAF1, E2F6, PHF8, P...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>YTHDF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32178833</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>138,127;146,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs750438424</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, PO2F2_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>10.00;6.45</td>\n",
       "      <td>NR2F2, WRNIP1, MBD4</td>\n",
       "      <td>promoter</td>\n",
       "      <td>TXLNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32221069</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>113,132;115,128</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs187445970</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...</td>\n",
       "      <td>5.57;1.94</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>EIF3I, TMEM234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32221885</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>51,35;73,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs199528110</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>1</td>\n",
       "      <td>ZBT7A_HUMAN.H11MO.0.A, ZN341_HUMAN.H11MO.1.C, ...</td>\n",
       "      <td>3.79;1.86</td>\n",
       "      <td>ZNF263, FOSL2, FOS, POLR2A, HMGN3, JUND, CHD1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>EIF3I, TMEM234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33081220</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>40,35;39,49</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs559970186</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C, BCL6B_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.22;1.44</td>\n",
       "      <td>REST, MXI1, TBP, MAZ, CCNT2, IRF1, PHF8, POLR2...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>AZIN2, AK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35268566</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>43,44;37,37</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs764543206</td>\n",
       "      <td>0.001400</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>2</td>\n",
       "      <td>ZN341_HUMAN.H11MO.1.C, CEBPD_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>3.87;1.52</td>\n",
       "      <td>ZNF143, REST, ZNF263, TBP, BACH1, MAFK, BHLHE4...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ZMYM4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36324190</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>54,76;76,73</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs575257092</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>0</td>\n",
       "      <td>OVOL2_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>6.36;2.29</td>\n",
       "      <td>MXI1, TBP, RAD21, FOSL2, HDAC1, HNF4A, POLR2A,...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>EVA1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37753888</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>47,48;56,59</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs770976519</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...</td>\n",
       "      <td>4.69;1.95</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>EPHA10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>38005745</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47,7;22,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>2</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, S...</td>\n",
       "      <td>4.94;2.34</td>\n",
       "      <td>RBBP5, MAZ, POLR2A, SIN3A, ZBTB7A</td>\n",
       "      <td>promoter</td>\n",
       "      <td>FHL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>40318303</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7,13;1,10</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.048</td>\n",
       "      <td>rs867492341</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>1</td>\n",
       "      <td>FOXF2_HUMAN.H11MO.0.D, IRX3_HUMAN.H11MO.0.D, P...</td>\n",
       "      <td>4.34;1.72</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>COL9A2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>40980295</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6,6;8,8</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs943593550</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>KAISO_HUMAN.H11MO.0.A, ZN322_HUMAN.H11MO.0.B</td>\n",
       "      <td>3.18;1.31</td>\n",
       "      <td>SAP30, RBBP5, HDAC1, CCNT2, E2F6, PHF8, POLR2A...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CTPS1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS REF ALT             VAFs   AC     AF     avsnp150  \\\n",
       "0   chr1    939561   G   A      72,69;91,70  2.0  0.043  rs888928964   \n",
       "1   chr1   1034902   A   T      7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2   chr1   6180416   G   A        10,11;8,6  2.0  0.043  rs151212360   \n",
       "3   chr1  14924017   C   G        6,5;12,14  2.0  0.043  rs531269337   \n",
       "4   chr1  21653564   C   T      13,15;30,24  2.0  0.043  rs890141715   \n",
       "5   chr1  25268534   A   G   13,11;0,7;15,7  4.0  0.105  rs148017678   \n",
       "6   chr1  26889274   C   T  115,126;109,108  2.0  0.043  rs781750092   \n",
       "7   chr1  26901253   C   T         6,7;3,12  2.0  0.043  rs556268801   \n",
       "8   chr1  27490564   C   T      65,44;52,41  2.0  0.043  rs530994873   \n",
       "9   chr1  28737136   C   G      41,33;38,43  2.0  0.043  rs199887916   \n",
       "10  chr1  32178833   T   C   138,127;146,95  2.0  0.043  rs750438424   \n",
       "11  chr1  32221069   C   T  113,132;115,128  2.0  0.043  rs187445970   \n",
       "12  chr1  32221885   G   C      51,35;73,70  2.0  0.043  rs199528110   \n",
       "13  chr1  33081220   A   C      40,35;39,49  2.0  0.043  rs559970186   \n",
       "14  chr1  35268566   G   A      43,44;37,37  2.0  0.043  rs764543206   \n",
       "15  chr1  36324190   G   C      54,76;76,73  2.0  0.043  rs575257092   \n",
       "16  chr1  37753888   G   A      47,48;56,59  2.0  0.043  rs770976519   \n",
       "17  chr1  38005745   T   C        47,7;22,6  2.0  0.043            .   \n",
       "18  chr1  40318303   T   C        7,13;1,10  2.0  0.048  rs867492341   \n",
       "19  chr1  40980295   G   C          6,6;8,8  2.0  0.043  rs943593550   \n",
       "\n",
       "    gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0            0.000052           0.000000                     0   \n",
       "1            0.001300           0.001700                     3   \n",
       "2            0.000300           0.000500                     2   \n",
       "3            0.001300           0.002000                     0   \n",
       "4            0.000600           0.001300                     2   \n",
       "5            0.004100           0.006900                     3   \n",
       "6            0.000200           0.000300                     0   \n",
       "7            0.001100           0.001900                     1   \n",
       "8            0.000200           0.000400                     2   \n",
       "9            0.000800           0.001400                     0   \n",
       "10           0.000200           0.000300                     0   \n",
       "11           0.000900           0.001200                     0   \n",
       "12           0.000400           0.000600                     1   \n",
       "13           0.000200           0.000300                     0   \n",
       "14           0.001400           0.002400                     2   \n",
       "15           0.001300           0.002500                     0   \n",
       "16           0.000065           0.000067                     0   \n",
       "17           0.000700           0.000600                     2   \n",
       "18           0.000700           0.000900                     1   \n",
       "19           0.000033           0.000067                     0   \n",
       "\n",
       "                                     PWMEnrich motifs  \\\n",
       "0   HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...   \n",
       "1   UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "2   UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...   \n",
       "3   MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "4   ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...   \n",
       "5   FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...   \n",
       "6   IKZF1_HUMAN.H11MO.0.C, STA5A_HUMAN.H11MO.0.A, ...   \n",
       "7   NFIA_HUMAN.H11MO.1.D, PRRX1_HUMAN.H11MO.0.D, H...   \n",
       "8   NKX25_HUMAN.H11MO.0.B, E2F8_HUMAN.H11MO.0.D, R...   \n",
       "9          MYB_HUMAN.H11MO.0.A, NR1H4_HUMAN.H11MO.0.B   \n",
       "10  ZN335_HUMAN.H11MO.1.A, PO2F2_HUMAN.H11MO.0.A, ...   \n",
       "11  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...   \n",
       "12  ZBT7A_HUMAN.H11MO.0.A, ZN341_HUMAN.H11MO.1.C, ...   \n",
       "13  ZBT14_HUMAN.H11MO.0.C, BCL6B_HUMAN.H11MO.0.D, ...   \n",
       "14  ZN341_HUMAN.H11MO.1.C, CEBPD_HUMAN.H11MO.0.C, ...   \n",
       "15  OVOL2_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "16  NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...   \n",
       "17  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, S...   \n",
       "18  FOXF2_HUMAN.H11MO.0.D, IRX3_HUMAN.H11MO.0.D, P...   \n",
       "19       KAISO_HUMAN.H11MO.0.A, ZN322_HUMAN.H11MO.0.B   \n",
       "\n",
       "   PWMEnrich best -log10pval and diff  \\\n",
       "0                           3.94;1.09   \n",
       "1                           4.79;1.83   \n",
       "2                          10.00;6.71   \n",
       "3                           5.20;2.79   \n",
       "4                           5.97;2.29   \n",
       "5                           4.37;1.50   \n",
       "6                          10.00;6.89   \n",
       "7                           4.79;1.91   \n",
       "8                           3.37;1.51   \n",
       "9                           3.58;1.29   \n",
       "10                         10.00;6.45   \n",
       "11                          5.57;1.94   \n",
       "12                          3.79;1.86   \n",
       "13                          4.22;1.44   \n",
       "14                          3.87;1.52   \n",
       "15                          6.36;2.29   \n",
       "16                          4.69;1.95   \n",
       "17                          4.94;2.34   \n",
       "18                          4.34;1.72   \n",
       "19                          3.18;1.31   \n",
       "\n",
       "                                               ENCODE genomic element  \\\n",
       "0                                                EZH2        promoter   \n",
       "1                                                            promoter   \n",
       "2             ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1        promoter   \n",
       "3   MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...        promoter   \n",
       "4                                      ZNF263, POLR2A        promoter   \n",
       "5                                                            promoter   \n",
       "6                                                            promoter   \n",
       "7                                                            promoter   \n",
       "8   USF2, TBL1XR1, REST, SRF, ATF2, MXI1, TBP, JUN...        promoter   \n",
       "9   SAP30, TBP, RBBP5, UBTF, BCLAF1, E2F6, PHF8, P...        promoter   \n",
       "10                                NR2F2, WRNIP1, MBD4        promoter   \n",
       "11                                                           promoter   \n",
       "12      ZNF263, FOSL2, FOS, POLR2A, HMGN3, JUND, CHD1        promoter   \n",
       "13  REST, MXI1, TBP, MAZ, CCNT2, IRF1, PHF8, POLR2...        promoter   \n",
       "14  ZNF143, REST, ZNF263, TBP, BACH1, MAFK, BHLHE4...        promoter   \n",
       "15  MXI1, TBP, RAD21, FOSL2, HDAC1, HNF4A, POLR2A,...        promoter   \n",
       "16                                               EZH2        promoter   \n",
       "17                  RBBP5, MAZ, POLR2A, SIN3A, ZBTB7A        promoter   \n",
       "18                                                           promoter   \n",
       "19  SAP30, RBBP5, HDAC1, CCNT2, E2F6, PHF8, POLR2A...        promoter   \n",
       "\n",
       "              Gene  \n",
       "0           SAMD11  \n",
       "1             AGRN  \n",
       "2             CHD5  \n",
       "3             KAZN  \n",
       "4          RAP1GAP  \n",
       "5            RSRP1  \n",
       "6             GPN2  \n",
       "7          GPATCH3  \n",
       "8            WASF2  \n",
       "9           YTHDF2  \n",
       "10           TXLNA  \n",
       "11  EIF3I, TMEM234  \n",
       "12  EIF3I, TMEM234  \n",
       "13      AZIN2, AK2  \n",
       "14           ZMYM4  \n",
       "15           EVA1B  \n",
       "16          EPHA10  \n",
       "17            FHL3  \n",
       "18          COL9A2  \n",
       "19           CTPS1  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode[\"Gene\"] = promoters_motifs_encode.apply(add_gene, args = (promoters_motifs_encode_intersection_df,), axis=1)\n",
    "promoters_motifs_encode = promoters_motifs_encode.drop(labels = [\"POS-1\"], axis=1)\n",
    "promoters_motifs_encode[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>chr8</td>\n",
       "      <td>142449370</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0,9;0,3;0,7;0,6;0,9;0,8;0,7;0,3</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.174</td>\n",
       "      <td>rs62511436</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>6</td>\n",
       "      <td>KLF8_HUMAN.H11MO.0.C, ZIC1_HUMAN.H11MO.0.B, GL...</td>\n",
       "      <td>6.28;2.56</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ADGRB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>chr20</td>\n",
       "      <td>36461763</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6,6;2,1;11,4;8,2;8,1;2,2;8,7;19,2</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.250</td>\n",
       "      <td>rs955371902</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>5</td>\n",
       "      <td>ZBT7A_HUMAN.H11MO.0.A, GLIS3_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.95;2.03</td>\n",
       "      <td>EGR1, ZNF263, PHF8, POLR2A, GABPA, E2F1, TAF1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>DLGAP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>chr7</td>\n",
       "      <td>128531249</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6,1;9,2;15,5;9,1;12,1;8,2;8,4</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.152</td>\n",
       "      <td>rs1035985757</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>4</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...</td>\n",
       "      <td>5.48;2.64</td>\n",
       "      <td>ZNF143, REST, EGR1, MXI1, TBP, ZBTB33, FOS, MA...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP11-212P7.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>207495374</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>7,3;6,3;1,2;8,1;0,1;6,3</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.184</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0026</td>\n",
       "      <td>0.0038</td>\n",
       "      <td>4</td>\n",
       "      <td>ZIC1_HUMAN.H11MO.0.B, SALL4_HUMAN.H11MO.0.B, E...</td>\n",
       "      <td>6.36;1.97</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>chr19</td>\n",
       "      <td>39445098</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>33,8;29,11;65,14;31,4;65,14;50,12</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.130</td>\n",
       "      <td>rs201114119</td>\n",
       "      <td>0.0054</td>\n",
       "      <td>0.0048</td>\n",
       "      <td>3</td>\n",
       "      <td>SOX13_HUMAN.H11MO.0.D, TBP_HUMAN.H11MO.0.A, AN...</td>\n",
       "      <td>4.89;1.68</td>\n",
       "      <td>PHF8, POLR2A, GATA2, CHD1, GATA1, SMARCB1, TCF7L2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SUPT5H</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                               VAFs   AC     AF  \\\n",
       "159   chr8  142449370   A   C    0,9;0,3;0,7;0,6;0,9;0,8;0,7;0,3  8.0  0.174   \n",
       "385  chr20   36461763   G   C  6,6;2,1;11,4;8,2;8,1;2,2;8,7;19,2  8.0  0.250   \n",
       "147   chr7  128531249   T   C      6,1;9,2;15,5;9,1;12,1;8,2;8,4  7.0  0.152   \n",
       "39    chr1  207495374   A   C            7,3;6,3;1,2;8,1;0,1;6,3  7.0  0.184   \n",
       "369  chr19   39445098   T   A  33,8;29,11;65,14;31,4;65,14;50,12  6.0  0.130   \n",
       "\n",
       "         avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "159    rs62511436             0.0011             0.0010                     6   \n",
       "385   rs955371902             0.0014             0.0017                     5   \n",
       "147  rs1035985757             0.0011             0.0004                     4   \n",
       "39              .             0.0026             0.0038                     4   \n",
       "369   rs201114119             0.0054             0.0048                     3   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "159  KLF8_HUMAN.H11MO.0.C, ZIC1_HUMAN.H11MO.0.B, GL...   \n",
       "385  ZBT7A_HUMAN.H11MO.0.A, GLIS3_HUMAN.H11MO.0.D, ...   \n",
       "147  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...   \n",
       "39   ZIC1_HUMAN.H11MO.0.B, SALL4_HUMAN.H11MO.0.B, E...   \n",
       "369  SOX13_HUMAN.H11MO.0.D, TBP_HUMAN.H11MO.0.A, AN...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "159                          6.28;2.56   \n",
       "385                          4.95;2.03   \n",
       "147                          5.48;2.64   \n",
       "39                           6.36;1.97   \n",
       "369                          4.89;1.68   \n",
       "\n",
       "                                                ENCODE genomic element  \\\n",
       "159                                               EZH2        promoter   \n",
       "385      EGR1, ZNF263, PHF8, POLR2A, GABPA, E2F1, TAF1        promoter   \n",
       "147  ZNF143, REST, EGR1, MXI1, TBP, ZBTB33, FOS, MA...        promoter   \n",
       "39                                                CTCF        promoter   \n",
       "369  PHF8, POLR2A, GATA2, CHD1, GATA1, SMARCB1, TCF7L2        promoter   \n",
       "\n",
       "             Gene  \n",
       "159        ADGRB1  \n",
       "385        DLGAP4  \n",
       "147  RP11-212P7.2  \n",
       "39            CR1  \n",
       "369        SUPT5H  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode[~(promoters_motifs_encode[\"PWMEnrich motifs\"]==\"\")].sort_values(by = \"AC\", ascending = False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intronic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir in_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t19482\t22400\tENSG00000227232\n",
      "chr1\t183942\t184568\tENSG00000279928\n",
      "chr1\t189803\t193125\tENSG00000279457\n",
      "chr1\t1024210\t1025994\tENSG00000188157\n",
      "chr1\t1031391\t1031884\tENSG00000188157\n",
      "chr1\t1032778\t1033104\tENSG00000188157\n",
      "chr1\t1134548\t1135587\t.\n",
      "chr1\t1156466\t1159847\t.\n",
      "chr1\t1290755\t1292028\tENSG00000162572\n",
      "chr1\t1427223\t1428530\tENSG00000205116\n"
     ]
    }
   ],
   "source": [
    "#copied from /mnt/chr3/People/Magda/Symfonia/brain_regulatory_regions/active_regions_outside_promoters/version_for_02.2018/H3K27ac_peaks_outside_promoters_with_gene_names.bed\n",
    "!head in_files/H3K27ac_peaks_outside_promoters_with_gene_names.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183942</td>\n",
       "      <td>184568</td>\n",
       "      <td>ENSG00000279928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>ENSG00000279457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1031391</td>\n",
       "      <td>1031884</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1032778</td>\n",
       "      <td>1033104</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1290755</td>\n",
       "      <td>1292028</td>\n",
       "      <td>ENSG00000162572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1427223</td>\n",
       "      <td>1428530</td>\n",
       "      <td>ENSG00000205116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1550199</td>\n",
       "      <td>1551447</td>\n",
       "      <td>ENSG00000160075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1567417</td>\n",
       "      <td>1568787</td>\n",
       "      <td>ENSG00000160075</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr    start      end               ID\n",
       "0   chr1    19482    22400  ENSG00000227232\n",
       "1   chr1   183942   184568  ENSG00000279928\n",
       "2   chr1   189803   193125  ENSG00000279457\n",
       "3   chr1  1024210  1025994  ENSG00000188157\n",
       "4   chr1  1031391  1031884  ENSG00000188157\n",
       "5   chr1  1032778  1033104  ENSG00000188157\n",
       "8   chr1  1290755  1292028  ENSG00000162572\n",
       "9   chr1  1427223  1428530  ENSG00000205116\n",
       "11  chr1  1550199  1551447  ENSG00000160075\n",
       "12  chr1  1567417  1568787  ENSG00000160075"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_genes = pd.read_csv('in_files/H3K27ac_peaks_outside_promoters_with_gene_names.bed', \n",
    "                        sep='\\t', names = ['chr', 'start', 'end', 'ID'])\n",
    "enh_in_genes = enh_genes[enh_genes['ID'] != '.']\n",
    "enh_in_genes[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>chr1</td>\n",
       "      <td>51582186</td>\n",
       "      <td>51585240</td>\n",
       "      <td>ENSG00000117859,ENSG00000227742</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr     start       end                               ID\n",
       "1176  chr1  51582186  51585240  ENSG00000117859,ENSG00000227742"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_in_genes[enh_in_genes[\"start\"] == 51582186]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482.0</td>\n",
       "      <td>22400.0</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183942.0</td>\n",
       "      <td>184568.0</td>\n",
       "      <td>ENSG00000279928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803.0</td>\n",
       "      <td>193125.0</td>\n",
       "      <td>ENSG00000279457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210.0</td>\n",
       "      <td>1025994.0</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1031391.0</td>\n",
       "      <td>1031884.0</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1032778.0</td>\n",
       "      <td>1033104.0</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1290755.0</td>\n",
       "      <td>1292028.0</td>\n",
       "      <td>ENSG00000162572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1427223.0</td>\n",
       "      <td>1428530.0</td>\n",
       "      <td>ENSG00000205116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1550199.0</td>\n",
       "      <td>1551447.0</td>\n",
       "      <td>ENSG00000160075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1567417.0</td>\n",
       "      <td>1568787.0</td>\n",
       "      <td>ENSG00000160075</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      start        end               ID\n",
       "0  chr1    19482.0    22400.0  ENSG00000227232\n",
       "1  chr1   183942.0   184568.0  ENSG00000279928\n",
       "2  chr1   189803.0   193125.0  ENSG00000279457\n",
       "3  chr1  1024210.0  1025994.0  ENSG00000188157\n",
       "4  chr1  1031391.0  1031884.0  ENSG00000188157\n",
       "5  chr1  1032778.0  1033104.0  ENSG00000188157\n",
       "6  chr1  1290755.0  1292028.0  ENSG00000162572\n",
       "7  chr1  1427223.0  1428530.0  ENSG00000205116\n",
       "8  chr1  1550199.0  1551447.0  ENSG00000160075\n",
       "9  chr1  1567417.0  1568787.0  ENSG00000160075"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reformat to have one gene ID in cell\n",
    "enh_in_gene = pd.DataFrame() \n",
    "for i, row in enh_in_genes.iterrows(): \n",
    "    genes = row['ID'].split(',') \n",
    "    if len(genes) == 1: \n",
    "        enh_in_gene = enh_in_gene.append(row, ignore_index=True) \n",
    "    else: \n",
    "        for gene in genes: \n",
    "            new_row = row \n",
    "            new_row['ID'] = gene \n",
    "            enh_in_gene = enh_in_gene.append(new_row, ignore_index=True) \n",
    "            enh_in_gene = enh_in_gene.reindex_axis(enh_in_genes.columns, axis=1) \n",
    "enh_in_gene[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183942</td>\n",
       "      <td>184568</td>\n",
       "      <td>ENSG00000279928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>ENSG00000279457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1031391</td>\n",
       "      <td>1031884</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end               ID\n",
       "0  chr1    19482    22400  ENSG00000227232\n",
       "1  chr1   183942   184568  ENSG00000279928\n",
       "2  chr1   189803   193125  ENSG00000279457\n",
       "3  chr1  1024210  1025994  ENSG00000188157\n",
       "4  chr1  1031391  1031884  ENSG00000188157"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_in_gene[\"start\"] = enh_in_gene[\"start\"].astype(int)\n",
    "enh_in_gene[\"end\"] = enh_in_gene[\"end\"].astype(int)\n",
    "enh_in_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>905</th>\n",
       "      <td>chr1</td>\n",
       "      <td>51582186</td>\n",
       "      <td>51585240</td>\n",
       "      <td>ENSG00000117859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>906</th>\n",
       "      <td>chr1</td>\n",
       "      <td>51582186</td>\n",
       "      <td>51585240</td>\n",
       "      <td>ENSG00000227742</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr     start       end               ID\n",
       "905  chr1  51582186  51585240  ENSG00000117859\n",
       "906  chr1  51582186  51585240  ENSG00000227742"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_in_gene[enh_in_gene[\"start\"] == 51582186]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\thavana\tgene\t11869\t14409\t.\t+\t.\tgene_id \"ENSG00000223972\"; gene_version \"5\"; gene_name \"DDX11L1\"; gene_source \"havana\"; gene_biotype \"transcribed_unprocessed_pseudogene\"; havana_gene \"OTTHUMG00000000961\"; havana_gene_version \"2\";\n",
      "chr1\thavana\tgene\t14404\t29570\t.\t-\t.\tgene_id \"ENSG00000227232\"; gene_version \"5\"; gene_name \"WASH7P\"; gene_source \"havana\"; gene_biotype \"unprocessed_pseudogene\"; havana_gene \"OTTHUMG00000000958\"; havana_gene_version \"1\";\n",
      "chr1\tensembl\tgene\t17369\t17436\t.\t-\t.\tgene_id \"ENSG00000278267\"; gene_version \"1\"; gene_name \"MIR6859-1\"; gene_source \"ensembl\"; gene_biotype \"miRNA\";\n",
      "chr1\thavana\tgene\t29554\t31109\t.\t+\t.\tgene_id \"ENSG00000243485\"; gene_version \"3\"; gene_name \"RP11-34P13.3\"; gene_source \"havana\"; gene_biotype \"lincRNA\"; havana_gene \"OTTHUMG00000000959\"; havana_gene_version \"2\";\n",
      "chr1\tensembl\tgene\t30366\t30503\t.\t+\t.\tgene_id \"ENSG00000274890\"; gene_version \"1\"; gene_name \"MIR1302-2\"; gene_source \"ensembl\"; gene_biotype \"miRNA\";\n",
      "chr1\thavana\tgene\t34554\t36081\t.\t-\t.\tgene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"havana\"; gene_biotype \"lincRNA\"; havana_gene \"OTTHUMG00000000960\"; havana_gene_version \"1\";\n",
      "chr1\thavana\tgene\t52473\t53312\t.\t+\t.\tgene_id \"ENSG00000268020\"; gene_version \"3\"; gene_name \"OR4G4P\"; gene_source \"havana\"; gene_biotype \"unprocessed_pseudogene\"; havana_gene \"OTTHUMG00000185779\"; havana_gene_version \"1\";\n",
      "chr1\thavana\tgene\t62948\t63887\t.\t+\t.\tgene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"unprocessed_pseudogene\"; havana_gene \"OTTHUMG00000001095\"; havana_gene_version \"2\";\n",
      "chr1\tensembl_havana\tgene\t69091\t70008\t.\t+\t.\tgene_id \"ENSG00000186092\"; gene_version \"4\"; gene_name \"OR4F5\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; havana_gene \"OTTHUMG00000001094\"; havana_gene_version \"2\";\n",
      "chr1\tensembl_havana\tgene\t89295\t133723\t.\t-\t.\tgene_id \"ENSG00000238009\"; gene_version \"6\"; gene_name \"RP11-34P13.7\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\"; havana_gene \"OTTHUMG00000001096\"; havana_gene_version \"2\";\n"
     ]
    }
   ],
   "source": [
    "!head in_files/hg38_full.genes.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000237613\"; gene_version \"2\"; g...</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>ENSG00000237613:FAM138A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52473</td>\n",
       "      <td>53312</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000268020\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "      <td>ENSG00000268020:OR4G4P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>62948</td>\n",
       "      <td>63887</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000240361\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000240361</td>\n",
       "      <td>ENSG00000240361:OR4G11P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69091</td>\n",
       "      <td>70008</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000186092\"; gene_version \"4\"; g...</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENSG00000186092:OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>89295</td>\n",
       "      <td>133723</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000238009\"; gene_version \"6\"; g...</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009:RP11-34P13.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>89551</td>\n",
       "      <td>91105</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000239945\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000239945</td>\n",
       "      <td>ENSG00000239945:RP11-34P13.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>131025</td>\n",
       "      <td>134836</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000233750\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000233750</td>\n",
       "      <td>ENSG00000233750:CICP27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135141</td>\n",
       "      <td>135895</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000268903\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000268903</td>\n",
       "      <td>ENSG00000268903:RP11-34P13.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>137682</td>\n",
       "      <td>137965</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000269981\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000269981</td>\n",
       "      <td>ENSG00000269981:RP11-34P13.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>139790</td>\n",
       "      <td>140339</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000239906\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>ENSG00000239906:RP11-34P13.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>141474</td>\n",
       "      <td>173862</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000241860\"; gene_version \"6\"; g...</td>\n",
       "      <td>ENSG00000241860</td>\n",
       "      <td>ENSG00000241860:RP11-34P13.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>157784</td>\n",
       "      <td>157887</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000222623\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000222623</td>\n",
       "      <td>ENSG00000222623:RNU6-1100P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>160446</td>\n",
       "      <td>161525</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000241599\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000241599</td>\n",
       "      <td>ENSG00000241599:RP11-34P13.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182393</td>\n",
       "      <td>184158</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000279928\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000279928</td>\n",
       "      <td>ENSG00000279928:FO538757.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>184923</td>\n",
       "      <td>200322</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000279457\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000279457</td>\n",
       "      <td>ENSG00000279457:FO538757.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr   start     end strand  \\\n",
       "0   chr1   11869   14409      +   \n",
       "1   chr1   14404   29570      -   \n",
       "2   chr1   17369   17436      -   \n",
       "3   chr1   29554   31109      +   \n",
       "4   chr1   30366   30503      +   \n",
       "5   chr1   34554   36081      -   \n",
       "6   chr1   52473   53312      +   \n",
       "7   chr1   62948   63887      +   \n",
       "8   chr1   69091   70008      +   \n",
       "9   chr1   89295  133723      -   \n",
       "10  chr1   89551   91105      -   \n",
       "11  chr1  131025  134836      +   \n",
       "12  chr1  135141  135895      -   \n",
       "13  chr1  137682  137965      -   \n",
       "14  chr1  139790  140339      -   \n",
       "15  chr1  141474  173862      -   \n",
       "16  chr1  157784  157887      -   \n",
       "17  chr1  160446  161525      +   \n",
       "18  chr1  182393  184158      +   \n",
       "19  chr1  184923  200322      -   \n",
       "\n",
       "                                                 info               ID  \\\n",
       "0   gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1   gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2   gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3   gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4   gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "5   gene_id \"ENSG00000237613\"; gene_version \"2\"; g...  ENSG00000237613   \n",
       "6   gene_id \"ENSG00000268020\"; gene_version \"3\"; g...  ENSG00000268020   \n",
       "7   gene_id \"ENSG00000240361\"; gene_version \"1\"; g...  ENSG00000240361   \n",
       "8   gene_id \"ENSG00000186092\"; gene_version \"4\"; g...  ENSG00000186092   \n",
       "9   gene_id \"ENSG00000238009\"; gene_version \"6\"; g...  ENSG00000238009   \n",
       "10  gene_id \"ENSG00000239945\"; gene_version \"1\"; g...  ENSG00000239945   \n",
       "11  gene_id \"ENSG00000233750\"; gene_version \"3\"; g...  ENSG00000233750   \n",
       "12  gene_id \"ENSG00000268903\"; gene_version \"1\"; g...  ENSG00000268903   \n",
       "13  gene_id \"ENSG00000269981\"; gene_version \"1\"; g...  ENSG00000269981   \n",
       "14  gene_id \"ENSG00000239906\"; gene_version \"1\"; g...  ENSG00000239906   \n",
       "15  gene_id \"ENSG00000241860\"; gene_version \"6\"; g...  ENSG00000241860   \n",
       "16  gene_id \"ENSG00000222623\"; gene_version \"1\"; g...  ENSG00000222623   \n",
       "17  gene_id \"ENSG00000241599\"; gene_version \"1\"; g...  ENSG00000241599   \n",
       "18  gene_id \"ENSG00000279928\"; gene_version \"1\"; g...  ENSG00000279928   \n",
       "19  gene_id \"ENSG00000279457\"; gene_version \"3\"; g...  ENSG00000279457   \n",
       "\n",
       "                             Gene  \n",
       "0         ENSG00000223972:DDX11L1  \n",
       "1          ENSG00000227232:WASH7P  \n",
       "2       ENSG00000278267:MIR6859-1  \n",
       "3    ENSG00000243485:RP11-34P13.3  \n",
       "4       ENSG00000274890:MIR1302-2  \n",
       "5         ENSG00000237613:FAM138A  \n",
       "6          ENSG00000268020:OR4G4P  \n",
       "7         ENSG00000240361:OR4G11P  \n",
       "8           ENSG00000186092:OR4F5  \n",
       "9    ENSG00000238009:RP11-34P13.7  \n",
       "10   ENSG00000239945:RP11-34P13.8  \n",
       "11         ENSG00000233750:CICP27  \n",
       "12  ENSG00000268903:RP11-34P13.15  \n",
       "13  ENSG00000269981:RP11-34P13.16  \n",
       "14  ENSG00000239906:RP11-34P13.14  \n",
       "15  ENSG00000241860:RP11-34P13.13  \n",
       "16     ENSG00000222623:RNU6-1100P  \n",
       "17   ENSG00000241599:RP11-34P13.9  \n",
       "18     ENSG00000279928:FO538757.3  \n",
       "19     ENSG00000279457:FO538757.2  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#from /mnt/chr3/People/Magda/Symfonia/brain_regulatory_regions/hg38_full.genes.gtf\n",
    "genes_info = pd.read_csv('in_files/hg38_full.genes.gtf', sep='\\t', usecols = [0, 3, 4, 6, 8], \n",
    "                            names = [\"chr\", \"start\", \"end\", \"strand\", 'info'])\n",
    "genes_info['ID'] = genes_info['info'].str.split('\"').str[1]\n",
    "genes_info['Gene'] = genes_info['ID'] + \":\" + genes_info['info'].str.split('\"').str[5]\n",
    "\n",
    "genes_info[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, start, end, strand, info, ID, Gene]\n",
       "Index: []"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info[genes_info[\"Gene\"].str.count(\":\")>1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183942</td>\n",
       "      <td>184568</td>\n",
       "      <td>ENSG00000279928</td>\n",
       "      <td>ENSG00000279928:FO538757.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>ENSG00000279457</td>\n",
       "      <td>ENSG00000279457:FO538757.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENSG00000188157:AGRN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1031391</td>\n",
       "      <td>1031884</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENSG00000188157:AGRN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end               ID                        Gene\n",
       "0  chr1    19482    22400  ENSG00000227232      ENSG00000227232:WASH7P\n",
       "1  chr1   183942   184568  ENSG00000279928  ENSG00000279928:FO538757.3\n",
       "2  chr1   189803   193125  ENSG00000279457  ENSG00000279457:FO538757.2\n",
       "3  chr1  1024210  1025994  ENSG00000188157        ENSG00000188157:AGRN\n",
       "4  chr1  1031391  1031884  ENSG00000188157        ENSG00000188157:AGRN"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_in_genes_names = pd.merge(enh_in_gene, genes_info[[\"ID\", \"Gene\"]], on = \"ID\", how = \"left\")\n",
    "enh_in_genes_names.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32713 32713\n"
     ]
    }
   ],
   "source": [
    "print len(enh_in_gene), len(enh_in_genes_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t19482\t22400\tENSG00000227232\tENSG00000227232:WASH7P\n",
      " chr1\t183942\t184568\tENSG00000279928\tENSG00000279928:FO538757.3\n",
      " chr1\t189803\t193125\tENSG00000279457\tENSG00000279457:FO538757.2\n",
      " chr1\t1024210\t1025994\tENSG00000188157\tENSG00000188157:AGRN\n",
      " chr1\t1031391\t1031884\tENSG00000188157\tENSG00000188157:AGRN\n",
      " chr1\t1032778\t1033104\tENSG00000188157\tENSG00000188157:AGRN\n",
      " chr1\t1290755\t1292028\tENSG00000162572\tENSG00000162572:SCNN1D\n",
      " chr1\t1427223\t1428530\tENSG00000205116\tENSG00000205116:TMEM88B\n",
      " chr1\t1550199\t1551447\tENSG00000160075\tENSG00000160075:SSU72\n",
      " chr1\t1567417\t1568787\tENSG00000160075\tENSG00000160075:SSU72\n",
      " "
     ]
    }
   ],
   "source": [
    "enhancers_motifs_encode[\"POS-1\"] = enhancers_motifs_encode[\"POS\"] - 1\n",
    "enhancer_snps_bedtool = pbt.BedTool.from_dataframe(enhancers_motifs_encode[[\"CHROM\", \"POS-1\", \"POS\"]])\n",
    "\n",
    "enh_in_genes_bedtool = pbt.BedTool.from_dataframe(enh_in_genes_names)\n",
    "enh_in_genes_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2227532\t2227533\t.\t-1\t-1\t.\t.\n",
      " chr1\t5896825\t5896826\tchr1\t5895777\t5897516\tENSG00000131697\tENSG00000131697:NPHP4\n",
      " chr1\t7041937\t7041938\tchr1\t7040690\t7043676\tENSG00000171735\tENSG00000171735:CAMTA1\n",
      " chr1\t16912415\t16912416\tchr1\t16912238\t16915295\tENSG00000058453\tENSG00000058453:CROCC\n",
      " chr1\t16913967\t16913968\tchr1\t16912238\t16915295\tENSG00000058453\tENSG00000058453:CROCC\n",
      " chr1\t16913969\t16913970\tchr1\t16912238\t16915295\tENSG00000058453\tENSG00000058453:CROCC\n",
      " chr1\t21254093\t21254094\tchr1\t21253526\t21255567\tENSG00000117298\tENSG00000117298:ECE1\n",
      " chr1\t21264260\t21264261\tchr1\t21260958\t21264403\tENSG00000117298\tENSG00000117298:ECE1\n",
      " chr1\t21625235\t21625236\tchr1\t21623828\t21625754\tENSG00000076864\tENSG00000076864:RAP1GAP\n",
      " chr1\t24705403\t24705404\t.\t-1\t-1\t.\t.\n",
      " "
     ]
    }
   ],
   "source": [
    "enhancers_motifs_encode_intersection = enhancer_snps_bedtool.intersect(enh_in_genes_bedtool, wa=True, wb=True, loj=True)\n",
    "enhancers_motifs_encode_intersection.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>5895777</td>\n",
       "      <td>5897516</td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>7040690</td>\n",
       "      <td>7043676</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS  enh_start   enh_end                    Gene\n",
       "0  chr1   2227533         -1        -1                       .\n",
       "1  chr1   5896826    5895777   5897516   ENSG00000131697:NPHP4\n",
       "2  chr1   7041938    7040690   7043676  ENSG00000171735:CAMTA1\n",
       "3  chr1  16912416   16912238  16915295   ENSG00000058453:CROCC\n",
       "4  chr1  16913968   16912238  16915295   ENSG00000058453:CROCC"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#enhancers_motifs_encode_intersection_df = enhancers_motifs_encode_intersection.to_dataframe(names = [\"CHROM\", \"POS\", \"Gene\"], usecols = [0, 2, 7]).drop_duplicates()\n",
    "enhancers_motifs_encode_intersection_df = enhancers_motifs_encode_intersection.to_dataframe(names = [\"CHROM\", \"POS\", \"enh_start\", \"enh_end\", \"Gene\"], usecols = [0, 2, 4,5, 7]).drop_duplicates()\n",
    "enhancers_motifs_encode_intersection_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913970</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>18,8;24,14;15,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs537954556</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>HMX3_HUMAN.H11MO.0.D, CTCFL_HUMAN.H11MO.0.A, C...</td>\n",
       "      <td>4.54;1.46</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1,2;17,3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.048</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>2</td>\n",
       "      <td>PPARA_HUMAN.H11MO.1.B, SOX17_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.47;1.66</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000117298:ECE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21264261</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>82,117;62,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs212534</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>1</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, NR1I3_HUMAN.H11MO.0.C, N...</td>\n",
       "      <td>3.37;1.07</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000117298:ECE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21625236</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>127,104;109,137</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs989937565</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0</td>\n",
       "      <td>NR2C1_HUMAN.H11MO.0.C, PPARD_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.46;1.19</td>\n",
       "      <td>ZNF143, REST, USF1, RCOR1, EP300</td>\n",
       "      <td>ENSG00000076864:RAP1GAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24705404</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>81,58;84,65</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs759876216</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>PBX3_HUMAN.H11MO.1.A, SOX9_HUMAN.H11MO.1.B, HX...</td>\n",
       "      <td>4.97;1.85</td>\n",
       "      <td>EGR1, TBP, SP2, NFYA, BACH1, FOS, TEAD4, SPI1,...</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27971997</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>44,62;100,80</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs1024768231</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>2</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, ZKSC3_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.67;2.24</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000158161:EYA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27980435</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>36,51;67,58</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs778827085</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>2</td>\n",
       "      <td>GATA4_HUMAN.H11MO.0.A, ZKSC3_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.71;1.37</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000158161:EYA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31811750</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>78,81;17,15</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs553603564</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>1</td>\n",
       "      <td>PAX3_HUMAN.H11MO.0.D, PAX8_HUMAN.H11MO.0.D, IR...</td>\n",
       "      <td>4.25;1.17</td>\n",
       "      <td>RAD21, SIN3A, MAX</td>\n",
       "      <td>ENSG00000134668:SPOCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33075149</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>112,96;104,84</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs554147217</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0</td>\n",
       "      <td>HIF1A_HUMAN.H11MO.0.C, MAFG_HUMAN.H11MO.1.A, A...</td>\n",
       "      <td>3.52;2.00</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000004455:AK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37234892</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>108,82;98,86;74,75;52,47</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs72655842</td>\n",
       "      <td>0.005700</td>\n",
       "      <td>0.009800</td>\n",
       "      <td>1</td>\n",
       "      <td>RORA_HUMAN.H11MO.0.C, NR1D1_HUMAN.H11MO.0.B</td>\n",
       "      <td>3.78;1.34</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43542202</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>36,35;8,8;23,28</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs561178520</td>\n",
       "      <td>0.004200</td>\n",
       "      <td>0.007700</td>\n",
       "      <td>2</td>\n",
       "      <td>MYF6_HUMAN.H11MO.0.C, SNAI1_HUMAN.H11MO.0.C, M...</td>\n",
       "      <td>4.09;1.96</td>\n",
       "      <td>HNF4G, RXRA, HNF4A, ZBTB7A, SP1</td>\n",
       "      <td>ENSG00000142949:PTPRF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65692298</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40,34;94,96</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs192789815</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>1</td>\n",
       "      <td>HXB7_HUMAN.H11MO.0.C, LHX6_HUMAN.H11MO.0.D, HX...</td>\n",
       "      <td>3.89;1.54</td>\n",
       "      <td>ZNF263</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>83694285</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>32,33;42,44;45,33</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs539011323</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>0.005400</td>\n",
       "      <td>1</td>\n",
       "      <td>E4F1_HUMAN.H11MO.0.D, HXB8_HUMAN.H11MO.0.C, SR...</td>\n",
       "      <td>3.94;1.55</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000233008:RP11-475O6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>94875371</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>143,154;132,131</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs771560806</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0</td>\n",
       "      <td>HMGA1_HUMAN.H11MO.0.D, SOX11_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.81;1.82</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000143036:SLC44A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100189355</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>9,5;43,10;5,6;42,6;9,2;33,6</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.130</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>4</td>\n",
       "      <td>BARX2_HUMAN.H11MO.0.D, HXB13_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>5.21;2.61</td>\n",
       "      <td>ESR1</td>\n",
       "      <td>ENSG00000137992:DBT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT                         VAFs   AC     AF  \\\n",
       "0   chr1    2227533   C   T                    4,4;13,11  2.0  0.043   \n",
       "1   chr1    5896826   C   T                  47,38;41,32  2.0  0.043   \n",
       "2   chr1    7041938   G   A              109,100;138,118  2.0  0.043   \n",
       "3   chr1   16912416   G   A                189,90;160,95  2.0  0.043   \n",
       "4   chr1   16913968   T   A             18,9;24,13;17,11  3.0  0.065   \n",
       "5   chr1   16913970   A   G             18,8;24,14;15,11  3.0  0.065   \n",
       "6   chr1   21254094   A   G                     1,2;17,3  2.0  0.048   \n",
       "7   chr1   21264261   A   T                 82,117;62,70  2.0  0.043   \n",
       "8   chr1   21625236   G   A              127,104;109,137  2.0  0.043   \n",
       "9   chr1   24705404   T   A                  81,58;84,65  2.0  0.043   \n",
       "10  chr1   27971997   G   C                 44,62;100,80  2.0  0.043   \n",
       "11  chr1   27980435   A   G                  36,51;67,58  2.0  0.043   \n",
       "12  chr1   31811750   A   T                  78,81;17,15  2.0  0.043   \n",
       "13  chr1   33075149   C   G                112,96;104,84  2.0  0.043   \n",
       "14  chr1   37234892   C   T     108,82;98,86;74,75;52,47  4.0  0.087   \n",
       "15  chr1   43542202   G   A              36,35;8,8;23,28  3.0  0.065   \n",
       "16  chr1   65692298   G   A                  40,34;94,96  2.0  0.043   \n",
       "17  chr1   83694285   G   T            32,33;42,44;45,33  3.0  0.065   \n",
       "18  chr1   94875371   C   T              143,154;132,131  2.0  0.043   \n",
       "19  chr1  100189355   A   T  9,5;43,10;5,6;42,6;9,2;33,6  6.0  0.130   \n",
       "\n",
       "        avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0              .           0.000100           0.000300                     2   \n",
       "1    rs373612069           0.000700           0.000700                     2   \n",
       "2    rs542608175           0.000032           0.000067                     1   \n",
       "3              .           0.000097           0.000067                     0   \n",
       "4    rs752881304           0.004400           0.007600                     2   \n",
       "5    rs537954556           0.004400           0.007600                     2   \n",
       "6              .           0.000037           0.000073                     2   \n",
       "7       rs212534           0.001200           0.001800                     1   \n",
       "8    rs989937565           0.000065           0.000100                     0   \n",
       "9    rs759876216           0.000032           0.000000                     0   \n",
       "10  rs1024768231           0.000032           0.000067                     2   \n",
       "11   rs778827085           0.000065           0.000100                     2   \n",
       "12   rs553603564           0.001700           0.002500                     1   \n",
       "13   rs554147217           0.000800           0.001500                     0   \n",
       "14    rs72655842           0.005700           0.009800                     1   \n",
       "15   rs561178520           0.004200           0.007700                     2   \n",
       "16   rs192789815           0.000500           0.000900                     1   \n",
       "17   rs539011323           0.003300           0.005400                     1   \n",
       "18   rs771560806           0.000100           0.000200                     0   \n",
       "19             .           0.000200           0.000100                     4   \n",
       "\n",
       "                                     PWMEnrich motifs  \\\n",
       "0   AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1   OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2   PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3   MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4   TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "5   HMX3_HUMAN.H11MO.0.D, CTCFL_HUMAN.H11MO.0.A, C...   \n",
       "6   PPARA_HUMAN.H11MO.1.B, SOX17_HUMAN.H11MO.0.C, ...   \n",
       "7   MAFF_HUMAN.H11MO.1.B, NR1I3_HUMAN.H11MO.0.C, N...   \n",
       "8   NR2C1_HUMAN.H11MO.0.C, PPARD_HUMAN.H11MO.0.D, ...   \n",
       "9   PBX3_HUMAN.H11MO.1.A, SOX9_HUMAN.H11MO.1.B, HX...   \n",
       "10  PRRX1_HUMAN.H11MO.0.D, ZKSC3_HUMAN.H11MO.0.D, ...   \n",
       "11  GATA4_HUMAN.H11MO.0.A, ZKSC3_HUMAN.H11MO.0.D, ...   \n",
       "12  PAX3_HUMAN.H11MO.0.D, PAX8_HUMAN.H11MO.0.D, IR...   \n",
       "13  HIF1A_HUMAN.H11MO.0.C, MAFG_HUMAN.H11MO.1.A, A...   \n",
       "14        RORA_HUMAN.H11MO.0.C, NR1D1_HUMAN.H11MO.0.B   \n",
       "15  MYF6_HUMAN.H11MO.0.C, SNAI1_HUMAN.H11MO.0.C, M...   \n",
       "16  HXB7_HUMAN.H11MO.0.C, LHX6_HUMAN.H11MO.0.D, HX...   \n",
       "17  E4F1_HUMAN.H11MO.0.D, HXB8_HUMAN.H11MO.0.C, SR...   \n",
       "18  HMGA1_HUMAN.H11MO.0.D, SOX11_HUMAN.H11MO.0.D, ...   \n",
       "19  BARX2_HUMAN.H11MO.0.D, HXB13_HUMAN.H11MO.0.A, ...   \n",
       "\n",
       "   PWMEnrich best -log10pval and diff  \\\n",
       "0                           4.41;1.32   \n",
       "1                           4.48;1.41   \n",
       "2                           4.16;1.45   \n",
       "3                           4.15;1.47   \n",
       "4                           4.75;1.76   \n",
       "5                           4.54;1.46   \n",
       "6                           4.47;1.66   \n",
       "7                           3.37;1.07   \n",
       "8                           4.46;1.19   \n",
       "9                           4.97;1.85   \n",
       "10                          4.67;2.24   \n",
       "11                          3.71;1.37   \n",
       "12                          4.25;1.17   \n",
       "13                          3.52;2.00   \n",
       "14                          3.78;1.34   \n",
       "15                          4.09;1.96   \n",
       "16                          3.89;1.54   \n",
       "17                          3.94;1.55   \n",
       "18                          3.81;1.82   \n",
       "19                          5.21;2.61   \n",
       "\n",
       "                                               ENCODE  \\\n",
       "0   NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...   \n",
       "1                                                       \n",
       "2                                                       \n",
       "3                                                CTCF   \n",
       "4   PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   \n",
       "5   PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   \n",
       "6                                                       \n",
       "7                                                       \n",
       "8                    ZNF143, REST, USF1, RCOR1, EP300   \n",
       "9   EGR1, TBP, SP2, NFYA, BACH1, FOS, TEAD4, SPI1,...   \n",
       "10                                                      \n",
       "11                                                      \n",
       "12                                  RAD21, SIN3A, MAX   \n",
       "13                                                      \n",
       "14                                                      \n",
       "15                    HNF4G, RXRA, HNF4A, ZBTB7A, SP1   \n",
       "16                                             ZNF263   \n",
       "17                                                      \n",
       "18                                                      \n",
       "19                                               ESR1   \n",
       "\n",
       "                            Gene  \n",
       "0                              .  \n",
       "1          ENSG00000131697:NPHP4  \n",
       "2         ENSG00000171735:CAMTA1  \n",
       "3          ENSG00000058453:CROCC  \n",
       "4          ENSG00000058453:CROCC  \n",
       "5          ENSG00000058453:CROCC  \n",
       "6           ENSG00000117298:ECE1  \n",
       "7           ENSG00000117298:ECE1  \n",
       "8        ENSG00000076864:RAP1GAP  \n",
       "9                              .  \n",
       "10          ENSG00000158161:EYA3  \n",
       "11          ENSG00000158161:EYA3  \n",
       "12        ENSG00000134668:SPOCD1  \n",
       "13           ENSG00000004455:AK2  \n",
       "14                             .  \n",
       "15         ENSG00000142949:PTPRF  \n",
       "16                             .  \n",
       "17  ENSG00000233008:RP11-475O6.1  \n",
       "18       ENSG00000143036:SLC44A3  \n",
       "19           ENSG00000137992:DBT  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode[\"Gene\"] = enhancers_motifs_encode.apply(add_gene, args = (enhancers_motifs_encode_intersection_df,), axis=1)\n",
    "enhancers_motifs_encode = enhancers_motifs_encode.drop(labels = [\"POS-1\"], axis=1)\n",
    "enhancers_motifs_encode[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "351"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_motifs_encode)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000100           0.000300                     2   \n",
       "1           0.000700           0.000700                     2   \n",
       "2           0.000032           0.000067                     1   \n",
       "3           0.000097           0.000067                     0   \n",
       "4           0.004400           0.007600                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          4.41;1.32   \n",
       "1                          4.48;1.41   \n",
       "2                          4.16;1.45   \n",
       "3                          4.15;1.47   \n",
       "4                          4.75;1.76   \n",
       "\n",
       "                                              ENCODE                    Gene  \\\n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...                       .   \n",
       "1                                                      ENSG00000131697:NPHP4   \n",
       "2                                                     ENSG00000171735:CAMTA1   \n",
       "3                                               CTCF   ENSG00000058453:CROCC   \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   ENSG00000058453:CROCC   \n",
       "\n",
       "       genomic element  \n",
       "0  enhancer intergenic  \n",
       "1    enhancer intronic  \n",
       "2    enhancer intronic  \n",
       "3    enhancer intronic  \n",
       "4    enhancer intronic  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode[\"genomic element\"] = enhancers_motifs_encode.Gene.apply(lambda x: \"enhancer intergenic\" if x == \".\" else \"enhancer intronic\")\n",
    "enhancers_motifs_encode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>chr14</td>\n",
       "      <td>66471764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>16,8;8,8;10,9</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs138793190</td>\n",
       "      <td>0.0033</td>\n",
       "      <td>0.0045</td>\n",
       "      <td>3</td>\n",
       "      <td>NF2L1_HUMAN.H11MO.0.C, DUX4_HUMAN.H11MO.0.A, D...</td>\n",
       "      <td>4.51;1.99</td>\n",
       "      <td>TEAD4, TAL1, GATA2, EP300</td>\n",
       "      <td>ENSG00000258561:RP11-72M17.1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS REF ALT           VAFs   AC     AF     avsnp150  \\\n",
       "264  chr14  66471764   C   T  16,8;8,8;10,9  3.0  0.065  rs138793190   \n",
       "\n",
       "     gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "264             0.0033             0.0045                     3   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "264  NF2L1_HUMAN.H11MO.0.C, DUX4_HUMAN.H11MO.0.A, D...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff                     ENCODE  \\\n",
       "264                          4.51;1.99  TEAD4, TAL1, GATA2, EP300   \n",
       "\n",
       "                             Gene    genomic element  \n",
       "264  ENSG00000258561:RP11-72M17.1  enhancer intronic  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode[enhancers_motifs_encode[\"POS\"] == 66471764]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intergenic"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Closest gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end strand  \\\n",
       "0  chr1  11869  14409      +   \n",
       "1  chr1  14404  29570      -   \n",
       "2  chr1  17369  17436      -   \n",
       "3  chr1  29554  31109      +   \n",
       "4  chr1  30366  30503      +   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "\n",
       "                           Gene  \n",
       "0       ENSG00000223972:DDX11L1  \n",
       "1        ENSG00000227232:WASH7P  \n",
       "2     ENSG00000278267:MIR6859-1  \n",
       "3  ENSG00000243485:RP11-34P13.3  \n",
       "4     ENSG00000274890:MIR1302-2  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_tss(row):\n",
    "    if row['strand'] == '+':\n",
    "        return row['start']\n",
    "    else:\n",
    "        return row['end']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "      <td>11869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "      <td>29570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "      <td>29554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "      <td>30366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end strand  \\\n",
       "0  chr1  11869  14409      +   \n",
       "1  chr1  14404  29570      -   \n",
       "2  chr1  17369  17436      -   \n",
       "3  chr1  29554  31109      +   \n",
       "4  chr1  30366  30503      +   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "\n",
       "                           Gene    tss  \n",
       "0       ENSG00000223972:DDX11L1  11869  \n",
       "1        ENSG00000227232:WASH7P  29570  \n",
       "2     ENSG00000278267:MIR6859-1  17436  \n",
       "3  ENSG00000243485:RP11-34P13.3  29554  \n",
       "4     ENSG00000274890:MIR1302-2  30366  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info[\"tss\"] = genes_info.apply(find_tss, axis=1)\n",
    "genes_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t11869\t11869\tENSG00000223972:DDX11L1\n",
      " chr1\t17436\t17436\tENSG00000278267:MIR6859-1\n",
      " chr1\t29554\t29554\tENSG00000243485:RP11-34P13.3\n",
      " chr1\t29570\t29570\tENSG00000227232:WASH7P\n",
      " chr1\t30366\t30366\tENSG00000274890:MIR1302-2\n",
      " chr1\t36081\t36081\tENSG00000237613:FAM138A\n",
      " chr1\t52473\t52473\tENSG00000268020:OR4G4P\n",
      " chr1\t62948\t62948\tENSG00000240361:OR4G11P\n",
      " chr1\t69091\t69091\tENSG00000186092:OR4F5\n",
      " chr1\t91105\t91105\tENSG00000239945:RP11-34P13.8\n",
      " "
     ]
    }
   ],
   "source": [
    "genes_info_tss_bed = pbt.BedTool.from_dataframe(genes_info[['chr', 'tss', 'tss', 'Gene']])\n",
    "genes_info_tss_bed_sorted = genes_info_tss_bed.sort()\n",
    "genes_info_tss_bed_sorted.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t19482\t22400\tchr1\t17436\t17436\tENSG00000278267:MIR6859-1\t2046\n",
      " chr1\t183942\t184568\tchr1\t182393\t182393\tENSG00000279928:FO538757.3\t1549\n",
      " chr1\t189803\t193125\tchr1\t187958\t187958\tENSG00000273874:MIR6859-2\t1845\n",
      " chr1\t1024210\t1025994\tchr1\t1020123\t1020123\tENSG00000188157:AGRN\t4087\n",
      " chr1\t1031391\t1031884\tchr1\t1020123\t1020123\tENSG00000188157:AGRN\t11268\n",
      " chr1\t1032778\t1033104\tchr1\t1020123\t1020123\tENSG00000188157:AGRN\t12655\n",
      " chr1\t1134548\t1135587\tchr1\t1137017\t1137017\tENSG00000223823:LINC01342\t1430\n",
      " chr1\t1156466\t1159847\tchr1\t1167092\t1167092\tENSG00000207730:MIR200B\t7245\n",
      " chr1\t1290755\t1292028\tchr1\t1296170\t1296170\tENSG00000278073:MIR6726\t4142\n",
      " chr1\t1427223\t1428530\tchr1\t1426128\t1426128\tENSG00000205116:TMEM88B\t1095\n",
      " "
     ]
    }
   ],
   "source": [
    "#find closest gene tss for all enhancers, report all genes for ties\n",
    "enhancers_bed = pbt.BedTool.from_dataframe(enh_genes[['chr', 'start', 'end']]).sort()\n",
    "tss_closest_to_enh = enhancers_bed.closest(genes_info_tss_bed_sorted, t='all', d=True)\n",
    "tss_closest_to_enh.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(45548, 45578)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_bed), len(tss_closest_to_enh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2227532\t2227533\n",
      " chr1\t5896825\t5896826\n",
      " chr1\t7041937\t7041938\n",
      " chr1\t16912415\t16912416\n",
      " chr1\t16913967\t16913968\n",
      " chr1\t16913969\t16913970\n",
      " chr1\t21254093\t21254094\n",
      " chr1\t21264260\t21264261\n",
      " chr1\t21625235\t21625236\n",
      " chr1\t24705403\t24705404\n",
      " "
     ]
    }
   ],
   "source": [
    "enhancer_snps_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "351"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancer_snps_bedtool)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2227532\t2227533\tchr1\t2227002\t2227695\tchr1\t2228695\t2228695\tENSG00000157933:SKI\t1000\n",
      " chr1\t5896825\t5896826\tchr1\t5895777\t5897516\tchr1\t5893924\t5893924\tENSG00000266687:AL356261.1\t1853\n",
      " chr1\t7041937\t7041938\tchr1\t7040690\t7043676\tchr1\t7014279\t7014279\tENSG00000237365:RP11-334N17.1\t26411\n",
      " chr1\t16912415\t16912416\tchr1\t16912238\t16915295\tchr1\t16904776\t16904776\tENSG00000272426:RP11-108M9.6\t7462\n",
      " chr1\t16913967\t16913968\tchr1\t16912238\t16915295\tchr1\t16904776\t16904776\tENSG00000272426:RP11-108M9.6\t7462\n",
      " chr1\t16913969\t16913970\tchr1\t16912238\t16915295\tchr1\t16904776\t16904776\tENSG00000272426:RP11-108M9.6\t7462\n",
      " chr1\t21254093\t21254094\tchr1\t21253526\t21255567\tchr1\t21266082\t21266082\tENSG00000236936:RP3-329E20.2\t10515\n",
      " chr1\t21264260\t21264261\tchr1\t21260958\t21264403\tchr1\t21266082\t21266082\tENSG00000236936:RP3-329E20.2\t1679\n",
      " chr1\t21625235\t21625236\tchr1\t21623828\t21625754\tchr1\t21592548\t21592548\tENSG00000221781:AL359815.1\t31280\n",
      " chr1\t24705403\t24705404\tchr1\t24704096\t24706797\tchr1\t24745357\t24745357\tENSG00000169504:CLIC4\t38560\n",
      " "
     ]
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest_bedtool = enhancer_snps_bedtool.intersect(tss_closest_to_enh, wa = True, wb = True, loj=True)\n",
    "enhancers_motifs_encode_closest_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>2227002</td>\n",
       "      <td>2227695</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>5895777</td>\n",
       "      <td>5897516</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>7040690</td>\n",
       "      <td>7043676</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS     start       end                   Closest gene  \\\n",
       "0  chr1   2227533   2227002   2227695            ENSG00000157933:SKI   \n",
       "1  chr1   5896826   5895777   5897516     ENSG00000266687:AL356261.1   \n",
       "2  chr1   7041938   7040690   7043676  ENSG00000237365:RP11-334N17.1   \n",
       "3  chr1  16912416  16912238  16915295   ENSG00000272426:RP11-108M9.6   \n",
       "4  chr1  16913968  16912238  16915295   ENSG00000272426:RP11-108M9.6   \n",
       "\n",
       "   dist. to closest gene  \n",
       "0                   1000  \n",
       "1                   1853  \n",
       "2                  26411  \n",
       "3                   7462  \n",
       "4                   7462  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# change enhancers_motifs_encode_closest_bedtool to df and merge with enhancers_motifs_encode\n",
    "enhancers_closest = enhancers_motifs_encode_closest_bedtool.to_dataframe(names = [\"CHROM\", \"POS\", \"start\", \"end\", \"Closest gene\", \"dist. to closest gene\"], usecols = [0,2,4,5,9,10])\n",
    "enhancers_closest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "351 351\n"
     ]
    }
   ],
   "source": [
    "print len(enhancer_snps_bedtool), len(enhancers_closest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The number of records has not changed - for enhancers with SNPs always only one gene per enhancer was found."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000100           0.000300                     2   \n",
       "1           0.000700           0.000700                     2   \n",
       "2           0.000032           0.000067                     1   \n",
       "3           0.000097           0.000067                     0   \n",
       "4           0.004400           0.007600                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          4.41;1.32   \n",
       "1                          4.48;1.41   \n",
       "2                          4.16;1.45   \n",
       "3                          4.15;1.47   \n",
       "4                          4.75;1.76   \n",
       "\n",
       "                                              ENCODE                    Gene  \\\n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...                       .   \n",
       "1                                                      ENSG00000131697:NPHP4   \n",
       "2                                                     ENSG00000171735:CAMTA1   \n",
       "3                                               CTCF   ENSG00000058453:CROCC   \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   ENSG00000058453:CROCC   \n",
       "\n",
       "       genomic element                   Closest gene  dist. to closest gene  \n",
       "0  enhancer intergenic            ENSG00000157933:SKI                   1000  \n",
       "1    enhancer intronic     ENSG00000266687:AL356261.1                   1853  \n",
       "2    enhancer intronic  ENSG00000237365:RP11-334N17.1                  26411  \n",
       "3    enhancer intronic   ENSG00000272426:RP11-108M9.6                   7462  \n",
       "4    enhancer intronic   ENSG00000272426:RP11-108M9.6                   7462  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest = pd.merge(enhancers_motifs_encode, enhancers_closest[[\"CHROM\", \"POS\", \"Closest gene\", \"dist. to closest gene\"]],\n",
    "                                          on = [\"CHROM\", \"POS\"], how = \"outer\")\n",
    "\n",
    "enhancers_motifs_encode_closest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(351, 351)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_motifs_encode), len(enhancers_motifs_encode_closest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "enhancer intronic      224\n",
       "enhancer intergenic    127\n",
       "Name: genomic element, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest[\"genomic element\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7f50a8a21f90>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYQAAAD8CAYAAAB3u9PLAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvIxREBQAAEuVJREFUeJzt3X2wXHddx/H3h6SlD2Cb0GvMUEpa7RQ7I5R4rWVABEorFKXVwdqOSgY7xlFUGHUkRUbrjDrFkSefG2k1Ig8tj62AYhqLjjNOS0pDn2tKDZqQNhekFBiG2vL1j/1d3cZ7b8692bObe3m/Znb2nN85Z8/3l7N3PzkPezZVhSRJT5p0AZKkI4OBIEkCDARJUmMgSJIAA0GS1BgIkiTAQJAkNb0FQpIzkuwaejyS5PVJ1ibZnmR3e17TVw2SpO4yji+mJVkF7AO+H3gt8F9VdWWSLcCaqnpD70VIkhY0rkA4H/itqnp+kvuAF1XV/iTrgU9W1RkLLX/SSSfVhg0beq9TklaSW2+99QtVNdV1/tV9FjPkEuC9bXhdVe1vww8C6+ZaIMlmYDPAKaecws6dO3svUpJWkiSfW8z8vZ9UTnI08Erg/QdPq8HuyZy7KFW1taqmq2p6aqpzwEmSlmgcVxm9HPh0VT3Uxh9qh4pozwfGUIMk6RDGEQiX8n+HiwBuADa14U3A9WOoQZJ0CL0GQpLjgfOADw01Xwmcl2Q38NI2LkmasF5PKlfV14CnHdT2ReDcPtcrSVo8v6ksSQIMBElSYyBIkgADQZLUjOubyhOzYcvHlrzsnitfMcJKJOnI5h6CJAkwECRJjYEgSQIMBElSYyBIkgADQZLUGAiSJMBAkCQ1BoIkCTAQJEmNgSBJAgwESVJjIEiSAANBktQYCJIkwECQJDUGgiQJ6DkQkpyY5ANJ7k1yT5LnJVmbZHuS3e15TZ81SJK66XsP4R3A31fVs4DnAPcAW4AdVXU6sKONS5ImrLdASHIC8ELgaoCqerSqHgYuBLa12bYBF/VVgySpuz73EE4FZoC/THJbkncmOR5YV1X72zwPAuvmWjjJ5iQ7k+ycmZnpsUxJEvQbCKuBjcCfVdVzga9x0OGhqiqg5lq4qrZW1XRVTU9NTfVYpiQJ+g2EvcDeqrq5jX+AQUA8lGQ9QHs+0GMNkqSOeguEqnoQ+M8kZ7Smc4G7gRuATa1tE3B9XzVIkrpb3fPr/xLw7iRHAw8Ar2EQQtcluQz4HHBxzzVIkjroNRCqahcwPcekc/tcryRp8fymsiQJMBAkSY2BIEkCDARJUmMgSJIAA0GS1BgIkiTAQJAkNQaCJAkwECRJjYEgSQIMBElSYyBIkgADQZLUGAiSJMBAkCQ1BoIkCTAQJEmNgSBJAgwESVJjIEiSAANBktSs7vPFk+wBvgI8DjxWVdNJ1gLXAhuAPcDFVfWlPuuQJB3aOPYQXlxVZ1XVdBvfAuyoqtOBHW1ckjRhkzhkdCGwrQ1vAy6aQA2SpIP0HQgF/EOSW5Nsbm3rqmp/G34QWDfXgkk2J9mZZOfMzEzPZUqSej2HALygqvYl+XZge5J7hydWVSWpuRasqq3AVoDp6ek555EkjU6vewhVta89HwA+DJwNPJRkPUB7PtBnDZKkbnoLhCTHJ3nq7DBwPnAncAOwqc22Cbi+rxokSd31echoHfDhJLPreU9V/X2STwHXJbkM+BxwcY81SJI66i0QquoB4DlztH8ROLev9UqSlsZvKkuSAANBktQYCJIkwECQJDUGgiQJMBAkSY2BIEkCDARJUmMgSJIAA0GS1BgIkiTAQJAkNQaCJAkwECRJjYEgSQIMBElS0ykQknxP34VIkiar6x7Cnya5JckvJDmh14okSRPRKRCq6geAnwSeAdya5D1Jzuu1MknSWHU+h1BVu4E3AW8AfhD4wyT3JvmxvoqTJI1P13MIz07yNuAe4CXAj1TVd7fht/VYnyRpTFZ3nO+PgHcCb6yqr882VtXnk7ypl8okSWPV9ZDRK4D3zIZBkiclOQ6gqt610IJJViW5LclH2/ipSW5Ocn+Sa5McfTgdkCSNRtdAuBE4dmj8uNbWxesYHGqa9WbgbVX1XcCXgMs6vo4kqUddA+GYqvrq7EgbPu5QCyU5mcHexTvbeBicd/hAm2UbcNFiCpYk9aNrIHwtycbZkSTfC3x9gflnvR34deCbbfxpwMNV9Vgb3ws8fa4Fk2xOsjPJzpmZmY5lSpKWqutJ5dcD70/yeSDAdwA/sdACSX4YOFBVtyZ50WILq6qtwFaA6enpWuzykqTF6RQIVfWpJM8CzmhN91XVfx9isecDr0xyAXAM8G3AO4ATk6xuewknA/uWVrokaZQWc3O77wOeDWwELk3y6oVmrqrLq+rkqtoAXAL8Y1X9JHAT8Ko22ybg+kVXLUkauU57CEneBXwnsAt4vDUX8NdLWOcbgPcl+R3gNuDqJbyGJGnEup5DmAbOrKolHcuvqk8Cn2zDDwBnL+V1JEn96XrI6E4GJ5IlSStU1z2Ek4C7k9wCfGO2sape2UtVkqSx6xoIV/RZhCRp8rpedvpPSZ4JnF5VN7b7GK3qtzRJ0jh1vf31zzK43cRVrenpwEf6KkqSNH5dTyq/lsEXzR6B//2xnG/vqyhJ0vh1DYRvVNWjsyNJVjP4HoIkaYXoGgj/lOSNwLHtt5TfD/xtf2VJksatayBsAWaAO4CfAz7O4PeVJUkrRNerjL4J/EV7SJJWoK73Mvp35jhnUFWnjbwiSdJELOZeRrOOAX4cWDv6ciRJk9LpHEJVfXHosa+q3s7gpzElSStE10NGG4dGn8Rgj6Hr3oUkaRno+qH+lqHhx4A9wMUjr0aSNDFdrzJ6cd+FSJImq+sho19ZaHpVvXU05UiSJmUxVxl9H3BDG/8R4BZgdx9FSZLGr2sgnAxsrKqvACS5AvhYVf1UX4VJksar660r1gGPDo0/2tokSStE1z2EvwZuSfLhNn4RsK2fkiRJk9D1i2m/C7wG+FJ7vKaqfm+hZZIck+SWJJ9JcleS327tpya5Ocn9Sa5NcvThdkKSdPi6HjICOA54pKreAexNcuoh5v8G8JKqeg5wFvCyJOcAbwbeVlXfxSBcLltC3ZKkEev6E5q/BbwBuLw1HQX8zULL1MBXh+Y/isEN8l7C4Oc4YXDY6aJF1ixJ6kHXPYQfBV4JfA2gqj4PPPVQCyVZlWQXcADYDnwWeLiqHmuz7GXw+8ySpAnrGgiPVlXRboGd5PguC1XV41V1FoPLVs8GntW1sCSbk+xMsnNmZqbrYpKkJeoaCNcluQo4McnPAjeyiB/LqaqHgZuA57XXmL266WRg3zzLbK2q6aqanpqa6roqSdISdb3K6A8YHPf/IHAG8JtV9UcLLZNkKsmJbfhY4DzgHgbB8Ko22ybg+qWVLkkapUN+DyHJKuDGdoO77Yt47fXAtrb8k4DrquqjSe4G3pfkd4DbgKuXULckacQOGQhV9XiSbyY5oaq+3PWFq+p24LlztD/A4HyCJOkI0vWbyl8F7kiynXalEUBV/XIvVUmSxq5rIHyoPSRJK9SCgZDklKr6j6ryvkWStMId6iqjj8wOJPlgz7VIkiboUIGQoeHT+ixEkjRZhwqEmmdYkrTCHOqk8nOSPMJgT+HYNkwbr6r6tl6rkySNzYKBUFWrxlWIJGmyFvN7CJKkFcxAkCQBBoIkqTEQJEmAgSBJagwESRJgIEiSGgNBkgQYCJKkxkCQJAEGgiSpMRAkSYCBIElqDARJEtBjICR5RpKbktyd5K4kr2vta5NsT7K7Pa/pqwZJUnd97iE8BvxqVZ0JnAO8NsmZwBZgR1WdDuxo45KkCestEKpqf1V9ug1/BbgHeDpwIbCtzbYNuKivGiRJ3Y3lHEKSDcBzgZuBdVW1v016EFg3jhokSQs71G8qH7YkTwE+CLy+qh5J8r/TqqqS1DzLbQY2A5xyyil9lzmnDVs+tuRl91z5ihFWIkn963UPIclRDMLg3VX1odb8UJL1bfp64MBcy1bV1qqarqrpqampPsuUJNHvVUYBrgbuqaq3Dk26AdjUhjcB1/dVgySpuz4PGT0f+GngjiS7WtsbgSuB65JcBnwOuLjHGiRJHfUWCFX1L0DmmXxuX+uVJC2N31SWJAEGgiSpMRAkSYCBIElqDARJEmAgSJIaA0GSBBgIkqTGQJAkAQaCJKkxECRJgIEgSWoMBEkSYCBIkpref0JTS+PPd0oaN/cQJEmAgSBJajxk1JPDOeQjSZPgHoIkCTAQJEmNgSBJAgwESVLTWyAkuSbJgSR3DrWtTbI9ye72vKav9UuSFqfPPYS/Al52UNsWYEdVnQ7saOOSpCNAb4FQVf8M/NdBzRcC29rwNuCivtYvSVqccZ9DWFdV+9vwg8C6Ma9fkjSPiZ1UrqoCar7pSTYn2Zlk58zMzBgrk6RvTeMOhIeSrAdozwfmm7GqtlbVdFVNT01Nja1ASfpWNe5AuAHY1IY3AdePef2SpHn0ednpe4F/Bc5IsjfJZcCVwHlJdgMvbeOSpCNAbze3q6pL55l0bl/rlCQtnd9UliQBBoIkqTEQJEmAgSBJagwESRJgIEiSGgNBkgQYCJKkxkCQJAEGgiSpMRAkSYCBIElqDARJEtDj3U61PG3Y8rElL7vnyleMsBJJ4+YegiQJMBAkSY2BIEkCPIewIh3OeQBJ37rcQ5AkAQaCJKnxkJFG5nAPVU3qslUvtZUG3EOQJAETCoQkL0tyX5L7k2yZRA2SpCca+yGjJKuAPwHOA/YCn0pyQ1XdPe5aJC2eh9gWZzn9e01iD+Fs4P6qeqCqHgXeB1w4gTokSUMmEQhPB/5zaHxva5MkTdARe5VRks3A5jb61ST3LeFlTgK+MLqqJmol9QXm6E/ePKFKDkOreSVtm177MoFtvKy3zUH/XkvpyzMXM/MkAmEf8Iyh8ZNb2xNU1VZg6+GsKMnOqpo+nNc4UqykvsDK6o99OXKtpP6Moy+TOGT0KeD0JKcmORq4BLhhAnVIkoaMfQ+hqh5L8ovAJ4BVwDVVdde465AkPdFEziFU1ceBj49hVYd1yOkIs5L6AiurP/blyLWS+tN7X1JVfa9DkrQMeOsKSRKwggPhSL09RpI9Se5IsivJzta2Nsn2JLvb85rWniR/2Ppwe5KNQ6+zqc2/O8mmofbvba9/f1s2I67/miQHktw51NZ7/fOto4e+XJFkX9s+u5JcMDTt8lbXfUl+aKh9zvdau3Di5tZ+bbuIgiRPbuP3t+kbRtCXZyS5KcndSe5K8rrWvuy2zQJ9Wa7b5pgktyT5TOvPby+1hlH1c15VteIeDE5WfxY4DTga+Axw5qTrarXtAU46qO33gS1teAvw5jZ8AfB3QIBzgJtb+1rggfa8pg2vadNuafOmLfvyEdf/QmAjcOc4659vHT305Qrg1+aY98z2PnoycGp7f61a6L0GXAdc0ob/HPj5NvwLwJ+34UuAa0fQl/XAxjb8VODfWs3Lbtss0Jflum0CPKUNHwXc3P4dF1XDKPs5b62j+JA40h7A84BPDI1fDlw+6bpaLXv4/4FwH7C+Da8H7mvDVwGXHjwfcClw1VD7Va1tPXDvUPsT5hthHzbwxA/R3uufbx099OUK5v7QecJ7iMFVcs+b773WPgS+AKw++D05u2wbXt3my4i30fUM7he2bLfNHH1Z9tsGOA74NPD9i61hlP2c77FSDxkdybfHKOAfktyawbexAdZV1f42/CCwrg3P14+F2vfO0d63cdQ/3zr68IvtMMo1Q4c/FtuXpwEPV9VjB7U/4bXa9C+3+UeiHWJ4LoP/iS7rbXNQX2CZbpskq5LsAg4A2xn8j36xNYyyn3NaqYFwJHtBVW0EXg68NskLhyfWIMqX7aVf46i/53X8GfCdwFnAfuAtPa2nF0meAnwQeH1VPTI8bbltmzn6smy3TVU9XlVnMbgzw9nAsyZc0pxWaiB0uj3GJFTVvvZ8APgwgzfHQ0nWA7TnA232+fqxUPvJc7T3bRz1z7eOkaqqh9of7zeBv2CwfThEzXO1fxE4Mcnqg9qf8Fpt+glt/sOS5CgGH6DvrqoPteZluW3m6sty3jazquph4CYGh28WW8Mo+zmnlRoIR+TtMZIcn+Sps8PA+cCdDGqbvZpjE4NjprT2V7crQs4Bvtx2zT8BnJ9kTdttPp/BscH9wCNJzmlXgLx66LX6NI7651vHSM1+sDU/ymD7zK7/knYFyKnA6QxOss75Xmv/U74JeNUcNQ/35VXAP7b5D6fuAFcD91TVW4cmLbttM19flvG2mUpyYhs+lsH5kHuWUMMo+zm3UZ78OZIeDK6i+DcGx+p+Y9L1tJpOY3AFwGeAu2brYnCsbwewG7gRWNvaw+DHhD4L3AFMD73WzwD3t8drhtqnGfyhfBb4Y0Z/svK9DHbX/5vBMcnLxlH/fOvooS/varXe3v4A1w/N/xutrvsYunprvvda2963tD6+H3hyaz+mjd/fpp82gr68gMGhmtuBXe1xwXLcNgv0Zblum2cDt7W67wR+c6k1jKqf8z38prIkCVi5h4wkSYtkIEiSAANBktQYCJIkwECQJDUGgiQJMBAkSY2BIEkC4H8ABKt/setZO8AAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intergenic = enhancers_motifs_encode_closest[enhancers_motifs_encode_closest[\"genomic element\"] == \"enhancer intergenic\"]\n",
    "\n",
    "intergenic[\"dist. to closest gene\"].plot.hist(bins = 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7f50a8a21e90>"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYwAAAD8CAYAAABkbJM/AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvIxREBQAAFPxJREFUeJzt3X+wX3V95/HnS1AUdQXkms0AacBJ7WLXRrzL0vFHrVSL2BbtdClMVxEZo63O6rQzbdCOurvjju2qVLdbJa6s0iqCRZRVugrU1enMCiYYIfySYOOaGElEK/4aFHjvH9/Pxa/pvTefm+T7I7nPx8x37jmfc873vD/J9+aV8znne06qCkmS9uYRky5AknRwMDAkSV0MDElSFwNDktTFwJAkdTEwJEldDAxJUhcDQ5LUxcCQJHU5fNIF7I9jjz22Vq9ePekyJOmgsmnTpm9V1cxStzuoA2P16tVs3Lhx0mVI0kElydf2ZTuHpCRJXQwMSVIXA0OS1MXAkCR1MTAkSV0MDElSFwNDktTFwJAkdTEwJEldDupveu+v1es/tc/bbnvbiw5gJZI0/TzCkCR1MTAkSV0MDElSFwNDktTFwJAkdTEwJEldRhYYSU5I8tkktyW5NcnrWvsxSa5Nclf7eXRrT5J3J9ma5OYkp4yqNknS0o3yCOMB4I+q6mTgNOA1SU4G1gPXV9Ua4Po2D/BCYE17rQPeM8LaJElLNLLAqKqdVXVTm/4ecDtwHHAW8MG22geBF7fps4BLa+ALwFFJVo6qPknS0ozlHEaS1cDTgRuAFVW1sy36JrCiTR8HfH1os+2tTZI0BUYeGEkeB1wJvL6q7hteVlUF1BLfb12SjUk27t69+wBWKklazEgDI8kjGYTFh6rqY635nrmhpvZzV2vfAZwwtPnxre1nVNWGqpqtqtmZmZnRFS9J+hmjvEoqwPuB26vqnUOLrgbOa9PnAZ8Yan9Zu1rqNOC7Q0NXkqQJG+Xdap8JvBS4Jcnm1vYG4G3AFUkuAL4GnN2WXQOcCWwFfgicP8LaJElLNLLAqKp/ALLA4tPnWb+A14yqHknS/vGb3pKkLgaGJKmLgSFJ6mJgSJK6GBiSpC4GhiSpi4EhSepiYEiSuhgYkqQuBoYkqYuBIUnqYmBIkroYGJKkLgaGJKmLgSFJ6jLKJ+5dkmRXki1DbZcn2dxe2+YerJRkdZIfDS1776jqkiTtm1E+ce8DwF8Cl841VNXvzk0neQfw3aH1766qtSOsR5K0H0b5xL3PJ1k937L2vO+zgeeNav+SpANrUucwng3cU1V3DbWdmORLST6X5NkTqkuStIBRDkkt5lzgsqH5ncCqqro3yTOAjyd5alXdt+eGSdYB6wBWrVo1lmIlSRM4wkhyOPDbwOVzbVV1f1Xd26Y3AXcDPz/f9lW1oapmq2p2ZmZmHCVLkpjMkNSvAXdU1fa5hiQzSQ5r0ycBa4CvTqA2SdICRnlZ7WXA/wWekmR7kgvaonP42eEogOcAN7fLbP8WeHVVfXtUtUmSlm6UV0mdu0D7y+dpuxK4clS1SJL2n9/0liR1MTAkSV0MDElSFwNDktTFwJAkdTEwJEldDAxJUhcDQ5LUxcCQJHUxMCRJXQwMSVIXA0OS1MXAkCR1MTAkSV0MDElSFwNDktRllE/cuyTJriRbhtrekmRHks3tdebQsguTbE1yZ5JfH1VdkqR9M8ojjA8AZ8zTflFVrW2vawCSnMzg0a1Pbdv81dwzviVJ02FkgVFVnwd6n8t9FvCRqrq/qv4R2AqcOqraJElLN4lzGK9NcnMbsjq6tR0HfH1one2tTZI0JcYdGO8BngysBXYC71jqGyRZl2Rjko27d+8+0PVJkhYw1sCoqnuq6sGqegh4Hz8ddtoBnDC06vGtbb732FBVs1U1OzMzM9qCJUkPG2tgJFk5NPsSYO4KqquBc5IckeREYA1w4zhrkyQt7vBRvXGSy4DnAscm2Q68GXhukrVAAduAVwFU1a1JrgBuAx4AXlNVD46qNknS0o0sMKrq3Hma37/I+m8F3jqqeiRJ+8dvekuSuhgYkqQuBoYkqYuBIUnqYmBIkroYGJKkLgaGJKmLgSFJ6mJgSJK6dAVGkn896kIkSdOt9wjjr5LcmOQPkjxhpBVJkqZSV2BU1bOB32NwC/JNST6c5PkjrUySNFW6z2FU1V3AnwJ/AvwK8O4kdyT57VEVJ0maHr3nMJ6W5CLgduB5wG9W1b9q0xeNsD5J0pTovb35fwP+B/CGqvrRXGNVfSPJn46kMknSVOkdknoR8OG5sEjyiCRHAlTVX8+3QZJLkuxKsmWo7b+2Yaybk1yV5KjWvjrJj5Jsbq/37l+3JEkHWm9gXAc8Zmj+yNa2mA8AZ+zRdi3wi1X1NOArwIVDy+6uqrXt9erOuiRJY9IbGI+uqu/PzbTpIxfboKo+D3x7j7bPVNUDbfYLwPFLqFWSNEG9gfGDJKfMzSR5BvCjRdbv8Qrg74bmT0zypSSfS/Ls/XxvSdIB1nvS+/XAR5N8AwjwL4Hf3dedJnkj8ADwoda0E1hVVfe2MPp4kqdW1X3zbLsOWAewatWqfS1BkrREXYFRVV9M8gvAU1rTnVX1k33ZYZKXA78BnF5V1d7/fuD+Nr0pyd3AzwMb56llA7ABYHZ2tvalBknS0vUeYQD8G2B12+aUJFTVpUvZWZIzgD8GfqWqfjjUPgN8u6oeTHISsAb46lLeW5I0Wl2BkeSvgScDm4EHW3MBCwZGksuA5wLHJtkOvJnBVVFHANcmAfhCuyLqOcB/SvIT4CHg1VX17XnfWJI0Eb1HGLPAyXNDSD2q6tx5mt+/wLpXAlf2vrckafx6r5LawuBEtyRpmeo9wjgWuC3JjbST0wBV9VsjqUqSNHV6A+MtoyxCkjT9ei+r/VySnwPWVNV17T5Sh422NEnSNOm9vfkrgb8FLm5NxwEfH1VRkqTp03vS+zXAM4H74OGHKT1pVEVJkqZPb2DcX1U/nptJcjiD72FIkpaJ3sD4XJI3AI9pz/L+KPC/RleWJGna9AbGemA3cAvwKuAaBs/3liQtE71XST0EvK+9JEnLUO+9pP6Rec5ZVNVJB7wiSdJUWsq9pOY8Gvh3wDEHvhxJ0rTqOodRVfcOvXZU1V8ALxpxbZKkKdI7JHXK0OwjGBxxLOVZGpKkg1zvP/rvGJp+ANgGnH3Aq5EkTa3eq6R+ddSFSJKmW++Q1B8utryq3rnAdpcweH73rqr6xdZ2DHA5g8e9bgPOrqrvZPAIvncBZwI/BF5eVTf1dUOSNGq9X9ybBX6fwU0HjwNeDZwCPL69FvIB4Iw92tYD11fVGuD6Ng/wQgbP8l4DrAPe01mbJGkMes9hHA+cUlXfA0jyFuBTVfXvF9uoqj6fZPUezWcxeNY3wAeB/wP8SWu/tD0G9gtJjkqysqp2dtYoSRqh3iOMFcCPh+Z/3Nr2xYqhEPjm0PscB3x9aL3trU2SNAV6jzAuBW5MclWbfzGDo4P9UlWVZEl3vU2yjsGQFatWrdrfEiRJnXq/uPdW4HzgO+11flX9l33c5z1JVgK0n7ta+w7ghKH1jm9te9ayoapmq2p2ZmZmH0uQJC1V75AUwJHAfVX1LmB7khP3cZ9XA+e16fOATwy1vywDpwHf9fyFJE2P3stq38zgSqmnAP8TeCTwNwyewrfYdpcxOMF9bJLtwJuBtwFXJLkA+Bo//QLgNQwuqd3K4LLa85fYF0nSCPWew3gJ8HTgJoCq+kaSxS6npa137gKLTp9n3WLwKFhJ0hTqHZL6cfsHvQCSPHZ0JUmSplFvYFyR5GLgqCSvBK7DhylJ0rLSey+pt7dned/H4DzGm6rq2pFWJkmaKnsNjCSHAde1GxAaEpK0TO11SKqqHgQeSvKEMdQjSZpSvVdJfR+4Jcm1wA/mGqvqP4ykKknS1OkNjI+1lyRpmVo0MJKsqqr/V1X7fd8oSdLBbW/nMD4+N5HkyhHXIkmaYnsLjAxNnzTKQiRJ021vgVELTEuSlpm9nfT+pST3MTjSeEybps1XVf2LkVYnSZoaiwZGVR02rkIkSdNtKc/DkCQtYwaGJKmLgSFJ6tL7Te8DJslTgMuHmk4C3gQcBbwS2N3a31BV14y5PEnSAsYeGFV1J7AWHr4T7g7gKgaPZL2oqt4+7pokSXs36SGp04G7q+prE65DkrQXkw6Mc4DLhuZfm+TmJJckOXq+DZKsS7Ixycbdu3fPt4okaQQyeFT3BHacPAr4BvDUqronyQrgWwy+Uf6fgZVV9YrF3mN2drY2bty4zzWsXv+pfd52f2x724smsl9JAkiyqapml7rdJI8wXgjcVFX3AFTVPVX1YFU9xOB54adOsDZJ0h4mGRjnMjQclWTl0LKXAFvGXpEkaUFjv0oKIMljgecDrxpq/vMkaxkMSW3bY5kkacImEhhV9QPgiXu0vXQStUiS+kz6KilJ0kHCwJAkdTEwJEldDAxJUhcDQ5LUxcCQJHUxMCRJXQwMSVIXA0OS1MXAkCR1MTAkSV0MDElSFwNDktTFwJAkdTEwJEldJvI8DIAk24DvAQ8CD1TVbJJjgMuB1QweonR2VX1nUjVKkn5q0kcYv1pVa4ceRr4euL6q1gDXt3lJ0hSYdGDs6Szgg236g8CLJ1iLJGnIJAOjgM8k2ZRkXWtbUVU72/Q3gRWTKU2StKeJncMAnlVVO5I8Cbg2yR3DC6uqktSeG7VwWQewatWq8VQqSZrcEUZV7Wg/dwFXAacC9yRZCdB+7ppnuw1VNVtVszMzM+MsWZKWtYkERpLHJnn83DTwAmALcDVwXlvtPOATk6hPkvTPTWpIagVwVZK5Gj5cVf87yReBK5JcAHwNOHtC9UmS9jCRwKiqrwK/NE/7vcDp469IkrQ303ZZrSRpShkYkqQuBoYkqYuBIUnqYmBIkroYGJKkLgaGJKmLgSFJ6mJgSJK6GBiSpC4GhiSpi4EhSepiYEiSuhgYkqQuk3xE67K1ev2n9nnbbW970QGsRJL6eYQhSeoy9sBIckKSzya5LcmtSV7X2t+SZEeSze115rhrkyQtbBJDUg8Af1RVN7Xnem9Kcm1bdlFVvX0CNUmS9mLsgVFVO4Gdbfp7SW4Hjht3HZKkpZnoOYwkq4GnAze0ptcmuTnJJUmOXmCbdUk2Jtm4e/fuMVUqSZpYYCR5HHAl8Pqqug94D/BkYC2DI5B3zLddVW2oqtmqmp2ZmRlbvZK03E0kMJI8kkFYfKiqPgZQVfdU1YNV9RDwPuDUSdQmSZrfJK6SCvB+4PaqeudQ+8qh1V4CbBl3bZKkhU3iKqlnAi8FbkmyubW9ATg3yVqggG3AqyZQmyRpAZO4SuofgMyz6Jpx1yJJ6uc3vSVJXQwMSVIXA0OS1MXAkCR1MTAkSV0MDElSFx+gdJDx4UuSJsUjDElSF48wlpH9OToBj1Ck5c4jDElSFwNDktTFwJAkdTEwJEldDAxJUhevklI3vwMiLW9Td4SR5IwkdybZmmT9pOuRJA1MVWAkOQz478ALgZMZPIXv5MlWJUmC6RuSOhXYWlVfBUjyEeAs4LaJVqX9tr9fGtxXDoVJB860BcZxwNeH5rcD/3ZCtegQsBy/3e65poPHwfZ3NW2BsVdJ1gHr2uz3k9y5j291LPCtA1PVQcV+L0H+bASVjM+S+3yQ93fOsviMz/N3tZR+/9y+7HPaAmMHcMLQ/PGt7WFVtQHYsL87SrKxqmb3930ONvZ7+ViOfQb7Pcp9TNVJb+CLwJokJyZ5FHAOcPWEa5IkMWVHGFX1QJLXAp8GDgMuqapbJ1yWJIkpCwyAqroGuGYMu9rvYa2DlP1ePpZjn8F+j0yqatT7kCQdAqbtHIYkaUotu8A4WG89kuSSJLuSbBlqOybJtUnuaj+Pbu1J8u7Wx5uTnDK0zXlt/buSnDfU/owkt7Rt3p0ki+1jTH0+Iclnk9yW5NYkr1sm/X50khuTfLn1+z+29hOT3NBqvbxdGEKSI9r81rZ89dB7Xdja70zy60Pt8/4eLLSPcUpyWJIvJfnkYjUdSv1Osq19Djcn2djapu9zXlXL5sXgRPrdwEnAo4AvAydPuq7O2p8DnAJsGWr7c2B9m14P/FmbPhP4OyDAacANrf0Y4Kvt59Ft+ui27Ma2btq2L1xsH2Pq80rglDb9eOArDG4Zc6j3O8Dj2vQjgRtajVcA57T29wK/36b/AHhvmz4HuLxNn9w+40cAJ7bP/mGL/R4stI8xf9b/EPgw8MnFajqU+g1sA47do23qPudj/SBM+gX8MvDpofkLgQsnXdcS6l/NzwbGncDKNr0SuLNNXwycu+d6wLnAxUPtF7e2lcAdQ+0Pr7fQPibU/08Az19O/QaOBG5icMeDbwGH7/lZZnBV4S+36cPbetnz8z233kK/B22befcxxv4eD1wPPA/45GI1HWL93sY/D4yp+5wvtyGp+W49ctyEajkQVlTVzjb9TWBFm16on4u1b5+nfbF9jFUbbng6g/9tH/L9bsMym4FdwLUM/mf8T1X1wDy1Pty/tvy7wBNZ+p/HExfZx7j8BfDHwENtfrGaDqV+F/CZJJsyuJsFTOHnfOouq9W+qapKMtJL3saxj/kkeRxwJfD6qrqvDb+OraZJ9LuqHgTWJjkKuAr4hXHufxKS/Aawq6o2JXnupOsZs2dV1Y4kTwKuTXLH8MJp+ZwvtyOMvd565CBzT5KVAO3nrta+UD8Xaz9+nvbF9jEWSR7JICw+VFUf20tNh0y/51TVPwGfZTBMclSSuf/kDdf6cP/a8icA97L0P497F9nHODwT+K0k24CPMBiWetciNR0q/aaqdrSfuxj8B+FUpvBzvtwC41C79cjVwNyVEOcxGOOfa39Zu5riNOC77bDz08ALkhzdroZ4AYOx2p3AfUlOa1dPvGyP95pvHyPXank/cHtVvXNo0aHe75l2ZEGSxzA4b3M7g+D4nXlqGq71d4C/r8Gg9NXAOe1qohOBNQxOfs77e9C2WWgfI1dVF1bV8VW1utX091X1e4vUdEj0O8ljkzx+bprB53ML0/g5H+eJnWl4MbjC4CsMxoTfOOl6llD3ZcBO4CcMxiAvYDD2ej1wF3AdcExbNwweRHU3cAswO/Q+rwC2ttf5Q+2z7UN6N/CX/PRLnfPuY0x9fhaDsd2bgc3tdeYy6PfTgC+1fm8B3tTaT2LwD99W4KPAEa390W1+a1t+0tB7vbH17U7alTGL/R4stI8JfN6fy0+vkjqk+932/eX2unWurmn8nPtNb0lSl+U2JCVJ2kcGhiSpi4EhSepiYEiSuhgYkqQuBoYkqYuBIUnqYmBIkrr8fwxv81kVm34nAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest[\"dist. to closest gene\"].plot.hist(bins = 20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There is an intronic enhancer with ~500kb to closest gene!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>chr7</td>\n",
       "      <td>70093290</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>19,14;67,66</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs569183497</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>2</td>\n",
       "      <td>MYF6_HUMAN.H11MO.0.C, ASCL2_HUMAN.H11MO.0.D, O...</td>\n",
       "      <td>10.00;7.40</td>\n",
       "      <td>EP300</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "      <td>493419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr6</td>\n",
       "      <td>90885089</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>13,26;21,19</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs747544862</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>1</td>\n",
       "      <td>KLF4_HUMAN.H11MO.0.A, GLI3_HUMAN.H11MO.0.B</td>\n",
       "      <td>3.49;1.12</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000135341:MAP3K7</td>\n",
       "      <td>297631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>chr6</td>\n",
       "      <td>82710722</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>32,23;26,30</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs186524540</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002100</td>\n",
       "      <td>2</td>\n",
       "      <td>MSX2_HUMAN.H11MO.0.D, NF2L1_HUMAN.H11MO.0.C, H...</td>\n",
       "      <td>4.15;1.58</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000220091:LAP3P1</td>\n",
       "      <td>212088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>chr9</td>\n",
       "      <td>77848708</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>83,73;40,36;87,72</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs182563279</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>3</td>\n",
       "      <td>NFYB_HUMAN.H11MO.0.A, RUNX2_HUMAN.H11MO.0.A</td>\n",
       "      <td>4.34;1.14</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000156052:GNAQ</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000156052:GNAQ</td>\n",
       "      <td>181239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chr12</td>\n",
       "      <td>11828474</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>22,25;47,35;121,106</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs140107009</td>\n",
       "      <td>0.004900</td>\n",
       "      <td>0.008000</td>\n",
       "      <td>2</td>\n",
       "      <td>KAISO_HUMAN.H11MO.1.A, SRBP1_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>3.72;1.60</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000139083:ETV6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000139083:ETV6</td>\n",
       "      <td>176520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>chr5</td>\n",
       "      <td>107973351</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>6,2;7,2;12,6;9,3;20,9</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.114</td>\n",
       "      <td>rs963187462</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0.002200</td>\n",
       "      <td>2</td>\n",
       "      <td>SRY_HUMAN.H11MO.0.B, BARX2_HUMAN.H11MO.0.D, IR...</td>\n",
       "      <td>5.37;2.25</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000145743:FBXL17</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000251732:RN7SKP122</td>\n",
       "      <td>160828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35979734</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>28,24;18,13</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs777315289</td>\n",
       "      <td>0.001600</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>1</td>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.82;1.54</td>\n",
       "      <td>JUN, FOS, MYC</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>150095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35842056</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40,21;32,41;22,8</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs181047757</td>\n",
       "      <td>0.004000</td>\n",
       "      <td>0.006400</td>\n",
       "      <td>2</td>\n",
       "      <td>NFAC3_HUMAN.H11MO.0.B, STAT4_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>4.44;1.43</td>\n",
       "      <td>REST</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236036:LINC00445</td>\n",
       "      <td>141923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35841849</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>61,55;44,28</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs142256254</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>1</td>\n",
       "      <td>DPRX_HUMAN.H11MO.0.D, BHE23_HUMAN.H11MO.0.D</td>\n",
       "      <td>3.30;1.55</td>\n",
       "      <td>REST</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236036:LINC00445</td>\n",
       "      <td>141923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>chr5</td>\n",
       "      <td>15795255</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>42,51;108,76;53,56</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs576805372</td>\n",
       "      <td>0.003500</td>\n",
       "      <td>0.005700</td>\n",
       "      <td>1</td>\n",
       "      <td>E4F1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, H...</td>\n",
       "      <td>4.82;1.42</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000183580:FBXL7</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000216077:MIR887</td>\n",
       "      <td>139082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>chr2</td>\n",
       "      <td>105605799</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0,5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>rs557383407</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>ZN708_HUMAN.H11MO.1.D</td>\n",
       "      <td>3.37;1.20</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000071051:NCK2</td>\n",
       "      <td>138671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>chr10</td>\n",
       "      <td>78864401</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>139,120;123,127</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs1034121075</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, HTF4_HUMAN.H11MO.0.A, T...</td>\n",
       "      <td>4.87;1.28</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000229569:RP11-481G8.2</td>\n",
       "      <td>136510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>chr11</td>\n",
       "      <td>120744299</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>131,109;102,105</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs140746993</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, TYY1_HUMAN.H11MO.0.A, T...</td>\n",
       "      <td>3.18;1.42</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000149403:GRIK4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000255006:TCEB1P22</td>\n",
       "      <td>123536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>chr13</td>\n",
       "      <td>94312722</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>85,9;74,8;83,47;48,50</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs72647627</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>1</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C, BHE40_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>4.72;1.81</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000183098:GPC6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236520:GPC6-AS1</td>\n",
       "      <td>123509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>chr7</td>\n",
       "      <td>154743157</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>94,92;95,89;18,24</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs766837968</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>1</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, ZN554_HUMAN.H11MO.1.D, Z...</td>\n",
       "      <td>5.74;1.74</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000130226:DPP6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236408:RP11-476H24.1</td>\n",
       "      <td>121132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>chr14</td>\n",
       "      <td>67943009</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>85,75;57,57</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs747732406</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0</td>\n",
       "      <td>OVOL2_HUMAN.H11MO.0.D, NR4A2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>3.48;1.33</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000182185:RAD51B</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000182185:RAD51B</td>\n",
       "      <td>120301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>chr11</td>\n",
       "      <td>71138813</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11,2;7,2;27,3;8,4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXP1_HUMAN.H11MO.0.A, GATA3_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>5.13;1.69</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000162105:SHANK2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000162105:SHANK2</td>\n",
       "      <td>111174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>chr7</td>\n",
       "      <td>50676547</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>49,52;54,50</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs956599362</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000106070:GRB10</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000132437:DDC</td>\n",
       "      <td>110988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>chr5</td>\n",
       "      <td>78534216</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>60,44;56,68</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs545034134</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN784_HUMAN.H11MO.0.D, RXRG_HUMAN.H11MO.0.B, T...</td>\n",
       "      <td>3.90;1.27</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000145685:LHFPL2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000248909:HMGB1P21</td>\n",
       "      <td>108674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>chr3</td>\n",
       "      <td>131217019</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>15,12;60,28</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs572945195</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>ESR2_HUMAN.H11MO.1.A, HIC1_HUMAN.H11MO.0.C, NR...</td>\n",
       "      <td>4.36;1.85</td>\n",
       "      <td>NR3C1, CREB1, FOXA1, ATF3, JUN, RAD21, FOSL2, ...</td>\n",
       "      <td>ENSG00000114670:NEK11</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000212073:AC121332.1</td>\n",
       "      <td>105161</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                   VAFs   AC     AF  \\\n",
       "149   chr7   70093290   G   A            19,14;67,66  2.0  0.043   \n",
       "130   chr6   90885089   C   G            13,26;21,19  2.0  0.043   \n",
       "129   chr6   82710722   T   G            32,23;26,30  2.0  0.043   \n",
       "176   chr9   77848708   T   C      83,73;40,36;87,72  3.0  0.065   \n",
       "227  chr12   11828474   G   A    22,25;47,35;121,106  3.0  0.065   \n",
       "109   chr5  107973351   G   T  6,2;7,2;12,6;9,3;20,9  5.0  0.114   \n",
       "250  chr13   35979734   A   C            28,24;18,13  2.0  0.043   \n",
       "249  chr13   35842056   C   T       40,21;32,41;22,8  3.0  0.065   \n",
       "248  chr13   35841849   C   T            61,55;44,28  2.0  0.043   \n",
       "95    chr5   15795255   G   C     42,51;108,76;53,56  3.0  0.065   \n",
       "61    chr2  105605799   C   T                    0,5  2.0  0.045   \n",
       "196  chr10   78864401   T   G        139,120;123,127  2.0  0.043   \n",
       "222  chr11  120744299   G   C        131,109;102,105  2.0  0.043   \n",
       "255  chr13   94312722   A   G  85,9;74,8;83,47;48,50  4.0  0.087   \n",
       "159   chr7  154743157   G   A      94,92;95,89;18,24  3.0  0.065   \n",
       "265  chr14   67943009   A   G            85,75;57,57  2.0  0.043   \n",
       "217  chr11   71138813   G   A      11,2;7,2;27,3;8,4  4.0  0.087   \n",
       "146   chr7   50676547   T   C            49,52;54,50  2.0  0.043   \n",
       "105   chr5   78534216   G   A            60,44;56,68  2.0  0.043   \n",
       "83    chr3  131217019   G   C            15,12;60,28  2.0  0.043   \n",
       "\n",
       "         avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "149   rs569183497           0.000700           0.001300                     2   \n",
       "130   rs747544862           0.000300           0.000600                     1   \n",
       "129   rs186524540           0.001300           0.002100                     2   \n",
       "176   rs182563279           0.001300           0.002300                     3   \n",
       "227   rs140107009           0.004900           0.008000                     2   \n",
       "109   rs963187462           0.001500           0.002200                     2   \n",
       "250   rs777315289           0.001600           0.001500                     1   \n",
       "249   rs181047757           0.004000           0.006400                     2   \n",
       "248   rs142256254           0.000700           0.001200                     1   \n",
       "95    rs576805372           0.003500           0.005700                     1   \n",
       "61    rs557383407           0.000032           0.000067                     1   \n",
       "196  rs1034121075           0.000600           0.001100                     0   \n",
       "222   rs140746993           0.000800           0.001300                     0   \n",
       "255    rs72647627           0.001800           0.001500                     1   \n",
       "159   rs766837968           0.000600           0.000900                     1   \n",
       "265   rs747732406           0.000500           0.000800                     0   \n",
       "217             .           0.000073           0.000100                     3   \n",
       "146   rs956599362           0.000500           0.000700                     2   \n",
       "105   rs545034134           0.000032           0.000000                     0   \n",
       "83    rs572945195           0.000500           0.000700                     2   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "149  MYF6_HUMAN.H11MO.0.C, ASCL2_HUMAN.H11MO.0.D, O...   \n",
       "130         KLF4_HUMAN.H11MO.0.A, GLI3_HUMAN.H11MO.0.B   \n",
       "129  MSX2_HUMAN.H11MO.0.D, NF2L1_HUMAN.H11MO.0.C, H...   \n",
       "176        NFYB_HUMAN.H11MO.0.A, RUNX2_HUMAN.H11MO.0.A   \n",
       "227  KAISO_HUMAN.H11MO.1.A, SRBP1_HUMAN.H11MO.0.A, ...   \n",
       "109  SRY_HUMAN.H11MO.0.B, BARX2_HUMAN.H11MO.0.D, IR...   \n",
       "250  CXXC1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, ...   \n",
       "249  NFAC3_HUMAN.H11MO.0.B, STAT4_HUMAN.H11MO.0.A, ...   \n",
       "248        DPRX_HUMAN.H11MO.0.D, BHE23_HUMAN.H11MO.0.D   \n",
       "95   E4F1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, H...   \n",
       "61                               ZN708_HUMAN.H11MO.1.D   \n",
       "196  ZN335_HUMAN.H11MO.1.A, HTF4_HUMAN.H11MO.0.A, T...   \n",
       "222  ZN335_HUMAN.H11MO.1.A, TYY1_HUMAN.H11MO.0.A, T...   \n",
       "255  ZBT14_HUMAN.H11MO.0.C, BHE40_HUMAN.H11MO.0.A, ...   \n",
       "159  MAFF_HUMAN.H11MO.1.B, ZN554_HUMAN.H11MO.1.D, Z...   \n",
       "265  OVOL2_HUMAN.H11MO.0.D, NR4A2_HUMAN.H11MO.0.C, ...   \n",
       "217  FOXP1_HUMAN.H11MO.0.A, GATA3_HUMAN.H11MO.0.A, ...   \n",
       "146                                                      \n",
       "105  ZN784_HUMAN.H11MO.0.D, RXRG_HUMAN.H11MO.0.B, T...   \n",
       "83   ESR2_HUMAN.H11MO.1.A, HIC1_HUMAN.H11MO.0.C, NR...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "149                         10.00;7.40   \n",
       "130                          3.49;1.12   \n",
       "129                          4.15;1.58   \n",
       "176                          4.34;1.14   \n",
       "227                          3.72;1.60   \n",
       "109                          5.37;2.25   \n",
       "250                          3.82;1.54   \n",
       "249                          4.44;1.43   \n",
       "248                          3.30;1.55   \n",
       "95                           4.82;1.42   \n",
       "61                           3.37;1.20   \n",
       "196                          4.87;1.28   \n",
       "222                          3.18;1.42   \n",
       "255                          4.72;1.81   \n",
       "159                          5.74;1.74   \n",
       "265                          3.48;1.33   \n",
       "217                          5.13;1.69   \n",
       "146                                      \n",
       "105                          3.90;1.27   \n",
       "83                           4.36;1.85   \n",
       "\n",
       "                                                ENCODE  \\\n",
       "149                                              EP300   \n",
       "130                                                      \n",
       "129                                                      \n",
       "176                                                      \n",
       "227                                                      \n",
       "109                                                      \n",
       "250                                      JUN, FOS, MYC   \n",
       "249                                               REST   \n",
       "248                                               REST   \n",
       "95                                                       \n",
       "61                                                       \n",
       "196                                                      \n",
       "222                                                      \n",
       "255                                                      \n",
       "159                                                      \n",
       "265                                                      \n",
       "217                                                      \n",
       "146                                                      \n",
       "105                                                      \n",
       "83   NR3C1, CREB1, FOXA1, ATF3, JUN, RAD21, FOSL2, ...   \n",
       "\n",
       "                       Gene      genomic element  \\\n",
       "149   ENSG00000158321:AUTS2    enhancer intronic   \n",
       "130                       .  enhancer intergenic   \n",
       "129                       .  enhancer intergenic   \n",
       "176    ENSG00000156052:GNAQ    enhancer intronic   \n",
       "227    ENSG00000139083:ETV6    enhancer intronic   \n",
       "109  ENSG00000145743:FBXL17    enhancer intronic   \n",
       "250   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "249   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "248   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "95    ENSG00000183580:FBXL7    enhancer intronic   \n",
       "61                        .  enhancer intergenic   \n",
       "196                       .  enhancer intergenic   \n",
       "222   ENSG00000149403:GRIK4    enhancer intronic   \n",
       "255    ENSG00000183098:GPC6    enhancer intronic   \n",
       "159    ENSG00000130226:DPP6    enhancer intronic   \n",
       "265  ENSG00000182185:RAD51B    enhancer intronic   \n",
       "217  ENSG00000162105:SHANK2    enhancer intronic   \n",
       "146   ENSG00000106070:GRB10    enhancer intronic   \n",
       "105  ENSG00000145685:LHFPL2    enhancer intronic   \n",
       "83    ENSG00000114670:NEK11    enhancer intronic   \n",
       "\n",
       "                      Closest gene  dist. to closest gene  \n",
       "149          ENSG00000158321:AUTS2                 493419  \n",
       "130         ENSG00000135341:MAP3K7                 297631  \n",
       "129         ENSG00000220091:LAP3P1                 212088  \n",
       "176           ENSG00000156052:GNAQ                 181239  \n",
       "227           ENSG00000139083:ETV6                 176520  \n",
       "109      ENSG00000251732:RN7SKP122                 160828  \n",
       "250          ENSG00000133083:DCLK1                 150095  \n",
       "249      ENSG00000236036:LINC00445                 141923  \n",
       "248      ENSG00000236036:LINC00445                 141923  \n",
       "95          ENSG00000216077:MIR887                 139082  \n",
       "61            ENSG00000071051:NCK2                 138671  \n",
       "196   ENSG00000229569:RP11-481G8.2                 136510  \n",
       "222       ENSG00000255006:TCEB1P22                 123536  \n",
       "255       ENSG00000236520:GPC6-AS1                 123509  \n",
       "159  ENSG00000236408:RP11-476H24.1                 121132  \n",
       "265         ENSG00000182185:RAD51B                 120301  \n",
       "217         ENSG00000162105:SHANK2                 111174  \n",
       "146            ENSG00000132437:DDC                 110988  \n",
       "105       ENSG00000248909:HMGB1P21                 108674  \n",
       "83      ENSG00000212073:AC121332.1                 105161  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest.sort_values(by = \"dist. to closest gene\", ascending=False)[:20]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Protein coding genes only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gene_id \"ENSG00000223972\"; gene_version \"5\"; gene_name \"DDX11L1\"; gene_source \"havana\"; gene_biotype \"transcribed_unprocessed_pseudogene\"; havana_gene \"OTTHUMG00000000961\"; havana_gene_version \"2\";'"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info[\"info\"].loc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "      <td>11869</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "      <td>29570</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "      <td>29554</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "      <td>30366</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000237613\"; gene_version \"2\"; g...</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>ENSG00000237613:FAM138A</td>\n",
       "      <td>36081</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52473</td>\n",
       "      <td>53312</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000268020\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "      <td>ENSG00000268020:OR4G4P</td>\n",
       "      <td>52473</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>62948</td>\n",
       "      <td>63887</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000240361\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000240361</td>\n",
       "      <td>ENSG00000240361:OR4G11P</td>\n",
       "      <td>62948</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69091</td>\n",
       "      <td>70008</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000186092\"; gene_version \"4\"; g...</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENSG00000186092:OR4F5</td>\n",
       "      <td>69091</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>89295</td>\n",
       "      <td>133723</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000238009\"; gene_version \"6\"; g...</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009:RP11-34P13.7</td>\n",
       "      <td>133723</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start     end strand  \\\n",
       "0  chr1  11869   14409      +   \n",
       "1  chr1  14404   29570      -   \n",
       "2  chr1  17369   17436      -   \n",
       "3  chr1  29554   31109      +   \n",
       "4  chr1  30366   30503      +   \n",
       "5  chr1  34554   36081      -   \n",
       "6  chr1  52473   53312      +   \n",
       "7  chr1  62948   63887      +   \n",
       "8  chr1  69091   70008      +   \n",
       "9  chr1  89295  133723      -   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "5  gene_id \"ENSG00000237613\"; gene_version \"2\"; g...  ENSG00000237613   \n",
       "6  gene_id \"ENSG00000268020\"; gene_version \"3\"; g...  ENSG00000268020   \n",
       "7  gene_id \"ENSG00000240361\"; gene_version \"1\"; g...  ENSG00000240361   \n",
       "8  gene_id \"ENSG00000186092\"; gene_version \"4\"; g...  ENSG00000186092   \n",
       "9  gene_id \"ENSG00000238009\"; gene_version \"6\"; g...  ENSG00000238009   \n",
       "\n",
       "                           Gene     tss                        gene_biotype  \n",
       "0       ENSG00000223972:DDX11L1   11869  transcribed_unprocessed_pseudogene  \n",
       "1        ENSG00000227232:WASH7P   29570              unprocessed_pseudogene  \n",
       "2     ENSG00000278267:MIR6859-1   17436                               miRNA  \n",
       "3  ENSG00000243485:RP11-34P13.3   29554                             lincRNA  \n",
       "4     ENSG00000274890:MIR1302-2   30366                               miRNA  \n",
       "5       ENSG00000237613:FAM138A   36081                             lincRNA  \n",
       "6        ENSG00000268020:OR4G4P   52473              unprocessed_pseudogene  \n",
       "7       ENSG00000240361:OR4G11P   62948              unprocessed_pseudogene  \n",
       "8         ENSG00000186092:OR4F5   69091                      protein_coding  \n",
       "9  ENSG00000238009:RP11-34P13.7  133723                             lincRNA  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info[\"gene_biotype\"] = genes_info[\"info\"].str.split('\"').str[9]\n",
    "genes_info[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2227532\t2227533\tchr1\t2227002\t2227695\tchr1\t2228695\t2228695\tENSG00000157933:SKI\t1000\n",
      " chr1\t5896825\t5896826\tchr1\t5895777\t5897516\tchr1\t5991466\t5991466\tENSG00000069424:KCNAB2\t93950\n",
      " chr1\t7041937\t7041938\tchr1\t7040690\t7043676\tchr1\t6785324\t6785324\tENSG00000171735:CAMTA1\t255366\n",
      " chr1\t16912415\t16912416\tchr1\t16912238\t16915295\tchr1\t16980835\t16980835\tENSG00000117122:MFAP2\t65540\n",
      " chr1\t16913967\t16913968\tchr1\t16912238\t16915295\tchr1\t16980835\t16980835\tENSG00000117122:MFAP2\t65540\n",
      " chr1\t16913969\t16913970\tchr1\t16912238\t16915295\tchr1\t16980835\t16980835\tENSG00000117122:MFAP2\t65540\n",
      " chr1\t21254093\t21254094\tchr1\t21253526\t21255567\tchr1\t21176888\t21176888\tENSG00000075151:EIF4G3\t76638\n",
      " chr1\t21264260\t21264261\tchr1\t21260958\t21264403\tchr1\t21345504\t21345504\tENSG00000117298:ECE1\t81101\n",
      " chr1\t21625235\t21625236\tchr1\t21623828\t21625754\tchr1\t21669363\t21669363\tENSG00000076864:RAP1GAP\t43609\n",
      " chr1\t24705403\t24705404\tchr1\t24704096\t24706797\tchr1\t24745357\t24745357\tENSG00000169504:CLIC4\t38560\n",
      " "
     ]
    }
   ],
   "source": [
    "protein_coding_genes = genes_info[genes_info[\"gene_biotype\"] == \"protein_coding\"]\n",
    "protein_coding_genes_info_tss_bed = pbt.BedTool.from_dataframe(protein_coding_genes[['chr', 'tss', 'tss', 'Gene']])\n",
    "protein_coding_genes_info_tss_bed_sorted = protein_coding_genes_info_tss_bed.sort()\n",
    "\n",
    "#find closest gene tss for all enhancers, report all genes for ties\n",
    "protein_coding_tss_closest_to_enh = enhancers_bed.closest(protein_coding_genes_info_tss_bed_sorted, t='all', d=True)\n",
    "enhancers_motifs_encode_closest_protein_coding_bedtool = enhancer_snps_bedtool.intersect(protein_coding_tss_closest_to_enh, wa = True, wb = True, loj=True)\n",
    "enhancers_motifs_encode_closest_protein_coding_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>2227002</td>\n",
       "      <td>2227695</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>5895777</td>\n",
       "      <td>5897516</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>7040690</td>\n",
       "      <td>7043676</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>16912238</td>\n",
       "      <td>16915295</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS     start       end    Closest protein gene  \\\n",
       "0  chr1   2227533   2227002   2227695     ENSG00000157933:SKI   \n",
       "1  chr1   5896826   5895777   5897516  ENSG00000069424:KCNAB2   \n",
       "2  chr1   7041938   7040690   7043676  ENSG00000171735:CAMTA1   \n",
       "3  chr1  16912416  16912238  16915295   ENSG00000117122:MFAP2   \n",
       "4  chr1  16913968  16912238  16915295   ENSG00000117122:MFAP2   \n",
       "\n",
       "   dist. to closest protein gene  \n",
       "0                           1000  \n",
       "1                          93950  \n",
       "2                         255366  \n",
       "3                          65540  \n",
       "4                          65540  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# change enhancers_motifs_encode_closest_protein_coding_bedtool to df and merge with enhancers_motifs_encode\n",
    "enhancers_closest_protein_coding = enhancers_motifs_encode_closest_protein_coding_bedtool.to_dataframe(names = [\"CHROM\", \"POS\", \"start\", \"end\", \"Closest protein gene\", \"dist. to closest protein gene\"], usecols = [0,2,4,5,9,10])\n",
    "enhancers_closest_protein_coding.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>2</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>2</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000100           0.000300                     2   \n",
       "1           0.000700           0.000700                     2   \n",
       "2           0.000032           0.000067                     1   \n",
       "3           0.000097           0.000067                     0   \n",
       "4           0.004400           0.007600                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          4.41;1.32   \n",
       "1                          4.48;1.41   \n",
       "2                          4.16;1.45   \n",
       "3                          4.15;1.47   \n",
       "4                          4.75;1.76   \n",
       "\n",
       "                                              ENCODE                    Gene  \\\n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...                       .   \n",
       "1                                                      ENSG00000131697:NPHP4   \n",
       "2                                                     ENSG00000171735:CAMTA1   \n",
       "3                                               CTCF   ENSG00000058453:CROCC   \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   ENSG00000058453:CROCC   \n",
       "\n",
       "       genomic element                   Closest gene  dist. to closest gene  \\\n",
       "0  enhancer intergenic            ENSG00000157933:SKI                   1000   \n",
       "1    enhancer intronic     ENSG00000266687:AL356261.1                   1853   \n",
       "2    enhancer intronic  ENSG00000237365:RP11-334N17.1                  26411   \n",
       "3    enhancer intronic   ENSG00000272426:RP11-108M9.6                   7462   \n",
       "4    enhancer intronic   ENSG00000272426:RP11-108M9.6                   7462   \n",
       "\n",
       "     Closest protein gene  dist. to closest protein gene  \n",
       "0     ENSG00000157933:SKI                           1000  \n",
       "1  ENSG00000069424:KCNAB2                          93950  \n",
       "2  ENSG00000171735:CAMTA1                         255366  \n",
       "3   ENSG00000117122:MFAP2                          65540  \n",
       "4   ENSG00000117122:MFAP2                          65540  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest = pd.merge(enhancers_motifs_encode_closest, enhancers_closest_protein_coding[[\"CHROM\", \"POS\", \"Closest protein gene\", \"dist. to closest protein gene\"]],\n",
    "                                          on = [\"CHROM\", \"POS\"], how = \"outer\")\n",
    "\n",
    "enhancers_motifs_encode_closest.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Hi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>-log10(qval)</th>\n",
       "      <th>confirmed_both_ways</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000606034</td>\n",
       "      <td>ENSG00000272512</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end             ENST             ENSG  -log10(qval)  \\\n",
       "1  chr1  1024210  1025994  ENST00000606034  ENSG00000272512      8.697366   \n",
       "2  chr1  1024210  1025994  ENST00000484667  ENSG00000188290      8.697366   \n",
       "3  chr1  1024210  1025994  ENST00000481869  ENSG00000188290      8.697366   \n",
       "4  chr1  1024210  1025994  ENST00000304952  ENSG00000188290      8.697366   \n",
       "5  chr1  1024210  1025994  ENST00000428771  ENSG00000188290      8.697366   \n",
       "\n",
       "   confirmed_both_ways  \n",
       "1                    1  \n",
       "2                    1  \n",
       "3                    1  \n",
       "4                    1  \n",
       "5                    1  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contacts\n",
    "contacts = pd.read_csv('/mnt/chr7/data/hansiu/Magdowe/luty2018/E-enhancers_1_hg19-P-TSSs_hg19-all-rem.bed', \n",
    "                       sep=' ')\n",
    "contacts_to_genes = contacts[contacts[\"ENSG\"] != '-']\n",
    "contacts_to_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t1024210\t1025994\tENSG00000272512\n",
      " chr1\t1024210\t1025994\tENSG00000188290\n",
      " chr1\t1024210\t1025994\tENSG00000187608\n",
      " chr1\t1024210\t1025994\tENSG00000184163\n",
      " chr1\t1024210\t1025994\tENSG00000131584\n",
      " chr1\t1024210\t1025994\tENSG00000169972\n",
      " chr1\t1024210\t1025994\tENSG00000127054\n",
      " chr1\t1024210\t1025994\tENSG00000224051\n",
      " chr1\t1024210\t1025994\tENSG00000162576\n",
      " chr1\t1024210\t1025994\tENSG00000175756\n",
      " "
     ]
    }
   ],
   "source": [
    "contacts_bedtool = pbt.BedTool.from_dataframe(contacts_to_genes[[\"chr\", \"start\", \"end\", \"ENSG\"]].drop_duplicates())\n",
    "contacts_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2227532\t2227533\tchr1\t2227002\t2227695\tENSG00000157933\n",
      " chr1\t2227532\t2227533\tchr1\t2227002\t2227695\tENSG00000116151\n",
      " chr1\t2227532\t2227533\tchr1\t2227002\t2227695\tENSG00000157916\n",
      " chr1\t5896825\t5896826\t.\t-1\t-1\t.\n",
      " chr1\t7041937\t7041938\t.\t-1\t-1\t.\n",
      " chr1\t16912415\t16912416\tchr1\t16912238\t16915295\tENSG00000206652\n",
      " chr1\t16913967\t16913968\tchr1\t16912238\t16915295\tENSG00000206652\n",
      " chr1\t16913969\t16913970\tchr1\t16912238\t16915295\tENSG00000206652\n",
      " chr1\t21254093\t21254094\tchr1\t21253526\t21255567\tENSG00000076864\n",
      " chr1\t21264260\t21264261\tchr1\t21260958\t21264403\tENSG00000076864\n",
      " "
     ]
    }
   ],
   "source": [
    "#intersect variants with contacts\n",
    "enhancer_snps_contacts_bedtool = enhancer_snps_bedtool.intersect(contacts_bedtool, wa=True, wb=True, loj=True)\n",
    "enhancer_snps_contacts_bedtool.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000116151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS               ID\n",
       "0  chr1  2227533  ENSG00000157933\n",
       "1  chr1  2227533  ENSG00000116151\n",
       "2  chr1  2227533  ENSG00000157916\n",
       "3  chr1  5896826                .\n",
       "4  chr1  7041938                ."
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancer_snps_contacts_df = enhancer_snps_contacts_bedtool.to_dataframe(usecols = [0,2,6], names = [\"CHROM\", \"POS\", \"ID\"])\n",
    "enhancer_snps_contacts_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "      <td>11869</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "      <td>29570</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "      <td>29554</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "      <td>30366</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end strand  \\\n",
       "0  chr1  11869  14409      +   \n",
       "1  chr1  14404  29570      -   \n",
       "2  chr1  17369  17436      -   \n",
       "3  chr1  29554  31109      +   \n",
       "4  chr1  30366  30503      +   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "\n",
       "                           Gene    tss                        gene_biotype  \n",
       "0       ENSG00000223972:DDX11L1  11869  transcribed_unprocessed_pseudogene  \n",
       "1        ENSG00000227232:WASH7P  29570              unprocessed_pseudogene  \n",
       "2     ENSG00000278267:MIR6859-1  17436                               miRNA  \n",
       "3  ENSG00000243485:RP11-34P13.3  29554                             lincRNA  \n",
       "4     ENSG00000274890:MIR1302-2  30366                               miRNA  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.2287e+06</td>\n",
       "      <td>2.31012e+06</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000157933\"; gene_version \"9\"; g...</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>2.2287e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000116151</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.32125e+06</td>\n",
       "      <td>2.39171e+06</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000116151\"; gene_version \"13\"; ...</td>\n",
       "      <td>ENSG00000116151:MORN1</td>\n",
       "      <td>2.39171e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157916</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.39178e+06</td>\n",
       "      <td>2.40544e+06</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000157916\"; gene_version \"18\"; ...</td>\n",
       "      <td>ENSG00000157916:RER1</td>\n",
       "      <td>2.39178e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS               ID   chr        start          end strand  \\\n",
       "0  chr1  2227533  ENSG00000157933  chr1   2.2287e+06  2.31012e+06      +   \n",
       "1  chr1  2227533  ENSG00000116151  chr1  2.32125e+06  2.39171e+06      -   \n",
       "2  chr1  2227533  ENSG00000157916  chr1  2.39178e+06  2.40544e+06      +   \n",
       "3  chr1  5896826                .     .            .            .      .   \n",
       "4  chr1  7041938                .     .            .            .      .   \n",
       "\n",
       "                                                info                   Gene  \\\n",
       "0  gene_id \"ENSG00000157933\"; gene_version \"9\"; g...    ENSG00000157933:SKI   \n",
       "1  gene_id \"ENSG00000116151\"; gene_version \"13\"; ...  ENSG00000116151:MORN1   \n",
       "2  gene_id \"ENSG00000157916\"; gene_version \"18\"; ...   ENSG00000157916:RER1   \n",
       "3                                                  .                      .   \n",
       "4                                                  .                      .   \n",
       "\n",
       "           tss    gene_biotype  \n",
       "0   2.2287e+06  protein_coding  \n",
       "1  2.39171e+06  protein_coding  \n",
       "2  2.39178e+06  protein_coding  \n",
       "3            .               .  \n",
       "4            .               .  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancer_snps_contacts_gene_df = pd.merge(enhancer_snps_contacts_df, genes_info, on = 'ID', how = \"left\").fillna('.')\n",
    "enhancer_snps_contacts_gene_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>chr6</td>\n",
       "      <td>150419419</td>\n",
       "      <td>ENSG00000120278</td>\n",
       "      <td>chr6</td>\n",
       "      <td>1.506e+08</td>\n",
       "      <td>1.50844e+08</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000120278\"; gene_version \"14\"; ...</td>\n",
       "      <td>ENSG00000120278:PLEKHG1</td>\n",
       "      <td>1.506e+08</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>chr6</td>\n",
       "      <td>150419419</td>\n",
       "      <td>ENSG00000120254</td>\n",
       "      <td>chr6</td>\n",
       "      <td>1.50866e+08</td>\n",
       "      <td>1.51102e+08</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000120254\"; gene_version \"15\"; ...</td>\n",
       "      <td>ENSG00000120254:MTHFD1L</td>\n",
       "      <td>1.50866e+08</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS               ID   chr        start          end strand  \\\n",
       "277  chr6  150419419  ENSG00000120278  chr6    1.506e+08  1.50844e+08      +   \n",
       "278  chr6  150419419  ENSG00000120254  chr6  1.50866e+08  1.51102e+08      +   \n",
       "\n",
       "                                                  info  \\\n",
       "277  gene_id \"ENSG00000120278\"; gene_version \"14\"; ...   \n",
       "278  gene_id \"ENSG00000120254\"; gene_version \"15\"; ...   \n",
       "\n",
       "                        Gene          tss    gene_biotype  \n",
       "277  ENSG00000120278:PLEKHG1    1.506e+08  protein_coding  \n",
       "278  ENSG00000120254:MTHFD1L  1.50866e+08  protein_coding  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancer_snps_contacts_gene_df[enhancer_snps_contacts_gene_df[\"POS\"]==150419419]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0           0.000100           0.000300   \n",
       "1           0.000700           0.000700   \n",
       "2           0.000032           0.000067   \n",
       "3           0.000097           0.000067   \n",
       "4           0.004400           0.007600   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "2                        ...                           \n",
       "3                        ...                           \n",
       "4                        ...                           \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  AHR_HUMAN.H11MO.0.B, E4F1_HUMAN.H11MO.0.D, CR3...   \n",
       "1  OVOL1_HUMAN.H11MO.0.C, HIC2_HUMAN.H11MO.0.D, N...   \n",
       "2  PRRX1_HUMAN.H11MO.0.D, OTX1_HUMAN.H11MO.0.D, R...   \n",
       "3  MAFF_HUMAN.H11MO.1.B, E2F5_HUMAN.H11MO.0.B, E2...   \n",
       "4  TGIF1_HUMAN.H11MO.0.A, NKX28_HUMAN.H11MO.0.C, ...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          4.41;1.32   \n",
       "1                          4.48;1.41   \n",
       "2                          4.16;1.45   \n",
       "3                          4.15;1.47   \n",
       "4                          4.75;1.76   \n",
       "\n",
       "                                              ENCODE                    Gene  \\\n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...                       .   \n",
       "1                                                      ENSG00000131697:NPHP4   \n",
       "2                                                     ENSG00000171735:CAMTA1   \n",
       "3                                               CTCF   ENSG00000058453:CROCC   \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   ENSG00000058453:CROCC   \n",
       "\n",
       "       genomic element                   Closest gene dist. to closest gene  \\\n",
       "0  enhancer intergenic            ENSG00000157933:SKI                  1000   \n",
       "1    enhancer intronic     ENSG00000266687:AL356261.1                  1853   \n",
       "2    enhancer intronic  ENSG00000237365:RP11-334N17.1                 26411   \n",
       "3    enhancer intronic   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "4    enhancer intronic   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "\n",
       "     Closest protein gene dist. to closest protein gene  \\\n",
       "0     ENSG00000157933:SKI                          1000   \n",
       "1  ENSG00000069424:KCNAB2                         93950   \n",
       "2  ENSG00000171735:CAMTA1                        255366   \n",
       "3   ENSG00000117122:MFAP2                         65540   \n",
       "4   ENSG00000117122:MFAP2                         65540   \n",
       "\n",
       "                                     Contacting gene  \n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...  \n",
       "1                                                  .  \n",
       "2                                                  .  \n",
       "3                             ENSG00000206652:RNU1-1  \n",
       "4                             ENSG00000206652:RNU1-1  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest[\"Contacting gene\"] = enhancers_motifs_encode_closest.apply(add_gene, args=(enhancer_snps_contacts_gene_df, ), axis=1)\n",
    "enhancers_motifs_encode_closest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>chrX</td>\n",
       "      <td>56805338</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0,62</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs773102002</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>E4F1_HUMAN.H11MO.0.D, MEIS3_HUMAN.H11MO.0.D, T...</td>\n",
       "      <td>4.11;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000204272:LINC01420</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000237748:UQCRBP1</td>\n",
       "      <td>65192</td>\n",
       "      <td>ENSG00000204272:LINC01420</td>\n",
       "      <td>73510</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>chrX</td>\n",
       "      <td>103693042</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0,56</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888867223</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>TGIF1_HUMAN.H11MO.0.A, OVOL2_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.95;2.11</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000123562:MORF4L2</td>\n",
       "      <td>4235</td>\n",
       "      <td>ENSG00000123562:MORF4L2</td>\n",
       "      <td>4235</td>\n",
       "      <td>ENSG00000123570:RAB9B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>chrX</td>\n",
       "      <td>129932525</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0,2;2,3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.075</td>\n",
       "      <td>rs1014404153</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>...</td>\n",
       "      <td>GCM2_HUMAN.H11MO.0.D, NR2C2_HUMAN.H11MO.0.B, P...</td>\n",
       "      <td>5.59;1.78</td>\n",
       "      <td>RBBP5, MAZ, E2F6, PHF8, POLR2A, E2F1, CHD1</td>\n",
       "      <td>ENSG00000235189:RP4-537K23.4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000235189:RP4-537K23.4</td>\n",
       "      <td>24078</td>\n",
       "      <td>ENSG00000156697:UTP14A</td>\n",
       "      <td>24753</td>\n",
       "      <td>ENSG00000235189:RP4-537K23.4, ENSG00000085185:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>chrX</td>\n",
       "      <td>131753299</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0,27</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs111850126</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>...</td>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D, HXC11_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.27;1.85</td>\n",
       "      <td>ATF3, RAD21, ZBTB33, MAZ, YY1, CTCF, SP1, ETS1...</td>\n",
       "      <td>ENSG00000213468:FIRRE</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000241418:FAM195CP</td>\n",
       "      <td>42842</td>\n",
       "      <td>ENSG00000147255:IGSF1</td>\n",
       "      <td>173686</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153380834</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0,23;8,14</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs947743537</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>...</td>\n",
       "      <td>AP2B_HUMAN.H11MO.0.B, AP2C_HUMAN.H11MO.0.A</td>\n",
       "      <td>3.21;1.36</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000214897:U82695.5</td>\n",
       "      <td>18453</td>\n",
       "      <td>ENSG00000214897:U82695.5</td>\n",
       "      <td>18453</td>\n",
       "      <td>ENSG00000183837:PNMA3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS REF ALT       VAFs   AC     AF      avsnp150  \\\n",
       "346  chrX   56805338   A   G       0,62  2.0  0.043   rs773102002   \n",
       "347  chrX  103693042   A   G       0,56  2.0  0.043   rs888867223   \n",
       "348  chrX  129932525   G   A    0,2;2,3  3.0  0.075  rs1014404153   \n",
       "349  chrX  131753299   T   A       0,27  2.0  0.043   rs111850126   \n",
       "350  chrX  153380834   C   T  0,23;8,14  3.0  0.065   rs947743537   \n",
       "\n",
       "     gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "346           0.000400             0.0006   \n",
       "347           0.000093             0.0002   \n",
       "348           0.000700             0.0013   \n",
       "349           0.000400             0.0007   \n",
       "350           0.000500             0.0008   \n",
       "\n",
       "                           ...                          \\\n",
       "346                        ...                           \n",
       "347                        ...                           \n",
       "348                        ...                           \n",
       "349                        ...                           \n",
       "350                        ...                           \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "346  E4F1_HUMAN.H11MO.0.D, MEIS3_HUMAN.H11MO.0.D, T...   \n",
       "347  TGIF1_HUMAN.H11MO.0.A, OVOL2_HUMAN.H11MO.0.D, ...   \n",
       "348  GCM2_HUMAN.H11MO.0.D, NR2C2_HUMAN.H11MO.0.B, P...   \n",
       "349  CXXC1_HUMAN.H11MO.0.D, HXC11_HUMAN.H11MO.0.D, ...   \n",
       "350         AP2B_HUMAN.H11MO.0.B, AP2C_HUMAN.H11MO.0.A   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "346                          4.11;1.45   \n",
       "347                          3.95;2.11   \n",
       "348                          5.59;1.78   \n",
       "349                          3.27;1.85   \n",
       "350                          3.21;1.36   \n",
       "\n",
       "                                                ENCODE  \\\n",
       "346                                                      \n",
       "347                                                      \n",
       "348         RBBP5, MAZ, E2F6, PHF8, POLR2A, E2F1, CHD1   \n",
       "349  ATF3, RAD21, ZBTB33, MAZ, YY1, CTCF, SP1, ETS1...   \n",
       "350                                                      \n",
       "\n",
       "                             Gene      genomic element  \\\n",
       "346     ENSG00000204272:LINC01420    enhancer intronic   \n",
       "347                             .  enhancer intergenic   \n",
       "348  ENSG00000235189:RP4-537K23.4    enhancer intronic   \n",
       "349         ENSG00000213468:FIRRE    enhancer intronic   \n",
       "350                             .  enhancer intergenic   \n",
       "\n",
       "                     Closest gene dist. to closest gene  \\\n",
       "346       ENSG00000237748:UQCRBP1                 65192   \n",
       "347       ENSG00000123562:MORF4L2                  4235   \n",
       "348  ENSG00000235189:RP4-537K23.4                 24078   \n",
       "349      ENSG00000241418:FAM195CP                 42842   \n",
       "350      ENSG00000214897:U82695.5                 18453   \n",
       "\n",
       "          Closest protein gene dist. to closest protein gene  \\\n",
       "346  ENSG00000204272:LINC01420                         73510   \n",
       "347    ENSG00000123562:MORF4L2                          4235   \n",
       "348     ENSG00000156697:UTP14A                         24753   \n",
       "349      ENSG00000147255:IGSF1                        173686   \n",
       "350   ENSG00000214897:U82695.5                         18453   \n",
       "\n",
       "                                       Contacting gene  \n",
       "346                                                  .  \n",
       "347                              ENSG00000123570:RAB9B  \n",
       "348  ENSG00000235189:RP4-537K23.4, ENSG00000085185:...  \n",
       "349                                                  .  \n",
       "350                              ENSG00000183837:PNMA3  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>chr7</td>\n",
       "      <td>70093290</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>19,14;67,66</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs569183497</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>...</td>\n",
       "      <td>MYF6_HUMAN.H11MO.0.C, ASCL2_HUMAN.H11MO.0.D, O...</td>\n",
       "      <td>10.00;7.40</td>\n",
       "      <td>EP300</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "      <td>493419</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "      <td>493419</td>\n",
       "      <td>ENSG00000158321:AUTS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr6</td>\n",
       "      <td>90885089</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>13,26;21,19</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs747544862</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>...</td>\n",
       "      <td>KLF4_HUMAN.H11MO.0.A, GLI3_HUMAN.H11MO.0.B</td>\n",
       "      <td>3.49;1.12</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000135341:MAP3K7</td>\n",
       "      <td>297631</td>\n",
       "      <td>ENSG00000135341:MAP3K7</td>\n",
       "      <td>297631</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>chr6</td>\n",
       "      <td>82710722</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>32,23;26,30</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs186524540</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002100</td>\n",
       "      <td>...</td>\n",
       "      <td>MSX2_HUMAN.H11MO.0.D, NF2L1_HUMAN.H11MO.0.C, H...</td>\n",
       "      <td>4.15;1.58</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000220091:LAP3P1</td>\n",
       "      <td>212088</td>\n",
       "      <td>ENSG00000146242:TPBG</td>\n",
       "      <td>347157</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>chr9</td>\n",
       "      <td>77848708</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>83,73;40,36;87,72</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs182563279</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.002300</td>\n",
       "      <td>...</td>\n",
       "      <td>NFYB_HUMAN.H11MO.0.A, RUNX2_HUMAN.H11MO.0.A</td>\n",
       "      <td>4.34;1.14</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000156052:GNAQ</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000156052:GNAQ</td>\n",
       "      <td>181239</td>\n",
       "      <td>ENSG00000156052:GNAQ</td>\n",
       "      <td>181239</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chr12</td>\n",
       "      <td>11828474</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>22,25;47,35;121,106</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs140107009</td>\n",
       "      <td>0.004900</td>\n",
       "      <td>0.008000</td>\n",
       "      <td>...</td>\n",
       "      <td>KAISO_HUMAN.H11MO.1.A, SRBP1_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>3.72;1.60</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000139083:ETV6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000139083:ETV6</td>\n",
       "      <td>176520</td>\n",
       "      <td>ENSG00000139083:ETV6</td>\n",
       "      <td>176520</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>chr5</td>\n",
       "      <td>107973351</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>6,2;7,2;12,6;9,3;20,9</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.114</td>\n",
       "      <td>rs963187462</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0.002200</td>\n",
       "      <td>...</td>\n",
       "      <td>SRY_HUMAN.H11MO.0.B, BARX2_HUMAN.H11MO.0.D, IR...</td>\n",
       "      <td>5.37;2.25</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000145743:FBXL17</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000251732:RN7SKP122</td>\n",
       "      <td>160828</td>\n",
       "      <td>ENSG00000184349:EFNA5</td>\n",
       "      <td>300562</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35979734</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>28,24;18,13</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs777315289</td>\n",
       "      <td>0.001600</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>...</td>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.82;1.54</td>\n",
       "      <td>JUN, FOS, MYC</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>150095</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>150095</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35842056</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40,21;32,41;22,8</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs181047757</td>\n",
       "      <td>0.004000</td>\n",
       "      <td>0.006400</td>\n",
       "      <td>...</td>\n",
       "      <td>NFAC3_HUMAN.H11MO.0.B, STAT4_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>4.44;1.43</td>\n",
       "      <td>REST</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236036:LINC00445</td>\n",
       "      <td>141923</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>289015</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>chr13</td>\n",
       "      <td>35841849</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>61,55;44,28</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs142256254</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>...</td>\n",
       "      <td>DPRX_HUMAN.H11MO.0.D, BHE23_HUMAN.H11MO.0.D</td>\n",
       "      <td>3.30;1.55</td>\n",
       "      <td>REST</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236036:LINC00445</td>\n",
       "      <td>141923</td>\n",
       "      <td>ENSG00000133083:DCLK1</td>\n",
       "      <td>289015</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>chr5</td>\n",
       "      <td>15795255</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>42,51;108,76;53,56</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs576805372</td>\n",
       "      <td>0.003500</td>\n",
       "      <td>0.005700</td>\n",
       "      <td>...</td>\n",
       "      <td>E4F1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, H...</td>\n",
       "      <td>4.82;1.42</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000183580:FBXL7</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000216077:MIR887</td>\n",
       "      <td>139082</td>\n",
       "      <td>ENSG00000183580:FBXL7</td>\n",
       "      <td>293506</td>\n",
       "      <td>ENSG00000183580:FBXL7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>chr2</td>\n",
       "      <td>105605799</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0,5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.045</td>\n",
       "      <td>rs557383407</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>ZN708_HUMAN.H11MO.1.D</td>\n",
       "      <td>3.37;1.20</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000071051:NCK2</td>\n",
       "      <td>138671</td>\n",
       "      <td>ENSG00000071051:NCK2</td>\n",
       "      <td>138671</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>chr10</td>\n",
       "      <td>78864401</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>139,120;123,127</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs1034121075</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>...</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, HTF4_HUMAN.H11MO.0.A, T...</td>\n",
       "      <td>4.87;1.28</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000229569:RP11-481G8.2</td>\n",
       "      <td>136510</td>\n",
       "      <td>ENSG00000108175:ZMIZ1</td>\n",
       "      <td>201742</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>chr11</td>\n",
       "      <td>120744299</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>131,109;102,105</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs140746993</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>...</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, TYY1_HUMAN.H11MO.0.A, T...</td>\n",
       "      <td>3.18;1.42</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000149403:GRIK4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000255006:TCEB1P22</td>\n",
       "      <td>123536</td>\n",
       "      <td>ENSG00000149403:GRIK4</td>\n",
       "      <td>232044</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>chr13</td>\n",
       "      <td>94312722</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>85,9;74,8;83,47;48,50</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs72647627</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>...</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C, BHE40_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>4.72;1.81</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000183098:GPC6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236520:GPC6-AS1</td>\n",
       "      <td>123509</td>\n",
       "      <td>ENSG00000080166:DCT</td>\n",
       "      <td>165738</td>\n",
       "      <td>ENSG00000282993:RP11-118A1.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>chr7</td>\n",
       "      <td>154743157</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>94,92;95,89;18,24</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs766837968</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.000900</td>\n",
       "      <td>...</td>\n",
       "      <td>MAFF_HUMAN.H11MO.1.B, ZN554_HUMAN.H11MO.1.D, Z...</td>\n",
       "      <td>5.74;1.74</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000130226:DPP6</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000236408:RP11-476H24.1</td>\n",
       "      <td>121132</td>\n",
       "      <td>ENSG00000157212:PAXIP1</td>\n",
       "      <td>258733</td>\n",
       "      <td>ENSG00000130226:DPP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>chr14</td>\n",
       "      <td>67943009</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>85,75;57,57</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs747732406</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>...</td>\n",
       "      <td>OVOL2_HUMAN.H11MO.0.D, NR4A2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>3.48;1.33</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000182185:RAD51B</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000182185:RAD51B</td>\n",
       "      <td>120301</td>\n",
       "      <td>ENSG00000182185:RAD51B</td>\n",
       "      <td>120301</td>\n",
       "      <td>ENSG00000185650:ZFP36L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>chr11</td>\n",
       "      <td>71138813</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11,2;7,2;27,3;8,4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>...</td>\n",
       "      <td>FOXP1_HUMAN.H11MO.0.A, GATA3_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>5.13;1.69</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000162105:SHANK2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000162105:SHANK2</td>\n",
       "      <td>111174</td>\n",
       "      <td>ENSG00000162105:SHANK2</td>\n",
       "      <td>111174</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>chr7</td>\n",
       "      <td>50676547</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>49,52;54,50</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs956599362</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000106070:GRB10</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000132437:DDC</td>\n",
       "      <td>110988</td>\n",
       "      <td>ENSG00000132437:DDC</td>\n",
       "      <td>110988</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>chr5</td>\n",
       "      <td>78534216</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>60,44;56,68</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs545034134</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>ZN784_HUMAN.H11MO.0.D, RXRG_HUMAN.H11MO.0.B, T...</td>\n",
       "      <td>3.90;1.27</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000145685:LHFPL2</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000248909:HMGB1P21</td>\n",
       "      <td>108674</td>\n",
       "      <td>ENSG00000085365:SCAMP1</td>\n",
       "      <td>173182</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>chr3</td>\n",
       "      <td>131217019</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>15,12;60,28</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs572945195</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>ESR2_HUMAN.H11MO.1.A, HIC1_HUMAN.H11MO.0.C, NR...</td>\n",
       "      <td>4.36;1.85</td>\n",
       "      <td>NR3C1, CREB1, FOXA1, ATF3, JUN, RAD21, FOSL2, ...</td>\n",
       "      <td>ENSG00000114670:NEK11</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000212073:AC121332.1</td>\n",
       "      <td>105161</td>\n",
       "      <td>ENSG00000198585:NUDT16</td>\n",
       "      <td>162905</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                   VAFs   AC     AF  \\\n",
       "149   chr7   70093290   G   A            19,14;67,66  2.0  0.043   \n",
       "130   chr6   90885089   C   G            13,26;21,19  2.0  0.043   \n",
       "129   chr6   82710722   T   G            32,23;26,30  2.0  0.043   \n",
       "176   chr9   77848708   T   C      83,73;40,36;87,72  3.0  0.065   \n",
       "227  chr12   11828474   G   A    22,25;47,35;121,106  3.0  0.065   \n",
       "109   chr5  107973351   G   T  6,2;7,2;12,6;9,3;20,9  5.0  0.114   \n",
       "250  chr13   35979734   A   C            28,24;18,13  2.0  0.043   \n",
       "249  chr13   35842056   C   T       40,21;32,41;22,8  3.0  0.065   \n",
       "248  chr13   35841849   C   T            61,55;44,28  2.0  0.043   \n",
       "95    chr5   15795255   G   C     42,51;108,76;53,56  3.0  0.065   \n",
       "61    chr2  105605799   C   T                    0,5  2.0  0.045   \n",
       "196  chr10   78864401   T   G        139,120;123,127  2.0  0.043   \n",
       "222  chr11  120744299   G   C        131,109;102,105  2.0  0.043   \n",
       "255  chr13   94312722   A   G  85,9;74,8;83,47;48,50  4.0  0.087   \n",
       "159   chr7  154743157   G   A      94,92;95,89;18,24  3.0  0.065   \n",
       "265  chr14   67943009   A   G            85,75;57,57  2.0  0.043   \n",
       "217  chr11   71138813   G   A      11,2;7,2;27,3;8,4  4.0  0.087   \n",
       "146   chr7   50676547   T   C            49,52;54,50  2.0  0.043   \n",
       "105   chr5   78534216   G   A            60,44;56,68  2.0  0.043   \n",
       "83    chr3  131217019   G   C            15,12;60,28  2.0  0.043   \n",
       "\n",
       "         avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "149   rs569183497           0.000700           0.001300   \n",
       "130   rs747544862           0.000300           0.000600   \n",
       "129   rs186524540           0.001300           0.002100   \n",
       "176   rs182563279           0.001300           0.002300   \n",
       "227   rs140107009           0.004900           0.008000   \n",
       "109   rs963187462           0.001500           0.002200   \n",
       "250   rs777315289           0.001600           0.001500   \n",
       "249   rs181047757           0.004000           0.006400   \n",
       "248   rs142256254           0.000700           0.001200   \n",
       "95    rs576805372           0.003500           0.005700   \n",
       "61    rs557383407           0.000032           0.000067   \n",
       "196  rs1034121075           0.000600           0.001100   \n",
       "222   rs140746993           0.000800           0.001300   \n",
       "255    rs72647627           0.001800           0.001500   \n",
       "159   rs766837968           0.000600           0.000900   \n",
       "265   rs747732406           0.000500           0.000800   \n",
       "217             .           0.000073           0.000100   \n",
       "146   rs956599362           0.000500           0.000700   \n",
       "105   rs545034134           0.000032           0.000000   \n",
       "83    rs572945195           0.000500           0.000700   \n",
       "\n",
       "                 ...               \\\n",
       "149              ...                \n",
       "130              ...                \n",
       "129              ...                \n",
       "176              ...                \n",
       "227              ...                \n",
       "109              ...                \n",
       "250              ...                \n",
       "249              ...                \n",
       "248              ...                \n",
       "95               ...                \n",
       "61               ...                \n",
       "196              ...                \n",
       "222              ...                \n",
       "255              ...                \n",
       "159              ...                \n",
       "265              ...                \n",
       "217              ...                \n",
       "146              ...                \n",
       "105              ...                \n",
       "83               ...                \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "149  MYF6_HUMAN.H11MO.0.C, ASCL2_HUMAN.H11MO.0.D, O...   \n",
       "130         KLF4_HUMAN.H11MO.0.A, GLI3_HUMAN.H11MO.0.B   \n",
       "129  MSX2_HUMAN.H11MO.0.D, NF2L1_HUMAN.H11MO.0.C, H...   \n",
       "176        NFYB_HUMAN.H11MO.0.A, RUNX2_HUMAN.H11MO.0.A   \n",
       "227  KAISO_HUMAN.H11MO.1.A, SRBP1_HUMAN.H11MO.0.A, ...   \n",
       "109  SRY_HUMAN.H11MO.0.B, BARX2_HUMAN.H11MO.0.D, IR...   \n",
       "250  CXXC1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, ...   \n",
       "249  NFAC3_HUMAN.H11MO.0.B, STAT4_HUMAN.H11MO.0.A, ...   \n",
       "248        DPRX_HUMAN.H11MO.0.D, BHE23_HUMAN.H11MO.0.D   \n",
       "95   E4F1_HUMAN.H11MO.0.D, GMEB2_HUMAN.H11MO.0.D, H...   \n",
       "61                               ZN708_HUMAN.H11MO.1.D   \n",
       "196  ZN335_HUMAN.H11MO.1.A, HTF4_HUMAN.H11MO.0.A, T...   \n",
       "222  ZN335_HUMAN.H11MO.1.A, TYY1_HUMAN.H11MO.0.A, T...   \n",
       "255  ZBT14_HUMAN.H11MO.0.C, BHE40_HUMAN.H11MO.0.A, ...   \n",
       "159  MAFF_HUMAN.H11MO.1.B, ZN554_HUMAN.H11MO.1.D, Z...   \n",
       "265  OVOL2_HUMAN.H11MO.0.D, NR4A2_HUMAN.H11MO.0.C, ...   \n",
       "217  FOXP1_HUMAN.H11MO.0.A, GATA3_HUMAN.H11MO.0.A, ...   \n",
       "146                                                      \n",
       "105  ZN784_HUMAN.H11MO.0.D, RXRG_HUMAN.H11MO.0.B, T...   \n",
       "83   ESR2_HUMAN.H11MO.1.A, HIC1_HUMAN.H11MO.0.C, NR...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "149                         10.00;7.40   \n",
       "130                          3.49;1.12   \n",
       "129                          4.15;1.58   \n",
       "176                          4.34;1.14   \n",
       "227                          3.72;1.60   \n",
       "109                          5.37;2.25   \n",
       "250                          3.82;1.54   \n",
       "249                          4.44;1.43   \n",
       "248                          3.30;1.55   \n",
       "95                           4.82;1.42   \n",
       "61                           3.37;1.20   \n",
       "196                          4.87;1.28   \n",
       "222                          3.18;1.42   \n",
       "255                          4.72;1.81   \n",
       "159                          5.74;1.74   \n",
       "265                          3.48;1.33   \n",
       "217                          5.13;1.69   \n",
       "146                                      \n",
       "105                          3.90;1.27   \n",
       "83                           4.36;1.85   \n",
       "\n",
       "                                                ENCODE  \\\n",
       "149                                              EP300   \n",
       "130                                                      \n",
       "129                                                      \n",
       "176                                                      \n",
       "227                                                      \n",
       "109                                                      \n",
       "250                                      JUN, FOS, MYC   \n",
       "249                                               REST   \n",
       "248                                               REST   \n",
       "95                                                       \n",
       "61                                                       \n",
       "196                                                      \n",
       "222                                                      \n",
       "255                                                      \n",
       "159                                                      \n",
       "265                                                      \n",
       "217                                                      \n",
       "146                                                      \n",
       "105                                                      \n",
       "83   NR3C1, CREB1, FOXA1, ATF3, JUN, RAD21, FOSL2, ...   \n",
       "\n",
       "                       Gene      genomic element  \\\n",
       "149   ENSG00000158321:AUTS2    enhancer intronic   \n",
       "130                       .  enhancer intergenic   \n",
       "129                       .  enhancer intergenic   \n",
       "176    ENSG00000156052:GNAQ    enhancer intronic   \n",
       "227    ENSG00000139083:ETV6    enhancer intronic   \n",
       "109  ENSG00000145743:FBXL17    enhancer intronic   \n",
       "250   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "249   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "248   ENSG00000133083:DCLK1    enhancer intronic   \n",
       "95    ENSG00000183580:FBXL7    enhancer intronic   \n",
       "61                        .  enhancer intergenic   \n",
       "196                       .  enhancer intergenic   \n",
       "222   ENSG00000149403:GRIK4    enhancer intronic   \n",
       "255    ENSG00000183098:GPC6    enhancer intronic   \n",
       "159    ENSG00000130226:DPP6    enhancer intronic   \n",
       "265  ENSG00000182185:RAD51B    enhancer intronic   \n",
       "217  ENSG00000162105:SHANK2    enhancer intronic   \n",
       "146   ENSG00000106070:GRB10    enhancer intronic   \n",
       "105  ENSG00000145685:LHFPL2    enhancer intronic   \n",
       "83    ENSG00000114670:NEK11    enhancer intronic   \n",
       "\n",
       "                      Closest gene dist. to closest gene  \\\n",
       "149          ENSG00000158321:AUTS2                493419   \n",
       "130         ENSG00000135341:MAP3K7                297631   \n",
       "129         ENSG00000220091:LAP3P1                212088   \n",
       "176           ENSG00000156052:GNAQ                181239   \n",
       "227           ENSG00000139083:ETV6                176520   \n",
       "109      ENSG00000251732:RN7SKP122                160828   \n",
       "250          ENSG00000133083:DCLK1                150095   \n",
       "249      ENSG00000236036:LINC00445                141923   \n",
       "248      ENSG00000236036:LINC00445                141923   \n",
       "95          ENSG00000216077:MIR887                139082   \n",
       "61            ENSG00000071051:NCK2                138671   \n",
       "196   ENSG00000229569:RP11-481G8.2                136510   \n",
       "222       ENSG00000255006:TCEB1P22                123536   \n",
       "255       ENSG00000236520:GPC6-AS1                123509   \n",
       "159  ENSG00000236408:RP11-476H24.1                121132   \n",
       "265         ENSG00000182185:RAD51B                120301   \n",
       "217         ENSG00000162105:SHANK2                111174   \n",
       "146            ENSG00000132437:DDC                110988   \n",
       "105       ENSG00000248909:HMGB1P21                108674   \n",
       "83      ENSG00000212073:AC121332.1                105161   \n",
       "\n",
       "       Closest protein gene dist. to closest protein gene  \\\n",
       "149   ENSG00000158321:AUTS2                        493419   \n",
       "130  ENSG00000135341:MAP3K7                        297631   \n",
       "129    ENSG00000146242:TPBG                        347157   \n",
       "176    ENSG00000156052:GNAQ                        181239   \n",
       "227    ENSG00000139083:ETV6                        176520   \n",
       "109   ENSG00000184349:EFNA5                        300562   \n",
       "250   ENSG00000133083:DCLK1                        150095   \n",
       "249   ENSG00000133083:DCLK1                        289015   \n",
       "248   ENSG00000133083:DCLK1                        289015   \n",
       "95    ENSG00000183580:FBXL7                        293506   \n",
       "61     ENSG00000071051:NCK2                        138671   \n",
       "196   ENSG00000108175:ZMIZ1                        201742   \n",
       "222   ENSG00000149403:GRIK4                        232044   \n",
       "255     ENSG00000080166:DCT                        165738   \n",
       "159  ENSG00000157212:PAXIP1                        258733   \n",
       "265  ENSG00000182185:RAD51B                        120301   \n",
       "217  ENSG00000162105:SHANK2                        111174   \n",
       "146     ENSG00000132437:DDC                        110988   \n",
       "105  ENSG00000085365:SCAMP1                        173182   \n",
       "83   ENSG00000198585:NUDT16                        162905   \n",
       "\n",
       "                  Contacting gene  \n",
       "149         ENSG00000158321:AUTS2  \n",
       "130                             .  \n",
       "129                             .  \n",
       "176                             .  \n",
       "227                             .  \n",
       "109                             .  \n",
       "250         ENSG00000133083:DCLK1  \n",
       "249                             .  \n",
       "248                             .  \n",
       "95          ENSG00000183580:FBXL7  \n",
       "61                              .  \n",
       "196                             .  \n",
       "222                             .  \n",
       "255  ENSG00000282993:RP11-118A1.2  \n",
       "159          ENSG00000130226:DPP6  \n",
       "265       ENSG00000185650:ZFP36L1  \n",
       "217                             .  \n",
       "146                             .  \n",
       "105                             .  \n",
       "83                              .  \n",
       "\n",
       "[20 rows x 21 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest.sort_values(by = \"dist. to closest gene\", ascending=False)[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "351"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_motifs_encode_closest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Protein coding genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ENST</th>\n",
       "      <th>ENSG</th>\n",
       "      <th>-log10(qval)</th>\n",
       "      <th>confirmed_both_ways</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>ENSG00000187608</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end             ENST             ENSG  -log10(qval)  \\\n",
       "2  chr1  1024210  1025994  ENST00000484667  ENSG00000188290      8.697366   \n",
       "3  chr1  1024210  1025994  ENST00000481869  ENSG00000188290      8.697366   \n",
       "4  chr1  1024210  1025994  ENST00000304952  ENSG00000188290      8.697366   \n",
       "5  chr1  1024210  1025994  ENST00000428771  ENSG00000188290      8.697366   \n",
       "6  chr1  1024210  1025994  ENST00000624697  ENSG00000187608      8.697366   \n",
       "\n",
       "   confirmed_both_ways  \n",
       "2                    1  \n",
       "3                    1  \n",
       "4                    1  \n",
       "5                    1  \n",
       "6                    1  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "contacts_to_protein_genes = contacts[contacts[\"ENSG\"].isin(protein_coding_genes[\"ID\"])]\n",
    "contacts_to_protein_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000116151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS               ID\n",
       "0  chr1  2227533  ENSG00000157933\n",
       "1  chr1  2227533  ENSG00000116151\n",
       "2  chr1  2227533  ENSG00000157916\n",
       "3  chr1  5896826                .\n",
       "4  chr1  7041938                ."
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "contacts_to_protein_genes_bedtool = pbt.BedTool.from_dataframe(contacts_to_protein_genes[[\"chr\", \"start\", \"end\", \"ENSG\"]].drop_duplicates())\n",
    "\n",
    "#intersect variants with contacts\n",
    "enhancer_snps_contacts_to_protein_genes_bedtool = enhancer_snps_bedtool.intersect(contacts_to_protein_genes_bedtool, wa=True, wb=True, loj=True)\n",
    "\n",
    "enhancer_snps_contacts_to_protein_genes_df = enhancer_snps_contacts_to_protein_genes_bedtool.to_dataframe(usecols = [0,2,6], names = [\"CHROM\", \"POS\", \"ID\"])\n",
    "enhancer_snps_contacts_to_protein_genes_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972:DDX11L1</td>\n",
       "      <td>11869</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232:WASH7P</td>\n",
       "      <td>29570</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267:MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485:RP11-34P13.3</td>\n",
       "      <td>29554</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890:MIR1302-2</td>\n",
       "      <td>30366</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  start    end strand  \\\n",
       "0  chr1  11869  14409      +   \n",
       "1  chr1  14404  29570      -   \n",
       "2  chr1  17369  17436      -   \n",
       "3  chr1  29554  31109      +   \n",
       "4  chr1  30366  30503      +   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "\n",
       "                           Gene    tss                        gene_biotype  \n",
       "0       ENSG00000223972:DDX11L1  11869  transcribed_unprocessed_pseudogene  \n",
       "1        ENSG00000227232:WASH7P  29570              unprocessed_pseudogene  \n",
       "2     ENSG00000278267:MIR6859-1  17436                               miRNA  \n",
       "3  ENSG00000243485:RP11-34P13.3  29554                             lincRNA  \n",
       "4     ENSG00000274890:MIR1302-2  30366                               miRNA  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "      <th>gene_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.2287e+06</td>\n",
       "      <td>2.31012e+06</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000157933\"; gene_version \"9\"; g...</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>2.2287e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000116151</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.32125e+06</td>\n",
       "      <td>2.39171e+06</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000116151\"; gene_version \"13\"; ...</td>\n",
       "      <td>ENSG00000116151:MORN1</td>\n",
       "      <td>2.39171e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>ENSG00000157916</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2.39178e+06</td>\n",
       "      <td>2.40544e+06</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000157916\"; gene_version \"18\"; ...</td>\n",
       "      <td>ENSG00000157916:RER1</td>\n",
       "      <td>2.39178e+06</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS               ID   chr        start          end strand  \\\n",
       "0  chr1  2227533  ENSG00000157933  chr1   2.2287e+06  2.31012e+06      +   \n",
       "1  chr1  2227533  ENSG00000116151  chr1  2.32125e+06  2.39171e+06      -   \n",
       "2  chr1  2227533  ENSG00000157916  chr1  2.39178e+06  2.40544e+06      +   \n",
       "3  chr1  5896826                .     .            .            .      .   \n",
       "4  chr1  7041938                .     .            .            .      .   \n",
       "\n",
       "                                                info                   Gene  \\\n",
       "0  gene_id \"ENSG00000157933\"; gene_version \"9\"; g...    ENSG00000157933:SKI   \n",
       "1  gene_id \"ENSG00000116151\"; gene_version \"13\"; ...  ENSG00000116151:MORN1   \n",
       "2  gene_id \"ENSG00000157916\"; gene_version \"18\"; ...   ENSG00000157916:RER1   \n",
       "3                                                  .                      .   \n",
       "4                                                  .                      .   \n",
       "\n",
       "           tss    gene_biotype  \n",
       "0   2.2287e+06  protein_coding  \n",
       "1  2.39171e+06  protein_coding  \n",
       "2  2.39178e+06  protein_coding  \n",
       "3            .               .  \n",
       "4            .               .  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancer_snps_contacts_protein_gene_df = pd.merge(enhancer_snps_contacts_to_protein_genes_df, protein_coding_genes, on = 'ID', how = \"left\").fillna('.')\n",
    "enhancer_snps_contacts_protein_gene_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>Contacting protein gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>4.41;1.32</td>\n",
       "      <td>NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>4.48;1.41</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000131697:NPHP4</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>4.16;1.45</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>4.15;1.47</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>4.75;1.76</td>\n",
       "      <td>PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...</td>\n",
       "      <td>ENSG00000058453:CROCC</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0           0.000100           0.000300   \n",
       "1           0.000700           0.000700   \n",
       "2           0.000032           0.000067   \n",
       "3           0.000097           0.000067   \n",
       "4           0.004400           0.007600   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "2                        ...                           \n",
       "3                        ...                           \n",
       "4                        ...                           \n",
       "\n",
       "   PWMEnrich best -log10pval and diff  \\\n",
       "0                           4.41;1.32   \n",
       "1                           4.48;1.41   \n",
       "2                           4.16;1.45   \n",
       "3                           4.15;1.47   \n",
       "4                           4.75;1.76   \n",
       "\n",
       "                                              ENCODE                    Gene  \\\n",
       "0  NR3C1, CREB1, SAP30, BACH1, RBBP5, UBTF, MAZ, ...                       .   \n",
       "1                                                      ENSG00000131697:NPHP4   \n",
       "2                                                     ENSG00000171735:CAMTA1   \n",
       "3                                               CTCF   ENSG00000058453:CROCC   \n",
       "4  PAX5, RBBP5, MAZ, E2F6, IRF1, POLR2A, HMGN3, E...   ENSG00000058453:CROCC   \n",
       "\n",
       "       genomic element                   Closest gene dist. to closest gene  \\\n",
       "0  enhancer intergenic            ENSG00000157933:SKI                  1000   \n",
       "1    enhancer intronic     ENSG00000266687:AL356261.1                  1853   \n",
       "2    enhancer intronic  ENSG00000237365:RP11-334N17.1                 26411   \n",
       "3    enhancer intronic   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "4    enhancer intronic   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "\n",
       "     Closest protein gene  dist. to closest protein gene  \\\n",
       "0     ENSG00000157933:SKI                           1000   \n",
       "1  ENSG00000069424:KCNAB2                          93950   \n",
       "2  ENSG00000171735:CAMTA1                         255366   \n",
       "3   ENSG00000117122:MFAP2                          65540   \n",
       "4   ENSG00000117122:MFAP2                          65540   \n",
       "\n",
       "                                     Contacting gene  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...   \n",
       "1                                                  .   \n",
       "2                                                  .   \n",
       "3                             ENSG00000206652:RNU1-1   \n",
       "4                             ENSG00000206652:RNU1-1   \n",
       "\n",
       "                             Contacting protein gene  \n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...  \n",
       "1                                                  .  \n",
       "2                                                  .  \n",
       "3                                                  .  \n",
       "4                                                  .  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_closest[\"Contacting protein gene\"] = enhancers_motifs_encode_closest.apply(add_gene, args=(enhancer_snps_contacts_protein_gene_df, ), axis=1)\n",
    "enhancers_motifs_encode_closest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "351"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_motifs_encode_closest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add POLGENOM info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Promoters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1105 ../2019-07-03-Variants_in_ENCODE_TFBS/out_files/promoters_tf_polgenom_gnomad001.csv\n"
     ]
    }
   ],
   "source": [
    "!wc -l ../2019-07-03-Variants_in_ENCODE_TFBS/out_files/promoters_tf_polgenom_gnomad001.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16613478</td>\n",
       "      <td>4.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25268534</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34930565</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37753675</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37753679</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS  polgenom_freq_perc\n",
       "0   chr1  16613478                4.76\n",
       "61  chr1  25268534                0.51\n",
       "62  chr1  34930565                 NaN\n",
       "63  chr1  37753675                 NaN\n",
       "64  chr1  37753679                 NaN"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#these snps have been checked by Marcelina; if freq is missing then it was not available in polgenom\n",
    "promoters_with_polgenom = pd.read_csv(\"../2019-07-03-Variants_in_ENCODE_TFBS/out_files/promoters_tf_polgenom_gnomad001.csv\", sep=\"\\t\",\n",
    "                                      usecols = [\"CHROM\", \"POS\", \"polgenom_freq_perc\"]).drop_duplicates()\n",
    "promoters_with_polgenom.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "154"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoters_with_polgenom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_polgenom(df, polgenom_df):\n",
    "    polgenom_df[\"polgenom_freq_perc\"] = polgenom_df.polgenom_freq_perc.apply(lambda x: \"unknown\" if str(x) == 'nan' else x)\n",
    "    return pd.merge(df, polgenom_df[[\"CHROM\", \"POS\", \"polgenom_freq_perc\"]], on = [\"CHROM\", \"POS\"], how = \"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters_motifs_encode_polgenom = add_polgenom(promoters_motifs_encode, promoters_with_polgenom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>chrX</td>\n",
       "      <td>15676222</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0,136;141,169;0,147</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.109</td>\n",
       "      <td>rs868393078</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0</td>\n",
       "      <td>IRX3_HUMAN.H11MO.0.D, RORG_HUMAN.H11MO.0.C, NR...</td>\n",
       "      <td>4.30;1.80</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>CA5BP1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90249253</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>44,5;23,17;37,38;39,31</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs751435350</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>0.0055</td>\n",
       "      <td>1</td>\n",
       "      <td>NKX22_HUMAN.H11MO.0.D, TBX20_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.89;1.52</td>\n",
       "      <td>ZNF143, NFIC, POLR2A, YY1, ELF1, CHD2, SP1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>TTLL13P, RP11-697E2.6</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>chr12</td>\n",
       "      <td>125983445</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>31,44;77,69;77,84;31,34</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs549358675</td>\n",
       "      <td>0.0050</td>\n",
       "      <td>0.0093</td>\n",
       "      <td>1</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, A...</td>\n",
       "      <td>5.23;1.70</td>\n",
       "      <td>BHLHE40, STAT3, CTCF, E2F1, TAF1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP5-916L7.2, LINC00939</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>chr3</td>\n",
       "      <td>10115323</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>32,37;29,36;62,60;40,51</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs548922348</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0040</td>\n",
       "      <td>1</td>\n",
       "      <td>BHE22_HUMAN.H11MO.0.D, OLIG1_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.85;1.31</td>\n",
       "      <td>REST, MXI1, SAP30, TBP, TRIM28, BCLAF1, TEAD4,...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>BRK1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>chr3</td>\n",
       "      <td>9946995</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>36,23;23,18;46,36;65,56</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs367612791</td>\n",
       "      <td>0.0026</td>\n",
       "      <td>0.0047</td>\n",
       "      <td>1</td>\n",
       "      <td>ZN384_HUMAN.H11MO.0.C, ZN257_HUMAN.H11MO.0.C</td>\n",
       "      <td>4.13;1.53</td>\n",
       "      <td>POLR2A, YY1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>PRRT3-AS1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25268534</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>13,11;0,7;15,7</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs148017678</td>\n",
       "      <td>0.0041</td>\n",
       "      <td>0.0069</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.37;1.50</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RSRP1</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>chr2</td>\n",
       "      <td>202635226</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4,5;0,6;5,2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs193178985</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>0.0085</td>\n",
       "      <td>3</td>\n",
       "      <td>MBD2_HUMAN.H11MO.0.B, SRBP2_HUMAN.H11MO.0.B, C...</td>\n",
       "      <td>3.91;1.38</td>\n",
       "      <td>NR3C1, RBBP5, HDAC1, TFAP2A, TAF7, PHF8, POLR2...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>FAM117B</td>\n",
       "      <td>1.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>chr6</td>\n",
       "      <td>144008499</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5,8;10,13;0,2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>rs775783341</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...</td>\n",
       "      <td>4.66;1.76</td>\n",
       "      <td>EGR1, RBBP5, EZH2, E2F6, POLR2A, KDM5B, YY1, U...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>PLAGL1, HYMAI</td>\n",
       "      <td>1.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>chr18</td>\n",
       "      <td>37565974</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4,5;6,1;0,4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs145522218</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>1</td>\n",
       "      <td>GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...</td>\n",
       "      <td>10.00;7.20</td>\n",
       "      <td>EGR1, RBBP5, EZH2, GATA2, CHD1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CELF4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41835260</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>16,4;1,3;0,10</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.125</td>\n",
       "      <td>rs185331159</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>MAF_HUMAN.H11MO.1.B, MAX_HUMAN.H11MO.0.A, SNAI...</td>\n",
       "      <td>3.86;1.38</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP5-821D11.7</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                     VAFs   AC     AF  \\\n",
       "415   chrX   15676222   G   C      0,136;141,169;0,147  5.0  0.109   \n",
       "279  chr15   90249253   A   G   44,5;23,17;37,38;39,31  4.0  0.087   \n",
       "243  chr12  125983445   C   T  31,44;77,69;77,84;31,34  4.0  0.087   \n",
       "75    chr3   10115323   A   G  32,37;29,36;62,60;40,51  4.0  0.087   \n",
       "74    chr3    9946995   C   T  36,23;23,18;46,36;65,56  4.0  0.087   \n",
       "5     chr1   25268534   A   G           13,11;0,7;15,7  4.0  0.105   \n",
       "69    chr2  202635226   G   A              4,5;0,6;5,2  4.0  0.087   \n",
       "126   chr6  144008499   G   A            5,8;10,13;0,2  4.0  0.091   \n",
       "353  chr18   37565974   T   C              4,5;6,1;0,4  4.0  0.105   \n",
       "404  chr22   41835260   C   G            16,4;1,3;0,10  4.0  0.125   \n",
       "\n",
       "        avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "415  rs868393078             0.0003             0.0006                     0   \n",
       "279  rs751435350             0.0034             0.0055                     1   \n",
       "243  rs549358675             0.0050             0.0093                     1   \n",
       "75   rs548922348             0.0022             0.0040                     1   \n",
       "74   rs367612791             0.0026             0.0047                     1   \n",
       "5    rs148017678             0.0041             0.0069                     3   \n",
       "69   rs193178985             0.0052             0.0085                     3   \n",
       "126  rs775783341             0.0010             0.0013                     2   \n",
       "353  rs145522218             0.0035             0.0019                     1   \n",
       "404  rs185331159             0.0006             0.0005                     2   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "415  IRX3_HUMAN.H11MO.0.D, RORG_HUMAN.H11MO.0.C, NR...   \n",
       "279  NKX22_HUMAN.H11MO.0.D, TBX20_HUMAN.H11MO.0.D, ...   \n",
       "243  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, A...   \n",
       "75   BHE22_HUMAN.H11MO.0.D, OLIG1_HUMAN.H11MO.0.D, ...   \n",
       "74        ZN384_HUMAN.H11MO.0.C, ZN257_HUMAN.H11MO.0.C   \n",
       "5    FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...   \n",
       "69   MBD2_HUMAN.H11MO.0.B, SRBP2_HUMAN.H11MO.0.B, C...   \n",
       "126  NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...   \n",
       "353  GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...   \n",
       "404  MAF_HUMAN.H11MO.1.B, MAX_HUMAN.H11MO.0.A, SNAI...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "415                          4.30;1.80   \n",
       "279                          3.89;1.52   \n",
       "243                          5.23;1.70   \n",
       "75                           3.85;1.31   \n",
       "74                           4.13;1.53   \n",
       "5                            4.37;1.50   \n",
       "69                           3.91;1.38   \n",
       "126                          4.66;1.76   \n",
       "353                         10.00;7.20   \n",
       "404                          3.86;1.38   \n",
       "\n",
       "                                                ENCODE genomic element  \\\n",
       "415                                                           promoter   \n",
       "279         ZNF143, NFIC, POLR2A, YY1, ELF1, CHD2, SP1        promoter   \n",
       "243                   BHLHE40, STAT3, CTCF, E2F1, TAF1        promoter   \n",
       "75   REST, MXI1, SAP30, TBP, TRIM28, BCLAF1, TEAD4,...        promoter   \n",
       "74                                         POLR2A, YY1        promoter   \n",
       "5                                                             promoter   \n",
       "69   NR3C1, RBBP5, HDAC1, TFAP2A, TAF7, PHF8, POLR2...        promoter   \n",
       "126  EGR1, RBBP5, EZH2, E2F6, POLR2A, KDM5B, YY1, U...        promoter   \n",
       "353                     EGR1, RBBP5, EZH2, GATA2, CHD1        promoter   \n",
       "404                                                           promoter   \n",
       "\n",
       "                       Gene polgenom_freq_perc  \n",
       "415                  CA5BP1                0.4  \n",
       "279   TTLL13P, RP11-697E2.6                0.4  \n",
       "243  RP5-916L7.2, LINC00939                0.4  \n",
       "75                     BRK1                0.4  \n",
       "74                PRRT3-AS1                0.4  \n",
       "5                     RSRP1               0.51  \n",
       "69                  FAM117B               1.19  \n",
       "126           PLAGL1, HYMAI               1.21  \n",
       "353                   CELF4                1.4  \n",
       "404            RP5-821D11.7               1.59  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode_polgenom.sort_values(by = \"polgenom_freq_perc\")[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>chrX</td>\n",
       "      <td>15676222</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0,136;141,169;0,147</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.109</td>\n",
       "      <td>rs868393078</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0</td>\n",
       "      <td>IRX3_HUMAN.H11MO.0.D, RORG_HUMAN.H11MO.0.C, NR...</td>\n",
       "      <td>4.30;1.80</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>CA5BP1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90249253</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>44,5;23,17;37,38;39,31</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs751435350</td>\n",
       "      <td>0.003400</td>\n",
       "      <td>0.0055</td>\n",
       "      <td>1</td>\n",
       "      <td>NKX22_HUMAN.H11MO.0.D, TBX20_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.89;1.52</td>\n",
       "      <td>ZNF143, NFIC, POLR2A, YY1, ELF1, CHD2, SP1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>TTLL13P, RP11-697E2.6</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>chr12</td>\n",
       "      <td>125983445</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>31,44;77,69;77,84;31,34</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs549358675</td>\n",
       "      <td>0.005000</td>\n",
       "      <td>0.0093</td>\n",
       "      <td>1</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, A...</td>\n",
       "      <td>5.23;1.70</td>\n",
       "      <td>BHLHE40, STAT3, CTCF, E2F1, TAF1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP5-916L7.2, LINC00939</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>chr3</td>\n",
       "      <td>9946995</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>36,23;23,18;46,36;65,56</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs367612791</td>\n",
       "      <td>0.002600</td>\n",
       "      <td>0.0047</td>\n",
       "      <td>1</td>\n",
       "      <td>ZN384_HUMAN.H11MO.0.C, ZN257_HUMAN.H11MO.0.C</td>\n",
       "      <td>4.13;1.53</td>\n",
       "      <td>POLR2A, YY1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>PRRT3-AS1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>chr3</td>\n",
       "      <td>10115323</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>32,37;29,36;62,60;40,51</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs548922348</td>\n",
       "      <td>0.002200</td>\n",
       "      <td>0.0040</td>\n",
       "      <td>1</td>\n",
       "      <td>BHE22_HUMAN.H11MO.0.D, OLIG1_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>3.85;1.31</td>\n",
       "      <td>REST, MXI1, SAP30, TBP, TRIM28, BCLAF1, TEAD4,...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>BRK1</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25268534</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>13,11;0,7;15,7</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs148017678</td>\n",
       "      <td>0.004100</td>\n",
       "      <td>0.0069</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.37;1.50</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RSRP1</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>chr2</td>\n",
       "      <td>202635226</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4,5;0,6;5,2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs193178985</td>\n",
       "      <td>0.005200</td>\n",
       "      <td>0.0085</td>\n",
       "      <td>3</td>\n",
       "      <td>MBD2_HUMAN.H11MO.0.B, SRBP2_HUMAN.H11MO.0.B, C...</td>\n",
       "      <td>3.91;1.38</td>\n",
       "      <td>NR3C1, RBBP5, HDAC1, TFAP2A, TAF7, PHF8, POLR2...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>FAM117B</td>\n",
       "      <td>1.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>chr6</td>\n",
       "      <td>144008499</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5,8;10,13;0,2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>rs775783341</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...</td>\n",
       "      <td>4.66;1.76</td>\n",
       "      <td>EGR1, RBBP5, EZH2, E2F6, POLR2A, KDM5B, YY1, U...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>PLAGL1, HYMAI</td>\n",
       "      <td>1.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>chr18</td>\n",
       "      <td>37565974</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4,5;6,1;0,4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs145522218</td>\n",
       "      <td>0.003500</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>1</td>\n",
       "      <td>GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...</td>\n",
       "      <td>10.00;7.20</td>\n",
       "      <td>EGR1, RBBP5, EZH2, GATA2, CHD1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CELF4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41835260</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>16,4;1,3;0,10</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.125</td>\n",
       "      <td>rs185331159</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>MAF_HUMAN.H11MO.1.B, MAX_HUMAN.H11MO.0.A, SNAI...</td>\n",
       "      <td>3.86;1.38</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP5-821D11.7</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>chr12</td>\n",
       "      <td>100266981</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>68,54;44,43;26,38;69,64</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs138623297</td>\n",
       "      <td>0.004800</td>\n",
       "      <td>0.0083</td>\n",
       "      <td>4</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, HAND1_HUMAN.H11MO.1.D, ...</td>\n",
       "      <td>3.85;1.75</td>\n",
       "      <td>USF2, CREB1, TBP, SP2, NFYA, JUN, MAZ, E2F6, I...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SCYL2, DEPDC4</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>chr7</td>\n",
       "      <td>74964567</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>33,27;27,33;18,20;26,21;24,27</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.109</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>5</td>\n",
       "      <td>EGR4_HUMAN.H11MO.0.D, ATF6A_HUMAN.H11MO.0.B, Z...</td>\n",
       "      <td>3.74;1.35</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>GATSL2</td>\n",
       "      <td>1.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>chr16</td>\n",
       "      <td>21502017</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>83,28;63,29;43,18;31,14</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs867677341</td>\n",
       "      <td>0.003000</td>\n",
       "      <td>0.0043</td>\n",
       "      <td>3</td>\n",
       "      <td>FEV_HUMAN.H11MO.0.B, XBP1_HUMAN.H11MO.0.D, ZN2...</td>\n",
       "      <td>4.43;1.70</td>\n",
       "      <td>PAX5, REST, ATF2, ZNF263, RBBP5, UBTF, BCLAF1,...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SMG1P3</td>\n",
       "      <td>1.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>chr16</td>\n",
       "      <td>21303291</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>57,42;61,57;36,33;19,9</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs11649081</td>\n",
       "      <td>0.003600</td>\n",
       "      <td>0.0051</td>\n",
       "      <td>3</td>\n",
       "      <td>BARX1_HUMAN.H11MO.0.D</td>\n",
       "      <td>3.53;1.45</td>\n",
       "      <td>MAFK, BHLHE40, POLR2A, MYC, MAFF, MAX</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CRYM</td>\n",
       "      <td>1.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>chr7</td>\n",
       "      <td>72925331</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>27,35;40,32;55,39;18,16</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs533479712</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>4</td>\n",
       "      <td>ZN257_HUMAN.H11MO.0.C, AP2D_HUMAN.H11MO.0.D, T...</td>\n",
       "      <td>3.85;1.39</td>\n",
       "      <td>NR3C1, RELA, CCNT2, NFIC, E2F6, PHF8, BCL3, PO...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RP11-313P13.5</td>\n",
       "      <td>2.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>chr10</td>\n",
       "      <td>132943730</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17,21;43,58;48,52;46,53</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs184581268</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.0079</td>\n",
       "      <td>4</td>\n",
       "      <td>IKZF1_HUMAN.H11MO.0.C, NKX25_HUMAN.H11MO.0.B, ...</td>\n",
       "      <td>4.13;1.49</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>CFAP46</td>\n",
       "      <td>3.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>148522213</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>150,143;204,181;88,85;109,87</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>3</td>\n",
       "      <td>CXXC1_HUMAN.H11MO.0.D, SOX15_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.78;2.98</td>\n",
       "      <td>YY1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>U1</td>\n",
       "      <td>6.75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                           VAFs   AC     AF  \\\n",
       "415   chrX   15676222   G   C            0,136;141,169;0,147  5.0  0.109   \n",
       "279  chr15   90249253   A   G         44,5;23,17;37,38;39,31  4.0  0.087   \n",
       "243  chr12  125983445   C   T        31,44;77,69;77,84;31,34  4.0  0.087   \n",
       "74    chr3    9946995   C   T        36,23;23,18;46,36;65,56  4.0  0.087   \n",
       "75    chr3   10115323   A   G        32,37;29,36;62,60;40,51  4.0  0.087   \n",
       "5     chr1   25268534   A   G                 13,11;0,7;15,7  4.0  0.105   \n",
       "69    chr2  202635226   G   A                    4,5;0,6;5,2  4.0  0.087   \n",
       "126   chr6  144008499   G   A                  5,8;10,13;0,2  4.0  0.091   \n",
       "353  chr18   37565974   T   C                    4,5;6,1;0,4  4.0  0.105   \n",
       "404  chr22   41835260   C   G                  16,4;1,3;0,10  4.0  0.125   \n",
       "235  chr12  100266981   T   C        68,54;44,43;26,38;69,64  4.0  0.087   \n",
       "140   chr7   74964567   C   T  33,27;27,33;18,20;26,21;24,27  5.0  0.109   \n",
       "294  chr16   21502017   A   C        83,28;63,29;43,18;31,14  4.0  0.087   \n",
       "293  chr16   21303291   T   C         57,42;61,57;36,33;19,9  4.0  0.087   \n",
       "138   chr7   72925331   C   G        27,35;40,32;55,39;18,16  4.0  0.087   \n",
       "201  chr10  132943730   G   A        17,21;43,58;48,52;46,53  4.0  0.087   \n",
       "30    chr1  148522213   A   G   150,143;204,181;88,85;109,87  4.0  0.087   \n",
       "\n",
       "        avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "415  rs868393078           0.000300             0.0006                     0   \n",
       "279  rs751435350           0.003400             0.0055                     1   \n",
       "243  rs549358675           0.005000             0.0093                     1   \n",
       "74   rs367612791           0.002600             0.0047                     1   \n",
       "75   rs548922348           0.002200             0.0040                     1   \n",
       "5    rs148017678           0.004100             0.0069                     3   \n",
       "69   rs193178985           0.005200             0.0085                     3   \n",
       "126  rs775783341           0.001000             0.0013                     2   \n",
       "353  rs145522218           0.003500             0.0019                     1   \n",
       "404  rs185331159           0.000600             0.0005                     2   \n",
       "235  rs138623297           0.004800             0.0083                     4   \n",
       "140            .           0.000044             0.0000                     5   \n",
       "294  rs867677341           0.003000             0.0043                     3   \n",
       "293   rs11649081           0.003600             0.0051                     3   \n",
       "138  rs533479712           0.001700             0.0029                     4   \n",
       "201  rs184581268           0.004700             0.0079                     4   \n",
       "30             .           0.000500             0.0009                     3   \n",
       "\n",
       "                                      PWMEnrich motifs  \\\n",
       "415  IRX3_HUMAN.H11MO.0.D, RORG_HUMAN.H11MO.0.C, NR...   \n",
       "279  NKX22_HUMAN.H11MO.0.D, TBX20_HUMAN.H11MO.0.D, ...   \n",
       "243  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, A...   \n",
       "74        ZN384_HUMAN.H11MO.0.C, ZN257_HUMAN.H11MO.0.C   \n",
       "75   BHE22_HUMAN.H11MO.0.D, OLIG1_HUMAN.H11MO.0.D, ...   \n",
       "5    FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...   \n",
       "69   MBD2_HUMAN.H11MO.0.B, SRBP2_HUMAN.H11MO.0.B, C...   \n",
       "126  NR1I3_HUMAN.H11MO.1.D, RARG_HUMAN.H11MO.1.B, N...   \n",
       "353  GATA3_HUMAN.H11MO.0.A, BPTF_HUMAN.H11MO.0.D, F...   \n",
       "404  MAF_HUMAN.H11MO.1.B, MAX_HUMAN.H11MO.0.A, SNAI...   \n",
       "235  MECP2_HUMAN.H11MO.0.C, HAND1_HUMAN.H11MO.1.D, ...   \n",
       "140  EGR4_HUMAN.H11MO.0.D, ATF6A_HUMAN.H11MO.0.B, Z...   \n",
       "294  FEV_HUMAN.H11MO.0.B, XBP1_HUMAN.H11MO.0.D, ZN2...   \n",
       "293                              BARX1_HUMAN.H11MO.0.D   \n",
       "138  ZN257_HUMAN.H11MO.0.C, AP2D_HUMAN.H11MO.0.D, T...   \n",
       "201  IKZF1_HUMAN.H11MO.0.C, NKX25_HUMAN.H11MO.0.B, ...   \n",
       "30   CXXC1_HUMAN.H11MO.0.D, SOX15_HUMAN.H11MO.0.D, ...   \n",
       "\n",
       "    PWMEnrich best -log10pval and diff  \\\n",
       "415                          4.30;1.80   \n",
       "279                          3.89;1.52   \n",
       "243                          5.23;1.70   \n",
       "74                           4.13;1.53   \n",
       "75                           3.85;1.31   \n",
       "5                            4.37;1.50   \n",
       "69                           3.91;1.38   \n",
       "126                          4.66;1.76   \n",
       "353                         10.00;7.20   \n",
       "404                          3.86;1.38   \n",
       "235                          3.85;1.75   \n",
       "140                          3.74;1.35   \n",
       "294                          4.43;1.70   \n",
       "293                          3.53;1.45   \n",
       "138                          3.85;1.39   \n",
       "201                          4.13;1.49   \n",
       "30                           4.78;2.98   \n",
       "\n",
       "                                                ENCODE genomic element  \\\n",
       "415                                                           promoter   \n",
       "279         ZNF143, NFIC, POLR2A, YY1, ELF1, CHD2, SP1        promoter   \n",
       "243                   BHLHE40, STAT3, CTCF, E2F1, TAF1        promoter   \n",
       "74                                         POLR2A, YY1        promoter   \n",
       "75   REST, MXI1, SAP30, TBP, TRIM28, BCLAF1, TEAD4,...        promoter   \n",
       "5                                                             promoter   \n",
       "69   NR3C1, RBBP5, HDAC1, TFAP2A, TAF7, PHF8, POLR2...        promoter   \n",
       "126  EGR1, RBBP5, EZH2, E2F6, POLR2A, KDM5B, YY1, U...        promoter   \n",
       "353                     EGR1, RBBP5, EZH2, GATA2, CHD1        promoter   \n",
       "404                                                           promoter   \n",
       "235  USF2, CREB1, TBP, SP2, NFYA, JUN, MAZ, E2F6, I...        promoter   \n",
       "140                                                           promoter   \n",
       "294  PAX5, REST, ATF2, ZNF263, RBBP5, UBTF, BCLAF1,...        promoter   \n",
       "293              MAFK, BHLHE40, POLR2A, MYC, MAFF, MAX        promoter   \n",
       "138  NR3C1, RELA, CCNT2, NFIC, E2F6, PHF8, BCL3, PO...        promoter   \n",
       "201                                                           promoter   \n",
       "30                                                 YY1        promoter   \n",
       "\n",
       "                       Gene polgenom_freq_perc  \n",
       "415                  CA5BP1                0.4  \n",
       "279   TTLL13P, RP11-697E2.6                0.4  \n",
       "243  RP5-916L7.2, LINC00939                0.4  \n",
       "74                PRRT3-AS1                0.4  \n",
       "75                     BRK1                0.4  \n",
       "5                     RSRP1               0.51  \n",
       "69                  FAM117B               1.19  \n",
       "126           PLAGL1, HYMAI               1.21  \n",
       "353                   CELF4                1.4  \n",
       "404            RP5-821D11.7               1.59  \n",
       "235           SCYL2, DEPDC4               1.59  \n",
       "140                  GATSL2                1.6  \n",
       "294                  SMG1P3                1.6  \n",
       "293                    CRYM               1.98  \n",
       "138           RP11-313P13.5               2.03  \n",
       "201                  CFAP46               3.17  \n",
       "30                       U1               6.75  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoters_motifs_encode_polgenom[(promoters_motifs_encode_polgenom[\"AF\"]*100) > promoters_motifs_encode_polgenom[\"polgenom_freq_perc\"]].sort_values(by = \"polgenom_freq_perc\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enhancers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9562378</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9562381</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>37234892</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100189355</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>109529529</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM        POS  polgenom_freq_perc\n",
       "0  chr1    9562378                 NaN\n",
       "1  chr1    9562381                 NaN\n",
       "2  chr1   37234892                 NaN\n",
       "3  chr1  100189355                 NaN\n",
       "4  chr1  109529529                 NaN"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#these snps have been checked by Marcelina; if freq is missing then it was not available in polgenom\n",
    "enhancers_with_polgenom = pd.read_csv(\"../2019-07-03-Variants_in_ENCODE_TFBS/out_files/enhancers_tf_polgenom_gnomad001.csv\", sep=\"\\t\",\n",
    "                                      usecols = [\"CHROM\", \"POS\", \"polgenom_freq_perc\"]).drop_duplicates()\n",
    "enhancers_with_polgenom.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enhancers_with_polgenom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "enhancers_motifs_encode_genes_polgenom = add_polgenom(enhancers_motifs_encode_closest, enhancers_with_polgenom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>Contacting protein gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>chr6</td>\n",
       "      <td>150419419</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>39,36;25,33;23,17;7,15</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs139564959</td>\n",
       "      <td>0.0054</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>...</td>\n",
       "      <td>BACH1, E2F6, CBX3, POLR2A, SETDB1, GABPA, ZBTB...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000220598:SSR1P1</td>\n",
       "      <td>33240</td>\n",
       "      <td>ENSG00000009765:IYD</td>\n",
       "      <td>49468</td>\n",
       "      <td>ENSG00000120278:PLEKHG1, ENSG00000120254:MTHFD1L</td>\n",
       "      <td>ENSG00000120278:PLEKHG1, ENSG00000120254:MTHFD1L</td>\n",
       "      <td>0.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>chr15</td>\n",
       "      <td>51938615</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>18,27;22,27;32,17;11,6</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs542004831</td>\n",
       "      <td>0.0033</td>\n",
       "      <td>0.0058</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000138594:TMOD3, ENSG00000166477:LEO1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000259556:RP11-56B16.2</td>\n",
       "      <td>7979</td>\n",
       "      <td>ENSG00000069956:MAPK6</td>\n",
       "      <td>11853</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48110886</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>79,9;56,12;34,12;94,14</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>...</td>\n",
       "      <td>NFIC</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000136146:MED4</td>\n",
       "      <td>14480</td>\n",
       "      <td>ENSG00000136146:MED4</td>\n",
       "      <td>14480</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>chr5</td>\n",
       "      <td>107973351</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>6,2;7,2;12,6;9,3;20,9</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.114</td>\n",
       "      <td>rs963187462</td>\n",
       "      <td>0.0015</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000145743:FBXL17</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000251732:RN7SKP122</td>\n",
       "      <td>160828</td>\n",
       "      <td>ENSG00000184349:EFNA5</td>\n",
       "      <td>300562</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359964</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>9,2;3,4;5,3;19,7</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>rs200573895</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000258346:RP11-148B3.2</td>\n",
       "      <td>8430</td>\n",
       "      <td>ENSG00000123066:MED13L</td>\n",
       "      <td>81076</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>chr3</td>\n",
       "      <td>132788529</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>54,79;63,44;42,31;33,20</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs570976618</td>\n",
       "      <td>0.0023</td>\n",
       "      <td>0.0041</td>\n",
       "      <td>...</td>\n",
       "      <td>FOS, JUND, EP300, CEBPB</td>\n",
       "      <td>ENSG00000248724:NPHP3-AS1</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000274810:NPHP3-ACAD11</td>\n",
       "      <td>64550</td>\n",
       "      <td>ENSG00000274810:NPHP3-ACAD11</td>\n",
       "      <td>64550</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69297077</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0,192;25,31;61,50</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.087</td>\n",
       "      <td>rs77368051</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0.0096</td>\n",
       "      <td>...</td>\n",
       "      <td>ELK4, IRF1, PHF8, POLR2A, SETDB1, RUNX3, GTF2B...</td>\n",
       "      <td>ENSG00000172927:MYEOV</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ENSG00000172927:MYEOV</td>\n",
       "      <td>2609</td>\n",
       "      <td>ENSG00000172927:MYEOV</td>\n",
       "      <td>2609</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359336</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>15,4;22,7;12,4;4,4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.091</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>CREB1, PAX5, EGR1, NFYA, JUN, TRIM28, MAZ, TEA...</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000258346:RP11-148B3.2</td>\n",
       "      <td>8430</td>\n",
       "      <td>ENSG00000123066:MED13L</td>\n",
       "      <td>81076</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS REF ALT                     VAFs   AC     AF  \\\n",
       "136   chr6  150419419   C   T   39,36;25,33;23,17;7,15  4.0  0.087   \n",
       "273  chr15   51938615   A   G   18,27;22,27;32,17;11,6  4.0  0.087   \n",
       "252  chr13   48110886   A   G   79,9;56,12;34,12;94,14  4.0  0.087   \n",
       "109   chr5  107973351   G   T    6,2;7,2;12,6;9,3;20,9  5.0  0.114   \n",
       "242  chr12  116359964   T   C         9,2;3,4;5,3;19,7  4.0  0.091   \n",
       "84    chr3  132788529   G   A  54,79;63,44;42,31;33,20  4.0  0.087   \n",
       "214  chr11   69297077   C   T        0,192;25,31;61,50  4.0  0.087   \n",
       "241  chr12  116359336   G   T       15,4;22,7;12,4;4,4  4.0  0.091   \n",
       "\n",
       "        avsnp150  gnomAD_genome_ALL  gnomAD_genome_NFE         ...          \\\n",
       "136  rs139564959             0.0054             0.0086         ...           \n",
       "273  rs542004831             0.0033             0.0058         ...           \n",
       "252            .             0.0005             0.0004         ...           \n",
       "109  rs963187462             0.0015             0.0022         ...           \n",
       "242  rs200573895             0.0007             0.0012         ...           \n",
       "84   rs570976618             0.0023             0.0041         ...           \n",
       "214   rs77368051             0.0059             0.0096         ...           \n",
       "241            .             0.0004             0.0006         ...           \n",
       "\n",
       "                                                ENCODE  \\\n",
       "136  BACH1, E2F6, CBX3, POLR2A, SETDB1, GABPA, ZBTB...   \n",
       "273                                                      \n",
       "252                                               NFIC   \n",
       "109                                                      \n",
       "242                                                      \n",
       "84                             FOS, JUND, EP300, CEBPB   \n",
       "214  ELK4, IRF1, PHF8, POLR2A, SETDB1, RUNX3, GTF2B...   \n",
       "241  CREB1, PAX5, EGR1, NFYA, JUN, TRIM28, MAZ, TEA...   \n",
       "\n",
       "                                            Gene      genomic element  \\\n",
       "136                                            .  enhancer intergenic   \n",
       "273  ENSG00000138594:TMOD3, ENSG00000166477:LEO1    enhancer intronic   \n",
       "252                                            .  enhancer intergenic   \n",
       "109                       ENSG00000145743:FBXL17    enhancer intronic   \n",
       "242                                            .  enhancer intergenic   \n",
       "84                     ENSG00000248724:NPHP3-AS1    enhancer intronic   \n",
       "214                        ENSG00000172927:MYEOV    enhancer intronic   \n",
       "241                                            .  enhancer intergenic   \n",
       "\n",
       "                     Closest gene dist. to closest gene  \\\n",
       "136        ENSG00000220598:SSR1P1                 33240   \n",
       "273  ENSG00000259556:RP11-56B16.2                  7979   \n",
       "252          ENSG00000136146:MED4                 14480   \n",
       "109     ENSG00000251732:RN7SKP122                160828   \n",
       "242  ENSG00000258346:RP11-148B3.2                  8430   \n",
       "84   ENSG00000274810:NPHP3-ACAD11                 64550   \n",
       "214         ENSG00000172927:MYEOV                  2609   \n",
       "241  ENSG00000258346:RP11-148B3.2                  8430   \n",
       "\n",
       "             Closest protein gene dist. to closest protein gene  \\\n",
       "136           ENSG00000009765:IYD                         49468   \n",
       "273         ENSG00000069956:MAPK6                         11853   \n",
       "252          ENSG00000136146:MED4                         14480   \n",
       "109         ENSG00000184349:EFNA5                        300562   \n",
       "242        ENSG00000123066:MED13L                         81076   \n",
       "84   ENSG00000274810:NPHP3-ACAD11                         64550   \n",
       "214         ENSG00000172927:MYEOV                          2609   \n",
       "241        ENSG00000123066:MED13L                         81076   \n",
       "\n",
       "                                      Contacting gene  \\\n",
       "136  ENSG00000120278:PLEKHG1, ENSG00000120254:MTHFD1L   \n",
       "273                                                 .   \n",
       "252                                                 .   \n",
       "109                                                 .   \n",
       "242                                                 .   \n",
       "84                                                  .   \n",
       "214                                                 .   \n",
       "241                                                 .   \n",
       "\n",
       "                              Contacting protein gene  polgenom_freq_perc  \n",
       "136  ENSG00000120278:PLEKHG1, ENSG00000120254:MTHFD1L                0.79  \n",
       "273                                                 .                0.79  \n",
       "252                                                 .                0.98  \n",
       "109                                                 .                1.19  \n",
       "242                                                 .                1.59  \n",
       "84                                                  .                1.98  \n",
       "214                                                 .                2.38  \n",
       "241                                                 .                4.21  \n",
       "\n",
       "[8 rows x 23 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancers_motifs_encode_genes_polgenom[(enhancers_motifs_encode_genes_polgenom[\"AF\"]*100) > enhancers_motifs_encode_genes_polgenom[\"polgenom_freq_perc\"]].sort_values(by = \"polgenom_freq_perc\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analyze variants with PWMEnrich hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "enh_with_hits = enhancers_motifs_encode_genes_polgenom[enhancers_motifs_encode_genes_polgenom[\"PWMEnrich motifs\"]!='']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "450 344\n"
     ]
    }
   ],
   "source": [
    "prom_with_hits = promoters_motifs_encode_polgenom[promoters_motifs_encode_polgenom[\"PWMEnrich motifs\"]!='']\n",
    "print len(prom_with_hits), len(enh_with_hits)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check correlation between H3K27ac in enhancer and gene expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "... and choose genes with p-value < 0.15."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n",
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if sys.path[0] == '':\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "missing expression data for: ENSG00000237365:RP11-334N17.1\n",
      "missing expression data for: ENSG00000221781:AL359815.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/numpy/lib/function_base.py:3162: RuntimeWarning: invalid value encountered in true_divide\n",
      "  c /= stddev[:, None]\n",
      "/home/magda/.local/lib/python2.7/site-packages/numpy/lib/function_base.py:3163: RuntimeWarning: invalid value encountered in true_divide\n",
      "  c /= stddev[None, :]\n",
      "/home/magda/.local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "/home/magda/.local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "/home/magda/.local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "missing expression data for: ENSG00000239319:RN7SL854P\n",
      "missing expression data for: ENSG00000227109:CRIP1P3\n",
      "RN7SKP27 is not expressed\n",
      "RN7SKP27 is not expressed\n",
      "missing expression data for: ENSG00000207456:RNU6-997P\n",
      "missing expression data for: ENSG00000231062:AC103563.9\n",
      "missing expression data for: ENSG00000271532:RP11-426K3.1\n",
      "missing expression data for: ENSG00000270930:GRAMD4P8\n",
      "missing expression data for: ENSG00000270776:RP11-17G11.1\n",
      "missing expression data for: ENSG00000234172:AC093639.1\n",
      "MIR4269 is not expressed\n",
      "CHCHD4P4 is not expressed\n",
      "missing expression data for: ENSG00000226621:AC083855.5\n",
      "missing expression data for: ENSG00000242364:RP11-451J24.1\n",
      "missing expression data for: ENSG00000221795:MIR1324\n",
      "RP11-197K3.1 is not expressed\n",
      "missing expression data for: ENSG00000212073:AC121332.1\n",
      "BTF3L4P4 is not expressed\n",
      "missing expression data for: ENSG00000248846:CTD-2016O11.1\n",
      "missing expression data for: ENSG00000251044:CTD-2582M21.1\n",
      "missing expression data for: ENSG00000281845:AC138942.1\n",
      "missing expression data for: ENSG00000253073:RN7SKP295\n",
      "missing expression data for: ENSG00000251732:RN7SKP122\n",
      "missing expression data for: ENSG00000248955:RP11-114H7.3\n",
      "missing expression data for: ENSG00000207775:MIR548A1\n",
      "missing expression data for: ENSG00000231912:RP1-249H1.2\n",
      "missing expression data for: ENSG00000219806:ATP5F1P6\n",
      "missing expression data for: ENSG00000220598:SSR1P1\n",
      "missing expression data for: ENSG00000234768:RP11-503C24.1\n",
      "missing expression data for: ENSG00000266430:AC004520.1\n",
      "AC105443.2 is not expressed\n",
      "missing expression data for: ENSG00000254091:RP11-10A14.8\n",
      "missing expression data for: ENSG00000265599:MIR4471\n",
      "missing expression data for: ENSG00000233713:RP11-379P1.4\n",
      "missing expression data for: ENSG00000236495:RP11-7C6.1\n",
      "IFIT6P is not expressed\n",
      "missing expression data for: ENSG00000238380:RNU7-165P\n",
      "AP000439.3 is not expressed\n",
      "RP11-497G19.2 is not expressed\n",
      "missing expression data for: ENSG00000229011:LINC01038\n",
      "AL121819.1 is not expressed\n",
      "AL121819.1 is not expressed\n",
      "missing expression data for: ENSG00000274379:RP11-179A9.3\n",
      "RP11-100M12.3 is not expressed\n",
      "AC104002.1 is not expressed\n",
      "missing expression data for: ENSG00000279523:RP11-461O7.2\n",
      "missing expression data for: ENSG00000206730:RNU6-468P\n",
      "missing expression data for: ENSG00000265883:RP11-169I9.2\n",
      "missing expression data for: ENSG00000281059:AF241734.1\n",
      "missing expression data for: ENSG00000233033:CASK-AS1\n"
     ]
    }
   ],
   "source": [
    "# H3K27ac coverage\n",
    "h3k27ac_cov = pd.read_csv('/home/magda/Symfonia/experiments/data/h3k27ac_np_normalized_cov_with_noise.csv', sep='\\t')\n",
    "\n",
    "# Genes/transcripts normalized counts\n",
    "counts = pd.read_csv('/home/magda/Symfonia/experiments/data/transcripts_normalized_counts_with_noise.csv', sep='\\t')\n",
    "counts['Gene'] = counts.Transcript.apply(lambda x: '_'.join(x.split('_')[1:]))\n",
    "\n",
    "#collect all assigned genes in one field\n",
    "enh_with_hits[\"all_genes\"] = enh_with_hits[\"Gene\"] + ', ' + enh_with_hits[\"Closest gene\"] + ', ' + enh_with_hits[\"Closest protein gene\"] + ', ' + enh_with_hits[\"Contacting gene\"]\n",
    "\n",
    "#intersect SNPs with H3K27ac coverage on enhancers\n",
    "enh_with_hits[\"POS-1\"] = enh_with_hits[\"POS\"] - 1\n",
    "snps_bedtool = pbt.BedTool.from_dataframe(enh_with_hits[[\"CHROM\", \"POS-1\", \"POS\", \"all_genes\"]])\n",
    "\n",
    "enh_bedtool = pbt.BedTool.from_dataframe(h3k27ac_cov)\n",
    "snps_enh_bedtool = snps_bedtool.intersect(enh_bedtool, wa=True, wb=True)\n",
    "snps_enh_df = snps_enh_bedtool.to_dataframe(names = [\"CHROM\", \"POS-1\", \"POS\", \"all_genes\"] + list(h3k27ac_cov.columns)).drop(labels = [\"POS-1\"], axis=1)\n",
    "\n",
    "#calculate correlations - each SNP-gene pair has separate output record in corr_df\n",
    "patients = snps_enh_df.columns[6:]\n",
    "corr_dict = {\"CHROM\": [],\n",
    "             \"POS\": [],\n",
    "             \"Transcript\" : [],\n",
    "             \"Spearman R\" : [],\n",
    "             \"Spearman p-value\" : []}\n",
    "\n",
    "for i,row in snps_enh_df.iterrows():\n",
    "    genes = set([el.strip() for el in row[\"all_genes\"].split(',')])\n",
    "    enh_act_vector = row[patients].values\n",
    "    \n",
    "    for gene in genes:        \n",
    "        if gene != '.':\n",
    "            gene_name = gene.split(':')[1]\n",
    "            gene_expr_rows = counts[counts[\"Gene\"] == gene_name]\n",
    "            \n",
    "            if len(gene_expr_rows) == 0:\n",
    "                print \"missing expression data for:\", gene\n",
    "            else:\n",
    "                gene_correlations = {}\n",
    "                for j, expr_row in gene_expr_rows.iterrows():\n",
    "                    expr_vector = expr_row[patients].values\n",
    "                    rho, pval = scipy.stats.spearmanr(enh_act_vector, expr_vector)\n",
    "                    if str(rho) != 'nan':\n",
    "                        gene_correlations[pval] = [rho, expr_row[\"Transcript\"]]\n",
    "                try:    \n",
    "                    min_pval = min(gene_correlations.keys())\n",
    "                    corr_dict[\"CHROM\"].append(row[\"CHROM\"])\n",
    "                    corr_dict[\"POS\"].append(row[\"POS\"])\n",
    "                    corr_dict[\"Transcript\"].append(gene_correlations[min_pval][1])\n",
    "                    corr_dict[\"Spearman R\"].append(gene_correlations[min_pval][0])\n",
    "                    corr_dict[\"Spearman p-value\"].append(min_pval)\n",
    "                except:\n",
    "                    print gene_name, \"is not expressed\"\n",
    "                \n",
    "corr_df = pd.DataFrame(corr_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Spearman R</th>\n",
       "      <th>Spearman p-value</th>\n",
       "      <th>Transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>0.728477</td>\n",
       "      <td>0.001372</td>\n",
       "      <td>ENST00000449373_MORN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>0.714706</td>\n",
       "      <td>0.001863</td>\n",
       "      <td>ENST00000488353_RER1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>-0.183959</td>\n",
       "      <td>0.495237</td>\n",
       "      <td>ENST00000508416_SKI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>-0.548933</td>\n",
       "      <td>0.027656</td>\n",
       "      <td>ENST00000481789_KCNAB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>0.620588</td>\n",
       "      <td>0.010314</td>\n",
       "      <td>ENST00000378161_NPHP4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS  Spearman R  Spearman p-value              Transcript\n",
       "0  chr1  2227533    0.728477          0.001372   ENST00000449373_MORN1\n",
       "1  chr1  2227533    0.714706          0.001863    ENST00000488353_RER1\n",
       "2  chr1  2227533   -0.183959          0.495237     ENST00000508416_SKI\n",
       "3  chr1  5896826   -0.548933          0.027656  ENST00000481789_KCNAB2\n",
       "4  chr1  5896826    0.620588          0.010314   ENST00000378161_NPHP4"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Spearman R</th>\n",
       "      <th>Spearman p-value</th>\n",
       "      <th>Transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>chr14</td>\n",
       "      <td>66471764</td>\n",
       "      <td>0.286972</td>\n",
       "      <td>0.281191</td>\n",
       "      <td>ENST00000450299_LINC00238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>632</th>\n",
       "      <td>chr14</td>\n",
       "      <td>66471764</td>\n",
       "      <td>0.254862</td>\n",
       "      <td>0.340787</td>\n",
       "      <td>ENST00000556874_RP11-72M17.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>633</th>\n",
       "      <td>chr14</td>\n",
       "      <td>66471764</td>\n",
       "      <td>0.753495</td>\n",
       "      <td>0.000750</td>\n",
       "      <td>ENST00000555527_GPHN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS  Spearman R  Spearman p-value  \\\n",
       "631  chr14  66471764    0.286972          0.281191   \n",
       "632  chr14  66471764    0.254862          0.340787   \n",
       "633  chr14  66471764    0.753495          0.000750   \n",
       "\n",
       "                       Transcript  \n",
       "631     ENST00000450299_LINC00238  \n",
       "632  ENST00000556874_RP11-72M17.1  \n",
       "633          ENST00000555527_GPHN  "
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_df[corr_df[\"POS\"]==66471764]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# collect p-vales for all genes with p-value < 0.15\n",
    "def collect_correlations(row):\n",
    "    corr_data = corr_df[(corr_df[\"CHROM\"] == row[\"CHROM\"]) & (corr_df[\"POS\"] == row[\"POS\"]) & (corr_df[\"Spearman p-value\"] < 0.15)]\n",
    "    genes = []\n",
    "    for i, corr_row in corr_data.iterrows():\n",
    "        genes.append(corr_row[\"Transcript\"] + ':' + \"%.4f\" % corr_row[\"Spearman p-value\"] )\n",
    "    if len(genes) > 0:\n",
    "        return ','.join(genes)\n",
    "    else:\n",
    "        return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>Closest gene</th>\n",
       "      <th>dist. to closest gene</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>Contacting protein gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>POS-1</th>\n",
       "      <th>Target_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>., ENSG00000157933:SKI, ENSG00000157933:SKI, E...</td>\n",
       "      <td>2227532</td>\n",
       "      <td>ENST00000449373_MORN1:0.0014,ENST00000488353_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000266687:AL356261.1</td>\n",
       "      <td>1853</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000131697:NPHP4, ENSG00000266687:AL35626...</td>\n",
       "      <td>5896825</td>\n",
       "      <td>ENST00000481789_KCNAB2:0.0277,ENST00000378161_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000237365:RP11-334N17.1</td>\n",
       "      <td>26411</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...</td>\n",
       "      <td>7041937</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16912415</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000272426:RP11-108M9.6</td>\n",
       "      <td>7462</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16913967</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0           0.000100           0.000300   \n",
       "1           0.000700           0.000700   \n",
       "2           0.000032           0.000067   \n",
       "3           0.000097           0.000067   \n",
       "4           0.004400           0.007600   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "2                        ...                           \n",
       "3                        ...                           \n",
       "4                        ...                           \n",
       "\n",
       "                    Closest gene dist. to closest gene  \\\n",
       "0            ENSG00000157933:SKI                  1000   \n",
       "1     ENSG00000266687:AL356261.1                  1853   \n",
       "2  ENSG00000237365:RP11-334N17.1                 26411   \n",
       "3   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "4   ENSG00000272426:RP11-108M9.6                  7462   \n",
       "\n",
       "     Closest protein gene dist. to closest protein gene  \\\n",
       "0     ENSG00000157933:SKI                          1000   \n",
       "1  ENSG00000069424:KCNAB2                         93950   \n",
       "2  ENSG00000171735:CAMTA1                        255366   \n",
       "3   ENSG00000117122:MFAP2                         65540   \n",
       "4   ENSG00000117122:MFAP2                         65540   \n",
       "\n",
       "                                     Contacting gene  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...   \n",
       "1                                                  .   \n",
       "2                                                  .   \n",
       "3                             ENSG00000206652:RNU1-1   \n",
       "4                             ENSG00000206652:RNU1-1   \n",
       "\n",
       "                             Contacting protein gene polgenom_freq_perc  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...                NaN   \n",
       "1                                                  .                NaN   \n",
       "2                                                  .                NaN   \n",
       "3                                                  .                NaN   \n",
       "4                                                  .                NaN   \n",
       "\n",
       "                                           all_genes     POS-1  \\\n",
       "0  ., ENSG00000157933:SKI, ENSG00000157933:SKI, E...   2227532   \n",
       "1  ENSG00000131697:NPHP4, ENSG00000266687:AL35626...   5896825   \n",
       "2  ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...   7041937   \n",
       "3  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16912415   \n",
       "4  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16913967   \n",
       "\n",
       "                                        Target_genes  \n",
       "0  ENST00000449373_MORN1:0.0014,ENST00000488353_R...  \n",
       "1  ENST00000481789_KCNAB2:0.0277,ENST00000378161_...  \n",
       "2                                               None  \n",
       "3                       ENST00000467938_CROCC:0.0053  \n",
       "4                       ENST00000467938_CROCC:0.0053  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_with_hits[\"Target_genes\"] = enh_with_hits.apply(collect_correlations, axis=1)\n",
    "enh_with_hits.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Additional checks for selected SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>GB08</th>\n",
       "      <th>GB02</th>\n",
       "      <th>GB01</th>\n",
       "      <th>PA04</th>\n",
       "      <th>...</th>\n",
       "      <th>GB04</th>\n",
       "      <th>GB05</th>\n",
       "      <th>DA03</th>\n",
       "      <th>GB06</th>\n",
       "      <th>DA04</th>\n",
       "      <th>GB07</th>\n",
       "      <th>DA05</th>\n",
       "      <th>GB03</th>\n",
       "      <th>DA01</th>\n",
       "      <th>DA06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69212239</td>\n",
       "      <td>., ENSG00000261070:RP11-554A11.8, ENSG00000172...</td>\n",
       "      <td>chr11</td>\n",
       "      <td>69210399</td>\n",
       "      <td>69213156</td>\n",
       "      <td>0.595277</td>\n",
       "      <td>0.305932</td>\n",
       "      <td>0.314822</td>\n",
       "      <td>0.822222</td>\n",
       "      <td>...</td>\n",
       "      <td>0.762554</td>\n",
       "      <td>0.201166</td>\n",
       "      <td>0.430563</td>\n",
       "      <td>0.275623</td>\n",
       "      <td>0.33651</td>\n",
       "      <td>0.411306</td>\n",
       "      <td>0.873914</td>\n",
       "      <td>0.295754</td>\n",
       "      <td>0.780768</td>\n",
       "      <td>0.539959</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                                          all_genes  \\\n",
       "208  chr11  69212239  ., ENSG00000261070:RP11-554A11.8, ENSG00000172...   \n",
       "\n",
       "       chr     start       end      GB08      GB02      GB01      PA04  \\\n",
       "208  chr11  69210399  69213156  0.595277  0.305932  0.314822  0.822222   \n",
       "\n",
       "       ...         GB04      GB05      DA03      GB06     DA04      GB07  \\\n",
       "208    ...     0.762554  0.201166  0.430563  0.275623  0.33651  0.411306   \n",
       "\n",
       "         DA05      GB03      DA01      DA06  \n",
       "208  0.873914  0.295754  0.780768  0.539959  \n",
       "\n",
       "[1 rows x 22 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_snp_df = snps_enh_df[snps_enh_df[\"POS\"] == 69212239]\n",
    "test_snp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_corr_dict = {\"CHROM\": [],\n",
    "             \"POS\": [],\n",
    "             \"Transcript\" : [],\n",
    "             \"Spearman R\" : [],\n",
    "             \"Spearman p-value\" : []}\n",
    "\n",
    "for i,row in test_snp_df.iterrows():\n",
    "    genes = set([el.strip() for el in row[\"all_genes\"].split(',')])\n",
    "    enh_act_vector = row[patients].values\n",
    "    \n",
    "    for gene in [\"TPCN2\", \"IGHMBP2\", \"PPP6R3\"]:        \n",
    "        if gene != '.':\n",
    "            gene_name = gene\n",
    "            gene_expr_rows = counts[counts[\"Gene\"] == gene_name]\n",
    "            \n",
    "            if len(gene_expr_rows) == 0:\n",
    "                print \"missing expression data for:\", gene\n",
    "            else:\n",
    "                gene_correlations = {}\n",
    "                for j, expr_row in gene_expr_rows.iterrows():\n",
    "                    expr_vector = expr_row[patients].values\n",
    "                    rho, pval = scipy.stats.spearmanr(enh_act_vector, expr_vector)\n",
    "                    if str(rho) != 'nan':\n",
    "                        gene_correlations[pval] = [rho, expr_row[\"Transcript\"]]\n",
    "                try:    \n",
    "                    min_pval = min(gene_correlations.keys())\n",
    "                    test_corr_dict[\"CHROM\"].append(row[\"CHROM\"])\n",
    "                    test_corr_dict[\"POS\"].append(row[\"POS\"])\n",
    "                    test_corr_dict[\"Transcript\"].append(gene_correlations[min_pval][1])\n",
    "                    test_corr_dict[\"Spearman R\"].append(gene_correlations[min_pval][0])\n",
    "                    test_corr_dict[\"Spearman p-value\"].append(min_pval)\n",
    "                except:\n",
    "                    print gene_name, \"is not expressed\"\n",
    "                \n",
    "test_corr_df = pd.DataFrame(test_corr_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Spearman R</th>\n",
       "      <th>Spearman p-value</th>\n",
       "      <th>Transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69212239</td>\n",
       "      <td>0.722590</td>\n",
       "      <td>0.001567</td>\n",
       "      <td>ENST00000542467_TPCN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69212239</td>\n",
       "      <td>0.504072</td>\n",
       "      <td>0.046489</td>\n",
       "      <td>ENST00000536803_IGHMBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69212239</td>\n",
       "      <td>-0.220588</td>\n",
       "      <td>0.411666</td>\n",
       "      <td>ENST00000524845_PPP6R3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS  Spearman R  Spearman p-value               Transcript\n",
       "0  chr11  69212239    0.722590          0.001567    ENST00000542467_TPCN2\n",
       "1  chr11  69212239    0.504072          0.046489  ENST00000536803_IGHMBP2\n",
       "2  chr11  69212239   -0.220588          0.411666   ENST00000524845_PPP6R3"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_corr_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check expression of target genes in brain and lymphocytes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex = pd.read_csv('../2020-12-09-Full_Hocomoco_tissues/in_files/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct', sep='\\t', skiprows=[0,1])\n",
    "brain_columns = [col for col in list(gtex.columns) if \"Brain\" in col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_names(genes_string):\n",
    "    #promoters will have only gene names, comma separated\n",
    "    #enhancers will have comma-separated lists with the following format: ENST000_gene_name:0.00123\n",
    "    if genes_string:\n",
    "        if \":\" not in genes_string:\n",
    "            return genes_string.split(',')\n",
    "        else:    \n",
    "            return ['_'.join((el.split(':')[0].split('_')[1:])) for el in genes_string.split(',')]\n",
    "    else:\n",
    "        return \"\"\n",
    "\n",
    "def check_expression_in_brain(genes):\n",
    "    \n",
    "    gene_names_list = get_gene_names(genes)\n",
    "    expression_list = []\n",
    "    for gene in gene_names_list:\n",
    "        if gene != \"\" and gene != \".\":\n",
    "            try:    \n",
    "                mean_median_tpm = sum(gtex[gtex['Description'] == gene.strip()][brain_columns].values[0]) / float(len(brain_columns))\n",
    "                expression_list.append(gene + ':' + \"%.2f\" % mean_median_tpm)\n",
    "            except:\n",
    "                print \"no gtex brain data for:\", gene\n",
    "            \n",
    "    return \",\".join(expression_list)\n",
    "\n",
    "\n",
    "def check_expression_in_lymphocytes(genes):\n",
    "    gene_names_list = get_gene_names(genes)\n",
    "    expression_list = []\n",
    "    for gene in gene_names_list:\n",
    "        if gene != \"\" and gene != \".\":\n",
    "            try:    \n",
    "                median_tpm = gtex[gtex['Description'] == gene.strip()]['Cells - EBV-transformed lymphocytes'].values[0]\n",
    "                expression_list.append(gene + ':' + \"%.2f\" % median_tpm)\n",
    "            except:\n",
    "                print \"no gtex lymphocytes data for:\", gene\n",
    "            \n",
    "    return \",\".join(expression_list)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no gtex lymphocytes data for: LINC01057\n",
      "no gtex lymphocytes data for: RP11-76I14.1\n",
      "no gtex lymphocytes data for: XXbac-BPG32J3.22\n",
      "no gtex lymphocytes data for: ATP6V1G2-DDX39B\n",
      "no gtex lymphocytes data for: AC006014.7\n",
      "no gtex lymphocytes data for: GS1-393G12.12\n",
      "no gtex lymphocytes data for: RP11-613M10.9\n",
      "no gtex lymphocytes data for: AC073370.1\n",
      "no gtex lymphocytes data for: KCNA6\n",
      "no gtex lymphocytes data for: RP11-793H13.10\n",
      "no gtex lymphocytes data for: RP11-235E17.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no gtex brain data for: LINC01057\n",
      "no gtex brain data for: RP11-76I14.1\n",
      "no gtex brain data for: XXbac-BPG32J3.22\n",
      "no gtex brain data for: ATP6V1G2-DDX39B\n",
      "no gtex brain data for: AC006014.7\n",
      "no gtex brain data for: GS1-393G12.12\n",
      "no gtex brain data for: RP11-613M10.9\n",
      "no gtex brain data for: AC073370.1\n",
      "no gtex brain data for: KCNA6\n",
      "no gtex brain data for: RP11-793H13.10\n",
      "no gtex brain data for: RP11-235E17.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "#enhancers\n",
    "enh_with_hits[\"target_expr_in_lymphocytes\"] = enh_with_hits.Target_genes.apply(check_expression_in_lymphocytes)\n",
    "enh_with_hits[\"target_expr_in_brain\"] = enh_with_hits.Target_genes.apply(check_expression_in_brain)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no gtex lymphocytes data for:  TMEM56-RWDD3\n",
      "no gtex lymphocytes data for: RP11-396C23.2\n",
      "no gtex lymphocytes data for: C1orf95\n",
      "no gtex lymphocytes data for: RP13-539J13.1\n",
      "no gtex lymphocytes data for:  XXbac-BPG32J3.22\n",
      "no gtex lymphocytes data for: MLLT4\n",
      "no gtex lymphocytes data for:  MLLT4-AS1\n",
      "no gtex lymphocytes data for: C11orf31\n",
      "no gtex lymphocytes data for:  RNU2-2P\n",
      "no gtex lymphocytes data for: AL132780.1\n",
      "no gtex lymphocytes data for:  RP11-697E2.6\n",
      "no gtex lymphocytes data for: CTC-435M10.3\n",
      "no gtex lymphocytes data for:  STX16-NPEPL1\n",
      "no gtex lymphocytes data for:  AC234775.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no gtex brain data for:  TMEM56-RWDD3\n",
      "no gtex brain data for: RP11-396C23.2\n",
      "no gtex brain data for: C1orf95\n",
      "no gtex brain data for: RP13-539J13.1\n",
      "no gtex brain data for:  XXbac-BPG32J3.22\n",
      "no gtex brain data for: MLLT4\n",
      "no gtex brain data for:  MLLT4-AS1\n",
      "no gtex brain data for: C11orf31\n",
      "no gtex brain data for:  RNU2-2P\n",
      "no gtex brain data for: AL132780.1\n",
      "no gtex brain data for:  RP11-697E2.6\n",
      "no gtex brain data for: CTC-435M10.3\n",
      "no gtex brain data for:  STX16-NPEPL1\n",
      "no gtex brain data for:  AC234775.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "#promoters\n",
    "prom_with_hits[\"target_expr_in_lymphocytes\"] = prom_with_hits.Gene.apply(check_expression_in_lymphocytes)\n",
    "prom_with_hits[\"target_expr_in_brain\"] = prom_with_hits.Gene.apply(check_expression_in_brain)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>Closest protein gene</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>Contacting protein gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>POS-1</th>\n",
       "      <th>Target_genes</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000157933:SKI</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>., ENSG00000157933:SKI, ENSG00000157933:SKI, E...</td>\n",
       "      <td>2227532</td>\n",
       "      <td>ENST00000449373_MORN1:0.0014,ENST00000488353_R...</td>\n",
       "      <td>MORN1:1.19,RER1:55.89</td>\n",
       "      <td>MORN1:3.76,RER1:25.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000069424:KCNAB2</td>\n",
       "      <td>93950</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000131697:NPHP4, ENSG00000266687:AL35626...</td>\n",
       "      <td>5896825</td>\n",
       "      <td>ENST00000481789_KCNAB2:0.0277,ENST00000378161_...</td>\n",
       "      <td>KCNAB2:69.76,NPHP4:6.44</td>\n",
       "      <td>KCNAB2:59.77,NPHP4:5.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000171735:CAMTA1</td>\n",
       "      <td>255366</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...</td>\n",
       "      <td>7041937</td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16912415</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000117122:MFAP2</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16913967</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE           ...             \\\n",
       "0           0.000100           0.000300           ...              \n",
       "1           0.000700           0.000700           ...              \n",
       "2           0.000032           0.000067           ...              \n",
       "3           0.000097           0.000067           ...              \n",
       "4           0.004400           0.007600           ...              \n",
       "\n",
       "     Closest protein gene dist. to closest protein gene  \\\n",
       "0     ENSG00000157933:SKI                          1000   \n",
       "1  ENSG00000069424:KCNAB2                         93950   \n",
       "2  ENSG00000171735:CAMTA1                        255366   \n",
       "3   ENSG00000117122:MFAP2                         65540   \n",
       "4   ENSG00000117122:MFAP2                         65540   \n",
       "\n",
       "                                     Contacting gene  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...   \n",
       "1                                                  .   \n",
       "2                                                  .   \n",
       "3                             ENSG00000206652:RNU1-1   \n",
       "4                             ENSG00000206652:RNU1-1   \n",
       "\n",
       "                             Contacting protein gene polgenom_freq_perc  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...                NaN   \n",
       "1                                                  .                NaN   \n",
       "2                                                  .                NaN   \n",
       "3                                                  .                NaN   \n",
       "4                                                  .                NaN   \n",
       "\n",
       "                                           all_genes     POS-1  \\\n",
       "0  ., ENSG00000157933:SKI, ENSG00000157933:SKI, E...   2227532   \n",
       "1  ENSG00000131697:NPHP4, ENSG00000266687:AL35626...   5896825   \n",
       "2  ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...   7041937   \n",
       "3  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16912415   \n",
       "4  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16913967   \n",
       "\n",
       "                                        Target_genes  \\\n",
       "0  ENST00000449373_MORN1:0.0014,ENST00000488353_R...   \n",
       "1  ENST00000481789_KCNAB2:0.0277,ENST00000378161_...   \n",
       "2                                               None   \n",
       "3                       ENST00000467938_CROCC:0.0053   \n",
       "4                       ENST00000467938_CROCC:0.0053   \n",
       "\n",
       "  target_expr_in_lymphocytes     target_expr_in_brain  \n",
       "0      MORN1:1.19,RER1:55.89    MORN1:3.76,RER1:25.86  \n",
       "1    KCNAB2:69.76,NPHP4:6.44  KCNAB2:59.77,NPHP4:5.59  \n",
       "2                                                      \n",
       "3                 CROCC:1.52              CROCC:15.56  \n",
       "4                 CROCC:1.52              CROCC:15.56  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_with_hits.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...</td>\n",
       "      <td>3.94;1.09</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:0.08</td>\n",
       "      <td>SAMD11:1.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>3</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.79;1.83</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN:57.03</td>\n",
       "      <td>AGRN:18.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...</td>\n",
       "      <td>10.00;6.71</td>\n",
       "      <td>ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CHD5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHD5:0.20</td>\n",
       "      <td>CHD5:31.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "      <td>MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>KAZN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KAZN:0.54</td>\n",
       "      <td>KAZN:22.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...</td>\n",
       "      <td>5.97;2.29</td>\n",
       "      <td>ZNF263, POLR2A</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RAP1GAP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RAP1GAP:0.60</td>\n",
       "      <td>RAP1GAP:103.43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT         VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1    939561   G   A  72,69;91,70  2.0  0.043  rs888928964   \n",
       "1  chr1   1034902   A   T  7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2  chr1   6180416   G   A    10,11;8,6  2.0  0.043  rs151212360   \n",
       "3  chr1  14924017   C   G    6,5;12,14  2.0  0.043  rs531269337   \n",
       "4  chr1  21653564   C   T  13,15;30,24  2.0  0.043  rs890141715   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000052             0.0000                     0   \n",
       "1           0.001300             0.0017                     3   \n",
       "2           0.000300             0.0005                     2   \n",
       "3           0.001300             0.0020                     0   \n",
       "4           0.000600             0.0013                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...   \n",
       "1  UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "2  UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...   \n",
       "3  MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "4  ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          3.94;1.09   \n",
       "1                          4.79;1.83   \n",
       "2                         10.00;6.71   \n",
       "3                          5.20;2.79   \n",
       "4                          5.97;2.29   \n",
       "\n",
       "                                              ENCODE genomic element     Gene  \\\n",
       "0                                               EZH2        promoter   SAMD11   \n",
       "1                                                           promoter     AGRN   \n",
       "2            ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1        promoter     CHD5   \n",
       "3  MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...        promoter     KAZN   \n",
       "4                                     ZNF263, POLR2A        promoter  RAP1GAP   \n",
       "\n",
       "  polgenom_freq_perc target_expr_in_lymphocytes target_expr_in_brain  \n",
       "0                NaN                SAMD11:0.08          SAMD11:1.25  \n",
       "1                NaN                 AGRN:57.03           AGRN:18.92  \n",
       "2                NaN                  CHD5:0.20           CHD5:31.50  \n",
       "3                NaN                  KAZN:0.54           KAZN:22.48  \n",
       "4                NaN               RAP1GAP:0.60       RAP1GAP:103.43  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_with_hits.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check expression in ADNI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "... for target genes expressed in brain."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "lo = LiftOver('/home/magda/Symfonia/experiments/2017/2017-08-11-SNPs-enhancers-gene_expression/in_files/hg38ToHg19.over.chain')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "def liftover_snp(chrom, pos):\n",
    "    result = lo.convert_coordinate(chrom, pos, '-')\n",
    "    chrom_hg19 = result[0][0]\n",
    "    pos_hg19 = result[0][1]\n",
    "    return chrom_hg19, pos_hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "ADNI_HEADER = open(\"../2019-07-29-Variant-expression_corr_ADNI/infiles/rs139564959.ADNI.header\").read().lstrip(\"#\").rstrip().split(\"\\t\")\n",
    "\n",
    "#expr data from ADNI\n",
    "EXPR = pd.read_csv('../2019-07-29-Variant-expression_corr_ADNI/infiles/ADNI_Gene_Expression_Profile.csv', \n",
    "                   sep=',', skiprows = [0,1, 3,4,5,6,7,8])\n",
    "EXPR = EXPR.rename(columns = {'SubjectID' : 'Probe',\n",
    "                              'Unnamed: 1' : 'LOC',\n",
    "                              'Unnamed: 2' : 'Gene',\n",
    "                              'Unnamed: 747' : 'Gene long'})\n",
    "\n",
    "def find_variant_in_ADNI(chrom, pos, adni_already_checked):\n",
    "    \n",
    "    chrom_hg19, pos_hg19 = liftover_snp(chrom, pos)\n",
    "    print chrom_hg19, pos_hg19\n",
    "    chrom_hg19 = chrom_hg19.lstrip(\"chr\")\n",
    "    \n",
    "    if chrom_hg19 == \"X\":\n",
    "        chrom_hg19 = '23'\n",
    "    \n",
    "    command1 = 'grep -P -m 1 \"%s\\\\t%s\\\\t\" %s' % (chrom_hg19, pos_hg19, adni_already_checked) \n",
    "    command2 = 'grep -P -m 1 \"\\\\t%s\\\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr%s_SNPs.vcf' % (pos_hg19, chrom_hg19) \n",
    "    \n",
    "    with open(adni_already_checked, \"a\") as previous_results:\n",
    "        print command1\n",
    "        adni_var_line = !$command1\n",
    "\n",
    "        if len(adni_var_line) == 0: #no search for this variant has been performed yet\n",
    "            print \"no search for this variant has been performed yet\"\n",
    "            print command2\n",
    "            adni_var_line = !$command2\n",
    "            if len(adni_var_line) == 0:\n",
    "                print \"no variant found in ADNI\"\n",
    "                previous_results.write(\"%s\\t%s\\tno variant found in ADNI\\n\"  % (chrom_hg19, pos_hg19))\n",
    "                return [False, False]\n",
    "            else:\n",
    "                adni_var = [el.split(':')[0] for el in adni_var_line[0].split(\"\\t\")]\n",
    "                adni_var_dict = dict(zip(ADNI_HEADER, adni_var))\n",
    "                previous_results.write(adni_var_line[0].replace(\"X\\t\", \"23\\t\") + '\\n')\n",
    "                return ':'.join(adni_var_line[0].split(\"\\t\")[:2]), adni_var_dict\n",
    "\n",
    "        elif  \"%s\\t%s\\tno variant found in ADNI\" % (chrom_hg19, pos_hg19) in adni_var_line[0]:\n",
    "            print \"no variant found in ADNI\"\n",
    "            return [False, False]\n",
    "\n",
    "        else:\n",
    "            adni_var = [el.split(':')[0] for el in adni_var_line[0].split(\"\\t\")]\n",
    "            adni_var_dict = dict(zip(ADNI_HEADER, adni_var))\n",
    "            return ':'.join(adni_var_line[0].split(\"\\t\")[:2]), adni_var_dict\n",
    "\n",
    "\n",
    "def check_expression(var_dict, gene_name):\n",
    "    print gene_name\n",
    "    #check if expression data for gene of interest is available\n",
    "    gene_expr = EXPR[EXPR['Gene'] == gene_name]\n",
    "    if len(gene_expr) == 0:\n",
    "        print gene_name, \" - gene not found in expression data\"\n",
    "        return {\"gene not found\":1}\n",
    "    \n",
    "    #collect p-values and corr coeffs for all probes\n",
    "    pvals = {}\n",
    "    \n",
    "    #find intensities of gene-related probes for each sample    \n",
    "    for i, expr_row in gene_expr.iterrows():\n",
    "        probe = expr_row[\"Probe\"]\n",
    "        print \"\\nProbe:\", probe\n",
    "        het_expr = []\n",
    "        ref_expr = []\n",
    "        alt_expr = []\n",
    "    \n",
    "        for sample in var_dict.keys():\n",
    "            if sample in EXPR.columns[3:]:\n",
    "                sample_expr = expr_row[sample]\n",
    "                gt = var_dict[sample]\n",
    "\n",
    "                if gt == '0/0' or gt == '0|0':                \n",
    "                        ref_expr.append(sample_expr)\n",
    "\n",
    "                elif gt == '0/1' or gt == '0|1':\n",
    "                        het_expr.append(sample_expr)\n",
    "\n",
    "                elif gt == '1/1' or gt == '1|1':\n",
    "                        alt_expr.append(sample_expr)\n",
    "\n",
    "                elif gt == \"./.\":\n",
    "                    pass\n",
    "\n",
    "                else:\n",
    "                    print var_dict[sample]            \n",
    "\n",
    "        print \"Number of REF, HET and ALT samples with expression data:\", len(ref_expr), len(het_expr), len(alt_expr)\n",
    "\n",
    "        # calculate Spearmann-corr for probe\n",
    "        num_categories = sum([len(ref_expr) > 0, len(het_expr) > 0, len(alt_expr) > 0])\n",
    "        if  num_categories > 1:\n",
    "            print \"Calculate Spearmann correlation\"\n",
    "            genotype_vector = [0]*len(ref_expr) + [1]*len(het_expr) + [2]*len(alt_expr)\n",
    "            expression_vector = ref_expr + het_expr + alt_expr\n",
    "            rho, pval = scipy.stats.spearmanr(genotype_vector, expression_vector)\n",
    "            print \"R, p\", rho, pval\n",
    "            if pval < 0.15:\n",
    "                pvals[probe] = pval\n",
    "            \n",
    "        else:\n",
    "            print \"Samples represent only one genotype. Exit.\"\n",
    "    print pvals    \n",
    "    return pvals\n",
    "\n",
    "def check_variant_gene_pair(chrom, pos, genes, adni_file): \n",
    "    print \"**Check if variant is present in ADNI\"\n",
    "    print chrom, pos\n",
    "    coords, var_dict = find_variant_in_ADNI(chrom, pos, adni_file)\n",
    "    \n",
    "    \n",
    "    if var_dict:\n",
    "        print \"**Check correlation with target expression\"\n",
    "        gene_names_list = get_gene_names(genes)\n",
    "        output = coords + ':'\n",
    "        for gene in gene_names_list:\n",
    "            gene_pvals = check_expression(var_dict,gene)\n",
    "            if len(gene_pvals.keys()) > 0:\n",
    "                output += gene + \":\"\n",
    "                for probe, pval in gene_pvals.items():\n",
    "                    output += probe + ':' + \"%.4f\" % pval + ','\n",
    "                    \n",
    "        return output.rstrip(\",\")\n",
    "        \n",
    "    else:\n",
    "        return \"no variant in ADNI\"\n",
    "    \n",
    "def check_target_expr_in_brain(row):\n",
    "    ''' check if any target gene is expressed in brain \n",
    "    to avoid calculations for targets without expression'''\n",
    "    \n",
    "    if row[\"target_expr_in_brain\"]:\n",
    "        genes_expression_list = [float(gene_info.split(':')[1]) for gene_info in row[\"target_expr_in_brain\"].split(',')]\n",
    "        if max(genes_expression_list) > 0.5:\n",
    "            print \"\\n*Target gene(s) expressed in brain - proceed.\"\n",
    "            return True\n",
    "        else:\n",
    "            print \"\\n*Target gene(s) not expressed in brain - stop.\"\n",
    "            return False\n",
    "    else:\n",
    "        print \"\\n*No target genes - stop.\"\n",
    "        return False\n",
    "        \n",
    "def check_ADNI(row, adni_file):\n",
    "    if check_target_expr_in_brain(row):\n",
    "        if row[\"genomic element\"] == \"promoter\":\n",
    "            return check_variant_gene_pair(row[\"CHROM\"], row[\"POS\"], row[\"Gene\"], adni_file)\n",
    "        else:\n",
    "            return check_variant_gene_pair(row[\"CHROM\"], row[\"POS\"], row[\"Target_genes\"], adni_file)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 2227533\n",
      "chr1 2158972\n",
      "grep -P -m 1 \"1\\t2158972\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t2158972\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 5896826\n",
      "chr1 5956886\n",
      "grep -P -m 1 \"1\\t5956886\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t5956886\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KCNAB2\n",
      "\n",
      "Probe: 11737785_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0226763587512 0.54606347393\n",
      "\n",
      "Probe: 11737786_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00565586756764 0.880332913645\n",
      "\n",
      "Probe: 11745714_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00967311469582 0.796806889435\n",
      "\n",
      "Probe: 11761657_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0406483038555 0.279070217314\n",
      "\n",
      "Probe: 11763315_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0245792307902 0.512892376181\n",
      "{}\n",
      "NPHP4\n",
      "\n",
      "Probe: 11754699_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.052171364276 0.164642344543\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 16912416\n",
      "chr1 17238911\n",
      "grep -P -m 1 \"1\\t17238911\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17238911\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 16913968\n",
      "chr1 17240463\n",
      "grep -P -m 1 \"1\\t17240463\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17240463\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CROCC\n",
      "\n",
      "Probe: 11728097_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0234351077517 0.533869562965\n",
      "\n",
      "Probe: 11728098_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0343266881209 0.362094108583\n",
      "\n",
      "Probe: 11756333_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0421664382636 0.262842719254\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 16913970\n",
      "chr1 17240465\n",
      "grep -P -m 1 \"1\\t17240465\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17240465\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CROCC\n",
      "\n",
      "Probe: 11728097_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 697 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0236366672811 0.530064712247\n",
      "\n",
      "Probe: 11728098_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 697 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0345050589801 0.35926285356\n",
      "\n",
      "Probe: 11756333_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 697 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0423281180392 0.260677184351\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 21254094\n",
      "chr1 21580587\n",
      "grep -P -m 1 \"1\\t21580587\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t21580587\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 21264261\n",
      "chr1 21590754\n",
      "grep -P -m 1 \"1\\t21590754\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t21590754\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ECE1\n",
      "\n",
      "Probe: 11716255_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0555435384095 0.139824229943\n",
      "\n",
      "Probe: 11716256_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0439514942677 0.242818378249\n",
      "\n",
      "Probe: 11737800_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0845433527061 0.0244739606591\n",
      "\n",
      "Probe: 11737801_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.06361196219 0.090774433299\n",
      "\n",
      "Probe: 11739461_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0861175576047 0.0219259146347\n",
      "{'11739461_a_at': 0.021925914634734318, '11716255_a_at': 0.13982422994327867, '11737800_a_at': 0.024473960659060441, '11737801_a_at': 0.090774433298960774}\n",
      "RAP1GAP\n",
      "\n",
      "Probe: 11725497_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 159 356 193\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0307110199719 0.414548384855\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 21625236\n",
      "chr1 21951729\n",
      "grep -P -m 1 \"1\\t21951729\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t21951729\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 24705404\n",
      "chr1 25031895\n",
      "grep -P -m 1 \"1\\t25031895\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t25031895\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CLIC4\n",
      "\n",
      "Probe: 11725926_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0265485014682 0.479391029421\n",
      "\n",
      "Probe: 11725927_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.028738064539 0.44389172811\n",
      "\n",
      "Probe: 11754405_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0629500074151 0.0932633255395\n",
      "{'11754405_s_at': 0.093263325539542891}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 27971997\n",
      "chr1 28298508\n",
      "grep -P -m 1 \"1\\t28298508\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 33075149\n",
      "chr1 33540750\n",
      "grep -P -m 1 \"1\\t33540750\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "S100PBP\n",
      "\n",
      "Probe: 11720828_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0473801296042 0.206679966662\n",
      "\n",
      "Probe: 11720829_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00830583833647 0.824904149709\n",
      "\n",
      "Probe: 11737387_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00568725018649 0.879589712032\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 94875371\n",
      "chr1 95340927\n",
      "grep -P -m 1 \"1\\t95340927\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t95340927\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 111515686\n",
      "chr1 112058308\n",
      "grep -P -m 1 \"1\\t112058308\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t112058308\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TMIGD3\n",
      "TMIGD3  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 154766557\n",
      "chr1 154739033\n",
      "grep -P -m 1 \"1\\t154739033\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t154739033\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KCNN3\n",
      "\n",
      "Probe: 11731894_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 682 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0529335592246 0.16609610425\n",
      "\n",
      "Probe: 11732631_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 682 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00203032792167 0.957667676045\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 161727094\n",
      "chr1 161696884\n",
      "grep -P -m 1 \"1\\t161696884\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 164644793\n",
      "chr1 164614030\n",
      "grep -P -m 1 \"1\\t164614030\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t164614030\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PBX1\n",
      "\n",
      "Probe: 11719412_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00710116901419 0.849973468015\n",
      "\n",
      "Probe: 11719413_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0129757752755 0.72961097382\n",
      "\n",
      "Probe: 11719414_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00284046810146 0.939689501775\n",
      "\n",
      "Probe: 11749792_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0015493461599 0.967081368468\n",
      "\n",
      "Probe: 11752626_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0189794956645 0.613143222058\n",
      "\n",
      "Probe: 11757871_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0250477639592 0.504588145985\n",
      "\n",
      "Probe: 11758070_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00968341164821 0.796454985167\n",
      "\n",
      "Probe: 11761679_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0741752513462 0.0478733355436\n",
      "{'11761679_at': 0.047873335543558833}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 167716310\n",
      "chr1 167685547\n",
      "grep -P -m 1 \"1\\t167685547\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 180843474\n",
      "chr1 180812610\n",
      "grep -P -m 1 \"1\\t180812610\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t180812610\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "XPR1\n",
      "\n",
      "Probe: 11736152_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0561639056626 0.134344175044\n",
      "\n",
      "Probe: 11736153_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0441563890456 0.239298019537\n",
      "\n",
      "Probe: 11736154_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0333109704013 0.37479247605\n",
      "\n",
      "Probe: 11736155_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00826320603003 0.8257884567\n",
      "\n",
      "Probe: 11741308_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00774674090833 0.83651851597\n",
      "{'11736152_a_at': 0.13434417504404747}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 183954208\n",
      "chr1 183923342\n",
      "grep -P -m 1 \"1\\t183923342\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t183923342\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "COLGALT2\n",
      "\n",
      "Probe: 11724166_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0500862998403 0.181889801703\n",
      "\n",
      "Probe: 11724167_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0537355672815 0.152040980544\n",
      "\n",
      "Probe: 11724168_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0218044569294 0.561333286155\n",
      "\n",
      "Probe: 11754253_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0592095996295 0.1144465924\n",
      "\n",
      "Probe: 11763295_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0353068706493 0.34683667038\n",
      "{'11754253_a_at': 0.11444659240046033}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 201193704\n",
      "chr1 201162832\n",
      "grep -P -m 1 \"1\\t201162832\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t201162832\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "IGFN1\n",
      "\n",
      "Probe: 11731279_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0279472146327 0.456855316884\n",
      "\n",
      "Probe: 11731280_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0201194684615 0.592243657611\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 201552411\n",
      "chr1 201521539\n",
      "grep -P -m 1 \"1\\t201521539\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t201521539\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NAV1\n",
      "\n",
      "Probe: 11725409_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0663648239221 0.0767814256269\n",
      "\n",
      "Probe: 11725410_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0342871795237 0.360949577519\n",
      "\n",
      "Probe: 11725411_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0285180572192 0.447389578778\n",
      "\n",
      "Probe: 11754794_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0781486444582 0.0370882615553\n",
      "\n",
      "Probe: 11754947_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0618232376405 0.0992853394061\n",
      "\n",
      "Probe: 11757767_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0263086189941 0.483372047746\n",
      "{'11754947_a_at': 0.099285339406105941, '11725409_s_at': 0.076781425626882324, '11754794_a_at': 0.037088261555338065}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 201771524\n",
      "chr1 201740652\n",
      "grep -P -m 1 \"1\\t201740652\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t201740652\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 202085611\n",
      "chr1 202054739\n",
      "grep -P -m 1 \"1\\t202054739\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t202054739\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 203201980\n",
      "chr1 203171108\n",
      "grep -P -m 1 \"1\\t203171108\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t203171108\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 204506594\n",
      "chr1 204475722\n",
      "grep -P -m 1 \"1\\t204475722\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t204475722\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 219556082\n",
      "chr1 219729424\n",
      "grep -P -m 1 \"1\\t219729424\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t219729424\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 223707099\n",
      "chr1 223894801\n",
      "grep -P -m 1 \"1\\t223894801\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t223894801\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 231054690\n",
      "chr1 231190436\n",
      "grep -P -m 1 \"1\\t231190436\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 240123509\n",
      "chr1 240286809\n",
      "grep -P -m 1 \"1\\t240286809\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t240286809\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 2877259\n",
      "chr2 2881031\n",
      "grep -P -m 1 \"2\\t2881031\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t2881031\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 8684545\n",
      "chr2 8824675\n",
      "grep -P -m 1 \"2\\t8824675\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t8824675\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 10285627\n",
      "chr2 10425753\n",
      "grep -P -m 1 \"2\\t10425753\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "HPCAL1\n",
      "\n",
      "Probe: 11718146_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0469845285299 0.210499771299\n",
      "\n",
      "Probe: 11724552_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.019249980043 0.608092249873\n",
      "\n",
      "Probe: 11748002_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0193411938497 0.606393123659\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 25965203\n",
      "chr2 26188072\n",
      "grep -P -m 1 \"2\\t26188072\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t26188072\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KIF3C\n",
      "\n",
      "Probe: 11717811_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0437000979515 0.244192443291\n",
      "\n",
      "Probe: 11717812_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0318399405492 0.396260950969\n",
      "\n",
      "Probe: 11750686_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0246327340301 0.511677417943\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 28594772\n",
      "chr2 28817639\n",
      "grep -P -m 1 \"2\\t28817639\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t28817639\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 53545939\n",
      "chr2 53773077\n",
      "grep -P -m 1 \"2\\t53773077\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53773077\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 54667122\n",
      "chr2 54894259\n",
      "grep -P -m 1 \"2\\t54894259\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t54894259\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "EML6\n",
      "\n",
      "Probe: 11725575_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0243856497045 0.51592245485\n",
      "{}\n",
      "SPTBN1\n",
      "\n",
      "Probe: 11716418_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0443522994255 0.237217908072\n",
      "\n",
      "Probe: 11716419_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0194348222372 0.604651225564\n",
      "\n",
      "Probe: 11716420_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0575270962946 0.125131169052\n",
      "\n",
      "Probe: 11716421_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0324459668913 0.387328210711\n",
      "\n",
      "Probe: 11716422_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0227490166287 0.544493936911\n",
      "\n",
      "Probe: 11716423_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00265950415661 0.943525207667\n",
      "\n",
      "Probe: 11722689_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0522489800453 0.163717058668\n",
      "\n",
      "Probe: 11722690_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0557268613229 0.137406143818\n",
      "\n",
      "Probe: 11758629_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00797851154142 0.831699377433\n",
      "{'11722690_at': 0.13740614381822533, '11716420_s_at': 0.12513116905205759}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 74844977\n",
      "chr2 75072104\n",
      "grep -P -m 1 \"2\\t75072104\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75072104\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HK2\n",
      "\n",
      "Probe: 11749099_a_at\n",
      "0/2\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0449809782406 0.230957883212\n",
      "\n",
      "Probe: 11756138_a_at\n",
      "0/2\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0148107837007 0.69339638\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 95057420\n",
      "chr2 95723165\n",
      "grep -P -m 1 \"2\\t95723165\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t95723165\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MAL\n",
      "\n",
      "Probe: 11754728_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 688 21 3\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0569109084299 0.129232854497\n",
      "{'11754728_a_at': 0.12923285449740257}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 96926516\n",
      "chr2 97592253\n",
      "grep -P -m 1 \"2\\t97592253\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t97592253\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "Metazoa_SRP\n",
      "Metazoa_SRP  - gene not found in expression data\n",
      "FAM178B\n",
      "\n",
      "Probe: 11729090_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0521614386404 0.164425096917\n",
      "{}\n",
      "SEMA4C\n",
      "\n",
      "Probe: 11720896_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0424779474658 0.257646018997\n",
      "\n",
      "Probe: 11720897_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0600374665929 0.109459181267\n",
      "{'11720897_at': 0.10945918126683842}\n",
      "ANKRD23\n",
      "\n",
      "Probe: 11727911_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00865054100859 0.817762339246\n",
      "\n",
      "Probe: 11727912_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0155580615257 0.678554443426\n",
      "{}\n",
      "ANKRD39\n",
      "\n",
      "Probe: 11726350_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0362806076144 0.333691694269\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 101063590\n",
      "chr2 101680052\n",
      "grep -P -m 1 \"2\\t101680052\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101680052\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TBC1D8\n",
      "\n",
      "Probe: 11723525_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0790150744212 0.0350329555305\n",
      "{'11723525_at': 0.035032955530476598}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 104450699\n",
      "chr2 105067157\n",
      "grep -P -m 1 \"2\\t105067157\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t105067157\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LINC01102\n",
      "\n",
      "Probe: 11763346_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0138802511894 0.711771693518\n",
      "\n",
      "Probe: 11763420_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0184153375758 0.623981847961\n",
      "\n",
      "Probe: 11763535_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00760436832151 0.839593500795\n",
      "{}\n",
      "RP11-76I14.1\n",
      "RP11-76I14.1  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 131683708\n",
      "chr2 132441281\n",
      "grep -P -m 1 \"2\\t132441281\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t132441281\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CCDC74A\n",
      "\n",
      "Probe: 11754776_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00307946184771 0.934625951337\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 172234077\n",
      "chr2 173098805\n",
      "grep -P -m 1 \"2\\t173098805\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t173098805\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 183998084\n",
      "chr2 184862811\n",
      "grep -P -m 1 \"2\\t184862811\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t184862811\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 239291983\n",
      "chr2 240213678\n",
      "grep -P -m 1 \"2\\t240213678\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t240213678\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 11194924\n",
      "chr3 11236610\n",
      "grep -P -m 1 \"3\\t11236610\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t11236610\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 11368620\n",
      "chr3 11410094\n",
      "grep -P -m 1 \"3\\t11410094\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t11410094\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 40269057\n",
      "chr3 40310548\n",
      "grep -P -m 1 \"3\\t40310548\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40310548\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "EIF1B-AS1\n",
      "EIF1B-AS1  - gene not found in expression data\n",
      "EIF1B\n",
      "\n",
      "Probe: 11717247_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0060213200711 0.872575432481\n",
      "\n",
      "Probe: 11754107_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0165130079582 0.660025606412\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 73636559\n",
      "chr3 73685710\n",
      "grep -P -m 1 \"3\\t73685710\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73685710\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PDZRN3\n",
      "\n",
      "Probe: 11724102_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0108566642417 0.772434868645\n",
      "\n",
      "Probe: 11747559_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0645013924533 0.085454437478\n",
      "\n",
      "Probe: 11753143_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0606696704363 0.105767262567\n",
      "{'11753143_a_at': 0.1057672625665472, '11747559_a_at': 0.085454437477961567}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 123658868\n",
      "chr3 123377715\n",
      "grep -P -m 1 \"3\\t123377715\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t123377715\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HACD2\n",
      "HACD2  - gene not found in expression data\n",
      "MYLK-AS2\n",
      "MYLK-AS2  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 125603884\n",
      "chr3 125322728\n",
      "grep -P -m 1 \"3\\t125322728\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t125322728\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OSBPL11\n",
      "\n",
      "Probe: 11723792_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00777795723077 0.836669293684\n",
      "\n",
      "Probe: 11723793_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.023704239446 0.529766483875\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 125644635\n",
      "chr3 125363479\n",
      "grep -P -m 1 \"3\\t125363479\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t125363479\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 134693630\n",
      "chr3 134412472\n",
      "grep -P -m 1 \"3\\t134412472\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "EPHB1\n",
      "\n",
      "Probe: 11727656_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0366752387818 0.328456912245\n",
      "\n",
      "Probe: 11727657_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0270958786124 0.470374156407\n",
      "\n",
      "Probe: 11753520_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0354892811169 0.344349502808\n",
      "\n",
      "Probe: 11755617_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0169691726302 0.651246669807\n",
      "\n",
      "Probe: 11763796_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0566551973858 0.130965297685\n",
      "\n",
      "Probe: 11763797_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0623114488898 0.0966391334762\n",
      "{'11763797_s_at': 0.096639133476186689, '11763796_at': 0.13096529768519061}\n",
      "KY\n",
      "\n",
      "Probe: 11731265_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000729857087509 0.984489462537\n",
      "\n",
      "Probe: 11746756_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0287381436015 0.443890473918\n",
      "\n",
      "Probe: 11746757_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0294679982777 0.432398925978\n",
      "\n",
      "Probe: 11750842_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00784595358278 0.834454855385\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 185657070\n",
      "chr3 185374858\n",
      "grep -P -m 1 \"3\\t185374858\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t185374858\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 7910986\n",
      "chr4 7912713\n",
      "grep -P -m 1 \"4\\t7912713\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t7912713\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AFAP1\n",
      "\n",
      "Probe: 11733382_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0147833949871 0.693729737961\n",
      "\n",
      "Probe: 11733383_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0123947757694 0.741274958895\n",
      "\n",
      "Probe: 11739641_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0207870620152 0.579747721354\n",
      "\n",
      "Probe: 11755704_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0102644567005 0.784533582474\n",
      "{}\n",
      "AC097381.1\n",
      "AC097381.1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 112910955\n",
      "chr4 113832111\n",
      "grep -P -m 1 \"4\\t113832111\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t113832111\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 112910956\n",
      "chr4 113832112\n",
      "grep -P -m 1 \"4\\t113832112\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 164250571\n",
      "chr4 165171723\n",
      "grep -P -m 1 \"4\\t165171723\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t165171723\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MARCH1\n",
      "MARCH1  - gene not found in expression data\n",
      "ANP32C\n",
      "\n",
      "Probe: 11742397_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0233670641739 0.533612811059\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 168604221\n",
      "chr4 169525372\n",
      "grep -P -m 1 \"4\\t169525372\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t169525372\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PALLD\n",
      "\n",
      "Probe: 11733083_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0241764458939 0.519530885159\n",
      "\n",
      "Probe: 11733084_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0530057282381 0.157691752342\n",
      "\n",
      "Probe: 11755122_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0283731240007 0.449702360306\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 15795255\n",
      "chr5 15795364\n",
      "grep -P -m 1 \"5\\t15795364\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t15795364\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FBXL7\n",
      "\n",
      "Probe: 11721378_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00384983944251 0.918322973398\n",
      "\n",
      "Probe: 11752259_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00508330110392 0.892294088313\n",
      "\n",
      "Probe: 11756659_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0950128298137 0.0111952540962\n",
      "{'11756659_a_at': 0.011195254096222337}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 38678428\n",
      "chr5 38678530\n",
      "grep -P -m 1 \"5\\t38678530\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38678530\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LIFR\n",
      "\n",
      "Probe: 11722685_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.048546535563 0.195710310447\n",
      "\n",
      "Probe: 11722686_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.06507293787 0.0827140528126\n",
      "\n",
      "Probe: 11722687_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0342795601581 0.361056367459\n",
      "\n",
      "Probe: 11722688_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0320200487178 0.393593308044\n",
      "\n",
      "Probe: 11758145_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00955439812307 0.79910852823\n",
      "{'11722686_a_at': 0.082714052812635622}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 39399620\n",
      "chr5 39399722\n",
      "grep -P -m 1 \"5\\t39399722\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39399722\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 39399621\n",
      "chr5 39399723\n",
      "grep -P -m 1 \"5\\t39399723\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39399723\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 39400498\n",
      "chr5 39400600\n",
      "grep -P -m 1 \"5\\t39400600\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39400600\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 68345575\n",
      "chr5 67641403\n",
      "grep -P -m 1 \"5\\t67641403\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t67641403\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PIK3R1\n",
      "\n",
      "Probe: 11733017_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0426053171831 0.256220434895\n",
      "\n",
      "Probe: 11733018_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0309276287904 0.409940018865\n",
      "\n",
      "Probe: 11733019_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0447949456095 0.232565070467\n",
      "\n",
      "Probe: 11739539_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.022443066641 0.549920802661\n",
      "\n",
      "Probe: 11739540_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0550128449127 0.142523765277\n",
      "\n",
      "Probe: 11739541_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.03621908331 0.334512627059\n",
      "\n",
      "Probe: 11748670_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0367664801923 0.327254199954\n",
      "\n",
      "Probe: 11754631_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0312925643665 0.40443493097\n",
      "{'11739540_a_at': 0.14252376527741648}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 71933681\n",
      "chr5 71229508\n",
      "grep -P -m 1 \"5\\t71229508\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "MAP1B\n",
      "\n",
      "Probe: 11744047_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0429192996421 0.252729570377\n",
      "\n",
      "Probe: 11744048_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0112806220609 0.76380706521\n",
      "\n",
      "Probe: 11744049_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0350363024619 0.35054682739\n",
      "\n",
      "Probe: 11744050_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0434766396626 0.246614793978\n",
      "\n",
      "Probe: 11757766_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0134305288048 0.720524670336\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 72557298\n",
      "chr5 71853125\n",
      "grep -P -m 1 \"5\\t71853125\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t71853125\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 78534216\n",
      "chr5 77830039\n",
      "grep -P -m 1 \"5\\t77830039\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t77830039\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 80109354\n",
      "chr5 79405177\n",
      "grep -P -m 1 \"5\\t79405177\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "THBS4\n",
      "\n",
      "Probe: 11721212_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00820597603379 0.827577037529\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 83641942\n",
      "chr5 82937761\n",
      "grep -P -m 1 \"5\\t82937761\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t82937761\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HAPLN1\n",
      "\n",
      "Probe: 11725374_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0507428869522 0.176221347191\n",
      "\n",
      "Probe: 11725375_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00538022159932 0.886044559827\n",
      "\n",
      "Probe: 11725376_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0793845663463 0.0341864353973\n",
      "{'11725376_at': 0.034186435397262474}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 107973351\n",
      "chr5 107309052\n",
      "grep -P -m 1 \"5\\t107309052\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t107309052\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 131069663\n",
      "chr5 130405356\n",
      "grep -P -m 1 \"5\\t130405356\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t130405356\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LYRM7\n",
      "\n",
      "Probe: 11729685_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.027004653207 0.471870380193\n",
      "\n",
      "Probe: 11729686_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.016969147807 0.651247146235\n",
      "\n",
      "Probe: 11729687_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0250887794123 0.503890363167\n",
      "\n",
      "Probe: 11729688_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00629500550276 0.866836426669\n",
      "\n",
      "Probe: 11729689_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00757224940864 0.840150720869\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 138944295\n",
      "chr5 138279984\n",
      "grep -P -m 1 \"5\\t138279984\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t138279984\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 139864946\n",
      "chr5 139244531\n",
      "grep -P -m 1 \"5\\t139244531\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "PSD2\n",
      "\n",
      "Probe: 11722771_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0467388185441 0.212897787939\n",
      "{}\n",
      "NRG2\n",
      "\n",
      "Probe: 11734615_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0465451312238 0.21480191337\n",
      "\n",
      "Probe: 11740147_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0169783429715 0.651070674305\n",
      "\n",
      "Probe: 11744983_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0552602743782 0.140734065257\n",
      "{'11744983_a_at': 0.14073406525728008}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 142751933\n",
      "chr5 142131498\n",
      "grep -P -m 1 \"5\\t142131498\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t142131498\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ARHGAP26\n",
      "\n",
      "Probe: 11740507_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0169782674739 0.651072123164\n",
      "\n",
      "Probe: 11740508_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0489335800709 0.192166098014\n",
      "\n",
      "Probe: 11740509_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0591334275852 0.114914307813\n",
      "\n",
      "Probe: 11743506_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.015687141734 0.676038224074\n",
      "\n",
      "Probe: 11743507_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.073464980625 0.0500549310819\n",
      "\n",
      "Probe: 11743508_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0451893076926 0.228474519192\n",
      "{'11740509_a_at': 0.11491430781304954, '11743507_a_at': 0.050054931081856213}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 180129591\n",
      "chr5 179556591\n",
      "grep -P -m 1 \"5\\t179556591\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t179556591\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MAPK9\n",
      "\n",
      "Probe: 11720237_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0440848736409 0.240060535394\n",
      "\n",
      "Probe: 11720238_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.021523826491 0.566384291094\n",
      "\n",
      "Probe: 11720239_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00207458135786 0.955931702709\n",
      "\n",
      "Probe: 11740489_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.060746599953 0.105324820395\n",
      "{'11740489_at': 0.10532482039473172}\n",
      "RNF130\n",
      "\n",
      "Probe: 11715617_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0163697952293 0.662791484854\n",
      "{}\n",
      "RASGEF1C\n",
      "\n",
      "Probe: 11738050_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0339713688419 0.365392418383\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 3211059\n",
      "chr6 3211293\n",
      "grep -P -m 1 \"6\\t3211293\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "PSMG4\n",
      "\n",
      "Probe: 11731252_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0291030145967 0.438123966879\n",
      "\n",
      "Probe: 11738272_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0194324194356 0.604695900182\n",
      "\n",
      "Probe: 11738273_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0487180290236 0.194134069585\n",
      "\n",
      "Probe: 11738274_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0538269347026 0.151344543877\n",
      "\n",
      "Probe: 11743499_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0360366553569 0.336954433866\n",
      "\n",
      "Probe: 11743500_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00465283436951 0.901366237519\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 18597220\n",
      "chr6 18597451\n",
      "grep -P -m 1 \"6\\t18597451\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t18597451\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 31822065\n",
      "chr6 31789842\n",
      "grep -P -m 1 \"6\\t31789842\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31789842\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 34223953\n",
      "chr6 34191730\n",
      "grep -P -m 1 \"6\\t34191730\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t34191730\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "C6orf106\n",
      "C6orf106  - gene not found in expression data\n",
      "NUDT3\n",
      "NUDT3  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 36997152\n",
      "chr6 36964928\n",
      "grep -P -m 1 \"6\\t36964928\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t36964928\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 37561432\n",
      "chr6 37529208\n",
      "grep -P -m 1 \"6\\t37529208\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t37529208\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CCDC167\n",
      "\n",
      "Probe: 11717849_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 40995620\n",
      "chr6 40963359\n",
      "grep -P -m 1 \"6\\t40963359\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40963359\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 53360575\n",
      "chr6 53225373\n",
      "grep -P -m 1 \"6\\t53225373\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53225373\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ELOVL5\n",
      "\n",
      "Probe: 11716078_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0703060048433 0.0607896079875\n",
      "\n",
      "Probe: 11757300_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0308359875402 0.411329387192\n",
      "{'11716078_a_at': 0.060789607987497878}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 98844736\n",
      "chr6 99292612\n",
      "grep -P -m 1 \"6\\t99292612\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RP11-436D23.1\n",
      "RP11-436D23.1  - gene not found in expression data\n",
      "MMS22L\n",
      "\n",
      "Probe: 11733445_at\n",
      "Number of REF, HET and ALT samples with expression data: 705 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0926667723828 0.0133745255433\n",
      "\n",
      "Probe: 11733446_at\n",
      "Number of REF, HET and ALT samples with expression data: 705 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0110466105656 0.768565800972\n",
      "{'11733445_at': 0.013374525543297299}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 113826733\n",
      "chr6 114147936\n",
      "grep -P -m 1 \"6\\t114147936\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t114147936\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MARCKS\n",
      "\n",
      "Probe: 11742948_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0343990763545 0.35938355481\n",
      "\n",
      "Probe: 11757816_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.034823879901 0.353477195239\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 130996901\n",
      "chr6 131318041\n",
      "grep -P -m 1 \"6\\t131318041\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t131318041\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 150419419\n",
      "chr6 150740555\n",
      "grep -P -m 1 \"6\\t150740555\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "MTHFD1L\n",
      "\n",
      "Probe: 11728984_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0965817102145 0.0099194724655\n",
      "\n",
      "Probe: 11728985_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0594231200238 0.113143490635\n",
      "\n",
      "Probe: 11728986_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.053337091898 0.155106798628\n",
      "\n",
      "Probe: 11756196_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0044957384682 0.904680324851\n",
      "{'11728985_s_at': 0.1131434906346067, '11728984_a_at': 0.0099194724655025292}\n",
      "PLEKHG1\n",
      "\n",
      "Probe: 11723901_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00574965669741 0.878278663793\n",
      "\n",
      "Probe: 11723902_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0341309260693 0.363143516679\n",
      "\n",
      "Probe: 11723903_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00911378291316 0.80818836885\n",
      "\n",
      "Probe: 11752002_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00501567176896 0.893718486333\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 163181632\n",
      "chr6 163602664\n",
      "grep -P -m 1 \"6\\t163602664\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 218574\n",
      "chr7 218574\n",
      "grep -P -m 1 \"7\\t218574\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t218574\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FAM20C\n",
      "\n",
      "Probe: 11731678_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0204562457997 0.586059946458\n",
      "\n",
      "Probe: 11731679_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0268521132701 0.474689683098\n",
      "{}\n",
      "AC145676.2\n",
      "AC145676.2  - gene not found in expression data\n",
      "ABC7-481722F1.1\n",
      "ABC7-481722F1.1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 23231333\n",
      "chr7 23270952\n",
      "grep -P -m 1 \"7\\t23270952\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t23270952\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "GPNMB\n",
      "\n",
      "Probe: 11724463_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0670916178827 0.0735989772645\n",
      "\n",
      "Probe: 11748750_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0557664626053 0.137126500414\n",
      "\n",
      "Probe: 11748751_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0515057773457 0.169801446723\n",
      "\n",
      "Probe: 11752544_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00515804952982 0.890720152664\n",
      "\n",
      "Probe: 11759826_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0077370748735 0.836719633369\n",
      "{'11748750_a_at': 0.13712650041439811, '11724463_a_at': 0.073598977264501847}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 31043908\n",
      "chr7 31083523\n",
      "grep -P -m 1 \"7\\t31083523\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31083523\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 55085497\n",
      "chr7 55153190\n",
      "grep -P -m 1 \"7\\t55153190\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t55153190\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "EGFR\n",
      "\n",
      "Probe: 11725102_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0420396424408 0.262593655193\n",
      "\n",
      "Probe: 11725103_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0186197662416 0.619889210739\n",
      "\n",
      "Probe: 11725104_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0258460826707 0.491098452641\n",
      "\n",
      "Probe: 11733760_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.02252305028 0.548499508616\n",
      "\n",
      "Probe: 11738815_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0163516571421 0.6631421157\n",
      "\n",
      "Probe: 11742247_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0369230010225 0.325197651714\n",
      "\n",
      "Probe: 11751983_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0134505273908 0.720125975826\n",
      "\n",
      "Probe: 11762479_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0118153775131 0.752966183146\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 55129169\n",
      "chr7 55196862\n",
      "grep -P -m 1 \"7\\t55196862\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t55196862\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 74372405\n",
      "chr7 73786735\n",
      "grep -P -m 1 \"7\\t73786735\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73786735\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 75624931\n",
      "chr7 75254249\n",
      "grep -P -m 1 \"7\\t75254249\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75254249\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 75888142\n",
      "chr7 75517460\n",
      "grep -P -m 1 \"7\\t75517460\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75517460\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RHBDD2\n",
      "\n",
      "Probe: 11716553_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.01117744824 0.766064285315\n",
      "{}\n",
      "POR\n",
      "\n",
      "Probe: 11716114_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00183236779922 0.961099883727\n",
      "\n",
      "Probe: 11746973_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0682099664775 0.069108844778\n",
      "\n",
      "Probe: 11748368_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0370596588455 0.323750189746\n",
      "\n",
      "Probe: 11757517_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0513979420795 0.170999828614\n",
      "{'11746973_x_at': 0.069108844778008202}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 105211569\n",
      "chr7 104852016\n",
      "grep -P -m 1 \"7\\t104852016\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t104852016\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SRPK2\n",
      "\n",
      "Probe: 11721820_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0329141425492 0.380511979901\n",
      "\n",
      "Probe: 11721821_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0447822749049 0.232697352659\n",
      "\n",
      "Probe: 11754831_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0403514902768 0.28225915103\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 130272590\n",
      "chr7 129912430\n",
      "grep -P -m 1 \"7\\t129912430\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t129912430\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MEST\n",
      "\n",
      "Probe: 11717196_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0011644647493 0.975273252043\n",
      "\n",
      "Probe: 11723028_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0110624704272 0.768401717877\n",
      "\n",
      "Probe: 11723029_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0501370525012 0.181754868751\n",
      "\n",
      "Probe: 11751454_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00446380389385 0.90542062854\n",
      "\n",
      "Probe: 11756568_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0188902792505 0.615062687492\n",
      "\n",
      "Probe: 11758268_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0300821461302 0.423190815949\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 139682073\n",
      "chr7 139366819\n",
      "grep -P -m 1 \"7\\t139366819\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t139366819\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TBXAS1\n",
      "\n",
      "Probe: 11721661_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0217554540436 0.56221370926\n",
      "\n",
      "Probe: 11740678_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0169137146977 0.652311421284\n",
      "\n",
      "Probe: 11750999_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0248541181981 0.507889433341\n",
      "\n",
      "Probe: 11760667_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0245313364228 0.513417277747\n",
      "\n",
      "Probe: 11762065_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0616513124198 0.100230830392\n",
      "{'11762065_at': 0.10023083039205902}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 141533583\n",
      "chr7 141233383\n",
      "grep -P -m 1 \"7\\t141233383\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t141233383\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TMEM178B\n",
      "TMEM178B  - gene not found in expression data\n",
      "AGK\n",
      "\n",
      "Probe: 11739123_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0174253111703 0.642516226024\n",
      "\n",
      "Probe: 11739124_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0167866954326 0.654752760738\n",
      "{}\n",
      "MRPS33\n",
      "\n",
      "Probe: 11733134_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.061125432009 0.103167396125\n",
      "{'11733134_a_at': 0.10316739612507493}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 154743157\n",
      "chr7 154534867\n",
      "grep -P -m 1 \"7\\t154534867\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t154534867\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 155468099\n",
      "chr7 155260794\n",
      "grep -P -m 1 \"7\\t155260794\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t155260794\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 158863705\n",
      "chr7 158656396\n",
      "grep -P -m 1 \"7\\t158656396\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t158656396\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 4772808\n",
      "chr8 4630330\n",
      "grep -P -m 1 \"8\\t4630330\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4630330\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CSMD1\n",
      "\n",
      "Probe: 11729708_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.03266109245 0.384186991211\n",
      "\n",
      "Probe: 11729709_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00556516177243 0.882155519\n",
      "\n",
      "Probe: 11750586_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0435178724393 0.246166559661\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 11449345\n",
      "chr8 11306854\n",
      "grep -P -m 1 \"8\\t11306854\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t11306854\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FAM167A\n",
      "\n",
      "Probe: 11723788_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0164018689154 0.662618633413\n",
      "\n",
      "Probe: 11723789_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0145217705571 0.699286989059\n",
      "\n",
      "Probe: 11747054_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.028265739093 0.45205879995\n",
      "\n",
      "Probe: 11749822_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0450566151921 0.23050295047\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 55844201\n",
      "chr8 56756760\n",
      "grep -P -m 1 \"8\\t56756760\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 67109695\n",
      "chr8 68021930\n",
      "grep -P -m 1 \"8\\t68021930\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t68021930\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "COPS5\n",
      "\n",
      "Probe: 11721461_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0227867485422 0.543826501656\n",
      "\n",
      "Probe: 11754844_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0152966610453 0.68366098533\n",
      "{}\n",
      "CSPP1\n",
      "\n",
      "Probe: 11734373_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0978984231984 0.00895028312204\n",
      "\n",
      "Probe: 11743745_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0832876287327 0.0262597564778\n",
      "{'11743745_a_at': 0.026259756477788262, '11734373_a_at': 0.0089502831220359073}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 100366199\n",
      "chr8 101378427\n",
      "grep -P -m 1 \"8\\t101378427\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RNF19A\n",
      "\n",
      "Probe: 11717688_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11717689_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 140472488\n",
      "chr8 141482587\n",
      "grep -P -m 1 \"8\\t141482587\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t141482587\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 143171516\n",
      "chr8 144252933\n",
      "grep -P -m 1 \"8\\t144252933\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t144252933\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 144007388\n",
      "chr8 145081556\n",
      "grep -P -m 1 \"8\\t145081556\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t145081556\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 564124\n",
      "chr9 564124\n",
      "grep -P -m 1 \"9\\t564124\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t564124\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 37938528\n",
      "chr9 37938525\n",
      "grep -P -m 1 \"9\\t37938525\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t37938525\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SLC25A51\n",
      "\n",
      "Probe: 11731128_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00519626244261 0.889915690369\n",
      "{}\n",
      "RP11-613M10.9\n",
      "RP11-613M10.9  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 95463010\n",
      "chr9 98225292\n",
      "grep -P -m 1 \"9\\t98225292\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 96030827\n",
      "chr9 98793109\n",
      "grep -P -m 1 \"9\\t98793109\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t98793109\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ERCC6L2\n",
      "\n",
      "Probe: 11737417_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11737418_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11748716_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11748717_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 98552973\n",
      "chr9 101315255\n",
      "grep -P -m 1 \"9\\t101315255\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101315255\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 136024253\n",
      "chr9 138916099\n",
      "grep -P -m 1 \"9\\t138916099\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t138916099\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NACC2\n",
      "\n",
      "Probe: 11744032_at\n",
      "Number of REF, HET and ALT samples with expression data: 289 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.015109001103 0.797109310072\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 136785822\n",
      "chr9 139680274\n",
      "grep -P -m 1 \"9\\t139680274\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t139680274\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 31133801\n",
      "chr10 31422730\n",
      "grep -P -m 1 \"10\\t31422730\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31422730\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF438\n",
      "\n",
      "Probe: 11737606_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 678 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0153053505945 0.689251397959\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 44975552\n",
      "chr10 45471000\n",
      "grep -P -m 1 \"10\\t45471000\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45471000\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 66942338\n",
      "chr10 68702096\n",
      "grep -P -m 1 \"10\\t68702096\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t68702096\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CTNNA3\n",
      "\n",
      "Probe: 11736588_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0154297807984 0.681058626032\n",
      "\n",
      "Probe: 11738525_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0158511485152 0.672846369324\n",
      "\n",
      "Probe: 11760021_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0178608948981 0.634224928636\n",
      "\n",
      "Probe: 11760919_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00959497929289 0.798273605805\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 71923829\n",
      "chr10 73683587\n",
      "grep -P -m 1 \"10\\t73683587\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73683587\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AC073370.1\n",
      "AC073370.1  - gene not found in expression data\n",
      "CHST3\n",
      "\n",
      "Probe: 11721191_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0424230574553 0.258262064578\n",
      "\n",
      "Probe: 11721192_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0314752059315 0.401696416776\n",
      "\n",
      "Probe: 11721193_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0513637749282 0.170982985387\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 73077420\n",
      "chr10 74837178\n",
      "grep -P -m 1 \"10\\t74837178\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t74837178\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "P4HA1\n",
      "\n",
      "Probe: 11730853_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0056809439466 0.879722213147\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 73888112\n",
      "chr10 75647870\n",
      "grep -P -m 1 \"10\\t75647870\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75647870\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 78864401\n",
      "chr10 80624158\n",
      "grep -P -m 1 \"10\\t80624158\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t80624158\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZMIZ1\n",
      "\n",
      "Probe: 11720334_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0354892362811 0.344350112745\n",
      "\n",
      "Probe: 11720335_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0108566050467 0.77243607529\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 78973767\n",
      "chr10 80733524\n",
      "grep -P -m 1 \"10\\t80733524\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t80733524\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LINC00856\n",
      "LINC00856  - gene not found in expression data\n",
      "ZMIZ1\n",
      "\n",
      "Probe: 11720334_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 680 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0260315439827 0.497337402942\n",
      "\n",
      "Probe: 11720335_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 680 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0670759883581 0.0800395609951\n",
      "{'11720335_s_at': 0.080039560995089326}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 86866412\n",
      "chr10 88626169\n",
      "grep -P -m 1 \"10\\t88626169\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 97597450\n",
      "chr10 99357207\n",
      "grep -P -m 1 \"10\\t99357207\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t99357207\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HOGA1\n",
      "\n",
      "Probe: 11731830_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0595855162147 0.112160229775\n",
      "\n",
      "Probe: 11731831_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0017430221516 0.962969161171\n",
      "{'11731830_a_at': 0.1121602297752585}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 114375147\n",
      "chr10 116134906\n",
      "grep -P -m 1 \"10\\t116134906\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t116134906\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 132132499\n",
      "chr10 133946003\n",
      "grep -P -m 1 \"10\\t133946003\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t133946003\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "JAKMIP3\n",
      "\n",
      "Probe: 11755615_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0215426401175 0.566318812573\n",
      "\n",
      "Probe: 11763596_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00368749322027 0.921810962422\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 132133068\n",
      "chr10 133946572\n",
      "grep -P -m 1 \"10\\t133946572\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t133946572\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "JAKMIP3\n",
      "\n",
      "Probe: 11755615_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11763596_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 1460668\n",
      "chr11 1481898\n",
      "grep -P -m 1 \"11\\t1481898\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t1481898\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "BRSK2\n",
      "\n",
      "Probe: 11749010_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0294500894793 0.436926073967\n",
      "\n",
      "Probe: 11749011_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000375161215865 0.992100283021\n",
      "\n",
      "Probe: 11749012_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0513971272062 0.174675208757\n",
      "\n",
      "Probe: 11750262_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0408925384051 0.280294298881\n",
      "\n",
      "Probe: 11752414_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0609637741557 0.107308216661\n",
      "\n",
      "Probe: 11752415_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00403298392077 0.915236367613\n",
      "\n",
      "Probe: 11754882_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0649028213615 0.086404402822\n",
      "\n",
      "Probe: 11759074_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.056743141376 0.133943075984\n",
      "{'11752414_a_at': 0.10730821666107088, '11754882_x_at': 0.086404402821954718, '11759074_at': 0.1339430759836816}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 12330533\n",
      "chr11 12352080\n",
      "grep -P -m 1 \"11\\t12352080\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t12352080\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TEAD1\n",
      "\n",
      "Probe: 11743817_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0520036129257 0.165707408457\n",
      "\n",
      "Probe: 11752086_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0494259686746 0.187725594296\n",
      "\n",
      "Probe: 11752087_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0558089469105 0.136826985893\n",
      "{'11752087_s_at': 0.13682698589261399}\n",
      "PARVA\n",
      "\n",
      "Probe: 11721080_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0750801507712 0.0452099746477\n",
      "\n",
      "Probe: 11721081_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0247948360198 0.508902336893\n",
      "\n",
      "Probe: 11751491_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0553587311996 0.140026725097\n",
      "\n",
      "Probe: 11758841_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0564634593384 0.132276073322\n",
      "{'11751491_a_at': 0.14002672509651204, '11721080_at': 0.045209974647736233, '11758841_s_at': 0.13227607332239691}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 15651411\n",
      "chr11 15672957\n",
      "grep -P -m 1 \"11\\t15672957\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "SOX6\n",
      "\n",
      "Probe: 11726178_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0611633796843 0.102953232748\n",
      "\n",
      "Probe: 11726179_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0391298793703 0.297096426424\n",
      "\n",
      "Probe: 11726180_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0293931476555 0.433569477565\n",
      "\n",
      "Probe: 11741468_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0688432479101 0.0663697988258\n",
      "\n",
      "Probe: 11741717_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0501466569178 0.181363159855\n",
      "\n",
      "Probe: 11757957_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0540778805642 0.149444218669\n",
      "{'11726178_a_at': 0.10295323274842487, '11757957_s_at': 0.1494442186688035, '11741468_a_at': 0.066369798825800366}\n",
      "INSC\n",
      "\n",
      "Probe: 11732591_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0419641990186 0.263451829666\n",
      "\n",
      "Probe: 11747006_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00219420269126 0.953393511203\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 64688498\n",
      "chr11 64455970\n",
      "grep -P -m 1 \"11\\t64455970\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 70252958\n",
      "chr11 70099064\n",
      "grep -P -m 1 \"11\\t70099064\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70099064\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 71138813\n",
      "chr11 70849859\n",
      "grep -P -m 1 \"11\\t70849859\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 86588759\n",
      "chr11 86299801\n",
      "grep -P -m 1 \"11\\t86299801\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t86299801\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 112844798\n",
      "chr11 112715521\n",
      "grep -P -m 1 \"11\\t112715521\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t112715521\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 120744299\n",
      "chr11 120615008\n",
      "grep -P -m 1 \"11\\t120615008\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t120615008\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 122155721\n",
      "chr11 122026429\n",
      "grep -P -m 1 \"11\\t122026429\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t122026429\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 4269215\n",
      "chr12 4378381\n",
      "grep -P -m 1 \"12\\t4378381\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 4780807\n",
      "chr12 4889973\n",
      "grep -P -m 1 \"12\\t4889973\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "GALNT8\n",
      "\n",
      "Probe: 11737220_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 10 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0076231828622 0.839202026066\n",
      "{}\n",
      "KCNA6\n",
      "\n",
      "Probe: 11740479_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 10 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0270301840083 0.471763975189\n",
      "\n",
      "Probe: 11740480_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 10 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.076348171034 0.0418326277713\n",
      "{'11740480_at': 0.041832627771340579}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 22101550\n",
      "chr12 22254484\n",
      "grep -P -m 1 \"12\\t22254484\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t22254484\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CMAS\n",
      "\n",
      "Probe: 11716699_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0401148886604 0.286454926394\n",
      "\n",
      "Probe: 11748820_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0446232227337 0.23568570662\n",
      "{}\n",
      "ST8SIA1\n",
      "\n",
      "Probe: 11715292_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0613683698379 0.102772456577\n",
      "{'11715292_at': 0.10277245657679791}\n",
      "RP11-49G2.3\n",
      "RP11-49G2.3  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 27225695\n",
      "chr12 27378628\n",
      "grep -P -m 1 \"12\\t27378628\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27378628\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 32140072\n",
      "chr12 32293006\n",
      "grep -P -m 1 \"12\\t32293006\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32293006\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 47722940\n",
      "chr12 48116723\n",
      "grep -P -m 1 \"12\\t48116723\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48116723\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ENDOU\n",
      "\n",
      "Probe: 11730375_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0125709583519 0.737731552553\n",
      "\n",
      "Probe: 11748287_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0586952776358 0.117633859394\n",
      "\n",
      "Probe: 11753124_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0515183538547 0.169697098377\n",
      "{'11748287_a_at': 0.11763385939435055}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 53591118\n",
      "chr12 53984902\n",
      "grep -P -m 1 \"12\\t53984902\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53984902\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-793H13.10\n",
      "RP11-793H13.10  - gene not found in expression data\n",
      "ATF7\n",
      "\n",
      "Probe: 11731103_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11731104_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11749187_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11752966_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11759607_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 58819409\n",
      "chr12 59213191\n",
      "grep -P -m 1 \"12\\t59213191\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t59213191\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-767I20.1\n",
      "RP11-767I20.1  - gene not found in expression data\n",
      "LRIG3\n",
      "\n",
      "Probe: 11724097_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.075270399539 0.044666174136\n",
      "\n",
      "Probe: 11724098_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0450461829297 0.229953140371\n",
      "\n",
      "Probe: 11738147_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0744790822996 0.0469647731283\n",
      "\n",
      "Probe: 11752429_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0391912041262 0.296339413787\n",
      "{'11724097_a_at': 0.044666174135990111, '11738147_a_at': 0.046964773128286619}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 110043011\n",
      "chr12 110480816\n",
      "grep -P -m 1 \"12\\t110480816\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t110480816\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "C12orf76\n",
      "C12orf76  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 110543117\n",
      "chr12 110980922\n",
      "grep -P -m 1 \"12\\t110980922\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "PPTC7\n",
      "\n",
      "Probe: 11739437_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0615369364216 0.100863791429\n",
      "\n",
      "Probe: 11739438_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0358828841792 0.339021528191\n",
      "\n",
      "Probe: 11764020_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0590000214148 0.115737068909\n",
      "{'11739437_at': 0.10086379142852607, '11764020_s_at': 0.11573706890949016}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 110543132\n",
      "chr12 110980937\n",
      "grep -P -m 1 \"12\\t110980937\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t110980937\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PPTC7\n",
      "\n",
      "Probe: 11739437_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0133198949469 0.722731649406\n",
      "\n",
      "Probe: 11739438_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0429703178123 0.252165487442\n",
      "\n",
      "Probe: 11764020_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0619464740779 0.0986119892088\n",
      "{'11764020_s_at': 0.098611989208843301}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 116359336\n",
      "chr12 116797141\n",
      "grep -P -m 1 \"12\\t116797141\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t116797141\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 116359964\n",
      "chr12 116797769\n",
      "grep -P -m 1 \"12\\t116797769\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t116797769\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 116359970\n",
      "chr12 116797775\n",
      "grep -P -m 1 \"12\\t116797775\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t116797775\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-148B3.2\n",
      "RP11-148B3.2  - gene not found in expression data\n",
      "MED13L\n",
      "\n",
      "Probe: 11724814_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.030812268147 0.413342075568\n",
      "\n",
      "Probe: 11724815_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00884919173294 0.814299092579\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 116710846\n",
      "chr12 117148651\n",
      "grep -P -m 1 \"12\\t117148651\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t117148651\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MED13L\n",
      "\n",
      "Probe: 11724814_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0617157343729 0.0998757105359\n",
      "\n",
      "Probe: 11724815_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0200769666831 0.592765398714\n",
      "{'11724814_s_at': 0.099875710535940507}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 35979734\n",
      "chr13 36553871\n",
      "grep -P -m 1 \"13\\t36553871\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t36553871\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 48110886\n",
      "chr13 48685022\n",
      "grep -P -m 1 \"13\\t48685022\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48685022\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MED4\n",
      "\n",
      "Probe: 11720497_at\n",
      "Number of REF, HET and ALT samples with expression data: 641 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.043168561975 0.273635265123\n",
      "\n",
      "Probe: 11758836_at\n",
      "Number of REF, HET and ALT samples with expression data: 641 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00164401175779 0.966760366556\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 79782895\n",
      "chr13 80357030\n",
      "grep -P -m 1 \"13\\t80357030\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t80357030\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NDFIP2\n",
      "\n",
      "Probe: 11727796_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0258586221669 0.49088811807\n",
      "\n",
      "Probe: 11727797_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0399335484948 0.287277862191\n",
      "\n",
      "Probe: 11751348_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.031014003068 0.408633054896\n",
      "\n",
      "Probe: 11757700_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0342462868064 0.361522944966\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 94312722\n",
      "chr13 94964976\n",
      "grep -P -m 1 \"13\\t94964976\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t94964976\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "DCT\n",
      "\n",
      "Probe: 11729446_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 627 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0218090362008 0.585403545802\n",
      "\n",
      "Probe: 11758553_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 627 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0548527846059 0.169782813391\n",
      "\n",
      "Probe: 11759082_at\n",
      "Number of REF, HET and ALT samples with expression data: 627 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0284177455516 0.47716390229\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 114063595\n",
      "chr13 114829070\n",
      "grep -P -m 1 \"13\\t114829070\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t114829070\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RASA3\n",
      "\n",
      "Probe: 11735739_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0138795820752 0.711590735651\n",
      "\n",
      "Probe: 11735740_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0173010603034 0.644889560542\n",
      "\n",
      "Probe: 11735741_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0084568674706 0.821773196424\n",
      "\n",
      "Probe: 11748645_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0454476286323 0.225822934392\n",
      "\n",
      "Probe: 11751530_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0484171037147 0.196906135994\n",
      "{}\n",
      "CDC16\n",
      "\n",
      "Probe: 11716809_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0795332529322 0.033850744743\n",
      "\n",
      "Probe: 11719625_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0786294568175 0.0359353984094\n",
      "\n",
      "Probe: 11746106_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00832774183893 0.824449899173\n",
      "\n",
      "Probe: 11746107_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0162035896324 0.666007186038\n",
      "\n",
      "Probe: 11754804_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0759181448247 0.0428557949015\n",
      "{'11716809_a_at': 0.033850744742960258, '11754804_x_at': 0.042855794901476306, '11719625_a_at': 0.035935398409421601}\n",
      "GAS6-AS1\n",
      "\n",
      "Probe: 11757098_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0272426745068 0.467971997071\n",
      "{}\n",
      "RASA3-IT1\n",
      "RASA3-IT1  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 64693936\n",
      "chr14 65160654\n",
      "grep -P -m 1 \"14\\t65160654\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "PLEKHG3\n",
      "\n",
      "Probe: 11748409_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0128474495674 0.732181947708\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 66471764\n",
      "chr14 66938482\n",
      "grep -P -m 1 \"14\\t66938482\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "GPHN\n",
      "\n",
      "Probe: 11720066_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0546479227021 0.145195062169\n",
      "\n",
      "Probe: 11750722_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0453423116395 0.226901322069\n",
      "{'11720066_a_at': 0.14519506216860664}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 68599442\n",
      "chr14 69066159\n",
      "grep -P -m 1 \"14\\t69066159\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t69066159\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RAD51B\n",
      "\n",
      "Probe: 11734380_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0199478719318 0.595146291932\n",
      "\n",
      "Probe: 11734381_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0181402777232 0.628930829236\n",
      "\n",
      "Probe: 11736967_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.059908167036 0.110226637041\n",
      "\n",
      "Probe: 11737562_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00561638734526 0.88107881182\n",
      "{'11736967_at': 0.11022663704102792}\n",
      "ZFP36L1\n",
      "\n",
      "Probe: 11716115_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0 1.0\n",
      "\n",
      "Probe: 11726889_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0540980470314 0.14929229737\n",
      "\n",
      "Probe: 11746561_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0134276741966 0.720581586378\n",
      "\n",
      "Probe: 11746562_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0288566026882 0.44201359318\n",
      "\n",
      "Probe: 11749937_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0295667076489 0.430858054677\n",
      "\n",
      "Probe: 11758557_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0329237342388 0.38037310637\n",
      "{'11726889_at': 0.14929229737044863}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 77104630\n",
      "chr14 77570973\n",
      "grep -P -m 1 \"14\\t77570973\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t77570973\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "POMT2\n",
      "\n",
      "Probe: 11729527_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0672649650113 0.0728560511089\n",
      "\n",
      "Probe: 11729528_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0390742692915 0.297784009813\n",
      "{'11729527_a_at': 0.072856051108851544}\n",
      "GSTZ1\n",
      "\n",
      "Probe: 11727400_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0162025107629 0.666028079825\n",
      "\n",
      "Probe: 11741826_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00634189189173 0.865853943291\n",
      "\n",
      "Probe: 11741827_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0181255465759 0.629209501916\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 102850033\n",
      "chr14 103316370\n",
      "grep -P -m 1 \"14\\t103316370\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t103316370\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TRAF3\n",
      "\n",
      "Probe: 11742091_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0434108297757 0.247331383436\n",
      "\n",
      "Probe: 11743501_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0656700898775 0.0799272658612\n",
      "\n",
      "Probe: 11759502_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0515866827031 0.169131006497\n",
      "{'11743501_a_at': 0.079927265861204563}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 51938615\n",
      "chr15 52230812\n",
      "grep -P -m 1 \"15\\t52230812\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RP11-56B16.2\n",
      "RP11-56B16.2  - gene not found in expression data\n",
      "TMOD3\n",
      "\n",
      "Probe: 11743613_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0516840940504 0.168326417186\n",
      "\n",
      "Probe: 11761864_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.032003329235 0.393840488045\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 77808233\n",
      "chr15 78100575\n",
      "grep -P -m 1 \"15\\t78100575\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t78100575\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LINGO1\n",
      "\n",
      "Probe: 11728336_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.042204036476 0.261403378379\n",
      "\n",
      "Probe: 11728337_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0396756178803 0.291085858595\n",
      "{}\n",
      "LINGO1-AS1\n",
      "LINGO1-AS1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 83991917\n",
      "chr15 84660669\n",
      "grep -P -m 1 \"15\\t84660669\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t84660669\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "UBE2Q2L\n",
      "UBE2Q2L  - gene not found in expression data\n",
      "ADAMTSL3\n",
      "\n",
      "Probe: 11728117_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00291943982752 0.938016051861\n",
      "{}\n",
      "RP11-342A23.2\n",
      "RP11-342A23.2  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 100114163\n",
      "chr15 100654368\n",
      "grep -P -m 1 \"15\\t100654368\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t100654368\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 68497\n",
      "chr16 118495\n",
      "grep -P -m 1 \"16\\t118495\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RHBDF1\n",
      "\n",
      "Probe: 11722079_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0517741545852 0.167585097315\n",
      "\n",
      "Probe: 11746177_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0365388996967 0.330259409711\n",
      "\n",
      "Probe: 11746178_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0369262031744 0.325155665982\n",
      "\n",
      "Probe: 11746179_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0327300608613 0.383183230533\n",
      "{}\n",
      "AXIN1\n",
      "\n",
      "Probe: 11754511_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0488690373006 0.192753829258\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 24702112\n",
      "chr16 24713433\n",
      "grep -P -m 1 \"16\\t24713433\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t24713433\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TNRC6A\n",
      "\n",
      "Probe: 11717817_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00611565155549 0.884979725704\n",
      "\n",
      "Probe: 11717818_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0109301017608 0.795985787727\n",
      "\n",
      "Probe: 11748942_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0581637430189 0.168525560938\n",
      "\n",
      "Probe: 11751728_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0368240021946 0.3835786111\n",
      "\n",
      "Probe: 11755871_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00871805102096 0.836620175012\n",
      "\n",
      "Probe: 11762292_at\n",
      "Number of REF, HET and ALT samples with expression data: 561 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0372144701953 0.378552180905\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 70737977\n",
      "chr16 70771880\n",
      "grep -P -m 1 \"16\\t70771880\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70771880\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MTSS1L\n",
      "\n",
      "Probe: 11721225_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000453807616009 0.990369150549\n",
      "\n",
      "Probe: 11721226_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00525119361063 0.888915351303\n",
      "{}\n",
      "VAC14\n",
      "\n",
      "Probe: 11719751_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0171149678283 0.648913391512\n",
      "\n",
      "Probe: 11719752_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0527063169624 0.160645477318\n",
      "\n",
      "Probe: 11750690_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0117989545993 0.753635081731\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 73001404\n",
      "chr16 73035303\n",
      "grep -P -m 1 \"16\\t73035303\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73035303\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 2434687\n",
      "chr17 2337981\n",
      "grep -P -m 1 \"17\\t2337981\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t2337981\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-235E17.2\n",
      "RP11-235E17.2  - gene not found in expression data\n",
      "METTL16\n",
      "\n",
      "Probe: 11723683_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0249833102384 0.505685702873\n",
      "\n",
      "Probe: 11745145_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00180757118092 0.961598842935\n",
      "\n",
      "Probe: 11750880_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0453831323176 0.226482906808\n",
      "\n",
      "Probe: 11752652_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0146542709178 0.696271370847\n",
      "\n",
      "Probe: 11755254_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00413159559753 0.91236854247\n",
      "{}\n",
      "MNT\n",
      "\n",
      "Probe: 11744056_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00367970642734 0.921920677918\n",
      "\n",
      "Probe: 11752262_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0328591292831 0.381309091153\n",
      "{}\n",
      "SHPK\n",
      "\n",
      "Probe: 11727946_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00961886732289 0.797782237833\n",
      "\n",
      "Probe: 11751239_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00897333991651 0.811087993947\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 17032283\n",
      "chr17 16935597\n",
      "grep -P -m 1 \"17\\t16935597\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t16935597\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 17567008\n",
      "chr17 17470322\n",
      "grep -P -m 1 \"17\\t17470322\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17470322\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PEMT\n",
      "\n",
      "Probe: 11731113_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0507250199403 0.176373837007\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 50930610\n",
      "chr17 49007971\n",
      "grep -P -m 1 \"17\\t49007971\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t49007971\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 56578360\n",
      "chr17 54655721\n",
      "grep -P -m 1 \"17\\t54655721\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t54655721\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NOG\n",
      "\n",
      "Probe: 11740638_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0359144298485 0.338937442397\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 57576773\n",
      "chr17 55654134\n",
      "grep -P -m 1 \"17\\t55654134\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 57907861\n",
      "chr17 55985222\n",
      "grep -P -m 1 \"17\\t55985222\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t55985222\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 59380388\n",
      "chr17 57457749\n",
      "grep -P -m 1 \"17\\t57457749\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "YPEL2\n",
      "\n",
      "Probe: 11721957_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.032181474745 0.39154564229\n",
      "\n",
      "Probe: 11721958_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0127994922752 0.733324488496\n",
      "\n",
      "Probe: 11721959_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0321814682984 0.391545737212\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 66374447\n",
      "chr17 64370565\n",
      "grep -P -m 1 \"17\\t64370565\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t64370565\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PRKCA\n",
      "\n",
      "Probe: 11717471_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0357097549677 0.34135854141\n",
      "\n",
      "Probe: 11717472_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0253714150075 0.499095823761\n",
      "\n",
      "Probe: 11754557_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0769579198109 0.0400793071137\n",
      "{'11754557_s_at': 0.0400793071137273}\n",
      "RN7SL735P\n",
      "RN7SL735P  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 66518319\n",
      "chr17 64514437\n",
      "grep -P -m 1 \"17\\t64514437\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "PRKCA\n",
      "\n",
      "Probe: 11717471_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.043188079487 0.249767653727\n",
      "\n",
      "Probe: 11717472_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0355705667156 0.343244840864\n",
      "\n",
      "Probe: 11754557_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0524841474064 0.161826371647\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 72408934\n",
      "chr17 70405075\n",
      "grep -P -m 1 \"17\\t70405075\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RP11-57A1.1\n",
      "RP11-57A1.1  - gene not found in expression data\n",
      "LINC00511\n",
      "LINC00511  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 75826405\n",
      "chr17 73822486\n",
      "grep -P -m 1 \"17\\t73822486\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 77367028\n",
      "chr17 75363110\n",
      "grep -P -m 1 \"17\\t75363110\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75363110\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-936I5.1\n",
      "RP11-936I5.1  - gene not found in expression data\n",
      "SEPT9\n",
      "SEPT9  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 13576449\n",
      "chr18 13576448\n",
      "grep -P -m 1 \"18\\t13576448\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 13578665\n",
      "chr18 13578664\n",
      "grep -P -m 1 \"18\\t13578664\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t13578664\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 28086160\n",
      "chr18 25666124\n",
      "grep -P -m 1 \"18\\t25666124\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t25666124\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 2167742\n",
      "chr19 2167741\n",
      "grep -P -m 1 \"19\\t2167741\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t2167741\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 4058396\n",
      "chr19 4058394\n",
      "grep -P -m 1 \"19\\t4058394\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4058394\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZBTB7A\n",
      "\n",
      "Probe: 11726696_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0321815268554 0.391544875007\n",
      "\n",
      "Probe: 11726697_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0546721247617 0.145300385723\n",
      "\n",
      "Probe: 11726698_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0354728667479 0.344913074148\n",
      "\n",
      "Probe: 11726699_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00786252477928 0.834225527923\n",
      "{'11726697_at': 0.14530038572292245}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 5037839\n",
      "chr19 5037850\n",
      "grep -P -m 1 \"19\\t5037850\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t5037850\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KDM4B\n",
      "\n",
      "Probe: 11743512_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0562930232362 0.133449715761\n",
      "\n",
      "Probe: 11747909_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0682358858091 0.068807340676\n",
      "\n",
      "Probe: 11755420_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0156225822191 0.677296262963\n",
      "\n",
      "Probe: 11761992_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.043123617222 0.250475803146\n",
      "{'11747909_x_at': 0.068807340675974063, '11743512_x_at': 0.13344971576086992}\n",
      "TINCR\n",
      "TINCR  - gene not found in expression data\n",
      "SNRPEP4\n",
      "SNRPEP4  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 6515973\n",
      "chr19 6515984\n",
      "grep -P -m 1 \"19\\t6515984\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t6515984\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 7224268\n",
      "chr19 7224279\n",
      "grep -P -m 1 \"19\\t7224279\\t\" out_files/enh_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 10888846\n",
      "chr19 10999522\n",
      "grep -P -m 1 \"19\\t10999522\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t10999522\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CARM1\n",
      "\n",
      "Probe: 11754363_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0502687961814 0.180300897906\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 11458313\n",
      "chr19 11569128\n",
      "grep -P -m 1 \"19\\t11569128\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t11569128\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ELAVL3\n",
      "\n",
      "Probe: 11728871_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0212554880982 0.571777269542\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 38666228\n",
      "chr19 39156868\n",
      "grep -P -m 1 \"19\\t39156868\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39156868\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 437517\n",
      "chr20 418161\n",
      "grep -P -m 1 \"20\\t418161\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t418161\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "Y_RNA\n",
      "Y_RNA  - gene not found in expression data\n",
      "TBC1D20\n",
      "\n",
      "Probe: 11728738_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0592877863014 0.113968062878\n",
      "\n",
      "Probe: 11728739_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0140358303998 0.708491338736\n",
      "\n",
      "Probe: 11728740_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0682344549871 0.0688131685449\n",
      "{'11728740_a_at': 0.068813168544881773, '11728738_x_at': 0.11396806287847062}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 30284068\n",
      "chr20 29518744\n",
      "grep -P -m 1 \"20\\t29518744\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t29518744\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 33693104\n",
      "chr20 32280910\n",
      "grep -P -m 1 \"20\\t32280910\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32280910\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 34709316\n",
      "chr20 33297120\n",
      "grep -P -m 1 \"20\\t33297120\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t33297120\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 46206927\n",
      "chr20 44835566\n",
      "grep -P -m 1 \"20\\t44835566\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t44835566\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CDH22\n",
      "\n",
      "Probe: 11729010_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0260570966566 0.487565395807\n",
      "\n",
      "Probe: 11729011_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0347603770618 0.354356208349\n",
      "\n",
      "Probe: 11761606_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0353405426325 0.346376695996\n",
      "{}\n",
      "RP5-998H6.2\n",
      "RP5-998H6.2  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 50573586\n",
      "chr20 49190123\n",
      "grep -P -m 1 \"20\\t49190123\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t49190123\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LINC01273\n",
      "LINC01273  - gene not found in expression data\n",
      "PTPN1\n",
      "\n",
      "Probe: 11717146_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11717147_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 712 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "RP11-112L6.2\n",
      "RP11-112L6.2  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 42523896\n",
      "chr21 43944006\n",
      "grep -P -m 1 \"21\\t43944006\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t43944006\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 45554469\n",
      "chr21 46974383\n",
      "grep -P -m 1 \"21\\t46974383\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t46974383\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 28034954\n",
      "chr22 28430942\n",
      "grep -P -m 1 \"22\\t28430942\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t28430942\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 30872281\n",
      "chr22 31268268\n",
      "grep -P -m 1 \"22\\t31268268\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31268268\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OSBP2\n",
      "\n",
      "Probe: 11736205_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0021626264281 0.954063469023\n",
      "\n",
      "Probe: 11736206_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00358679481298 0.923886104153\n",
      "\n",
      "Probe: 11750705_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00490546970276 0.896040275211\n",
      "{}\n",
      "MORC2\n",
      "\n",
      "Probe: 11715125_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0570723290359 0.128148408772\n",
      "{'11715125_at': 0.12814840877152076}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 41412963\n",
      "chr22 41808967\n",
      "grep -P -m 1 \"22\\t41808967\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t41808967\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 42386115\n",
      "chr22 42782121\n",
      "grep -P -m 1 \"22\\t42782121\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42782121\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 49967521\n",
      "chr22 50361169\n",
      "grep -P -m 1 \"22\\t50361169\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t50361169\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PIM3\n",
      "\n",
      "Probe: 11716071_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00888163281881 0.813241705218\n",
      "{}\n",
      "MIR6821\n",
      "MIR6821  - gene not found in expression data\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 39689917\n",
      "chrX 39549171\n",
      "grep -P -m 1 \"23\\t39549171\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39549171\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "BCOR\n",
      "\n",
      "Probe: 11724716_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 680 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11741799_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 680 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11746812_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 680 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 47362705\n",
      "chrX 47222104\n",
      "grep -P -m 1 \"23\\t47222104\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t47222104\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF157\n",
      "\n",
      "Probe: 11740574_at\n",
      "Number of REF, HET and ALT samples with expression data: 678 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0374234153189 0.330200645192\n",
      "\n",
      "Probe: 11759353_at\n",
      "Number of REF, HET and ALT samples with expression data: 678 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0563308039051 0.14256464966\n",
      "{'11759353_at': 0.14256464966008286}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 53072145\n",
      "chrX 53101327\n",
      "grep -P -m 1 \"23\\t53101327\\t\" out_files/enh_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "GPR173\n",
      "\n",
      "Probe: 11740335_at\n",
      "Number of REF, HET and ALT samples with expression data: 643 2 2\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0108264290401 0.783424797978\n",
      "{}\n",
      "TSPYL2\n",
      "\n",
      "Probe: 11719030_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 643 2 2\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0336529318563 0.392778534521\n",
      "\n",
      "Probe: 11752279_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 643 2 2\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0585267429826 0.136990033911\n",
      "\n",
      "Probe: 11757628_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 643 2 2\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0711955013349 0.070336460131\n",
      "{'11752279_s_at': 0.13699003391063988, '11757628_s_at': 0.070336460131000741}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 129932525\n",
      "chrX 129066501\n",
      "grep -P -m 1 \"23\\t129066501\\t\" out_files/enh_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t129066501\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "#corrds:Gene:probe:pvalue\n",
    "enh_with_hits[\"ADNI_data\"] = enh_with_hits.apply(check_ADNI, args = (\"out_files/enh_var_in_adni.txt\",), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>Contacting protein gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>POS-1</th>\n",
       "      <th>Target_genes</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "      <th>ADNI_data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>1000</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>., ENSG00000157933:SKI, ENSG00000157933:SKI, E...</td>\n",
       "      <td>2227532</td>\n",
       "      <td>ENST00000449373_MORN1:0.0014,ENST00000488353_R...</td>\n",
       "      <td>MORN1:1.19,RER1:55.89</td>\n",
       "      <td>MORN1:3.76,RER1:25.86</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>93950</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000131697:NPHP4, ENSG00000266687:AL35626...</td>\n",
       "      <td>5896825</td>\n",
       "      <td>ENST00000481789_KCNAB2:0.0277,ENST00000378161_...</td>\n",
       "      <td>KCNAB2:69.76,NPHP4:6.44</td>\n",
       "      <td>KCNAB2:59.77,NPHP4:5.59</td>\n",
       "      <td>1:5956886:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>255366</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...</td>\n",
       "      <td>7041937</td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16912415</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>65540</td>\n",
       "      <td>ENSG00000206652:RNU1-1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000058453:CROCC, ENSG00000272426:RP11-10...</td>\n",
       "      <td>16913967</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "      <td>1:17240463:</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE         ...          \\\n",
       "0           0.000100           0.000300         ...           \n",
       "1           0.000700           0.000700         ...           \n",
       "2           0.000032           0.000067         ...           \n",
       "3           0.000097           0.000067         ...           \n",
       "4           0.004400           0.007600         ...           \n",
       "\n",
       "   dist. to closest protein gene  \\\n",
       "0                           1000   \n",
       "1                          93950   \n",
       "2                         255366   \n",
       "3                          65540   \n",
       "4                          65540   \n",
       "\n",
       "                                     Contacting gene  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...   \n",
       "1                                                  .   \n",
       "2                                                  .   \n",
       "3                             ENSG00000206652:RNU1-1   \n",
       "4                             ENSG00000206652:RNU1-1   \n",
       "\n",
       "                             Contacting protein gene polgenom_freq_perc  \\\n",
       "0  ENSG00000157933:SKI, ENSG00000116151:MORN1, EN...                NaN   \n",
       "1                                                  .                NaN   \n",
       "2                                                  .                NaN   \n",
       "3                                                  .                NaN   \n",
       "4                                                  .                NaN   \n",
       "\n",
       "                                           all_genes     POS-1  \\\n",
       "0  ., ENSG00000157933:SKI, ENSG00000157933:SKI, E...   2227532   \n",
       "1  ENSG00000131697:NPHP4, ENSG00000266687:AL35626...   5896825   \n",
       "2  ENSG00000171735:CAMTA1, ENSG00000237365:RP11-3...   7041937   \n",
       "3  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16912415   \n",
       "4  ENSG00000058453:CROCC, ENSG00000272426:RP11-10...  16913967   \n",
       "\n",
       "                                        Target_genes  \\\n",
       "0  ENST00000449373_MORN1:0.0014,ENST00000488353_R...   \n",
       "1  ENST00000481789_KCNAB2:0.0277,ENST00000378161_...   \n",
       "2                                               None   \n",
       "3                       ENST00000467938_CROCC:0.0053   \n",
       "4                       ENST00000467938_CROCC:0.0053   \n",
       "\n",
       "   target_expr_in_lymphocytes     target_expr_in_brain           ADNI_data  \n",
       "0       MORN1:1.19,RER1:55.89    MORN1:3.76,RER1:25.86  no variant in ADNI  \n",
       "1     KCNAB2:69.76,NPHP4:6.44  KCNAB2:59.77,NPHP4:5.59          1:5956886:  \n",
       "2                                                                     None  \n",
       "3                  CROCC:1.52              CROCC:15.56  no variant in ADNI  \n",
       "4                  CROCC:1.52              CROCC:15.56         1:17240463:  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_with_hits.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 939561\n",
      "chr1 874941\n",
      "grep -P -m 1 \"1\\t874941\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t874941\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 1034902\n",
      "chr1 970282\n",
      "grep -P -m 1 \"1\\t970282\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 6180416\n",
      "chr1 6240476\n",
      "grep -P -m 1 \"1\\t6240476\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 14924017\n",
      "chr1 15250513\n",
      "grep -P -m 1 \"1\\t15250513\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "KAZN\n",
      "\n",
      "Probe: 11719884_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 589 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0413282601 0.315855171733\n",
      "\n",
      "Probe: 11719885_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 589 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.01665084932 0.686241142546\n",
      "\n",
      "Probe: 11728168_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 589 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.012808370016 0.756004905299\n",
      "\n",
      "Probe: 11749857_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 589 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00128084035226 0.975212160543\n",
      "\n",
      "Probe: 11754962_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 589 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00495256405334 0.904368203555\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 21653564\n",
      "chr1 21980057\n",
      "grep -P -m 1 \"1\\t21980057\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 25268534\n",
      "chr1 25595025\n",
      "grep -P -m 1 \"1\\t25595025\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RSRP1\n",
      "RSRP1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 26889274\n",
      "chr1 27215765\n",
      "grep -P -m 1 \"1\\t27215765\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 26901253\n",
      "chr1 27227744\n",
      "grep -P -m 1 \"1\\t27227744\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27227744\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "GPATCH3\n",
      "\n",
      "Probe: 11719068_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0365784101034 0.329736398355\n",
      "\n",
      "Probe: 11763764_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.02557215769 0.495705236826\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 27490564\n",
      "chr1 27817075\n",
      "grep -P -m 1 \"1\\t27817075\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27817075\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 28737136\n",
      "chr1 29063648\n",
      "grep -P -m 1 \"1\\t29063648\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t29063648\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 32178833\n",
      "chr1 32644434\n",
      "grep -P -m 1 \"1\\t32644434\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TXLNA\n",
      "\n",
      "Probe: 11729720_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0763779348029 0.0416086320249\n",
      "\n",
      "Probe: 11750379_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00458899789403 0.902712725046\n",
      "\n",
      "Probe: 11756485_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0017406548655 0.963019418641\n",
      "{'11729720_a_at': 0.041608632024935671}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 32221069\n",
      "chr1 32686670\n",
      "grep -P -m 1 \"1\\t32686670\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32686670\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "EIF3I\n",
      "\n",
      "Probe: 11747337_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0269383603831 0.473271400197\n",
      "{}\n",
      " TMEM234\n",
      " TMEM234  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 32221885\n",
      "chr1 32687486\n",
      "grep -P -m 1 \"1\\t32687486\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32687486\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 33081220\n",
      "chr1 33546821\n",
      "grep -P -m 1 \"1\\t33546821\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t33546821\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AZIN2\n",
      "AZIN2  - gene not found in expression data\n",
      " AK2\n",
      " AK2  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 35268566\n",
      "chr1 35734167\n",
      "grep -P -m 1 \"1\\t35734167\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35734167\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 36324190\n",
      "chr1 36789791\n",
      "grep -P -m 1 \"1\\t36789791\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "EVA1B\n",
      "\n",
      "Probe: 11744562_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00957421273563 0.79870083104\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 37753888\n",
      "chr1 38219560\n",
      "grep -P -m 1 \"1\\t38219560\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38219560\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 38005745\n",
      "chr1 38471417\n",
      "grep -P -m 1 \"1\\t38471417\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38471417\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 40318303\n",
      "chr1 40783975\n",
      "grep -P -m 1 \"1\\t40783975\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40783975\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "COL9A2\n",
      "\n",
      "Probe: 11737065_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 574 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0585902414543 0.160589146824\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 40980295\n",
      "chr1 41445967\n",
      "grep -P -m 1 \"1\\t41445967\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t41445967\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 43172584\n",
      "chr1 43638255\n",
      "grep -P -m 1 \"1\\t43638255\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 44801300\n",
      "chr1 45266972\n",
      "grep -P -m 1 \"1\\t45266972\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45266972\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PLK3\n",
      "\n",
      "Probe: 11756358_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0415067860244 0.269033888727\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 46247507\n",
      "chr1 46713179\n",
      "grep -P -m 1 \"1\\t46713179\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 46247512\n",
      "chr1 46713184\n",
      "grep -P -m 1 \"1\\t46713184\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 89994867\n",
      "chr1 90460426\n",
      "grep -P -m 1 \"1\\t90460426\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t90460426\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF326\n",
      "\n",
      "Probe: 11732398_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0344483201208 0.359373976194\n",
      "\n",
      "Probe: 11735465_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0436104246795 0.24582884452\n",
      "\n",
      "Probe: 11746311_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00861207966926 0.8188099378\n",
      "\n",
      "Probe: 11746312_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0221715316886 0.5553178273\n",
      "{}\n",
      " GEMIN8P4\n",
      " GEMIN8P4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 93181611\n",
      "chr1 93647168\n",
      "grep -P -m 1 \"1\\t93647168\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t93647168\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 93847377\n",
      "chr1 94312933\n",
      "grep -P -m 1 \"1\\t94312933\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "BCAR3\n",
      "\n",
      "Probe: 11719608_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00281088163038 0.940358519128\n",
      "\n",
      "Probe: 11755006_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0268248884163 0.475137862892\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 95117558\n",
      "chr1 95583114\n",
      "grep -P -m 1 \"1\\t95583114\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t95583114\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 109249723\n",
      "chr1 109792345\n",
      "grep -P -m 1 \"1\\t109792345\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t109792345\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CELSR2\n",
      "\n",
      "Probe: 11734849_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00674650849974 0.85898433138\n",
      "\n",
      "Probe: 11734850_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0323966131935 0.393454648866\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 154956132\n",
      "chr1 154928608\n",
      "grep -P -m 1 \"1\\t154928608\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t154928608\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 160098877\n",
      "chr1 160068667\n",
      "grep -P -m 1 \"1\\t160068667\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 160098882\n",
      "chr1 160068672\n",
      "grep -P -m 1 \"1\\t160068672\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 184051639\n",
      "chr1 184020773\n",
      "grep -P -m 1 \"1\\t184020773\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t184020773\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TSEN15\n",
      "\n",
      "Probe: 11720555_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0317487030863 0.397616446558\n",
      "\n",
      "Probe: 11720556_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0307451562153 0.412709224617\n",
      "\n",
      "Probe: 11739912_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0190674614182 0.611498538611\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 198157062\n",
      "chr1 198126192\n",
      "grep -P -m 1 \"1\\t198126192\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t198126192\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr1_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NEK7\n",
      "\n",
      "Probe: 11717450_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0137729578892 0.714287798034\n",
      "\n",
      "Probe: 11717451_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0128633954315 0.732408003998\n",
      "\n",
      "Probe: 11717452_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00993989563314 0.791618368505\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr1 205344578\n",
      "chr1 205313706\n",
      "grep -P -m 1 \"1\\t205313706\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 8681782\n",
      "chr2 8821912\n",
      "grep -P -m 1 \"2\\t8821912\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 8681796\n",
      "chr2 8821926\n",
      "grep -P -m 1 \"2\\t8821926\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 8681803\n",
      "chr2 8821933\n",
      "grep -P -m 1 \"2\\t8821933\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 15591914\n",
      "chr2 15732038\n",
      "grep -P -m 1 \"2\\t15732038\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t15732038\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 17538730\n",
      "chr2 17719997\n",
      "grep -P -m 1 \"2\\t17719997\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "VSNL1\n",
      "\n",
      "Probe: 11723418_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.000968345832942 0.979422209014\n",
      "\n",
      "Probe: 11749996_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0770156976587 0.0399295963258\n",
      "\n",
      "Probe: 11749997_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00103290194708 0.978050696702\n",
      "{'11749996_a_at': 0.039929596325833705}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 27357824\n",
      "chr2 27580691\n",
      "grep -P -m 1 \"2\\t27580691\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27580691\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "GTF3C2\n",
      "\n",
      "Probe: 11743502_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00587462020498 0.875654410523\n",
      "\n",
      "Probe: 11743503_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.002453135722 0.947900976608\n",
      "\n",
      "Probe: 11744752_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0647499633101 0.0842537480128\n",
      "{'11744752_a_at': 0.084253748012780058}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 27495900\n",
      "chr2 27718767\n",
      "grep -P -m 1 \"2\\t27718767\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27718767\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FNDC4\n",
      "\n",
      "Probe: 11719330_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0636527163408 0.0896582104416\n",
      "{'11719330_s_at': 0.089658210441602063}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 36355201\n",
      "chr2 36582344\n",
      "grep -P -m 1 \"2\\t36582344\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t36582344\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP11-490M8.1\n",
      "RP11-490M8.1  - gene not found in expression data\n",
      " CRIM1\n",
      " CRIM1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 38602871\n",
      "chr2 38830013\n",
      "grep -P -m 1 \"2\\t38830013\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38830013\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HNRNPLL\n",
      "\n",
      "Probe: 11733135_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0134107039296 0.72167497416\n",
      "\n",
      "Probe: 11739651_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0209192983525 0.578413499396\n",
      "\n",
      "Probe: 11739652_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00977863902596 0.795065015931\n",
      "\n",
      "Probe: 11752508_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0400924376298 0.286724925866\n",
      "\n",
      "Probe: 11761935_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0579035281811 0.123735483645\n",
      "{'11761935_at': 0.1237354836454744}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 47905786\n",
      "chr2 48132925\n",
      "grep -P -m 1 \"2\\t48132925\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48132925\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 70190965\n",
      "chr2 70418097\n",
      "grep -P -m 1 \"2\\t70418097\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70418097\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 71453104\n",
      "chr2 71680234\n",
      "grep -P -m 1 \"2\\t71680234\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t71680234\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 84972001\n",
      "chr2 85199124\n",
      "grep -P -m 1 \"2\\t85199124\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "KCMF1\n",
      "\n",
      "Probe: 11718518_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0780337084582 0.0391548272522\n",
      "\n",
      "Probe: 11718519_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0613855020301 0.104895014437\n",
      "\n",
      "Probe: 11718520_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0533057201205 0.159188010508\n",
      "{'11718519_at': 0.10489501443706845, '11718518_at': 0.039154827252227062}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 85583623\n",
      "chr2 85810746\n",
      "grep -P -m 1 \"2\\t85810746\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t85810746\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 97995962\n",
      "chr2 98612425\n",
      "grep -P -m 1 \"2\\t98612425\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t98612425\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 102355017\n",
      "chr2 102971477\n",
      "grep -P -m 1 \"2\\t102971477\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t102971477\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "IL18R1\n",
      "\n",
      "Probe: 11735275_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0430430182594 0.251363188459\n",
      "\n",
      "Probe: 11735276_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00822400112828 0.826601864973\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 111884949\n",
      "chr2 112642526\n",
      "grep -P -m 1 \"2\\t112642526\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t112642526\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 120735263\n",
      "chr2 121492839\n",
      "grep -P -m 1 \"2\\t121492839\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t121492839\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 181657900\n",
      "chr2 182522627\n",
      "grep -P -m 1 \"2\\t182522627\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t182522627\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CERKL\n",
      "\n",
      "Probe: 11731984_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00484170811791 0.897384041389\n",
      "\n",
      "Probe: 11738856_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.079404210466 0.034141922608\n",
      "\n",
      "Probe: 11752268_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0316970544143 0.398385014818\n",
      "{'11738856_a_at': 0.034141922608024185}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 196800461\n",
      "chr2 197665185\n",
      "grep -P -m 1 \"2\\t197665185\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 200306434\n",
      "chr2 201171157\n",
      "grep -P -m 1 \"2\\t201171157\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t201171157\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SPATS2L\n",
      "\n",
      "Probe: 11719364_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0473493671777 0.206975186542\n",
      "\n",
      "Probe: 11719365_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.022899228816 0.541839264714\n",
      "\n",
      "Probe: 11736293_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.051637292803 0.168712623274\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 202635226\n",
      "chr2 203499949\n",
      "grep -P -m 1 \"2\\t203499949\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "FAM117B\n",
      "\n",
      "Probe: 11726459_at\n",
      "Number of REF, HET and ALT samples with expression data: 691 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0443461159908 0.240280976628\n",
      "\n",
      "Probe: 11726460_at\n",
      "Number of REF, HET and ALT samples with expression data: 691 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0217807863083 0.564250456906\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 206083841\n",
      "chr2 206948565\n",
      "grep -P -m 1 \"2\\t206948565\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t206948565\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 209578668\n",
      "chr2 210443392\n",
      "grep -P -m 1 \"2\\t210443392\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 218292312\n",
      "chr2 219157035\n",
      "grep -P -m 1 \"2\\t219157035\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr2 239402361\n",
      "chr2 240324055\n",
      "grep -P -m 1 \"2\\t240324055\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t240324055\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr2_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 9946995\n",
      "chr3 9988679\n",
      "grep -P -m 1 \"3\\t9988679\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t9988679\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PRRT3-AS1\n",
      "PRRT3-AS1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 10115323\n",
      "chr3 10157007\n",
      "grep -P -m 1 \"3\\t10157007\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t10157007\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 10141590\n",
      "chr3 10183274\n",
      "grep -P -m 1 \"3\\t10183274\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t10183274\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "VHL\n",
      "\n",
      "Probe: 11748633_a_at\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "Number of REF, HET and ALT samples with expression data: 16 148 499\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.198937898191 2.41146881943e-07\n",
      "\n",
      "Probe: 11752180_x_at\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "1/2\n",
      "Number of REF, HET and ALT samples with expression data: 16 148 499\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0358134733466 0.357202512606\n",
      "{'11748633_a_at': 2.4114688194263059e-07}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 10164287\n",
      "chr3 10205971\n",
      "grep -P -m 1 \"3\\t10205971\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 32570194\n",
      "chr3 32611686\n",
      "grep -P -m 1 \"3\\t32611686\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32611686\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "DYNC1LI1\n",
      "\n",
      "Probe: 11720786_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0734580147227 0.0505640375405\n",
      "\n",
      "Probe: 11720787_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0321173339849 0.393159670231\n",
      "{'11720786_a_at': 0.050564037540536601}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 37176944\n",
      "chr3 37218435\n",
      "grep -P -m 1 \"3\\t37218435\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 39155156\n",
      "chr3 39196647\n",
      "grep -P -m 1 \"3\\t39196647\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39196647\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CSRNP1\n",
      "\n",
      "Probe: 11721543_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0566479540886 0.1318319741\n",
      "\n",
      "Probe: 11757721_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0637175029678 0.0900111805042\n",
      "{'11757721_s_at': 0.09001118050423465, '11721543_a_at': 0.13183197409991415}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 42601962\n",
      "chr3 42643454\n",
      "grep -P -m 1 \"3\\t42643454\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42643454\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 48635663\n",
      "chr3 48673096\n",
      "grep -P -m 1 \"3\\t48673096\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48673096\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SLC26A6\n",
      "\n",
      "Probe: 11718818_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0521185179314 0.165071004177\n",
      "\n",
      "Probe: 11733413_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00549728495881 0.883664038567\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 50359509\n",
      "chr3 50396940\n",
      "grep -P -m 1 \"3\\t50396940\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TMEM115\n",
      "\n",
      "Probe: 11718421_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0136799901105 0.716132734631\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 50567523\n",
      "chr3 50604954\n",
      "grep -P -m 1 \"3\\t50604954\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t50604954\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 56556938\n",
      "chr3 56590966\n",
      "grep -P -m 1 \"3\\t56590966\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t56590966\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CCDC66\n",
      "\n",
      "Probe: 11727244_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 677 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.012309696858 0.748292397753\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 71584193\n",
      "chr3 71633344\n",
      "grep -P -m 1 \"3\\t71633344\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t71633344\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 99817392\n",
      "chr3 99536236\n",
      "grep -P -m 1 \"3\\t99536236\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t99536236\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 113746527\n",
      "chr3 113465374\n",
      "grep -P -m 1 \"3\\t113465374\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t113465374\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NAA50\n",
      "\n",
      "Probe: 11721799_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0189488248501 0.613717127871\n",
      "\n",
      "Probe: 11721800_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0119122841202 0.75100675628\n",
      "\n",
      "Probe: 11751697_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0349888798725 0.351199685241\n",
      "\n",
      "Probe: 11756128_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0300023247151 0.424096071067\n",
      "{}\n",
      " ATP6V1A\n",
      " ATP6V1A  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 138608079\n",
      "chr3 138326921\n",
      "grep -P -m 1 \"3\\t138326921\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t138326921\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FAIM\n",
      "\n",
      "Probe: 11732554_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 604 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0533523516106 0.190022047908\n",
      "\n",
      "Probe: 11741829_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 604 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0493917187689 0.225090187767\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 160398904\n",
      "chr3 160116692\n",
      "grep -P -m 1 \"3\\t160116692\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t160116692\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SMC4\n",
      "\n",
      "Probe: 11733874_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0287471234064 0.443748037871\n",
      "\n",
      "Probe: 11733875_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.070575561778 0.0598048580067\n",
      "\n",
      "Probe: 11733876_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0179867651495 0.631837446313\n",
      "\n",
      "Probe: 11754320_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 5.27469899968e-05 0.998878978726\n",
      "{'11733875_a_at': 0.059804858006663866}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 182980156\n",
      "chr3 182697944\n",
      "grep -P -m 1 \"3\\t182697944\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 184135238\n",
      "chr3 183853026\n",
      "grep -P -m 1 \"3\\t183853026\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t183853026\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr3 186570696\n",
      "chr3 186288485\n",
      "grep -P -m 1 \"3\\t186288485\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t186288485\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr3_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "DNAJB11\n",
      "\n",
      "Probe: 11723655_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 704 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0693003687189 0.0647710776573\n",
      "{'11723655_a_at': 0.064771077657293341}\n",
      " TBCCD1\n",
      " TBCCD1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 8441483\n",
      "chr4 8443210\n",
      "grep -P -m 1 \"4\\t8443210\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t8443210\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ACOX3\n",
      "\n",
      "Probe: 11737303_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0409285926941 0.275427264863\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 26319183\n",
      "chr4 26320805\n",
      "grep -P -m 1 \"4\\t26320805\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t26320805\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RBPJ\n",
      "\n",
      "Probe: 11739392_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00993671528774 0.791827595196\n",
      "\n",
      "Probe: 11739393_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.021253507104 0.572357050631\n",
      "\n",
      "Probe: 11757737_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0019321399657 0.959070446183\n",
      "\n",
      "Probe: 11762366_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0278780030089 0.458924697748\n",
      "\n",
      "Probe: 11762367_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0442552425431 0.239574522256\n",
      "\n",
      "Probe: 11764053_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0328463527617 0.382839824596\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 55125585\n",
      "chr4 55991752\n",
      "grep -P -m 1 \"4\\t55991752\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t55991752\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 55546494\n",
      "chr4 56412661\n",
      "grep -P -m 1 \"4\\t56412661\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 83456767\n",
      "chr4 84377920\n",
      "grep -P -m 1 \"4\\t84377920\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t84377920\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HELQ\n",
      "\n",
      "Probe: 11723889_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00738979146387 0.843952389577\n",
      "\n",
      "Probe: 11756856_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0199798178428 0.594556708311\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 86358480\n",
      "chr4 87279633\n",
      "grep -P -m 1 \"4\\t87279633\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t87279633\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 86593730\n",
      "chr4 87514883\n",
      "grep -P -m 1 \"4\\t87514883\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t87514883\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PTPN13\n",
      "\n",
      "Probe: 11722317_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0135935740188 0.717276398972\n",
      "\n",
      "Probe: 11730524_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0447949359293 0.232565171508\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr4 118278643\n",
      "chr4 119199798\n",
      "grep -P -m 1 \"4\\t119199798\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t119199798\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr4_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SNHG8\n",
      "\n",
      "Probe: 11757112_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0131013544728 0.727467527216\n",
      "\n",
      "Probe: 11763889_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.013651076643 0.71651508985\n",
      "{}\n",
      " SNORA24\n",
      " SNORA24  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 62404701\n",
      "chr5 61700528\n",
      "grep -P -m 1 \"5\\t61700528\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t61700528\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 62404707\n",
      "chr5 61700534\n",
      "grep -P -m 1 \"5\\t61700534\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 80408310\n",
      "chr5 79704129\n",
      "grep -P -m 1 \"5\\t79704129\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t79704129\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 80960575\n",
      "chr5 80256394\n",
      "grep -P -m 1 \"5\\t80256394\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t80256394\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RASGRF2\n",
      "\n",
      "Probe: 11736820_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0854312686437 0.022910162611\n",
      "{'11736820_a_at': 0.02291016261097422}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 80961674\n",
      "chr5 80257493\n",
      "grep -P -m 1 \"5\\t80257493\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t80257493\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 90528507\n",
      "chr5 89824324\n",
      "grep -P -m 1 \"5\\t89824324\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t89824324\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 144204348\n",
      "chr5 143583911\n",
      "grep -P -m 1 \"5\\t143583911\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t143583911\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KCTD16\n",
      "\n",
      "Probe: 11738606_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0102180753215 0.785483387102\n",
      "\n",
      "Probe: 11738607_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0227168739757 0.545062824632\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 147453962\n",
      "chr5 146833525\n",
      "grep -P -m 1 \"5\\t146833525\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr5 178940592\n",
      "chr5 178367593\n",
      "grep -P -m 1 \"5\\t178367593\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t178367593\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr5_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF454\n",
      "\n",
      "Probe: 11740453_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0133128651152 0.723246754991\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 4134591\n",
      "chr6 4134825\n",
      "grep -P -m 1 \"6\\t4134825\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4134825\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ECI2\n",
      "\n",
      "Probe: 11754652_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0190164348624 0.612452337628\n",
      "\n",
      "Probe: 11761967_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.045849822162 0.221738299359\n",
      "{}\n",
      " RP3-400B16.4\n",
      " RP3-400B16.4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 26382814\n",
      "chr6 26383042\n",
      "grep -P -m 1 \"6\\t26383042\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t26383042\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 26595976\n",
      "chr6 26596204\n",
      "grep -P -m 1 \"6\\t26596204\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t26596204\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 29628104\n",
      "chr6 29595881\n",
      "grep -P -m 1 \"6\\t29595881\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t29595881\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 31356312\n",
      "chr6 31324089\n",
      "grep -P -m 1 \"6\\t31324089\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31324089\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MIR6891\n",
      "MIR6891  - gene not found in expression data\n",
      " HLA-B\n",
      " HLA-B  - gene not found in expression data\n",
      " XXbac-BPG248L24.12\n",
      " XXbac-BPG248L24.12  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 31666030\n",
      "chr6 31633807\n",
      "grep -P -m 1 \"6\\t31633807\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31633807\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "GPANK1\n",
      "\n",
      "Probe: 11744087_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0283982116064 0.450903640404\n",
      "\n",
      "Probe: 11754350_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0466542277532 0.215348859987\n",
      "{}\n",
      " CSNK2B\n",
      " CSNK2B  - gene not found in expression data\n",
      " XXbac-BPG32J3.22\n",
      " XXbac-BPG32J3.22  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 42048635\n",
      "chr6 42016373\n",
      "grep -P -m 1 \"6\\t42016373\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42016373\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CCND3\n",
      "\n",
      "Probe: 11754434_a_at\n",
      "0/2\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0456353610215 0.225219961957\n",
      "\n",
      "Probe: 11762304_a_at\n",
      "0/2\n",
      "Number of REF, HET and ALT samples with expression data: 707 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.025485916663 0.498375586873\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 43427781\n",
      "chr6 43395519\n",
      "grep -P -m 1 \"6\\t43395519\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t43395519\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ABCC10\n",
      "\n",
      "Probe: 11724636_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00234376626494 0.950360994103\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 43688733\n",
      "chr6 43656470\n",
      "grep -P -m 1 \"6\\t43656470\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t43656470\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 53064185\n",
      "chr6 52928983\n",
      "grep -P -m 1 \"6\\t52928983\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t52928983\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FBXO9\n",
      "\n",
      "Probe: 11719191_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0301909429168 0.421842828335\n",
      "\n",
      "Probe: 11719192_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.018059287077 0.630945905335\n",
      "\n",
      "Probe: 11741750_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0316037550337 0.400440720309\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 84227481\n",
      "chr6 84937199\n",
      "grep -P -m 1 \"6\\t84937199\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t84937199\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CEP162\n",
      "CEP162  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 87322775\n",
      "chr6 88032493\n",
      "grep -P -m 1 \"6\\t88032493\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t88032493\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr6 144008499\n",
      "chr6 144329636\n",
      "grep -P -m 1 \"6\\t144329636\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t144329636\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr6_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PLAGL1\n",
      "\n",
      "Probe: 11734150_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0148784483206 0.693721946337\n",
      "\n",
      "Probe: 11737052_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.000185980567072 0.996072555974\n",
      "\n",
      "Probe: 11741152_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00418456207258 0.911812106915\n",
      "\n",
      "Probe: 11745820_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.020085909222 0.594956879124\n",
      "\n",
      "Probe: 11745821_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0183191492884 0.627752281243\n",
      "{}\n",
      " HYMAI\n",
      " HYMAI  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 7566622\n",
      "chr7 7606253\n",
      "grep -P -m 1 \"7\\t7606253\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t7606253\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 28180233\n",
      "chr7 28219852\n",
      "grep -P -m 1 \"7\\t28219852\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 44200775\n",
      "chr7 44240374\n",
      "grep -P -m 1 \"7\\t44240374\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t44240374\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 44574794\n",
      "chr7 44614393\n",
      "grep -P -m 1 \"7\\t44614393\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 45769578\n",
      "chr7 45809177\n",
      "grep -P -m 1 \"7\\t45809177\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45809177\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SEPT7P2\n",
      "SEPT7P2  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 66995291\n",
      "chr7 66460278\n",
      "grep -P -m 1 \"7\\t66460278\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t66460278\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SBDS\n",
      "SBDS  - gene not found in expression data\n",
      " TYW1\n",
      " TYW1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 72925331\n",
      "chr7 72395869\n",
      "grep -P -m 1 \"7\\t72395869\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RP11-313P13.5\n",
      "RP11-313P13.5  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 73522483\n",
      "chr7 72936813\n",
      "grep -P -m 1 \"7\\t72936813\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t72936813\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 74964567\n",
      "chr7 74653584\n",
      "grep -P -m 1 \"7\\t74653584\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 98253208\n",
      "chr7 97882520\n",
      "grep -P -m 1 \"7\\t97882520\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TECPR1\n",
      "\n",
      "Probe: 11754843_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00588702055667 0.875481201614\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 100895922\n",
      "chr7 100493543\n",
      "grep -P -m 1 \"7\\t100493543\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 100895925\n",
      "chr7 100493546\n",
      "grep -P -m 1 \"7\\t100493546\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 100895943\n",
      "chr7 100493564\n",
      "grep -P -m 1 \"7\\t100493564\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t100493564\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 102365587\n",
      "chr7 102006034\n",
      "grep -P -m 1 \"7\\t102006034\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t102006034\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 105692438\n",
      "chr7 105332885\n",
      "grep -P -m 1 \"7\\t105332885\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t105332885\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 128531249\n",
      "chr7 128171303\n",
      "grep -P -m 1 \"7\\t128171303\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 129224942\n",
      "chr7 128864783\n",
      "grep -P -m 1 \"7\\t128864783\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t128864783\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 130205309\n",
      "chr7 129845149\n",
      "grep -P -m 1 \"7\\t129845149\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TMEM209\n",
      "\n",
      "Probe: 11727860_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0174438943777 0.642396221003\n",
      "\n",
      "Probe: 11752484_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 8 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0283585375481 0.450255016083\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 131328120\n",
      "chr7 131012879\n",
      "grep -P -m 1 \"7\\t131012879\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t131012879\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MKLN1-AS\n",
      "MKLN1-AS  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 134986488\n",
      "chr7 134671239\n",
      "grep -P -m 1 \"7\\t134671239\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t134671239\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AGBL3\n",
      "\n",
      "Probe: 11740768_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0791192591502 0.0349206406352\n",
      "{'11740768_at': 0.03492064063517538}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr7 148884204\n",
      "chr7 148581296\n",
      "grep -P -m 1 \"7\\t148581296\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t148581296\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr7_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 42541539\n",
      "chr8 42396682\n",
      "grep -P -m 1 \"8\\t42396682\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42396682\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 64798880\n",
      "chr8 65711437\n",
      "grep -P -m 1 \"8\\t65711437\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t65711437\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CYP7B1\n",
      "\n",
      "Probe: 11740151_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.011215507297 0.765451870408\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 101205131\n",
      "chr8 102217359\n",
      "grep -P -m 1 \"8\\t102217359\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t102217359\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 140458967\n",
      "chr8 141469066\n",
      "grep -P -m 1 \"8\\t141469066\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t141469066\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 142449370\n",
      "chr8 143530731\n",
      "grep -P -m 1 \"8\\t143530731\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 143610457\n",
      "chr8 144692627\n",
      "grep -P -m 1 \"8\\t144692627\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t144692627\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PYCRL\n",
      "\n",
      "Probe: 11739164_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0269704074076 0.472432734051\n",
      "\n",
      "Probe: 11739165_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0359301383532 0.338385443778\n",
      "\n",
      "Probe: 11749442_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0629920525247 0.0930444052808\n",
      "\n",
      "Probe: 11749443_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0191993345071 0.609036585762\n",
      "\n",
      "Probe: 11751408_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000731402058199 0.984456633776\n",
      "{'11749442_x_at': 0.093044405280842193}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 143610460\n",
      "chr8 144692630\n",
      "grep -P -m 1 \"8\\t144692630\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t144692630\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PYCRL\n",
      "\n",
      "Probe: 11739164_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0269704074076 0.472432734051\n",
      "\n",
      "Probe: 11739165_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0359301383532 0.338385443778\n",
      "\n",
      "Probe: 11749442_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0629920525247 0.0930444052808\n",
      "\n",
      "Probe: 11749443_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0191993345071 0.609036585762\n",
      "\n",
      "Probe: 11751408_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000731402058199 0.984456633776\n",
      "{'11749442_x_at': 0.093044405280842193}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 143610462\n",
      "chr8 144692632\n",
      "grep -P -m 1 \"8\\t144692632\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t144692632\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PYCRL\n",
      "\n",
      "Probe: 11739164_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0246856307124 0.510770995497\n",
      "\n",
      "Probe: 11739165_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.039507635722 0.29245376131\n",
      "\n",
      "Probe: 11749442_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0796482739808 0.033592994114\n",
      "\n",
      "Probe: 11749443_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0535384516057 0.153551759062\n",
      "\n",
      "Probe: 11751408_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0196746903905 0.600198882459\n",
      "{'11749442_x_at': 0.033592994114035692}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 143944588\n",
      "chr8 145018756\n",
      "grep -P -m 1 \"8\\t145018756\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t145018756\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr8 144950488\n",
      "chr8 146175874\n",
      "grep -P -m 1 \"8\\t146175874\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t146175874\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr8_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF16\n",
      "\n",
      "Probe: 11734904_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 691 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0359690130537 0.344415392759\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 4742306\n",
      "chr9 4742306\n",
      "grep -P -m 1 \"9\\t4742306\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4742306\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AK3\n",
      "\n",
      "Probe: 11723438_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0353918474165 0.345676595355\n",
      "\n",
      "Probe: 11748857_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0438204049831 0.242895207921\n",
      "\n",
      "Probe: 11757768_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0409154325798 0.2755817921\n",
      "\n",
      "Probe: 11758884_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0281907248466 0.452622596334\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 17578340\n",
      "chr9 17578338\n",
      "grep -P -m 1 \"9\\t17578338\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17578338\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 27529965\n",
      "chr9 27529963\n",
      "grep -P -m 1 \"9\\t27529963\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t27529963\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 32551161\n",
      "chr9 32551159\n",
      "grep -P -m 1 \"9\\t32551159\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TOPORS-AS1\n",
      "TOPORS-AS1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 33264577\n",
      "chr9 33264575\n",
      "grep -P -m 1 \"9\\t33264575\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t33264575\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CHMP5\n",
      "\n",
      "Probe: 11718655_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 687 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0424544598472 0.265769687271\n",
      "\n",
      "Probe: 11760624_at\n",
      "Number of REF, HET and ALT samples with expression data: 687 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00922334243539 0.809037965445\n",
      "{}\n",
      " BAG1\n",
      " BAG1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 89498162\n",
      "chr9 92113077\n",
      "grep -P -m 1 \"9\\t92113077\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 95003181\n",
      "chr9 97765463\n",
      "grep -P -m 1 \"9\\t97765463\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t97765463\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 97039811\n",
      "chr9 99802093\n",
      "grep -P -m 1 \"9\\t99802093\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 113875959\n",
      "chr9 116638239\n",
      "grep -P -m 1 \"9\\t116638239\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 120843877\n",
      "chr9 123606155\n",
      "grep -P -m 1 \"9\\t123606155\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t123606155\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 123269003\n",
      "chr9 126031282\n",
      "grep -P -m 1 \"9\\t126031282\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t126031282\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "STRBP\n",
      "\n",
      "Probe: 11727157_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.064773834343 0.0845774637148\n",
      "\n",
      "Probe: 11727158_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0645905791119 0.0854628147214\n",
      "\n",
      "Probe: 11756674_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0537796669102 0.152282746884\n",
      "{'11727157_a_at': 0.084577463714843884, '11727158_a_at': 0.085462814721428179}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 131531342\n",
      "chr9 134406729\n",
      "grep -P -m 1 \"9\\t134406729\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t134406729\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "UCK1\n",
      "\n",
      "Probe: 11719617_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0451307925242 0.229735556443\n",
      "\n",
      "Probe: 11734900_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00503219896651 0.893519859186\n",
      "\n",
      "Probe: 11748851_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0388273699153 0.30153184464\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr9 136992242\n",
      "chr9 139886694\n",
      "grep -P -m 1 \"9\\t139886694\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t139886694\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr9_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 135080\n",
      "chr10 181020\n",
      "grep -P -m 1 \"10\\t181020\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t181020\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZMYND11\n",
      "\n",
      "Probe: 11743611_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.013159033772 0.728929299784\n",
      "\n",
      "Probe: 11744367_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0405458445045 0.285435073314\n",
      "\n",
      "Probe: 11744368_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0432845879381 0.254116445126\n",
      "\n",
      "Probe: 11747871_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0587814471197 0.121306381251\n",
      "{'11747871_a_at': 0.12130638125125159}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 3066951\n",
      "chr10 3109143\n",
      "grep -P -m 1 \"10\\t3109143\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t3109143\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 3067811\n",
      "chr10 3110003\n",
      "grep -P -m 1 \"10\\t3110003\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t3110003\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 43408029\n",
      "chr10 43903477\n",
      "grep -P -m 1 \"10\\t43903477\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "HNRNPF\n",
      "\n",
      "Probe: 11725970_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0461980268484 0.218244857852\n",
      "\n",
      "Probe: 11725971_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0329665419052 0.379753693357\n",
      "\n",
      "Probe: 11725972_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00908262676574 0.808831401936\n",
      "\n",
      "Probe: 11725973_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0285186734809 0.44737975917\n",
      "\n",
      "Probe: 11739142_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0646248903887 0.0848561825477\n",
      "\n",
      "Probe: 11741133_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0333028496587 0.374908986695\n",
      "\n",
      "Probe: 11749786_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0467960412777 0.212337569814\n",
      "{'11739142_x_at': 0.084856182547686787}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 47300103\n",
      "chr10 48439257\n",
      "grep -P -m 1 \"10\\t48439257\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48439257\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 69179246\n",
      "chr10 70939002\n",
      "grep -P -m 1 \"10\\t70939002\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70939002\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SUPV3L1\n",
      "\n",
      "Probe: 11718860_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.014506370806 0.699590391081\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 69450743\n",
      "chr10 71210499\n",
      "grep -P -m 1 \"10\\t71210499\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t71210499\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TSPAN15\n",
      "\n",
      "Probe: 11752612_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0551953949903 0.141201672566\n",
      "{'11752612_at': 0.14120167256633889}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 72217495\n",
      "chr10 73977253\n",
      "grep -P -m 1 \"10\\t73977253\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73977253\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 75111805\n",
      "chr10 76871563\n",
      "grep -P -m 1 \"10\\t76871563\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t76871563\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SAMD8\n",
      "\n",
      "Probe: 11741306_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0278448354769 0.459141493437\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 93482373\n",
      "chr10 95242130\n",
      "grep -P -m 1 \"10\\t95242130\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t95242130\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MYOF\n",
      "\n",
      "Probe: 11718168_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00834674248818 0.824055897599\n",
      "\n",
      "Probe: 11754928_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00810124986006 0.829149858397\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 96831659\n",
      "chr10 98591416\n",
      "grep -P -m 1 \"10\\t98591416\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t98591416\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LCOR\n",
      "\n",
      "Probe: 11728429_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0772085726548 0.0394332529763\n",
      "{'11728429_a_at': 0.039433252976319742}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 96832965\n",
      "chr10 98592722\n",
      "grep -P -m 1 \"10\\t98592722\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 96832970\n",
      "chr10 98592727\n",
      "grep -P -m 1 \"10\\t98592727\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t98592727\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 98268428\n",
      "chr10 100028185\n",
      "grep -P -m 1 \"10\\t100028185\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 98268431\n",
      "chr10 100028188\n",
      "grep -P -m 1 \"10\\t100028188\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t100028188\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 99731530\n",
      "chr10 101491287\n",
      "grep -P -m 1 \"10\\t101491287\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101491287\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr10_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CUTC\n",
      "\n",
      "Probe: 11728166_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0120019618412 0.749536802649\n",
      "\n",
      "Probe: 11753345_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0203391863136 0.588469361075\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 102056802\n",
      "chr10 103816559\n",
      "grep -P -m 1 \"10\\t103816559\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 132186178\n",
      "chr10 133999682\n",
      "grep -P -m 1 \"10\\t133999682\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "RP11-140A10.3\n",
      "RP11-140A10.3  - gene not found in expression data\n",
      " DPYSL4\n",
      " DPYSL4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr10 132943730\n",
      "chr10 134757234\n",
      "grep -P -m 1 \"10\\t134757234\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "CFAP46\n",
      "CFAP46  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 806131\n",
      "chr11 806131\n",
      "grep -P -m 1 \"11\\t806131\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 810639\n",
      "chr11 810639\n",
      "grep -P -m 1 \"11\\t810639\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 1310283\n",
      "chr11 1331513\n",
      "grep -P -m 1 \"11\\t1331513\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t1331513\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TOLLIP\n",
      "\n",
      "Probe: 11722171_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0354854921389 0.345081985452\n",
      "\n",
      "Probe: 11722172_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0140931962472 0.707748382645\n",
      "\n",
      "Probe: 11755301_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0423255176238 0.260031817079\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 10294294\n",
      "chr11 10315841\n",
      "grep -P -m 1 \"11\\t10315841\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t10315841\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SBF2\n",
      "\n",
      "Probe: 11739476_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0654654941562 0.0837063257599\n",
      "\n",
      "Probe: 11739477_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0211965541424 0.57584286863\n",
      "\n",
      "Probe: 11739478_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00347023176666 0.927028430582\n",
      "{'11739476_a_at': 0.083706325759939054}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 22831185\n",
      "chr11 22852731\n",
      "grep -P -m 1 \"11\\t22852731\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t22852731\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SVIP\n",
      "\n",
      "Probe: 11730791_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00316325190657 0.932851308979\n",
      "\n",
      "Probe: 11730792_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0553891463933 0.139808766944\n",
      "\n",
      "Probe: 11759039_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0211098502014 0.573874840588\n",
      "{'11730792_at': 0.13980876694398953}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 57596398\n",
      "chr11 57363871\n",
      "grep -P -m 1 \"11\\t57363871\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57363871\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 57648836\n",
      "chr11 57416309\n",
      "grep -P -m 1 \"11\\t57416309\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57416309\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CLP1\n",
      "\n",
      "Probe: 11735630_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00196564489391 0.958243565444\n",
      "{}\n",
      " AP000662.4\n",
      " AP000662.4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 57656862\n",
      "chr11 57424334\n",
      "grep -P -m 1 \"11\\t57424334\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57424334\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CLP1\n",
      "\n",
      "Probe: 11735630_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0583528474564 0.120055320538\n",
      "{'11735630_a_at': 0.12005532053799492}\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 61751995\n",
      "chr11 61519467\n",
      "grep -P -m 1 \"11\\t61519467\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t61519467\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MYRF\n",
      "\n",
      "Probe: 11719340_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00690261180086 0.854120753869\n",
      "\n",
      "Probe: 11748190_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0564093804566 0.132647599325\n",
      "{'11748190_a_at': 0.13264759932498579}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 62841646\n",
      "chr11 62609118\n",
      "grep -P -m 1 \"11\\t62609118\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t62609118\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 65833779\n",
      "chr11 65601250\n",
      "grep -P -m 1 \"11\\t65601250\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t65601250\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SNX32\n",
      "\n",
      "Probe: 11731749_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0539854661465 0.151006663795\n",
      "\n",
      "Probe: 11747462_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0321343085563 0.392909511705\n",
      "\n",
      "Probe: 11751825_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0184543462656 0.623737958862\n",
      "\n",
      "Probe: 11753110_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0349804180308 0.352336578848\n",
      "\n",
      "Probe: 11762337_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0149653763228 0.690775373829\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 65871991\n",
      "chr11 65639462\n",
      "grep -P -m 1 \"11\\t65639462\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t65639462\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "EFEMP2\n",
      "\n",
      "Probe: 11728351_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0203876801626 0.587316483637\n",
      "\n",
      "Probe: 11728352_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00265131268902 0.943738490067\n",
      "\n",
      "Probe: 11750280_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0362041195019 0.335053333778\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 66288392\n",
      "chr11 66055863\n",
      "grep -P -m 1 \"11\\t66055863\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t66055863\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 67022580\n",
      "chr11 66790051\n",
      "grep -P -m 1 \"11\\t66790051\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t66790051\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 67119871\n",
      "chr11 66887342\n",
      "grep -P -m 1 \"11\\t66887342\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t66887342\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "KDM2A\n",
      "\n",
      "Probe: 11717166_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0832425127088 0.0263414415911\n",
      "\n",
      "Probe: 11717167_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0479983067457 0.200811988053\n",
      "{'11717166_a_at': 0.026341441591136116}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 72782967\n",
      "chr11 72494012\n",
      "grep -P -m 1 \"11\\t72494012\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t72494012\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "STARD10\n",
      "\n",
      "Probe: 11715807_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0116109777571 0.757104291153\n",
      "\n",
      "Probe: 11758732_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0420098313447 0.26293252812\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 78962455\n",
      "chr11 78673500\n",
      "grep -P -m 1 \"11\\t78673500\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t78673500\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 102046984\n",
      "chr11 101917715\n",
      "grep -P -m 1 \"11\\t101917715\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101917715\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 102046989\n",
      "chr11 101917720\n",
      "grep -P -m 1 \"11\\t101917720\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101917720\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr11 111872089\n",
      "chr11 111742813\n",
      "grep -P -m 1 \"11\\t111742813\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t111742813\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr11_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ALG9\n",
      "\n",
      "Probe: 11726400_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0393210166002 0.294741218618\n",
      "\n",
      "Probe: 11726401_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0419666960267 0.263423395151\n",
      "\n",
      "Probe: 11726402_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00775471666587 0.836352575537\n",
      "\n",
      "Probe: 11747122_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0546479803869 0.145194636902\n",
      "{'11747122_x_at': 0.14519463690221537}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 389591\n",
      "chr12 498757\n",
      "grep -P -m 1 \"12\\t498757\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t498757\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CCDC77\n",
      "\n",
      "Probe: 11725486_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 165 354 192\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0110402634323 0.768853416214\n",
      "\n",
      "Probe: 11741269_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 165 354 192\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00979834716819 0.794234868395\n",
      "{}\n",
      " KDM5A\n",
      " KDM5A  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 10213010\n",
      "chr12 10365609\n",
      "grep -P -m 1 \"12\\t10365609\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 12685403\n",
      "chr12 12838337\n",
      "grep -P -m 1 \"12\\t12838337\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 48104277\n",
      "chr12 48498060\n",
      "grep -P -m 1 \"12\\t48498060\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 48104281\n",
      "chr12 48498064\n",
      "grep -P -m 1 \"12\\t48498064\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 57520951\n",
      "chr12 57914734\n",
      "grep -P -m 1 \"12\\t57914734\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57914734\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MBD6\n",
      "\n",
      "Probe: 11722250_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "\n",
      "Probe: 11751582_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 0 0\n",
      "Samples represent only one genotype. Exit.\n",
      "{}\n",
      " DDIT3\n",
      " DDIT3  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 57782965\n",
      "chr12 58176748\n",
      "grep -P -m 1 \"12\\t58176748\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t58176748\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 100266981\n",
      "chr12 100660759\n",
      "grep -P -m 1 \"12\\t100660759\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "SCYL2\n",
      "\n",
      "Probe: 11739335_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 19 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0391354078676 0.297028127995\n",
      "\n",
      "Probe: 11739336_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 19 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0217725111205 0.561907173334\n",
      "\n",
      "Probe: 11758463_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 19 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0473398715123 0.2070663759\n",
      "\n",
      "Probe: 11759298_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 19 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0020564094698 0.956317323522\n",
      "{}\n",
      " DEPDC4\n",
      " DEPDC4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 109880224\n",
      "chr12 110318029\n",
      "grep -P -m 1 \"12\\t110318029\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t110318029\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP1-7G5.6\n",
      "RP1-7G5.6  - gene not found in expression data\n",
      " GLTP\n",
      " GLTP  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 120302046\n",
      "chr12 120739849\n",
      "grep -P -m 1 \"12\\t120739849\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t120739849\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 120369619\n",
      "chr12 120807422\n",
      "grep -P -m 1 \"12\\t120807422\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t120807422\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 122499967\n",
      "chr12 122984514\n",
      "grep -P -m 1 \"12\\t122984514\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t122984514\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 122501440\n",
      "chr12 122985987\n",
      "grep -P -m 1 \"12\\t122985987\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t122985987\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 123585043\n",
      "chr12 124069590\n",
      "grep -P -m 1 \"12\\t124069590\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t124069590\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr12 130870920\n",
      "chr12 131355465\n",
      "grep -P -m 1 \"12\\t131355465\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t131355465\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr12_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 44576053\n",
      "chr13 45150189\n",
      "grep -P -m 1 \"13\\t45150189\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45150189\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TSC22D1-AS1\n",
      "TSC22D1-AS1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 45419149\n",
      "chr13 45993284\n",
      "grep -P -m 1 \"13\\t45993284\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45993284\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 60397285\n",
      "chr13 60971419\n",
      "grep -P -m 1 \"13\\t60971419\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t60971419\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 75635909\n",
      "chr13 76210045\n",
      "grep -P -m 1 \"13\\t76210045\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t76210045\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 78659327\n",
      "chr13 79233462\n",
      "grep -P -m 1 \"13\\t79233462\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t79233462\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 99086886\n",
      "chr13 99739140\n",
      "grep -P -m 1 \"13\\t99739140\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 108215003\n",
      "chr13 108867351\n",
      "grep -P -m 1 \"13\\t108867351\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t108867351\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr13_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr13 110561783\n",
      "chr13 111214130\n",
      "grep -P -m 1 \"13\\t111214130\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 20342721\n",
      "chr14 20810880\n",
      "grep -P -m 1 \"14\\t20810880\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t20810880\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*No target genes - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 31457439\n",
      "chr14 31926645\n",
      "grep -P -m 1 \"14\\t31926645\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 39171489\n",
      "chr14 39640693\n",
      "grep -P -m 1 \"14\\t39640693\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39640693\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 51988674\n",
      "chr14 52455392\n",
      "grep -P -m 1 \"14\\t52455392\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t52455392\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "C14orf166\n",
      "C14orf166  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 57391485\n",
      "chr14 57858203\n",
      "grep -P -m 1 \"14\\t57858203\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "NAA30\n",
      "\n",
      "Probe: 11739508_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0618030068366 0.0996349139654\n",
      "\n",
      "Probe: 11739509_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0412325079781 0.272213173585\n",
      "\n",
      "Probe: 11746938_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0616202137654 0.100642515548\n",
      "{'11746938_a_at': 0.10064251554835325, '11739508_a_at': 0.099634913965404595}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 60250005\n",
      "chr14 60716723\n",
      "grep -P -m 1 \"14\\t60716723\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t60716723\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 63852858\n",
      "chr14 64319576\n",
      "grep -P -m 1 \"14\\t64319576\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t64319576\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 77320823\n",
      "chr14 77787166\n",
      "grep -P -m 1 \"14\\t77787166\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t77787166\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 92793345\n",
      "chr14 93259690\n",
      "grep -P -m 1 \"14\\t93259690\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t93259690\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 94026240\n",
      "chr14 94492586\n",
      "grep -P -m 1 \"14\\t94492586\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t94492586\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OTUB2\n",
      "\n",
      "Probe: 11725831_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0473493789842 0.20697507318\n",
      "\n",
      "Probe: 11725832_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0469844730766 0.210500310288\n",
      "\n",
      "Probe: 11753488_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0470758588345 0.20961341983\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 100239112\n",
      "chr14 100705449\n",
      "grep -P -m 1 \"14\\t100705449\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t100705449\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 100569972\n",
      "chr14 101036309\n",
      "grep -P -m 1 \"14\\t101036309\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101036309\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "BEGAIN\n",
      "\n",
      "Probe: 11754479_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0195166394417 0.603130903766\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr14 103714128\n",
      "chr14 104180465\n",
      "grep -P -m 1 \"14\\t104180465\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t104180465\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr14_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 40357636\n",
      "chr15 40649837\n",
      "grep -P -m 1 \"15\\t40649837\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40649837\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 51971606\n",
      "chr15 52263803\n",
      "grep -P -m 1 \"15\\t52263803\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t52263803\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LEO1\n",
      "\n",
      "Probe: 11723233_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0118783426742 0.751692862081\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 52019279\n",
      "chr15 52311476\n",
      "grep -P -m 1 \"15\\t52311476\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t52311476\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MAPK6\n",
      "\n",
      "Probe: 11722796_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00220348604 0.953295493022\n",
      "{}\n",
      " CTD-2650P22.1\n",
      " CTD-2650P22.1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 58933663\n",
      "chr15 59225862\n",
      "grep -P -m 1 \"15\\t59225862\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t59225862\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 64703341\n",
      "chr15 64995540\n",
      "grep -P -m 1 \"15\\t64995540\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t64995540\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OAZ2\n",
      "\n",
      "Probe: 11715726_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0730949056551 0.051718346727\n",
      "\n",
      "Probe: 11715727_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00589219840387 0.875547167478\n",
      "{'11715726_a_at': 0.051718346726953288}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 69452829\n",
      "chr15 69745168\n",
      "grep -P -m 1 \"15\\t69745168\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t69745168\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RPLP1\n",
      "\n",
      "Probe: 11715538_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0865654362343 0.0208807789967\n",
      "\n",
      "Probe: 11715539_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0761001070087 0.0423585090174\n",
      "\n",
      "Probe: 11741896_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0866775416927 0.0207151494556\n",
      "\n",
      "Probe: 11744834_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 6 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0750160279686 0.0453945114874\n",
      "{'11715538_s_at': 0.02088077899666169, '11741896_x_at': 0.020715149455579715, '11744834_x_at': 0.04539451148743507, '11715539_x_at': 0.042358509017413601}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 74938064\n",
      "chr15 75230405\n",
      "grep -P -m 1 \"15\\t75230405\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75230405\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "COX5A\n",
      "\n",
      "Probe: 11715656_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0309016256757 0.410664022386\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 83067189\n",
      "chr15 83735941\n",
      "grep -P -m 1 \"15\\t83735941\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t83735941\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "BTBD1\n",
      "\n",
      "Probe: 11716590_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0420501496412 0.262810858207\n",
      "\n",
      "Probe: 11716591_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000841002712039 0.982140419836\n",
      "\n",
      "Probe: 11716592_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0393977812365 0.294139306459\n",
      "\n",
      "Probe: 11716593_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0227070969162 0.545519326801\n",
      "\n",
      "Probe: 11716594_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0656628952863 0.0801732627913\n",
      "\n",
      "Probe: 11741494_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0137795103447 0.713771320656\n",
      "{'11716594_a_at': 0.080173262791270364}\n",
      " MIR4515\n",
      " MIR4515  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 88546187\n",
      "chr15 89089418\n",
      "grep -P -m 1 \"15\\t89089418\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t89089418\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "DET1\n",
      "\n",
      "Probe: 11736589_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0315954942776 0.399898902286\n",
      "\n",
      "Probe: 11755361_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00348130669279 0.926118131823\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 90249253\n",
      "chr15 90792485\n",
      "grep -P -m 1 \"15\\t90792485\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t90792485\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TTLL13P\n",
      "TTLL13P  - gene not found in expression data\n",
      " RP11-697E2.6\n",
      " RP11-697E2.6  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 90352019\n",
      "chr15 90895251\n",
      "grep -P -m 1 \"15\\t90895251\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t90895251\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF774\n",
      "\n",
      "Probe: 11740636_at\n",
      "Number of REF, HET and ALT samples with expression data: 702 10 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0506363025682 0.177132468227\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 90870801\n",
      "chr15 91414031\n",
      "grep -P -m 1 \"15\\t91414031\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t91414031\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 90870904\n",
      "chr15 91414134\n",
      "grep -P -m 1 \"15\\t91414134\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t91414134\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FURIN\n",
      "\n",
      "Probe: 11716282_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0331172235066 0.377578306965\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr15 92809906\n",
      "chr15 93353136\n",
      "grep -P -m 1 \"15\\t93353136\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t93353136\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr15_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FAM174B\n",
      "\n",
      "Probe: 11720279_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0192804854845 0.608284106856\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 76736\n",
      "chr16 126735\n",
      "grep -P -m 1 \"16\\t126735\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t126735\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MPG\n",
      "\n",
      "Probe: 11726720_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 535 160 17\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0395339896152 0.292131693507\n",
      "\n",
      "Probe: 11726721_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 535 160 17\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0273644611733 0.465984245032\n",
      "{}\n",
      " RHBDF1\n",
      " RHBDF1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 275035\n",
      "chr16 325035\n",
      "grep -P -m 1 \"16\\t325035\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t325035\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RGS11\n",
      "\n",
      "Probe: 11749892_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0716383606307 0.0563974560054\n",
      "\n",
      "Probe: 11754903_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0364103673129 0.332646945242\n",
      "\n",
      "Probe: 11761420_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0246909190182 0.511276842835\n",
      "\n",
      "Probe: 11761802_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0239258468842 0.524455964832\n",
      "\n",
      "Probe: 11762560_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0222566052973 0.553801027146\n",
      "{'11749892_a_at': 0.056397456005395141}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 400930\n",
      "chr16 450930\n",
      "grep -P -m 1 \"16\\t450930\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t450930\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 649865\n",
      "chr16 699865\n",
      "grep -P -m 1 \"16\\t699865\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t699865\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 1474270\n",
      "chr16 1524271\n",
      "grep -P -m 1 \"16\\t1524271\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t1524271\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 2460185\n",
      "chr16 2510186\n",
      "grep -P -m 1 \"16\\t2510186\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t2510186\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 4614236\n",
      "chr16 4664237\n",
      "grep -P -m 1 \"16\\t4664237\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4664237\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 9090991\n",
      "chr16 9184848\n",
      "grep -P -m 1 \"16\\t9184848\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t9184848\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 21303291\n",
      "chr16 21314612\n",
      "grep -P -m 1 \"16\\t21314612\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t21314612\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CRYM\n",
      "\n",
      "Probe: 11721887_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00335506552731 0.928790035497\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 30394971\n",
      "chr16 30406292\n",
      "grep -P -m 1 \"16\\t30406292\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t30406292\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 31116944\n",
      "chr16 31128265\n",
      "grep -P -m 1 \"16\\t31128265\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31128265\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 56451366\n",
      "chr16 56485278\n",
      "grep -P -m 1 \"16\\t56485278\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t56485278\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OGFOD1\n",
      "\n",
      "Probe: 11723427_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00348516893205 0.92608839284\n",
      "\n",
      "Probe: 11758435_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0672843276514 0.0729754796021\n",
      "\n",
      "Probe: 11758449_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.02989045408 0.426152967331\n",
      "{'11758435_s_at': 0.072975479602148022}\n",
      " NUDT21\n",
      " NUDT21  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 70523819\n",
      "chr16 70557722\n",
      "grep -P -m 1 \"16\\t70557722\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70557722\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SF3B3\n",
      "\n",
      "Probe: 11722175_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0104580979662 0.780571641969\n",
      "\n",
      "Probe: 11722176_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0493855430709 0.188087291468\n",
      "\n",
      "Probe: 11753125_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0247250007581 0.510096905609\n",
      "{}\n",
      " COG4\n",
      " COG4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 70523900\n",
      "chr16 70557803\n",
      "grep -P -m 1 \"16\\t70557803\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t70557803\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SF3B3\n",
      "\n",
      "Probe: 11722175_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0104580979662 0.780571641969\n",
      "\n",
      "Probe: 11722176_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0493855430709 0.188087291468\n",
      "\n",
      "Probe: 11753125_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0247250007581 0.510096905609\n",
      "{}\n",
      " COG4\n",
      " COG4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 71289683\n",
      "chr16 71323586\n",
      "grep -P -m 1 \"16\\t71323586\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "CMTR2\n",
      "\n",
      "Probe: 11728310_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 690 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0606788050366 0.109473048882\n",
      "\n",
      "Probe: 11754232_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 690 7 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0519542076835 0.170657585615\n",
      "{'11728310_a_at': 0.10947304888179608}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 75000161\n",
      "chr16 75034059\n",
      "grep -P -m 1 \"16\\t75034059\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "ZNRF1\n",
      "\n",
      "Probe: 11732356_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00870327526787 0.816671065298\n",
      "\n",
      "Probe: 11732357_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0188834370153 0.614941439712\n",
      "\n",
      "Probe: 11732358_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0195164138749 0.603135092873\n",
      "\n",
      "Probe: 11732359_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0899337629579 0.0163779981122\n",
      "\n",
      "Probe: 11746939_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0172482876631 0.645898673926\n",
      "{'11732359_a_at': 0.016377998112212654}\n",
      " WDR59\n",
      " WDR59  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 84145072\n",
      "chr16 84178677\n",
      "grep -P -m 1 \"16\\t84178677\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t84178677\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 84618318\n",
      "chr16 84651924\n",
      "grep -P -m 1 \"16\\t84651924\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "COTL1\n",
      "\n",
      "Probe: 11715524_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0218927400843 0.567315252471\n",
      "\n",
      "Probe: 11754124_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0727926609654 0.0568823880603\n",
      "\n",
      "Probe: 11754125_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0525421058919 0.169567176481\n",
      "{'11754124_a_at': 0.05688238806029329}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 84618343\n",
      "chr16 84651949\n",
      "grep -P -m 1 \"16\\t84651949\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "COTL1\n",
      "\n",
      "Probe: 11715524_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 672 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0138794406691 0.719086658468\n",
      "\n",
      "Probe: 11754124_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 672 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0670832934429 0.0818043121113\n",
      "\n",
      "Probe: 11754125_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 672 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0574799689311 0.136032520055\n",
      "{'11754125_x_at': 0.13603252005537853, '11754124_a_at': 0.081804312111325259}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 87383764\n",
      "chr16 87417370\n",
      "grep -P -m 1 \"16\\t87417370\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr16 88786106\n",
      "chr16 88852514\n",
      "grep -P -m 1 \"16\\t88852514\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t88852514\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr16_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 1563724\n",
      "chr17 1467018\n",
      "grep -P -m 1 \"17\\t1467018\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t1467018\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 4940449\n",
      "chr17 4843744\n",
      "grep -P -m 1 \"17\\t4843744\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t4843744\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RNF167\n",
      "\n",
      "Probe: 11744262_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0156658950915 0.676452147317\n",
      "\n",
      "Probe: 11747757_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0101274462376 0.787340221469\n",
      "\n",
      "Probe: 11756574_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0154021529551 0.68159841284\n",
      "{}\n",
      " SLC25A11\n",
      " SLC25A11  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 5486310\n",
      "chr17 5389630\n",
      "grep -P -m 1 \"17\\t5389630\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t5389630\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 6556629\n",
      "chr17 6459949\n",
      "grep -P -m 1 \"17\\t6459949\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 6556632\n",
      "chr17 6459952\n",
      "grep -P -m 1 \"17\\t6459952\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t6459952\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 21043151\n",
      "chr17 20946464\n",
      "grep -P -m 1 \"17\\t20946464\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t20946464\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 21043153\n",
      "chr17 20946466\n",
      "grep -P -m 1 \"17\\t20946466\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t20946466\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 28661104\n",
      "chr17 26988122\n",
      "grep -P -m 1 \"17\\t26988122\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t26988122\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 28813130\n",
      "chr17 27140148\n",
      "grep -P -m 1 \"17\\t27140148\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 38605508\n",
      "chr17 36761761\n",
      "grep -P -m 1 \"17\\t36761761\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t36761761\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 39636349\n",
      "chr17 37792602\n",
      "grep -P -m 1 \"17\\t37792602\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t37792602\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "STARD3\n",
      "\n",
      "Probe: 11716353_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000422868627895 0.991019379209\n",
      "\n",
      "Probe: 11745302_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0142718126178 0.704018386244\n",
      "\n",
      "Probe: 11758753_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0277507451354 0.460027012559\n",
      "\n",
      "Probe: 11760262_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0590430247777 0.115728086706\n",
      "\n",
      "Probe: 11760263_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0497399082406 0.185244004103\n",
      "{'11760262_a_at': 0.11572808670605514}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 40818912\n",
      "chr17 38975164\n",
      "grep -P -m 1 \"17\\t38975164\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38975164\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 41811457\n",
      "chr17 39967709\n",
      "grep -P -m 1 \"17\\t39967709\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39967709\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 41865538\n",
      "chr17 40021791\n",
      "grep -P -m 1 \"17\\t40021791\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40021791\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 43025799\n",
      "chr17 41177816\n",
      "grep -P -m 1 \"17\\t41177816\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t41177816\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 44123129\n",
      "chr17 42200497\n",
      "grep -P -m 1 \"17\\t42200497\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42200497\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HDAC5\n",
      "\n",
      "Probe: 11717036_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.021912339664 0.559951475753\n",
      "\n",
      "Probe: 11717037_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00667743314865 0.859025977407\n",
      "\n",
      "Probe: 11760617_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0163370122803 0.663870830115\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 47821754\n",
      "chr17 45899120\n",
      "grep -P -m 1 \"17\\t45899120\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45899120\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "OSBPL7\n",
      "\n",
      "Probe: 11735603_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0500864609475 0.181888394467\n",
      "\n",
      "Probe: 11735604_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0431527490093 0.250155603197\n",
      "\n",
      "Probe: 11751119_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0254537737547 0.497703281185\n",
      "\n",
      "Probe: 11751120_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0446125300477 0.234474627992\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 47821802\n",
      "chr17 45899168\n",
      "grep -P -m 1 \"17\\t45899168\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "OSBPL7\n",
      "\n",
      "Probe: 11735603_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0548305337512 0.143853571599\n",
      "\n",
      "Probe: 11735604_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0280082324437 0.455554992451\n",
      "\n",
      "Probe: 11751119_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0483530469174 0.197499930158\n",
      "\n",
      "Probe: 11751120_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0041054475504 0.912920934963\n",
      "{'11735603_a_at': 0.1438535715987613}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 49677906\n",
      "chr17 47755268\n",
      "grep -P -m 1 \"17\\t47755268\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 57989256\n",
      "chr17 56066617\n",
      "grep -P -m 1 \"17\\t56066617\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t56066617\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "VEZF1\n",
      "\n",
      "Probe: 11718191_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.072271549942 0.0539076011906\n",
      "\n",
      "Probe: 11718192_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0258733641526 0.490640901246\n",
      "\n",
      "Probe: 11718193_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0756541922595 0.0435859004191\n",
      "\n",
      "Probe: 11718194_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.06719738202 0.0731449646947\n",
      "\n",
      "Probe: 11757754_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0291646054384 0.437154829042\n",
      "{'11718191_at': 0.053907601190645645, '11718193_at': 0.043585900419074459, '11718194_s_at': 0.073144964694682527}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 59893179\n",
      "chr17 57970540\n",
      "grep -P -m 1 \"17\\t57970540\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57970540\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 59964691\n",
      "chr17 58042052\n",
      "grep -P -m 1 \"17\\t58042052\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t58042052\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 60079259\n",
      "chr17 58156620\n",
      "grep -P -m 1 \"17\\t58156620\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 60079261\n",
      "chr17 58156622\n",
      "grep -P -m 1 \"17\\t58156622\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 64497155\n",
      "chr17 62493273\n",
      "grep -P -m 1 \"17\\t62493273\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t62493273\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 74776397\n",
      "chr17 72772536\n",
      "grep -P -m 1 \"17\\t72772536\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t72772536\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TMEM104\n",
      "\n",
      "Probe: 11723873_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0366961531518 0.330234713271\n",
      "\n",
      "Probe: 11723874_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0844950065143 0.0247600484694\n",
      "\n",
      "Probe: 11723875_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 12 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0387392369339 0.304002033294\n",
      "{'11723874_at': 0.024760048469403516}\n",
      " NAT9\n",
      " NAT9  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 75289603\n",
      "chr17 73285684\n",
      "grep -P -m 1 \"17\\t73285684\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73285684\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SLC25A19\n",
      "\n",
      "Probe: 11724064_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.000821088124693 0.982550960025\n",
      "\n",
      "Probe: 11751502_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 711 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0468020585065 0.212278722154\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 75766621\n",
      "chr17 73762702\n",
      "grep -P -m 1 \"17\\t73762702\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73762702\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "GALK1\n",
      "\n",
      "Probe: 11749866_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0607519572225 0.105294063976\n",
      "\n",
      "Probe: 11755574_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0511980589662 0.172369590249\n",
      "\n",
      "Probe: 11761150_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0251876512613 0.502210379197\n",
      "{'11749866_x_at': 0.10529406397581866}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 75779884\n",
      "chr17 73775965\n",
      "grep -P -m 1 \"17\\t73775965\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73775965\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "H3F3B\n",
      "\n",
      "Probe: 11754775_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0439768756336 0.241547157141\n",
      "\n",
      "Probe: 11757037_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0247372173959 0.510186204198\n",
      "\n",
      "Probe: 11757038_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0136512662398 0.716319705632\n",
      "\n",
      "Probe: 11758593_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0464963717145 0.21560681053\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 75855034\n",
      "chr17 73851115\n",
      "grep -P -m 1 \"17\\t73851115\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t73851115\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr17 77141354\n",
      "chr17 75137436\n",
      "grep -P -m 1 \"17\\t75137436\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t75137436\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr17_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 3772176\n",
      "chr18 3772176\n",
      "grep -P -m 1 \"18\\t3772176\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t3772176\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 3772179\n",
      "chr18 3772179\n",
      "grep -P -m 1 \"18\\t3772179\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t3772179\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 5238288\n",
      "chr18 5238287\n",
      "grep -P -m 1 \"18\\t5238287\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t5238287\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LINC00667\n",
      "LINC00667  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 5631364\n",
      "chr18 5631363\n",
      "grep -P -m 1 \"18\\t5631363\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "EPB41L3\n",
      "\n",
      "Probe: 11727141_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0244022322387 0.515636988685\n",
      "\n",
      "Probe: 11727142_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0449956601743 0.230476708954\n",
      "\n",
      "Probe: 11727143_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0251123298619 0.503489936665\n",
      "\n",
      "Probe: 11745192_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00103289779192 0.978050784978\n",
      "\n",
      "Probe: 11747508_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0261452403521 0.486093649166\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 12884427\n",
      "chr18 12884426\n",
      "grep -P -m 1 \"18\\t12884426\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t12884426\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PTPN2\n",
      "\n",
      "Probe: 11738863_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 689 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0405300753896 0.284237514721\n",
      "\n",
      "Probe: 11739325_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 689 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00397910835887 0.916304729259\n",
      "\n",
      "Probe: 11739326_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 689 11 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0289338130914 0.444683336665\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 34159044\n",
      "chr18 31739008\n",
      "grep -P -m 1 \"18\\t31739008\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t31739008\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 34708781\n",
      "chr18 32288745\n",
      "grep -P -m 1 \"18\\t32288745\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t32288745\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 37565974\n",
      "chr18 35145937\n",
      "grep -P -m 1 \"18\\t35145937\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35145937\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 47930651\n",
      "chr18 45457022\n",
      "grep -P -m 1 \"18\\t45457022\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t45457022\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 50559471\n",
      "chr18 48085841\n",
      "grep -P -m 1 \"18\\t48085841\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48085841\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MAPK4\n",
      "\n",
      "Probe: 11723181_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00592899869692 0.87512834209\n",
      "\n",
      "Probe: 11751129_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 696 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0146207199037 0.69835736608\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 50878910\n",
      "chr18 48405280\n",
      "grep -P -m 1 \"18\\t48405280\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48405280\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr18_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ME2\n",
      "\n",
      "Probe: 11739452_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.045709859309 0.225781370197\n",
      "\n",
      "Probe: 11739453_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 701 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0457634609802 0.225237077066\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr18 80247600\n",
      "chr18 78005483\n",
      "grep -P -m 1 \"18\\t78005483\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 589365\n",
      "chr19 589365\n",
      "grep -P -m 1 \"19\\t589365\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t589365\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HCN2\n",
      "\n",
      "Probe: 11732497_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00319986956998 0.93212363324\n",
      "\n",
      "Probe: 11732498_at\n",
      "Number of REF, HET and ALT samples with expression data: 710 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00749682826757 0.841831847279\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 12945283\n",
      "chr19 13056097\n",
      "grep -P -m 1 \"19\\t13056097\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t13056097\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 13865299\n",
      "chr19 13976113\n",
      "grep -P -m 1 \"19\\t13976113\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t13976113\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 13865303\n",
      "chr19 13976117\n",
      "grep -P -m 1 \"19\\t13976117\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t13976117\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 16896062\n",
      "chr19 17006873\n",
      "grep -P -m 1 \"19\\t17006873\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t17006873\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CPAMD8\n",
      "\n",
      "Probe: 11752397_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0482823562387 0.20134810865\n",
      "\n",
      "Probe: 11754808_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0288105824371 0.445975026524\n",
      "\n",
      "Probe: 11762554_at\n",
      "Number of REF, HET and ALT samples with expression data: 698 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0428657033855 0.25669543633\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 34364863\n",
      "chr19 34855768\n",
      "grep -P -m 1 \"19\\t34855768\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t34855768\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 34927008\n",
      "chr19 35417912\n",
      "grep -P -m 1 \"19\\t35417912\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35417912\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 35248691\n",
      "chr19 35739594\n",
      "grep -P -m 1 \"19\\t35739594\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35739594\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "LSR\n",
      "\n",
      "Probe: 11719463_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 689 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00812872237446 0.831217416221\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 38229464\n",
      "chr19 38720104\n",
      "grep -P -m 1 \"19\\t38720104\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t38720104\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "DPF1\n",
      "\n",
      "Probe: 11732879_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0343472871055 0.364201943309\n",
      "\n",
      "Probe: 11742102_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0298550163691 0.430314486888\n",
      "\n",
      "Probe: 11742248_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00729997686855 0.847118244914\n",
      "\n",
      "Probe: 11751137_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0367805693636 0.33119381904\n",
      "\n",
      "Probe: 11761855_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0605522874912 0.109450305576\n",
      "{'11761855_x_at': 0.10945030557619982}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 39321105\n",
      "chr19 39811745\n",
      "grep -P -m 1 \"19\\t39811745\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39811745\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 39445098\n",
      "chr19 39935738\n",
      "grep -P -m 1 \"19\\t39935738\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 41350302\n",
      "chr19 41856207\n",
      "grep -P -m 1 \"19\\t41856207\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t41856207\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 42269413\n",
      "chr19 42773565\n",
      "grep -P -m 1 \"19\\t42773565\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42773565\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 47484330\n",
      "chr19 47987587\n",
      "grep -P -m 1 \"19\\t47987587\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t47987587\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NAPA-AS1\n",
      "NAPA-AS1  - gene not found in expression data\n",
      " KPTN\n",
      " KPTN  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 49451032\n",
      "chr19 49954289\n",
      "grep -P -m 1 \"19\\t49954289\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t49954289\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 52822066\n",
      "chr19 53325319\n",
      "grep -P -m 1 \"19\\t53325319\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 52857562\n",
      "chr19 53360815\n",
      "grep -P -m 1 \"19\\t53360815\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53360815\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZNF28\n",
      "\n",
      "Probe: 11740337_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 684 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0242872850796 0.524484042596\n",
      "\n",
      "Probe: 11762350_at\n",
      "Number of REF, HET and ALT samples with expression data: 684 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0211062506169 0.58021574097\n",
      "\n",
      "Probe: 11762351_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 684 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00176243590477 0.963168472304\n",
      "{}\n",
      " ZNF468\n",
      " ZNF468  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 56764356\n",
      "chr19 57275724\n",
      "grep -P -m 1 \"19\\t57275724\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57275724\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "ZIM2-AS1\n",
      "ZIM2-AS1  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 57746913\n",
      "chr19 58258281\n",
      "grep -P -m 1 \"19\\t58258281\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t58258281\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr19 58574422\n",
      "chr19 59085789\n",
      "grep -P -m 1 \"19\\t59085789\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t59085789\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr19_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 346809\n",
      "chr20 327453\n",
      "grep -P -m 1 \"20\\t327453\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t327453\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "NRSN2\n",
      "\n",
      "Probe: 11720089_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 361 292 59\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0569641284705 0.128874531783\n",
      "\n",
      "Probe: 11720090_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 361 292 59\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0506028020503 0.177419564561\n",
      "{'11720089_a_at': 0.12887453178282349}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 462504\n",
      "chr20 443148\n",
      "grep -P -m 1 \"20\\t443148\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t443148\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TBC1D20\n",
      "\n",
      "Probe: 11728738_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0565105131977 0.136392521903\n",
      "\n",
      "Probe: 11728739_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0509661824552 0.179260212071\n",
      "\n",
      "Probe: 11728740_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 694 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0118231252377 0.755522631773\n",
      "{'11728738_x_at': 0.1363925219028706}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 6007060\n",
      "chr20 5987706\n",
      "grep -P -m 1 \"20\\t5987706\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "CRLS1\n",
      "\n",
      "Probe: 11726334_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.042999953796 0.251838219337\n",
      "\n",
      "Probe: 11726335_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.038137241499 0.309529288864\n",
      "\n",
      "Probe: 11726336_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 703 9 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00669772035017 0.858404563588\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 23638319\n",
      "chr20 23618956\n",
      "grep -P -m 1 \"20\\t23618956\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 36461763\n",
      "chr20 35090166\n",
      "grep -P -m 1 \"20\\t35090166\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 36461771\n",
      "chr20 35090174\n",
      "grep -P -m 1 \"20\\t35090174\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35090174\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 36461773\n",
      "chr20 35090176\n",
      "grep -P -m 1 \"20\\t35090176\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t35090176\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 50510408\n",
      "chr20 49126945\n",
      "grep -P -m 1 \"20\\t49126945\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t49126945\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PTPN1\n",
      "\n",
      "Probe: 11717146_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0034429726895 0.927032160861\n",
      "\n",
      "Probe: 11717147_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 706 4 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00527922434806 0.888326252989\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 54206990\n",
      "chr20 52823529\n",
      "grep -P -m 1 \"20\\t52823529\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t52823529\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PFDN4\n",
      "\n",
      "Probe: 11720949_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0224626260693 0.54957306814\n",
      "\n",
      "Probe: 11720950_at\n",
      "Number of REF, HET and ALT samples with expression data: 707 5 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0135020944347 0.719098278864\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 58650761\n",
      "chr20 57225817\n",
      "grep -P -m 1 \"20\\t57225817\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57225817\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr20 63657857\n",
      "chr20 62289210\n",
      "grep -P -m 1 \"20\\t62289210\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t62289210\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr20_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 26966649\n",
      "chr21 28338968\n",
      "grep -P -m 1 \"21\\t28338968\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t28338968\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 33403112\n",
      "chr21 34775418\n",
      "grep -P -m 1 \"21\\t34775418\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t34775418\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 42653720\n",
      "chr21 44073830\n",
      "grep -P -m 1 \"21\\t44073830\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 42653724\n",
      "chr21 44073834\n",
      "grep -P -m 1 \"21\\t44073834\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 45642964\n",
      "chr21 47062878\n",
      "grep -P -m 1 \"21\\t47062878\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t47062878\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr21 46324282\n",
      "chr21 47744196\n",
      "grep -P -m 1 \"21\\t47744196\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t47744196\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr21_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 19291974\n",
      "chr22 19279497\n",
      "grep -P -m 1 \"22\\t19279497\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t19279497\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 29305936\n",
      "chr22 29701926\n",
      "grep -P -m 1 \"22\\t29701926\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t29701926\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 30592309\n",
      "chr22 30988296\n",
      "grep -P -m 1 \"22\\t30988296\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t30988296\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PES1\n",
      "\n",
      "Probe: 11716242_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 709 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0533272937606 0.155182769984\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 39320421\n",
      "chr22 39716426\n",
      "grep -P -m 1 \"22\\t39716426\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39716426\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 39320435\n",
      "chr22 39716440\n",
      "grep -P -m 1 \"22\\t39716440\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39716440\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 39349983\n",
      "chr22 39745988\n",
      "grep -P -m 1 \"22\\t39745988\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t39745988\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SYNGR1\n",
      "\n",
      "Probe: 11721147_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0525255416747 0.173502007887\n",
      "\n",
      "Probe: 11721148_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0645807378413 0.0941300913572\n",
      "\n",
      "Probe: 11725137_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0562567735047 0.144876762274\n",
      "\n",
      "Probe: 11741994_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0461535153138 0.231799121125\n",
      "\n",
      "Probe: 11748695_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0453498185717 0.24003730821\n",
      "\n",
      "Probe: 11761351_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 670 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0426518127993 0.269190065614\n",
      "{'11725137_x_at': 0.1448767622736313, '11721148_at': 0.094130091357150072}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 41835260\n",
      "chr22 42231264\n",
      "grep -P -m 1 \"22\\t42231264\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42231264\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP5-821D11.7\n",
      "RP5-821D11.7  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 41835269\n",
      "chr22 42231273\n",
      "grep -P -m 1 \"22\\t42231273\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42231273\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "RP5-821D11.7\n",
      "RP5-821D11.7  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 42344176\n",
      "chr22 42740182\n",
      "grep -P -m 1 \"22\\t42740182\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 42581828\n",
      "chr22 42977834\n",
      "grep -P -m 1 \"22\\t42977834\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t42977834\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 42615287\n",
      "chr22 43011293\n",
      "grep -P -m 1 \"22\\t43011293\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t43011293\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 44498614\n",
      "chr22 44894494\n",
      "grep -P -m 1 \"22\\t44894494\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t44894494\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 44498615\n",
      "chr22 44894495\n",
      "grep -P -m 1 \"22\\t44894495\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t44894495\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr22_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 50600542\n",
      "chr22 51038971\n",
      "grep -P -m 1 \"22\\t51038971\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chr22 50600547\n",
      "chr22 51038976\n",
      "grep -P -m 1 \"22\\t51038976\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 6227646\n",
      "chrX 6145687\n",
      "grep -P -m 1 \"23\\t6145687\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t6145687\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 15676222\n",
      "chrX 15694345\n",
      "grep -P -m 1 \"23\\t15694345\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t15694345\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 18675961\n",
      "chrX 18694081\n",
      "grep -P -m 1 \"23\\t18694081\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t18694081\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 41085449\n",
      "chrX 40944702\n",
      "grep -P -m 1 \"23\\t40944702\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t40944702\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 48539949\n",
      "chrX 48398337\n",
      "grep -P -m 1 \"23\\t48398337\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "TBC1D25\n",
      "\n",
      "Probe: 11722611_at\n",
      "Number of REF, HET and ALT samples with expression data: 656 2 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0345769898764 0.375506736338\n",
      "\n",
      "Probe: 11748734_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 656 2 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0337628011634 0.386859898556\n",
      "\n",
      "Probe: 11748753_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 656 2 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0388261557058 0.319644296997\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 48695796\n",
      "chrX 48554187\n",
      "grep -P -m 1 \"23\\t48554187\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48554187\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SUV39H1\n",
      "\n",
      "Probe: 11721277_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.020444566171 0.593498867445\n",
      "\n",
      "Probe: 11721278_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0220917661618 0.564082483103\n",
      "\n",
      "Probe: 11721279_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0210259449966 0.583036521673\n",
      "\n",
      "Probe: 11746768_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0382731095844 0.317549087405\n",
      "\n",
      "Probe: 11752693_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 683 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00106583158297 0.977802415175\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 49043298\n",
      "chrX 48900824\n",
      "grep -P -m 1 \"23\\t48900824\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t48900824\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 51802590\n",
      "chrX 51545686\n",
      "grep -P -m 1 \"23\\t51545686\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t51545686\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 53225279\n",
      "chrX 53254461\n",
      "grep -P -m 1 \"23\\t53254461\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 53433980\n",
      "chrX 53460927\n",
      "grep -P -m 1 \"23\\t53460927\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53460927\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 53434858\n",
      "chrX 53461805\n",
      "grep -P -m 1 \"23\\t53461805\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t53461805\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "HSD17B10\n",
      "\n",
      "Probe: 11717515_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 2 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0266975995761 0.480361806025\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 54806646\n",
      "chrX 54833079\n",
      "grep -P -m 1 \"23\\t54833079\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t54833079\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "MAGED2\n",
      "\n",
      "Probe: 11715397_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 1 3\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0415883487511 0.270470583644\n",
      "\n",
      "Probe: 11748370_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 1 3\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0916330139205 0.0150117011606\n",
      "\n",
      "Probe: 11748489_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 1 3\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0255979935945 0.497711016921\n",
      "\n",
      "Probe: 11757043_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 1 3\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0471844079348 0.211147771271\n",
      "\n",
      "Probe: 11761242_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 700 1 3\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0505579594624 0.180268932011\n",
      "{'11748370_a_at': 0.015011701160587735}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 57121233\n",
      "chrX 57147666\n",
      "grep -P -m 1 \"23\\t57147666\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t57147666\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 68828777\n",
      "chrX 68048620\n",
      "grep -P -m 1 \"23\\t68048620\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t68048620\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 68828783\n",
      "chrX 68048626\n",
      "grep -P -m 1 \"23\\t68048626\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 69615560\n",
      "chrX 68835404\n",
      "grep -P -m 1 \"23\\t68835404\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "EDA\n",
      "\n",
      "Probe: 11728157_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0955471931512 0.0116104920262\n",
      "\n",
      "Probe: 11728158_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0490541243631 0.195831151721\n",
      "\n",
      "Probe: 11735891_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0238161218993 0.530184667502\n",
      "\n",
      "Probe: 11742320_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00418195440082 0.912243562068\n",
      "\n",
      "Probe: 11742321_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.017036324997 0.653433687981\n",
      "\n",
      "Probe: 11742331_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.044361051722 0.242146234643\n",
      "\n",
      "Probe: 11752858_x_at\n",
      "Number of REF, HET and ALT samples with expression data: 693 3 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0442264983316 0.24357889256\n",
      "{'11728157_at': 0.011610492026152542}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 73562927\n",
      "chrX 72782763\n",
      "grep -P -m 1 \"23\\t72782763\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t72782763\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 101623482\n",
      "chrX 100878472\n",
      "grep -P -m 1 \"23\\t100878472\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t100878472\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 102711900\n",
      "chrX 101966828\n",
      "grep -P -m 1 \"23\\t101966828\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t101966828\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 104166140\n",
      "chrX 103410821\n",
      "grep -P -m 1 \"23\\t103410821\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t103410821\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "FAM199X\n",
      "\n",
      "Probe: 11725538_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 691 0 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.051988922794 0.171918097168\n",
      "\n",
      "Probe: 11755408_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 691 0 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0149491449962 0.694643305691\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 107776416\n",
      "chrX 107019646\n",
      "grep -P -m 1 \"23\\t107019646\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t107019646\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "TSC22D3\n",
      "\n",
      "Probe: 11717829_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 636 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0315909204341 0.426060258757\n",
      "\n",
      "Probe: 11717830_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 636 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.012291336612 0.756848647904\n",
      "\n",
      "Probe: 11751415_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 636 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0176822603825 0.656002993352\n",
      "{}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 107825412\n",
      "chrX 107068642\n",
      "grep -P -m 1 \"23\\t107068642\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 109535925\n",
      "chrX 108779154\n",
      "grep -P -m 1 \"23\\t108779154\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t108779154\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 120630512\n",
      "chrX 119764367\n",
      "grep -P -m 1 \"23\\t119764367\\t\" out_files/prom_var_in_adni.txt\n",
      "**Check correlation with target expression\n",
      "C1GALT1C1\n",
      "\n",
      "Probe: 11726618_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 1\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00241066222256 0.949199664528\n",
      "\n",
      "Probe: 11726619_s_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00382018208689 0.919578602182\n",
      "\n",
      "Probe: 11743934_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 699 1 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0267086645305 0.48017983088\n",
      "{}\n",
      "\n",
      "*Target gene(s) not expressed in brain - stop.\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 133415561\n",
      "chrX 132549589\n",
      "grep -P -m 1 \"23\\t132549589\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t132549589\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 133415564\n",
      "chrX 132549592\n",
      "grep -P -m 1 \"23\\t132549592\\t\" out_files/prom_var_in_adni.txt\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 136497595\n",
      "chrX 135579754\n",
      "grep -P -m 1 \"23\\t135579754\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t135579754\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 148501051\n",
      "chrX 147582571\n",
      "grep -P -m 1 \"23\\t147582571\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t147582571\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "AFF2\n",
      "\n",
      "Probe: 11733487_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 650 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0424737685188 0.279203119524\n",
      "\n",
      "Probe: 11733488_at\n",
      "Number of REF, HET and ALT samples with expression data: 650 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0227500767446 0.562305225178\n",
      "\n",
      "Probe: 11745291_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 650 1 0\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.00688765955838 0.860764613918\n",
      "{}\n",
      " AC002368.4\n",
      " AC002368.4  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 149542024\n",
      "chrX 148623569\n",
      "grep -P -m 1 \"23\\t148623569\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t148623569\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "CXorf40A\n",
      "CXorf40A  - gene not found in expression data\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 153830460\n",
      "chrX 153095915\n",
      "grep -P -m 1 \"23\\t153095915\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t153095915\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "PDZD4\n",
      "\n",
      "Probe: 11724042_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 632 2 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0857088830656 0.0308102025025\n",
      "\n",
      "Probe: 11724043_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 632 2 1\n",
      "Calculate Spearmann correlation\n",
      "R, p -0.0719112853563 0.0701591326731\n",
      "{'11724042_a_at': 0.030810202502523386, '11724043_a_at': 0.070159132673114147}\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 154437141\n",
      "chrX 153665487\n",
      "grep -P -m 1 \"23\\t153665487\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t153665487\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 154479894\n",
      "chrX 153708233\n",
      "grep -P -m 1 \"23\\t153708233\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t153708233\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 154479895\n",
      "chrX 153708234\n",
      "grep -P -m 1 \"23\\t153708234\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t153708234\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "no variant found in ADNI\n",
      "\n",
      "*Target gene(s) expressed in brain - proceed.\n",
      "**Check if variant is present in ADNI\n",
      "chrX 155767382\n",
      "chrX 154997044\n",
      "grep -P -m 1 \"23\\t154997044\\t\" out_files/prom_var_in_adni.txt\n",
      "no search for this variant has been performed yet\n",
      "grep -P -m 1 \"\\t154997044\\t\" /mnt/chr11/Data/marlena/licencjat/wgs/files/adni_chr23_SNPs.vcf\n",
      "**Check correlation with target expression\n",
      "SPRY3\n",
      "\n",
      "Probe: 11729018_a_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.00422869478523 0.910380594994\n",
      "\n",
      "Probe: 11729019_at\n",
      "Number of REF, HET and ALT samples with expression data: 708 3 0\n",
      "Calculate Spearmann correlation\n",
      "R, p 0.0222005896228 0.554519929484\n",
      "{}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/magda/.local/lib/python2.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "#corrds:Gene:probe:pvalue\n",
    "prom_with_hits[\"ADNI_data\"] = prom_with_hits.apply(check_ADNI, args = (\"out_files/prom_var_in_adni.txt\",), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "      <th>ADNI_data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>72,69;91,70</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs888928964</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0</td>\n",
       "      <td>HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...</td>\n",
       "      <td>3.94;1.09</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>promoter</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:0.08</td>\n",
       "      <td>SAMD11:1.25</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>7,3;9,8;5,6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs751320169</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>3</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.79;1.83</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN:57.03</td>\n",
       "      <td>AGRN:18.92</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6180416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>10,11;8,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs151212360</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>2</td>\n",
       "      <td>UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...</td>\n",
       "      <td>10.00;6.71</td>\n",
       "      <td>ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>CHD5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHD5:0.20</td>\n",
       "      <td>CHD5:31.50</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "      <td>MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>KAZN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KAZN:0.54</td>\n",
       "      <td>KAZN:22.48</td>\n",
       "      <td>1:15250513:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21653564</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13,15;30,24</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs890141715</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>2</td>\n",
       "      <td>ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...</td>\n",
       "      <td>5.97;2.29</td>\n",
       "      <td>ZNF263, POLR2A</td>\n",
       "      <td>promoter</td>\n",
       "      <td>RAP1GAP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RAP1GAP:0.60</td>\n",
       "      <td>RAP1GAP:103.43</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT         VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1    939561   G   A  72,69;91,70  2.0  0.043  rs888928964   \n",
       "1  chr1   1034902   A   T  7,3;9,8;5,6  3.0  0.065  rs751320169   \n",
       "2  chr1   6180416   G   A    10,11;8,6  2.0  0.043  rs151212360   \n",
       "3  chr1  14924017   C   G    6,5;12,14  2.0  0.043  rs531269337   \n",
       "4  chr1  21653564   C   T  13,15;30,24  2.0  0.043  rs890141715   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "0           0.000052             0.0000                     0   \n",
       "1           0.001300             0.0017                     3   \n",
       "2           0.000300             0.0005                     2   \n",
       "3           0.001300             0.0020                     0   \n",
       "4           0.000600             0.0013                     2   \n",
       "\n",
       "                                    PWMEnrich motifs  \\\n",
       "0  HXA1_HUMAN.H11MO.0.C, ZKSC3_HUMAN.H11MO.0.D, E...   \n",
       "1  UBIP1_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "2  UBIP1_HUMAN.H11MO.0.D, SOX5_HUMAN.H11MO.0.C, R...   \n",
       "3  MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "4  ETV7_HUMAN.H11MO.0.D, ETV6_HUMAN.H11MO.0.D, ZN...   \n",
       "\n",
       "  PWMEnrich best -log10pval and diff  \\\n",
       "0                          3.94;1.09   \n",
       "1                          4.79;1.83   \n",
       "2                         10.00;6.71   \n",
       "3                          5.20;2.79   \n",
       "4                          5.97;2.29   \n",
       "\n",
       "                                              ENCODE genomic element     Gene  \\\n",
       "0                                               EZH2        promoter   SAMD11   \n",
       "1                                                           promoter     AGRN   \n",
       "2            ZNF143, RAD21, EZH2, CTCF, SIN3A, BRCA1        promoter     CHD5   \n",
       "3  MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...        promoter     KAZN   \n",
       "4                                     ZNF263, POLR2A        promoter  RAP1GAP   \n",
       "\n",
       "  polgenom_freq_perc target_expr_in_lymphocytes target_expr_in_brain  \\\n",
       "0                NaN                SAMD11:0.08          SAMD11:1.25   \n",
       "1                NaN                 AGRN:57.03           AGRN:18.92   \n",
       "2                NaN                  CHD5:0.20           CHD5:31.50   \n",
       "3                NaN                  KAZN:0.54           KAZN:22.48   \n",
       "4                NaN               RAP1GAP:0.60       RAP1GAP:103.43   \n",
       "\n",
       "            ADNI_data  \n",
       "0  no variant in ADNI  \n",
       "1  no variant in ADNI  \n",
       "2  no variant in ADNI  \n",
       "3         1:15250513:  \n",
       "4  no variant in ADNI  "
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_with_hits.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>num_samples_with_rna</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>PWMEnrich best -log10pval and diff</th>\n",
       "      <th>ENCODE</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "      <th>ADNI_data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14924017</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6,5;12,14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs531269337</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...</td>\n",
       "      <td>5.20;2.79</td>\n",
       "      <td>MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>KAZN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KAZN:0.54</td>\n",
       "      <td>KAZN:22.48</td>\n",
       "      <td>1:15250513:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25268534</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>13,11;0,7;15,7</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>rs148017678</td>\n",
       "      <td>0.0041</td>\n",
       "      <td>0.0069</td>\n",
       "      <td>3</td>\n",
       "      <td>FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...</td>\n",
       "      <td>4.37;1.50</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>RSRP1</td>\n",
       "      <td>0.51</td>\n",
       "      <td>RSRP1:42.77</td>\n",
       "      <td>RSRP1:35.43</td>\n",
       "      <td>1:25595025:RSRP1:gene not found:1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26901253</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6,7;3,12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs556268801</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>1</td>\n",
       "      <td>NFIA_HUMAN.H11MO.1.D, PRRX1_HUMAN.H11MO.0.D, H...</td>\n",
       "      <td>4.79;1.91</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>GPATCH3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GPATCH3:13.92</td>\n",
       "      <td>GPATCH3:10.03</td>\n",
       "      <td>1:27227744:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32178833</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>138,127;146,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs750438424</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0</td>\n",
       "      <td>ZN335_HUMAN.H11MO.1.A, PO2F2_HUMAN.H11MO.0.A, ...</td>\n",
       "      <td>10.00;6.45</td>\n",
       "      <td>NR2F2, WRNIP1, MBD4</td>\n",
       "      <td>promoter</td>\n",
       "      <td>TXLNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TXLNA:78.67</td>\n",
       "      <td>TXLNA:21.65</td>\n",
       "      <td>1:32644434:TXLNA:11729720_a_at:0.0416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32221069</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>113,132;115,128</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs187445970</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>0</td>\n",
       "      <td>PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...</td>\n",
       "      <td>5.57;1.94</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>EIF3I, TMEM234</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EIF3I:380.35, TMEM234:7.38</td>\n",
       "      <td>EIF3I:90.54, TMEM234:5.88</td>\n",
       "      <td>1:32686670: TMEM234:gene not found:1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33081220</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>40,35;39,49</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs559970186</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0</td>\n",
       "      <td>ZBT14_HUMAN.H11MO.0.C, BCL6B_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>4.22;1.44</td>\n",
       "      <td>REST, MXI1, TBP, MAZ, CCNT2, IRF1, PHF8, POLR2...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>AZIN2, AK2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AZIN2:0.12, AK2:137.47</td>\n",
       "      <td>AZIN2:10.28, AK2:18.43</td>\n",
       "      <td>1:33546821:AZIN2:gene not found:1.0000, AK2:ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36324190</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>54,76;76,73</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs575257092</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.0025</td>\n",
       "      <td>0</td>\n",
       "      <td>OVOL2_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...</td>\n",
       "      <td>6.36;2.29</td>\n",
       "      <td>MXI1, TBP, RAD21, FOSL2, HDAC1, HNF4A, POLR2A,...</td>\n",
       "      <td>promoter</td>\n",
       "      <td>EVA1B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EVA1B:9.22</td>\n",
       "      <td>EVA1B:6.70</td>\n",
       "      <td>1:36789791:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>40318303</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7,13;1,10</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.048</td>\n",
       "      <td>rs867492341</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>FOXF2_HUMAN.H11MO.0.D, IRX3_HUMAN.H11MO.0.D, P...</td>\n",
       "      <td>4.34;1.72</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>COL9A2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COL9A2:50.55</td>\n",
       "      <td>COL9A2:44.11</td>\n",
       "      <td>1:40783975:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44801300</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>11,10;9,6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs528752020</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>0</td>\n",
       "      <td>ELK1_HUMAN.H11MO.0.B, ETV1_HUMAN.H11MO.0.A, GC...</td>\n",
       "      <td>4.28;1.82</td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td>PLK3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLK3:14.53</td>\n",
       "      <td>PLK3:4.62</td>\n",
       "      <td>1:45266972:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>85048779</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>30,29;11,7;20,30</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs542693353</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>3</td>\n",
       "      <td>HIC1_HUMAN.H11MO.0.C, GLIS3_HUMAN.H11MO.0.D, P...</td>\n",
       "      <td>4.70;1.81</td>\n",
       "      <td>EZH2, POLR2A, CTCF, E2F1, ZEB1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>MCOLN3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCOLN3:3.75</td>\n",
       "      <td>MCOLN3:0.13</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "3   chr1  14924017   C   G         6,5;12,14  2.0  0.043  rs531269337   \n",
       "5   chr1  25268534   A   G    13,11;0,7;15,7  4.0  0.105  rs148017678   \n",
       "7   chr1  26901253   C   T          6,7;3,12  2.0  0.043  rs556268801   \n",
       "10  chr1  32178833   T   C    138,127;146,95  2.0  0.043  rs750438424   \n",
       "11  chr1  32221069   C   T   113,132;115,128  2.0  0.043  rs187445970   \n",
       "13  chr1  33081220   A   C       40,35;39,49  2.0  0.043  rs559970186   \n",
       "15  chr1  36324190   G   C       54,76;76,73  2.0  0.043  rs575257092   \n",
       "18  chr1  40318303   T   C         7,13;1,10  2.0  0.048  rs867492341   \n",
       "21  chr1  44801300   C   T         11,10;9,6  2.0  0.043  rs528752020   \n",
       "24  chr1  85048779   G   A  30,29;11,7;20,30  3.0  0.065  rs542693353   \n",
       "\n",
       "    gnomAD_genome_ALL  gnomAD_genome_NFE  num_samples_with_rna  \\\n",
       "3              0.0013             0.0020                     0   \n",
       "5              0.0041             0.0069                     3   \n",
       "7              0.0011             0.0019                     1   \n",
       "10             0.0002             0.0003                     0   \n",
       "11             0.0009             0.0012                     0   \n",
       "13             0.0002             0.0003                     0   \n",
       "15             0.0013             0.0025                     0   \n",
       "18             0.0007             0.0009                     1   \n",
       "21             0.0007             0.0012                     0   \n",
       "24             0.0008             0.0011                     3   \n",
       "\n",
       "                                     PWMEnrich motifs  \\\n",
       "3   MECP2_HUMAN.H11MO.0.C, AP2B_HUMAN.H11MO.0.B, N...   \n",
       "5   FOXO4_HUMAN.H11MO.0.C, FOXP2_HUMAN.H11MO.0.C, ...   \n",
       "7   NFIA_HUMAN.H11MO.1.D, PRRX1_HUMAN.H11MO.0.D, H...   \n",
       "10  ZN335_HUMAN.H11MO.1.A, PO2F2_HUMAN.H11MO.0.A, ...   \n",
       "11  PATZ1_HUMAN.H11MO.1.C, ZIC1_HUMAN.H11MO.0.B, G...   \n",
       "13  ZBT14_HUMAN.H11MO.0.C, BCL6B_HUMAN.H11MO.0.D, ...   \n",
       "15  OVOL2_HUMAN.H11MO.0.D, ZN652_HUMAN.H11MO.0.D, ...   \n",
       "18  FOXF2_HUMAN.H11MO.0.D, IRX3_HUMAN.H11MO.0.D, P...   \n",
       "21  ELK1_HUMAN.H11MO.0.B, ETV1_HUMAN.H11MO.0.A, GC...   \n",
       "24  HIC1_HUMAN.H11MO.0.C, GLIS3_HUMAN.H11MO.0.D, P...   \n",
       "\n",
       "   PWMEnrich best -log10pval and diff  \\\n",
       "3                           5.20;2.79   \n",
       "5                           4.37;1.50   \n",
       "7                           4.79;1.91   \n",
       "10                         10.00;6.45   \n",
       "11                          5.57;1.94   \n",
       "13                          4.22;1.44   \n",
       "15                          6.36;2.29   \n",
       "18                          4.34;1.72   \n",
       "21                          4.28;1.82   \n",
       "24                          4.70;1.81   \n",
       "\n",
       "                                               ENCODE genomic element  \\\n",
       "3   MXI1, RBBP5, BHLHE40, EZH2, MAZ, CCNT2, POLR2A...        promoter   \n",
       "5                                                            promoter   \n",
       "7                                                            promoter   \n",
       "10                                NR2F2, WRNIP1, MBD4        promoter   \n",
       "11                                                           promoter   \n",
       "13  REST, MXI1, TBP, MAZ, CCNT2, IRF1, PHF8, POLR2...        promoter   \n",
       "15  MXI1, TBP, RAD21, FOSL2, HDAC1, HNF4A, POLR2A,...        promoter   \n",
       "18                                                           promoter   \n",
       "21                                                           promoter   \n",
       "24                     EZH2, POLR2A, CTCF, E2F1, ZEB1        promoter   \n",
       "\n",
       "              Gene polgenom_freq_perc  target_expr_in_lymphocytes  \\\n",
       "3             KAZN                NaN                   KAZN:0.54   \n",
       "5            RSRP1               0.51                 RSRP1:42.77   \n",
       "7          GPATCH3                NaN               GPATCH3:13.92   \n",
       "10           TXLNA                NaN                 TXLNA:78.67   \n",
       "11  EIF3I, TMEM234                NaN  EIF3I:380.35, TMEM234:7.38   \n",
       "13      AZIN2, AK2                NaN      AZIN2:0.12, AK2:137.47   \n",
       "15           EVA1B                NaN                  EVA1B:9.22   \n",
       "18          COL9A2                NaN                COL9A2:50.55   \n",
       "21            PLK3                NaN                  PLK3:14.53   \n",
       "24          MCOLN3                NaN                 MCOLN3:3.75   \n",
       "\n",
       "         target_expr_in_brain  \\\n",
       "3                  KAZN:22.48   \n",
       "5                 RSRP1:35.43   \n",
       "7               GPATCH3:10.03   \n",
       "10                TXLNA:21.65   \n",
       "11  EIF3I:90.54, TMEM234:5.88   \n",
       "13     AZIN2:10.28, AK2:18.43   \n",
       "15                 EVA1B:6.70   \n",
       "18               COL9A2:44.11   \n",
       "21                  PLK3:4.62   \n",
       "24                MCOLN3:0.13   \n",
       "\n",
       "                                            ADNI_data  \n",
       "3                                         1:15250513:  \n",
       "5              1:25595025:RSRP1:gene not found:1.0000  \n",
       "7                                         1:27227744:  \n",
       "10              1:32644434:TXLNA:11729720_a_at:0.0416  \n",
       "11          1:32686670: TMEM234:gene not found:1.0000  \n",
       "13  1:33546821:AZIN2:gene not found:1.0000, AK2:ge...  \n",
       "15                                        1:36789791:  \n",
       "18                                        1:40783975:  \n",
       "21                                        1:45266972:  \n",
       "24                                               None  "
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_with_hits[~prom_with_hits[\"ADNI_data\"].isin([\"no variant in ADNI\", \"None\"])][:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "prom_with_hits.to_csv(\"out_files/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add previous manual annotations for enhancer SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>PWMEnrich motifs</th>\n",
       "      <th>...</th>\n",
       "      <th>dist. to closest protein gene</th>\n",
       "      <th>Contacting gene</th>\n",
       "      <th>polgenom_freq_perc</th>\n",
       "      <th>AF/polgenom</th>\n",
       "      <th>ADNI</th>\n",
       "      <th>genes with best correlation between H3K27ac and expression</th>\n",
       "      <th>variant reliability</th>\n",
       "      <th>logo</th>\n",
       "      <th>comments</th>\n",
       "      <th>test?</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>564124</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs1055969240</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>CEBPE_HUMAN.H11MO.0.A</td>\n",
       "      <td>...</td>\n",
       "      <td>91954</td>\n",
       "      <td>.</td>\n",
       "      <td>0.4</td>\n",
       "      <td>#ADR!</td>\n",
       "      <td>no data for variant</td>\n",
       "      <td>ENSG00000107104:KANK1</td>\n",
       "      <td>medium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low negative correlation between enh act and e...</td>\n",
       "      <td>rather not (check hits for CEBPG and CEBPZ)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>2167742</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C</td>\n",
       "      <td>...</td>\n",
       "      <td>2457</td>\n",
       "      <td>.</td>\n",
       "      <td>unknown</td>\n",
       "      <td>#ADR!</td>\n",
       "      <td>no data for variant</td>\n",
       "      <td>ENSG00000104885:DOT1L, ENSG00000065000:AP3D1</td>\n",
       "      <td>low</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>3356059</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>0.100</td>\n",
       "      <td>rs540473332</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.0089</td>\n",
       "      <td>WT1_HUMAN.H11MO.0.C</td>\n",
       "      <td>...</td>\n",
       "      <td>99400</td>\n",
       "      <td>.</td>\n",
       "      <td>unknown</td>\n",
       "      <td>#ADR!</td>\n",
       "      <td>no data for variant</td>\n",
       "      <td>-</td>\n",
       "      <td>low</td>\n",
       "      <td>NaN</td>\n",
       "      <td>filtered out in Symfonia data</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>4269215</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs907328134</td>\n",
       "      <td>0.000081</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>E2F4_HUMAN.H11MO.0.A, E2F7_HUMAN.H11MO.0.B</td>\n",
       "      <td>...</td>\n",
       "      <td>3582</td>\n",
       "      <td>ENSG00000256862:RP11-664D1.1, ENSG00000111224:...</td>\n",
       "      <td>unknown</td>\n",
       "      <td>#ADR!</td>\n",
       "      <td>no data for variant</td>\n",
       "      <td>ENSG00000118971:CCND2</td>\n",
       "      <td>low</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr7</td>\n",
       "      <td>5671835</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>SOX2_HUMAN.H11MO.0.A</td>\n",
       "      <td>...</td>\n",
       "      <td>77609</td>\n",
       "      <td>ENSG00000207973:MIR589, ENSG00000228974:AC0064...</td>\n",
       "      <td>unknown</td>\n",
       "      <td>#ADR!</td>\n",
       "      <td>no data for variant</td>\n",
       "      <td>ENSG00000011275:RNF216, ENSG00000075624:ACTB</td>\n",
       "      <td>low</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS REF ALT  AC     AF      avsnp150  gnomAD_genome_ALL  \\\n",
       "0   chr9   564124   A   G   2  0.043  rs1055969240           0.000400   \n",
       "1  chr19  2167742   T   C   2  0.043             .           0.000036   \n",
       "2   chr6  3356059   A   G   4  0.100   rs540473332           0.004400   \n",
       "3  chr12  4269215   C   G   2  0.043   rs907328134           0.000081   \n",
       "4   chr7  5671835   G   A   2  0.043             .           0.001300   \n",
       "\n",
       "   gnomAD_genome_NFE                            PWMEnrich motifs  \\\n",
       "0             0.0004                       CEBPE_HUMAN.H11MO.0.A   \n",
       "1             0.0000                       PRDM6_HUMAN.H11MO.0.C   \n",
       "2             0.0089                         WT1_HUMAN.H11MO.0.C   \n",
       "3             0.0001  E2F4_HUMAN.H11MO.0.A, E2F7_HUMAN.H11MO.0.B   \n",
       "4             0.0005                        SOX2_HUMAN.H11MO.0.A   \n",
       "\n",
       "                      ...                      dist. to closest protein gene  \\\n",
       "0                     ...                                              91954   \n",
       "1                     ...                                               2457   \n",
       "2                     ...                                              99400   \n",
       "3                     ...                                               3582   \n",
       "4                     ...                                              77609   \n",
       "\n",
       "                                     Contacting gene polgenom_freq_perc  \\\n",
       "0                                                  .                0.4   \n",
       "1                                                  .            unknown   \n",
       "2                                                  .            unknown   \n",
       "3  ENSG00000256862:RP11-664D1.1, ENSG00000111224:...            unknown   \n",
       "4  ENSG00000207973:MIR589, ENSG00000228974:AC0064...            unknown   \n",
       "\n",
       "  AF/polgenom                 ADNI  \\\n",
       "0       #ADR!  no data for variant   \n",
       "1       #ADR!  no data for variant   \n",
       "2       #ADR!  no data for variant   \n",
       "3       #ADR!  no data for variant   \n",
       "4       #ADR!  no data for variant   \n",
       "\n",
       "  genes with best correlation between H3K27ac and expression  \\\n",
       "0                              ENSG00000107104:KANK1           \n",
       "1       ENSG00000104885:DOT1L, ENSG00000065000:AP3D1           \n",
       "2                                                  -           \n",
       "3                              ENSG00000118971:CCND2           \n",
       "4       ENSG00000011275:RNF216, ENSG00000075624:ACTB           \n",
       "\n",
       "   variant reliability logo  \\\n",
       "0               medium  NaN   \n",
       "1                  low  NaN   \n",
       "2                  low  NaN   \n",
       "3                  low  NaN   \n",
       "4                  low  NaN   \n",
       "\n",
       "                                            comments  \\\n",
       "0  low negative correlation between enh act and e...   \n",
       "1                                                NaN   \n",
       "2                      filtered out in Symfonia data   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                         test?  \n",
       "0  rather not (check hits for CEBPG and CEBPZ)  \n",
       "1                                           NO  \n",
       "2                                           NO  \n",
       "3                                           NO  \n",
       "4                                           NO  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "manual_annotations = pd.read_csv(\"../2020-11-30-Collect_variants_annotations/out_files/SNPs_enhancers_AF_Gene_Encode_motifs_Polgenom_with_PWMEnrich_hits_only_manual_annot.txt\", sep = '\\t')\n",
    "manual_annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>VAFs</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>...</th>\n",
       "      <th>POS-1</th>\n",
       "      <th>Target_genes</th>\n",
       "      <th>target_expr_in_lymphocytes</th>\n",
       "      <th>target_expr_in_brain</th>\n",
       "      <th>ADNI_data</th>\n",
       "      <th>polgenom_freq_perc_y</th>\n",
       "      <th>ADNI</th>\n",
       "      <th>variant reliability</th>\n",
       "      <th>comments</th>\n",
       "      <th>test?</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2227533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4,4;13,11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000300</td>\n",
       "      <td>...</td>\n",
       "      <td>2227532</td>\n",
       "      <td>ENST00000449373_MORN1:0.0014,ENST00000488353_R...</td>\n",
       "      <td>MORN1:1.19,RER1:55.89</td>\n",
       "      <td>MORN1:3.76,RER1:25.86</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5896826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47,38;41,32</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs373612069</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>...</td>\n",
       "      <td>5896825</td>\n",
       "      <td>ENST00000481789_KCNAB2:0.0277,ENST00000378161_...</td>\n",
       "      <td>KCNAB2:69.76,NPHP4:6.44</td>\n",
       "      <td>KCNAB2:59.77,NPHP4:5.59</td>\n",
       "      <td>1:5956886:</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7041938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>109,100;138,118</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>rs542608175</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>7041937</td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16912416</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>189,90;160,95</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.043</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>...</td>\n",
       "      <td>16912415</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "      <td>no variant in ADNI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16913968</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>18,9;24,13;17,11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.065</td>\n",
       "      <td>rs752881304</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>...</td>\n",
       "      <td>16913967</td>\n",
       "      <td>ENST00000467938_CROCC:0.0053</td>\n",
       "      <td>CROCC:1.52</td>\n",
       "      <td>CROCC:15.56</td>\n",
       "      <td>1:17240463:</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT              VAFs   AC     AF     avsnp150  \\\n",
       "0  chr1   2227533   C   T         4,4;13,11  2.0  0.043            .   \n",
       "1  chr1   5896826   C   T       47,38;41,32  2.0  0.043  rs373612069   \n",
       "2  chr1   7041938   G   A   109,100;138,118  2.0  0.043  rs542608175   \n",
       "3  chr1  16912416   G   A     189,90;160,95  2.0  0.043            .   \n",
       "4  chr1  16913968   T   A  18,9;24,13;17,11  3.0  0.065  rs752881304   \n",
       "\n",
       "   gnomAD_genome_ALL  gnomAD_genome_NFE  ...       POS-1  \\\n",
       "0           0.000100           0.000300  ...     2227532   \n",
       "1           0.000700           0.000700  ...     5896825   \n",
       "2           0.000032           0.000067  ...     7041937   \n",
       "3           0.000097           0.000067  ...    16912415   \n",
       "4           0.004400           0.007600  ...    16913967   \n",
       "\n",
       "                                        Target_genes  \\\n",
       "0  ENST00000449373_MORN1:0.0014,ENST00000488353_R...   \n",
       "1  ENST00000481789_KCNAB2:0.0277,ENST00000378161_...   \n",
       "2                                               None   \n",
       "3                       ENST00000467938_CROCC:0.0053   \n",
       "4                       ENST00000467938_CROCC:0.0053   \n",
       "\n",
       "  target_expr_in_lymphocytes     target_expr_in_brain           ADNI_data  \\\n",
       "0      MORN1:1.19,RER1:55.89    MORN1:3.76,RER1:25.86  no variant in ADNI   \n",
       "1    KCNAB2:69.76,NPHP4:6.44  KCNAB2:59.77,NPHP4:5.59          1:5956886:   \n",
       "2                                                                    None   \n",
       "3                 CROCC:1.52              CROCC:15.56  no variant in ADNI   \n",
       "4                 CROCC:1.52              CROCC:15.56         1:17240463:   \n",
       "\n",
       "  polgenom_freq_perc_y ADNI  variant reliability comments  test?  \n",
       "0                  NaN  NaN                  NaN      NaN    NaN  \n",
       "1                  NaN  NaN                  NaN      NaN    NaN  \n",
       "2                  NaN  NaN                  NaN      NaN    NaN  \n",
       "3                  NaN  NaN                  NaN      NaN    NaN  \n",
       "4                  NaN  NaN                  NaN      NaN    NaN  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_with_hits_annot = pd.merge(enh_with_hits, manual_annotations[[\"CHROM\", \"POS\", \"polgenom_freq_perc\", \"ADNI\", \"variant reliability\", \"comments\", \"test?\"]], \n",
    "              on = [\"CHROM\", \"POS\"], how = \"left\")\n",
    "enh_with_hits_annot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "toc_cell": true,
   "toc_position": {
    "height": "969px",
    "left": "0px",
    "right": "2075.08px",
    "top": "111px",
    "width": "212px"
   },
   "toc_section_display": "block",
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
