{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The analysis provided in this notebook aims at selection and annotation of putative regulatory SNPs functional in human brain from whole-genome sequencing data (or targeted sequencing if putative regulatory regions were included as sequencing targets).  \n",
    "  \n",
    "These are the main steps of the analysis:\n",
    "1. Selection of SNPs located in promoters and non-promoter regulatory regions active in human brain.\n",
    "2. Selection of rare SNPs, enriched in the analyzed group of samples.\n",
    "3. Annotation with transcription factor motifs.\n",
    "4. Assignment of target genes which expression could be affected by the selected SNPs."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from scipy.stats import binom_test, spearmanr\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import pybedtools as pbt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Timestamp('2021-03-15 09:52:00.718381')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t1 = pd.to_datetime('today')\n",
    "t1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Paths to input and output files and 3rd party software"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Start by setting paths to input files, output folder and 3rd party software necessary to run this analysis.  \n",
    "Input files include your vcf file with variants and two interval_list files describing promoters and enhancers active in human brain. Common active promoters and common active enhancers identified and described in [Stepniak et al](https://www.biorxiv.org/content/10.1101/867861v2) are provided by default but you can replace them with your own regions files if you wish.  \n",
    "All intermediate output files will be saved to the output folder defined here.  \n",
    "If you use the VirtualBox Ubuntu image provided for this analysis the paths to software executables are already set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Input files\n",
    "#INPUT_VCF = \"data/test_variants_chr16.vcf.gz\"\n",
    "INPUT_VCF = \"data/Lib1-6.without113.norm.vcf.gz\"\n",
    "PROMOTER_REGIONS = \"data/brain_promoters_active.bed\" #last column should contain gene names, comma separated if promoters of several genes overlap\n",
    "ENHANCER_REGIONS = \"data/brain_enhancers_active.bed\" #last column should contain gene names if enhancer is located inside gene, comma separated, \".\" for intergenic enhancers = no gene overlaps\n",
    "\n",
    "### Output folder and files\n",
    "OUTPUT = \"output/\"\n",
    "ANNOTATED_PROMOTER_SNPs = OUTPUT + \"annotated_promoter_snps.csv\"\n",
    "ANNOTATED_ENHANCER_SNPs = OUTPUT + \"annotated_enhancer_snps.csv\"\n",
    "\n",
    "### Software and datasets paths\n",
    "GATK = \"/home/researcher/Programs/gatk-4.1.9.0/gatk\"\n",
    "ANNOVAR = \"/home/researcher/Programs/annovar/\"\n",
    "GTEX = 'data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct'\n",
    "ENHANCER_ACTIVITY = 'data/h3k27ac_coverage_quantile_normalized.csv'\n",
    "GENE_EXPRESSION = 'data/transcrtpts_rnaseq_quantile_normalized.csv'\n",
    "CHROMATIN_CONTACTS = 'data/predicted_contacts.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory 'output/': File exists\n"
     ]
    }
   ],
   "source": [
    "#Create output folder\n",
    "!mkdir $OUTPUT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Select biallelic SNPs located in promoters and enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the first step of the analysis biallelic SNPs located in promoter and enhancer regions are selected from the input .vcf files. Two vcf files are generated in this step: promoter_SNPs.vcf and enhancer_SNPs.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in the input file: 264271\n"
     ]
    }
   ],
   "source": [
    "count_before = !$GATK CountVariants -V $INPUT_VCF\n",
    "print(\"Number of variants in the input file:\", count_before[-4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/researcher/Programs/gatk-4.1.9.0/gatk SelectVariants -V data/Lib1-6.without113.norm.vcf.gz -L data/brain_promoters_active.bed --select-type-to-include SNP --restrict-alleles-to BIALLELIC -O output/promoter_SNPs.vcf\n",
      "Done\n",
      "/home/researcher/Programs/gatk-4.1.9.0/gatk CountVariants -V output/promoter_SNPs.vcf\n",
      "Done\n",
      "Number of biallelic SNPs in promoter regions: 83596 \n",
      "\n",
      "/home/researcher/Programs/gatk-4.1.9.0/gatk SelectVariants -V data/Lib1-6.without113.norm.vcf.gz -L data/brain_enhancers_active.bed --select-type-to-include SNP --restrict-alleles-to BIALLELIC -O output/enhancer_SNPs.vcf\n",
      "Done\n",
      "/home/researcher/Programs/gatk-4.1.9.0/gatk CountVariants -V output/enhancer_SNPs.vcf\n",
      "Done\n",
      "Number of biallelic SNPs in enhancer regions: 71681 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "select_logs = []\n",
    "count_logs = []\n",
    "for r, regions in [(\"promoter\", PROMOTER_REGIONS), (\"enhancer\", ENHANCER_REGIONS)]:\n",
    "    command1 =\"%s SelectVariants -V %s -L %s --select-type-to-include SNP --restrict-alleles-to BIALLELIC -O %s%s_SNPs.vcf\" % (GATK, INPUT_VCF, regions, OUTPUT, r)\n",
    "    print(command1)\n",
    "    log1 = !$command1\n",
    "    select_logs.append(log1)\n",
    "    print(\"Done\")\n",
    "    \n",
    "    command2 = \"%s CountVariants -V %s%s_SNPs.vcf\" % (GATK, OUTPUT, r)\n",
    "    print(command2)\n",
    "    log2 = !$command2\n",
    "    count_logs.append(log2)\n",
    "    print(\"Done\")\n",
    "    print(\"Number of biallelic SNPs in %s regions:\" % r, log2[-4], \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Terminal output from 3rd party software is stored in log variables. You can check them if you suspect that something could have gone wrong during the calculations:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['09:52:11.596 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/researcher/Programs/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so',\n",
       " 'Mar 15, 2021 9:52:11 AM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine',\n",
       " 'INFO: Failed to detect whether we are running on Google Compute Engine.',\n",
       " '09:52:11.749 INFO  SelectVariants - ------------------------------------------------------------',\n",
       " '09:52:11.750 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.1.9.0',\n",
       " '09:52:11.750 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/',\n",
       " '09:52:11.750 INFO  SelectVariants - Executing as researcher@brain-reg-var on Linux v5.8.0-44-generic amd64',\n",
       " '09:52:11.750 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v11.0.10+9-Ubuntu-0ubuntu1.20.04',\n",
       " '09:52:11.750 INFO  SelectVariants - Start Date/Time: March 15, 2021 at 9:52:11 AM CET',\n",
       " '09:52:11.751 INFO  SelectVariants - ------------------------------------------------------------',\n",
       " '09:52:11.751 INFO  SelectVariants - ------------------------------------------------------------',\n",
       " '09:52:11.751 INFO  SelectVariants - HTSJDK Version: 2.23.0',\n",
       " '09:52:11.752 INFO  SelectVariants - Picard Version: 2.23.3',\n",
       " '09:52:11.752 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2',\n",
       " '09:52:11.752 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false',\n",
       " '09:52:11.752 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true',\n",
       " '09:52:11.752 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false',\n",
       " '09:52:11.752 INFO  SelectVariants - Deflater: IntelDeflater',\n",
       " '09:52:11.752 INFO  SelectVariants - Inflater: IntelInflater',\n",
       " '09:52:11.752 INFO  SelectVariants - GCS max retries/reopens: 20',\n",
       " '09:52:11.752 INFO  SelectVariants - Requester pays: disabled',\n",
       " '09:52:11.753 INFO  SelectVariants - Initializing engine',\n",
       " '09:52:11.873 INFO  FeatureManager - Using codec VCFCodec to read file file:///home/researcher/brain_regulatory_variants_tool/brain_reg_var/data/Lib1-6.without113.norm.vcf.gz',\n",
       " '09:52:12.255 INFO  FeatureManager - Using codec BEDCodec to read file file:///home/researcher/brain_regulatory_variants_tool/brain_reg_var/data/brain_promoters_active.bed',\n",
       " '09:52:12.454 INFO  IntervalArgumentCollection - Processing 25573839 bp from intervals',\n",
       " '09:52:12.487 INFO  SelectVariants - Done initializing engine',\n",
       " '09:52:12.631 INFO  ProgressMeter - Starting traversal',\n",
       " '09:52:12.632 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute',\n",
       " '09:52:22.718 INFO  ProgressMeter -       chr15:89751516              0.2                 61000         362915.2',\n",
       " '09:52:25.726 INFO  ProgressMeter -       chrX:118345852              0.2                 85470         391645.0',\n",
       " '09:52:25.726 INFO  ProgressMeter - Traversal complete. Processed 85470 total variants in 0.2 minutes.',\n",
       " '09:52:26.128 INFO  SelectVariants - Shutting down engine',\n",
       " '[March 15, 2021 at 9:52:26 AM CET] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.24 minutes.',\n",
       " 'Runtime.totalMemory()=713031680',\n",
       " 'Using GATK jar /home/researcher/Programs/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar',\n",
       " 'Running:',\n",
       " '    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/researcher/Programs/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar SelectVariants -V data/Lib1-6.without113.norm.vcf.gz -L data/brain_promoters_active.bed --select-type-to-include SNP --restrict-alleles-to BIALLELIC -O output/promoter_SNPs.vcf']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "select_logs[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotate with allele frequencies from gnomAD genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will use ANNOVAR to annotate promoter and enhancer SNPs with population frequencies from the gnomAD genome resource."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /home/researcher/Programs/annovar/table_annovar.pl output/promoter_SNPs.vcf /home/researcher/Programs/annovar/humandb/ -buildver hg38 -remove -protocol gnomad_genome -operation f -nastring . -vcfinput -out output/promoter_SNPs\n",
      "Done\n",
      "perl /home/researcher/Programs/annovar/table_annovar.pl output/enhancer_SNPs.vcf /home/researcher/Programs/annovar/humandb/ -buildver hg38 -remove -protocol gnomad_genome -operation f -nastring . -vcfinput -out output/enhancer_SNPs\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "annovar_logs = []\n",
    "for r in [\"promoter\", \"enhancer\"]:\n",
    "    command = \"perl %stable_annovar.pl %s%s_SNPs.vcf %shumandb/ -buildver hg38 -remove -protocol gnomad_genome -operation f -nastring . -vcfinput -out %s%s_SNPs\" % (ANNOVAR, OUTPUT, r, ANNOVAR, OUTPUT, r)\n",
    "    print(command)\n",
    "    log = !$command\n",
    "    annovar_logs.append(log)\n",
    "    print(\"Done\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 798M\n",
      "-rw-rw-r-- 1 researcher researcher  471 Feb  4 13:55 promoter_SNPs.hg38_multianno.csv\n",
      "-rw-rw-r-- 1 researcher researcher  288 Feb  4 13:56 promoter_SNPs.csv\n",
      "-rw-rw-r-- 1 researcher researcher  92M Mar 12 10:38 promoter_SNPs.hg38_multianno.nomissing.vcf\n",
      "-rw-rw-r-- 1 researcher researcher  80M Mar 12 10:38 enhancer_SNPs.hg38_multianno.nomissing.vcf\n",
      "-rw-rw-r-- 1 researcher researcher 9.4M Mar 12 10:39 promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "-rw-rw-r-- 1 researcher researcher 136K Mar 12 10:39 promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf.idx\n",
      "-rw-rw-r-- 1 researcher researcher 7.1M Mar 12 10:39 enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "-rw-rw-r-- 1 researcher researcher 136K Mar 12 10:39 enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf.idx\n",
      "-rw-rw-r-- 1 researcher researcher 327K Mar 12 10:39 promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\n",
      "-rw-rw-r-- 1 researcher researcher 250K Mar 12 10:39 enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\n",
      "-rw-rw-r-- 1 researcher researcher  60K Mar 12 10:39 promoter_rare_enriched_SNPs.bed\n",
      "-rw-rw-r-- 1 researcher researcher  36K Mar 12 10:39 enhancer_rare_enriched_SNPs.bed\n",
      "-rw-rw-r-- 1 researcher researcher 2.8M Mar 12 12:33 promoter_rare_enriched_SNPs_motifbreakR-scores.csv\n",
      "-rw-rw-r-- 1 researcher researcher 807K Mar 12 13:49 enhancer_rare_enriched_SNPs_motifbreakR-scores.csv\n",
      "-rw-rw-r-- 1 researcher researcher 165K Mar 12 13:52 annotated_promoter_snps.csv\n",
      "-rw-rw-r-- 1 researcher researcher 143K Mar 12 13:52 annotated_enhancer_snps.csv\n",
      "-rw-rw-r-- 1 researcher researcher  74M Mar 15 09:52 promoter_SNPs.vcf\n",
      "-rw-rw-r-- 1 researcher researcher 1.5M Mar 15 09:52 promoter_SNPs.vcf.idx\n",
      "-rw-rw-r-- 1 researcher researcher  64M Mar 15 09:52 enhancer_SNPs.vcf\n",
      "-rw-rw-r-- 1 researcher researcher 1.9M Mar 15 09:52 enhancer_SNPs.vcf.idx\n",
      "-rw-rw-r-- 1 researcher researcher  77M Mar 15 09:53 promoter_SNPs.avinput\n",
      "-rw-rw-r-- 1 researcher researcher  81M Mar 15 09:55 promoter_SNPs.hg38_multianno.txt\n",
      "-rw-rw-r-- 1 researcher researcher  92M Mar 15 09:55 promoter_SNPs.hg38_multianno.vcf\n",
      "-rw-rw-r-- 1 researcher researcher  67M Mar 15 09:55 enhancer_SNPs.avinput\n",
      "-rw-rw-r-- 1 researcher researcher  71M Mar 15 09:56 enhancer_SNPs.hg38_multianno.txt\n",
      "-rw-rw-r-- 1 researcher researcher  80M Mar 15 09:56 enhancer_SNPs.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "!ls -lrth $OUTPUT/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Annovar has generated two \\*.hg38_multianno.vcf files which contain frequency annotations."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Select SNPs with MAF < 0.01"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the next step we will choose only rare SNPs - those with minor allele frequency (MAF) below 0.01 in all populations included in gnomAD genome.  \n",
    "  \n",
    "First we will replace \".\", which marks missing MAF values, with \"100.\". As a result all variants with missing frequency data will be filtered out in the next step. You can motify this behaviour by changing the value which is inserted instead of \".\" but it must be a float value for the filtering to work properly. For example if you would like to treat missing data as equal to very low frequency you may replace \"100.\" with \"0.0\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations = ['gnomAD_genome_ALL',\n",
    "             'gnomAD_genome_AFR',\n",
    "             'gnomAD_genome_AMR',\n",
    "             'gnomAD_genome_ASJ',\n",
    "             'gnomAD_genome_EAS',\n",
    "             'gnomAD_genome_FIN',\n",
    "             'gnomAD_genome_NFE',\n",
    "             'gnomAD_genome_OTH']\n",
    "for r in [\"promoter\", \"enhancer\"]:\n",
    "    with open('%s%s_SNPs.hg38_multianno.nomissing.vcf' % (OUTPUT, r), 'w') as o:\n",
    "        for line in open('%s%s_SNPs.hg38_multianno.vcf' % (OUTPUT, r)).readlines():    \n",
    "            for el in annotations:\n",
    "                if el + '=.' in line:\n",
    "                    line = line.replace(el + '=.', el + '=100.0')\n",
    "            o.write(line)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \\*hg38_multianno.nomissing.vcf files contain \".\" frequency values replaced by 100.0.  \n",
    "\n",
    "Now we select rare variants and save them in \\*.hg38_multianno.nomissing.gnomad_below_0.01.vcf files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "select_rare_logs = []\n",
    "for r in [\"promoter\", \"enhancer\"]:\n",
    "    command = \"%s SelectVariants -V %s%s_SNPs.hg38_multianno.nomissing.vcf\" \\\n",
    "            \" -select 'gnomAD_genome_ALL < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_AFR < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_AMR < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_ASJ < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_EAS < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_FIN < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_NFE < 0.01'\" \\\n",
    "            \" -select 'gnomAD_genome_OTH < 0.01'\" \\\n",
    "            \" -O %s%s_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\" % (GATK, OUTPUT, r, OUTPUT, r) \n",
    "    log = !$command\n",
    "    select_rare_logs.append(log)\n",
    "    print(\"Done\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's check how many variants have been selected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/researcher/Programs/gatk-4.1.9.0/gatk CountVariants -V output/promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "Done\n",
      "Number of rare SNPs in promoter regions: 8762\n",
      "/home/researcher/Programs/gatk-4.1.9.0/gatk CountVariants -V output/enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\n",
      "Done\n",
      "Number of rare SNPs in enhancer regions: 6489\n"
     ]
    }
   ],
   "source": [
    "count_rare_logs = []\n",
    "for r in [\"promoter\", \"enhancer\"]:\n",
    "    command = \"%s CountVariants -V %s%s_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf\" % (GATK, OUTPUT, r)\n",
    "    print(command)\n",
    "    log = !$command\n",
    "    count_rare_logs.append(log)\n",
    "    print(\"Done\")\n",
    "    print(\"Number of rare SNPs in %s regions:\" % r, log[-4])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Choose SNPs enriched in analyzed cohort compared to chosen population"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will now use binomial test to choose SNPs enriched in out analyzed cohort compared to population. Since the variants anayzed in the test example are from Polish population gnomad_NFE (non-Finnish European) population is chosen but you can modify this option according to your needs.  \n",
    "\n",
    "First we need to reformat vcf files to csv to be able to read them with pandas. For this we will use the VariantsToTable tool from the GATK package. You can specify fields from the vcf which will be present in the csv. Here I choose information about SNP position, REF and ALT alleles, allele counts and frequencies in the analyzed cohort and gnomAD genome frequencies in global population (gnomAD_genome_ALL) and non-Finnish Europeans (gnomAD_genome_NFE)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/researcher/Programs/gatk-4.1.9.0/gatk VariantsToTable  -V output/promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf -F CHROM -F POS -F REF -F ALT -F AC -F AF -F AN -F gnomAD_genome_ALL -F gnomAD_genome_NFE -O output/promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\n",
      "Done\n",
      "/home/researcher/Programs/gatk-4.1.9.0/gatk VariantsToTable  -V output/enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf -F CHROM -F POS -F REF -F ALT -F AC -F AF -F AN -F gnomAD_genome_ALL -F gnomAD_genome_NFE -O output/enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "totable_logs = []\n",
    "for r in [\"promoter\", \"enhancer\"]:\n",
    "    command = \"%s VariantsToTable  -V %s%s_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.vcf \" \\\n",
    "    \"-F CHROM -F POS -F REF -F ALT -F AC -F AF -F AN \" \\\n",
    "    \"-F gnomAD_genome_ALL -F gnomAD_genome_NFE \" \\\n",
    "    \"-O %s%s_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\" % (GATK, OUTPUT, r, OUTPUT, r)\n",
    "    print(command)\n",
    "    log = !$command\n",
    "    totable_logs.append(log)\n",
    "    print(\"Done\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read generated csv files with pandas and inspect their contents."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.067</td>\n",
       "      <td>30</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939354</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013784</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033651</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.026</td>\n",
       "      <td>38</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0  chr1    30570   C   T   2  0.067  30           0.000000             0.0000   \n",
       "1  chr1   939354   C   T   1  0.022  46           0.001200             0.0014   \n",
       "2  chr1   939561   G   A   2  0.043  46           0.000052             0.0000   \n",
       "3  chr1  1013784   C   T   1  0.022  46           0.000400             0.0006   \n",
       "4  chr1  1033651   A   C   1  0.026  38           0.000000             0.0000   \n",
       "\n",
       "   Unnamed: 9  \n",
       "0         NaN  \n",
       "1         NaN  \n",
       "2         NaN  \n",
       "3         NaN  \n",
       "4         NaN  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_promoter_snps = pd.read_csv(\"%s/promoter_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\" % OUTPUT, sep = '\\t')\n",
    "rare_promoter_snps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20184</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004300</td>\n",
       "      <td>0.0065</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20254</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1157576</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1158496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>0.0040</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1159286</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.0074</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0  chr1    20184   A   G   1  0.023  44           0.004300             0.0065   \n",
       "1  chr1    20254   G   A   1  0.023  44           0.004400             0.0029   \n",
       "2  chr1  1157576   G   C   1  0.022  46           0.000065             0.0001   \n",
       "3  chr1  1158496   C   T   1  0.022  46           0.002500             0.0040   \n",
       "4  chr1  1159286   C   G   1  0.022  46           0.004700             0.0074   \n",
       "\n",
       "   Unnamed: 9  \n",
       "0         NaN  \n",
       "1         NaN  \n",
       "2         NaN  \n",
       "3         NaN  \n",
       "4         NaN  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enhancer_snps = pd.read_csv(\"%s/enhancer_SNPs.hg38_multianno.nomissing.gnomad_below_0.01.csv\" % OUTPUT, sep = '\\t')\n",
    "rare_enhancer_snps.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Both csv files contain empty \"Unnamed:9\" column - remove it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20184</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004300</td>\n",
       "      <td>0.0065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20254</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.0029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1157576</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1158496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>0.0040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1159286</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.0074</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE\n",
       "0  chr1    20184   A   G   1  0.023  44           0.004300             0.0065\n",
       "1  chr1    20254   G   A   1  0.023  44           0.004400             0.0029\n",
       "2  chr1  1157576   G   C   1  0.022  46           0.000065             0.0001\n",
       "3  chr1  1158496   C   T   1  0.022  46           0.002500             0.0040\n",
       "4  chr1  1159286   C   G   1  0.022  46           0.004700             0.0074"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for df in [rare_enhancer_snps, rare_promoter_snps]:\n",
    "    for col in df.columns:\n",
    "        if \"Unnamed:\" in col:\n",
    "            df.drop(labels = col, axis=1, inplace = True)\n",
    "rare_enhancer_snps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.067</td>\n",
       "      <td>30</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939354</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.0014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013784</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033651</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.026</td>\n",
       "      <td>38</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE\n",
       "0  chr1    30570   C   T   2  0.067  30           0.000000             0.0000\n",
       "1  chr1   939354   C   T   1  0.022  46           0.001200             0.0014\n",
       "2  chr1   939561   G   A   2  0.043  46           0.000052             0.0000\n",
       "3  chr1  1013784   C   T   1  0.022  46           0.000400             0.0006\n",
       "4  chr1  1033651   A   C   1  0.026  38           0.000000             0.0000"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_promoter_snps.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calculate p-values for one-sided binomial test in which the number of successes is equal to the number of ALT alleles in the cohort (AC), the number of trials is equal to the total number of identified alleles (AN) and probability of success is equal to population frequency of ALT allele (gnomAD_genome_NFE). The alternative hypothesis is that observed frequency is greater than expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calc_binom_pval(row):\n",
    "        x = row['AC']\n",
    "        n = row['AN']\n",
    "        p = float(row['gnomAD_genome_NFE'])        \n",
    "\n",
    "        return binom_test(x,n,p, alternative = 'greater')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "for df in [rare_enhancer_snps, rare_promoter_snps]:            \n",
    "    df['binom_pval']=df.apply(calc_binom_pval, axis=1)\n",
    "    \n",
    "    #apply correction for multiple hypothesis testing with the Benjamini-Hochberg procedure, use FDR = 0.01\n",
    "    multipletests_correction = multipletests(df['binom_pval'], alpha=0.01, \n",
    "              method='fdr_bh', is_sorted=False, returnsorted=False)\n",
    "    df['B-H_reject_H0'] = multipletests_correction[0]\n",
    "    df['corrected_binom_pval'] = multipletests_correction[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.067</td>\n",
       "      <td>30</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939354</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0.062412</td>\n",
       "      <td>False</td>\n",
       "      <td>0.103377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013784</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.027231</td>\n",
       "      <td>False</td>\n",
       "      <td>0.057433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033651</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.026</td>\n",
       "      <td>38</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0  chr1    30570   C   T   2  0.067  30           0.000000             0.0000   \n",
       "1  chr1   939354   C   T   1  0.022  46           0.001200             0.0014   \n",
       "2  chr1   939561   G   A   2  0.043  46           0.000052             0.0000   \n",
       "3  chr1  1013784   C   T   1  0.022  46           0.000400             0.0006   \n",
       "4  chr1  1033651   A   C   1  0.026  38           0.000000             0.0000   \n",
       "\n",
       "   binom_pval  B-H_reject_H0  corrected_binom_pval  \n",
       "0    0.000000           True              0.000000  \n",
       "1    0.062412          False              0.103377  \n",
       "2    0.000000           True              0.000000  \n",
       "3    0.027231          False              0.057433  \n",
       "4    0.000000           True              0.000000  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_promoter_snps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20184</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004300</td>\n",
       "      <td>0.0065</td>\n",
       "      <td>0.249438</td>\n",
       "      <td>False</td>\n",
       "      <td>0.273069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20254</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023</td>\n",
       "      <td>44</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>0.119958</td>\n",
       "      <td>False</td>\n",
       "      <td>0.165396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1157576</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.004590</td>\n",
       "      <td>False</td>\n",
       "      <td>0.017124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1158496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>0.0040</td>\n",
       "      <td>0.168371</td>\n",
       "      <td>False</td>\n",
       "      <td>0.208415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1159286</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.0074</td>\n",
       "      <td>0.289414</td>\n",
       "      <td>False</td>\n",
       "      <td>0.305077</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  gnomAD_genome_NFE  \\\n",
       "0  chr1    20184   A   G   1  0.023  44           0.004300             0.0065   \n",
       "1  chr1    20254   G   A   1  0.023  44           0.004400             0.0029   \n",
       "2  chr1  1157576   G   C   1  0.022  46           0.000065             0.0001   \n",
       "3  chr1  1158496   C   T   1  0.022  46           0.002500             0.0040   \n",
       "4  chr1  1159286   C   G   1  0.022  46           0.004700             0.0074   \n",
       "\n",
       "   binom_pval  B-H_reject_H0  corrected_binom_pval  \n",
       "0    0.249438          False              0.273069  \n",
       "1    0.119958          False              0.165396  \n",
       "2    0.004590          False              0.017124  \n",
       "3    0.168371          False              0.208415  \n",
       "4    0.289414          False              0.305077  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enhancer_snps.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select SNPs significantly enriched in analyzed cohort at FDR = 0.01."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1313 SNPs in promoters are enriched in analyzed cohort.\n",
      "784 SNPs in enhancers are enriched in analyzed cohort.\n"
     ]
    }
   ],
   "source": [
    "rare_enriched_promoter_snps = rare_promoter_snps[rare_promoter_snps[\"B-H_reject_H0\"]]\n",
    "rare_enriched_enhancer_snps = rare_enhancer_snps[rare_enhancer_snps[\"B-H_reject_H0\"]]\n",
    "print(len(rare_enriched_promoter_snps), \"SNPs in promoters are enriched in analyzed cohort.\")\n",
    "print(len(rare_enriched_enhancer_snps), \"SNPs in enhancers are enriched in analyzed cohort.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotate with predicted TF binding sites"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use motifbreakR package and Hocomoco v11 full database of TF models to identify SNPs which may destroy or create a TF binding site.  \n",
    "\n",
    "We will first save SNPs to a bed file which will serve as input to motifbreakR."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "snps_bed_files = []\n",
    "for snps_df, r in [(rare_enriched_promoter_snps, \"promoter\"), (rare_enriched_enhancer_snps, \"enhancer\")]:\n",
    "    snps_bed = pd.DataFrame()\n",
    "    snps_bed[\"chromosome\"] = snps_df[\"CHROM\"]\n",
    "    snps_bed[\"start\"] = snps_df[\"POS\"] - 1\n",
    "    snps_bed[\"end\"] = snps_df[\"POS\"]\n",
    "    snps_bed[\"name\"] = snps_df[\"CHROM\"] + \":\" + snps_df[\"POS\"].astype(str) + \":\" + snps_df[\"REF\"] + \":\" + snps_df[\"ALT\"]\n",
    "    snps_bed[\"score\"] = 0\n",
    "    snps_bed[\"strand\"] = \"+\"\n",
    "    output_bed_path = \"%s%s_rare_enriched_SNPs.bed\" % (OUTPUT, r)\n",
    "    snps_bed.to_csv(output_bed_path, sep=\"\\t\", index=False, header=False)\n",
    "    snps_bed_files.append(output_bed_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 780K\n",
      "-rw-rw-r-- 1 researcher researcher 526K Feb 24 13:51 annotate_brain_regulatory_variants.html\n",
      "-rw-rw-r-- 1 researcher researcher 168K Feb 24 16:49 annotate_brain_regulatory_variants.pdf\n",
      "drwxrwxr-x 2 researcher researcher 4.0K Mar 12 10:33 data\n",
      "-rw-rw-r-- 1 researcher researcher 1.4K Mar 15 09:51 install_dependencies.sh\n",
      "-rw-rw-r-- 1 researcher researcher 6.9K Mar 15 09:51 README.md\n",
      "drwxrwxr-x 2 researcher researcher 4.0K Mar 15 09:56 output\n",
      "-rw-rw-r-- 1 researcher researcher  61K Mar 15 09:57 annotate_brain_regulatory_variants.ipynb\n"
     ]
    }
   ],
   "source": [
    "!ls -lrth $output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quick look at one of the bed files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t30569\t30570\tchr1:30570:C:T\t0\t+\n",
      "chr1\t939560\t939561\tchr1:939561:G:A\t0\t+\n",
      "chr1\t1033650\t1033651\tchr1:1033651:A:C\t0\t+\n",
      "chr1\t1033651\t1033652\tchr1:1033652:G:C\t0\t+\n",
      "chr1\t1033652\t1033653\tchr1:1033653:T:C\t0\t+\n",
      "chr1\t1033662\t1033663\tchr1:1033663:G:C\t0\t+\n",
      "chr1\t1033663\t1033664\tchr1:1033664:G:C\t0\t+\n",
      "chr1\t1034901\t1034902\tchr1:1034902:A:T\t0\t+\n",
      "chr1\t1117137\t1117138\tchr1:1117138:T:C\t0\t+\n",
      "chr1\t1470770\t1470771\tchr1:1470771:T:G\t0\t+\n"
     ]
    }
   ],
   "source": [
    "!head \"$OUTPUT\"promoter_rare_enriched_SNPs.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To analyze our SNPs with motifbreakR we load hg38 as a reference genome and we choose all human TF models from HOCOMOCO v11."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Loading required package: grid\n",
      "\n",
      "R[write to console]: Loading required package: MotifDb\n",
      "\n",
      "R[write to console]: Loading required package: BiocGenerics\n",
      "\n",
      "R[write to console]: Loading required package: parallel\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from package:parallel:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from package:base:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: S4Vectors\n",
      "\n",
      "R[write to console]: Loading required package: stats4\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: S4Vectors\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from package:base:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: IRanges\n",
      "\n",
      "R[write to console]: Loading required package: GenomicRanges\n",
      "\n",
      "R[write to console]: Loading required package: GenomeInfoDb\n",
      "\n",
      "R[write to console]: Loading required package: Biostrings\n",
      "\n",
      "R[write to console]: Loading required package: XVector\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: Biostrings\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from package:base:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "R[write to console]: See system.file(\"LICENSE\", package=\"MotifDb\") for use restrictions.\n",
      "\n",
      "R[write to console]: Loading required package: BSgenome\n",
      "\n",
      "R[write to console]: Loading required package: rtracklayer\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 768\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "\n",
    "#load libraries and select TF motifs\n",
    "library(motifbreakR)\n",
    "library(BSgenome.Hsapiens.UCSC.hg38)\n",
    "library(MotifDb)\n",
    "\n",
    "motifs <- query(MotifDb, andStrings=c(\"hocomocov11\", \"hsapiens\"))\n",
    "length(motifs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "motifbreakR implements three methods for calculation of motif match scores: \"log\", \"default\" and \"ic\". We will use the \"log\" method with uniform background and filter results with p-value threshold = 1e-4."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "score_snps <- function(snps_file, out_file) {\n",
    "    #read SNPs from input bed file\n",
    "    snps.mb.frombed <- snps.from.file(file = snps_file, search.genome = BSgenome.Hsapiens.UCSC.hg38, format = \"bed\")\n",
    "\n",
    "    #calculate scores\n",
    "    results_log <- motifbreakR(snpList = snps.mb.frombed, filterp = TRUE,\n",
    "                           pwmList = motifs,\n",
    "                           threshold = 1e-5,\n",
    "                           method = \"log\",\n",
    "                           bkg = c(A=0.25, C=0.25, G=0.25, T=0.25),\n",
    "                           BPPARAM = BiocParallel::bpparam())\n",
    "\n",
    "    #reformat results to dataframe and save to file\n",
    "    results_log_df <- data.frame(results_log)\n",
    "    write.table(results_log_df, out_file, quote=F, sep=\"\\t\", row.names=F)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculate scores for input: output/promoter_rare_enriched_SNPs.bed, save output to: output/promoter_rare_enriched_SNPs_motifbreakR-scores.csv\n",
      "Done\n",
      "Calculate scores for input: output/enhancer_rare_enriched_SNPs.bed, save output to: output/enhancer_rare_enriched_SNPs_motifbreakR-scores.csv\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "for snp_bed in snps_bed_files:\n",
    "    snp_scores_csv = snp_bed.replace(\".bed\", \"_motifbreakR-scores.csv\")\n",
    "    print(\"Calculate scores for input: %s, save output to: %s\" % (snp_bed, snp_scores_csv))\n",
    "    %Rpush snp_bed\n",
    "    %Rpush snp_scores_csv\n",
    "    %R snp_scores = score_snps(snp_bed, snp_scores_csv)\n",
    "    print(\"Done\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Inspect results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>SNP_id</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varType</th>\n",
       "      <th>motifPos</th>\n",
       "      <th>...</th>\n",
       "      <th>seqMatch</th>\n",
       "      <th>pctRef</th>\n",
       "      <th>pctAlt</th>\n",
       "      <th>scoreRef</th>\n",
       "      <th>scoreAlt</th>\n",
       "      <th>Refpvalue</th>\n",
       "      <th>Altpvalue</th>\n",
       "      <th>altPos</th>\n",
       "      <th>alleleDiff</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>45783029</td>\n",
       "      <td>45783029</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>chr19:45783029:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-10, 7)</td>\n",
       "      <td>...</td>\n",
       "      <td>aggaggtggggaaGggggggtgaggacaggaccag</td>\n",
       "      <td>0.857928</td>\n",
       "      <td>0.881182</td>\n",
       "      <td>8.465864</td>\n",
       "      <td>9.665273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.199409</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>222320241</td>\n",
       "      <td>222320241</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2:222320241:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-11, 4)</td>\n",
       "      <td>...</td>\n",
       "      <td>gcgaccgcctcCccctcccgcctccccgtcc</td>\n",
       "      <td>0.923301</td>\n",
       "      <td>0.894667</td>\n",
       "      <td>10.236474</td>\n",
       "      <td>8.806286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.430189</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>6556629</td>\n",
       "      <td>6556629</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:6556629:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-7, 3)</td>\n",
       "      <td>...</td>\n",
       "      <td>caccccCcgccccgccgggga</td>\n",
       "      <td>0.820873</td>\n",
       "      <td>0.957512</td>\n",
       "      <td>2.275853</td>\n",
       "      <td>9.838534</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>7.562681</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr14</td>\n",
       "      <td>92106635</td>\n",
       "      <td>92106635</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14:92106635:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-4, 14)</td>\n",
       "      <td>...</td>\n",
       "      <td>ctctcctccgcccaCcccccccctccccggcccgccc</td>\n",
       "      <td>0.907857</td>\n",
       "      <td>0.916432</td>\n",
       "      <td>11.693748</td>\n",
       "      <td>12.290103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0.596354</td>\n",
       "      <td>weak</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>26692643</td>\n",
       "      <td>26692643</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2:26692643:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-3, 12)</td>\n",
       "      <td>...</td>\n",
       "      <td>gaggcggcctgCggggggggggcggggggcg</td>\n",
       "      <td>0.900518</td>\n",
       "      <td>0.951450</td>\n",
       "      <td>9.098499</td>\n",
       "      <td>11.642407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2.543908</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqnames      start        end  width strand              SNP_id REF ALT  \\\n",
       "0    chr19   45783029   45783029      1      +  chr19:45783029:T:G   T   G   \n",
       "1     chr2  222320241  222320241      1      -  chr2:222320241:T:C   T   C   \n",
       "2    chr17    6556629    6556629      1      -   chr17:6556629:G:C   G   C   \n",
       "3    chr14   92106635   92106635      1      -  chr14:92106635:G:C   G   C   \n",
       "4     chr2   26692643   26692643      1      +   chr2:26692643:C:G   C   G   \n",
       "\n",
       "  varType   motifPos  ...                                           seqMatch  \\\n",
       "0     SNV  c(-10, 7)  ...        aggaggtggggaaGggggggtgaggacaggaccag           \n",
       "1     SNV  c(-11, 4)  ...          gcgaccgcctcCccctcccgcctccccgtcc             \n",
       "2     SNV   c(-7, 3)  ...               caccccCcgccccgccgggga                  \n",
       "3     SNV  c(-4, 14)  ...       ctctcctccgcccaCcccccccctccccggcccgccc          \n",
       "4     SNV  c(-3, 12)  ...          gaggcggcctgCggggggggggcggggggcg             \n",
       "\n",
       "     pctRef    pctAlt   scoreRef   scoreAlt  Refpvalue  Altpvalue  altPos  \\\n",
       "0  0.857928  0.881182   8.465864   9.665273        NaN        NaN       1   \n",
       "1  0.923301  0.894667  10.236474   8.806286        NaN        NaN       1   \n",
       "2  0.820873  0.957512   2.275853   9.838534        NaN        NaN       1   \n",
       "3  0.907857  0.916432  11.693748  12.290103        NaN        NaN       1   \n",
       "4  0.900518  0.951450   9.098499  11.642407        NaN        NaN       1   \n",
       "\n",
       "   alleleDiff  effect  \n",
       "0    1.199409  strong  \n",
       "1   -1.430189  strong  \n",
       "2    7.562681  strong  \n",
       "3    0.596354    weak  \n",
       "4    2.543908  strong  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_SNPs_motifbreakr = pd.read_csv(\"%spromoter_rare_enriched_SNPs_motifbreakR-scores.csv\" % OUTPUT, sep = \"\\t\")\n",
    "promoter_SNPs_motifbreakr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>SNP_id</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varType</th>\n",
       "      <th>motifPos</th>\n",
       "      <th>...</th>\n",
       "      <th>seqMatch</th>\n",
       "      <th>pctRef</th>\n",
       "      <th>pctAlt</th>\n",
       "      <th>scoreRef</th>\n",
       "      <th>scoreAlt</th>\n",
       "      <th>Refpvalue</th>\n",
       "      <th>Altpvalue</th>\n",
       "      <th>altPos</th>\n",
       "      <th>alleleDiff</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>90397298</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2:90397298:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-10, 9)</td>\n",
       "      <td>...</td>\n",
       "      <td>atcatcttcgagtggAaccgaaaggaatcgccaaatgga</td>\n",
       "      <td>0.858048</td>\n",
       "      <td>0.890445</td>\n",
       "      <td>10.935724</td>\n",
       "      <td>14.099791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3.164068</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>90397732</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2:90397732:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-10, 6)</td>\n",
       "      <td>...</td>\n",
       "      <td>tcgaatggaattGaatagaatcaacgaatggaa</td>\n",
       "      <td>0.903968</td>\n",
       "      <td>0.918037</td>\n",
       "      <td>9.979373</td>\n",
       "      <td>10.774472</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0.795099</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>116359966</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr12:116359966:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-7, 8)</td>\n",
       "      <td>...</td>\n",
       "      <td>ccttccttcctTcttcctgtttcttttagac</td>\n",
       "      <td>0.869172</td>\n",
       "      <td>0.960194</td>\n",
       "      <td>6.350885</td>\n",
       "      <td>12.108658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>5.757773</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>26475911</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>chr13:26475911:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-15, 4)</td>\n",
       "      <td>...</td>\n",
       "      <td>atggggcaagggggcAaggctgggaacaaggctgtggcc</td>\n",
       "      <td>0.847194</td>\n",
       "      <td>0.867085</td>\n",
       "      <td>8.574766</td>\n",
       "      <td>9.675056</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.100290</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>4857069</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>chr4:4857069:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>c(-9, 2)</td>\n",
       "      <td>...</td>\n",
       "      <td>gccccccGccccccccccccccc</td>\n",
       "      <td>0.958712</td>\n",
       "      <td>0.924593</td>\n",
       "      <td>10.942699</td>\n",
       "      <td>8.989977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.952722</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqnames      start        end  width strand               SNP_id REF ALT  \\\n",
       "0     chr2   90397298   90397298      1      +    chr2:90397298:T:G   T   G   \n",
       "1     chr2   90397732   90397732      1      -    chr2:90397732:G:A   G   A   \n",
       "2    chr12  116359966  116359966      1      -  chr12:116359966:T:C   T   C   \n",
       "3    chr13   26475911   26475911      1      +   chr13:26475911:A:C   A   C   \n",
       "4     chr4    4857069    4857069      1      -     chr4:4857069:C:A   C   A   \n",
       "\n",
       "  varType   motifPos  ...                                           seqMatch  \\\n",
       "0     SNV  c(-10, 9)  ...      atcatcttcgagtggAaccgaaaggaatcgccaaatgga         \n",
       "1     SNV  c(-10, 6)  ...         tcgaatggaattGaatagaatcaacgaatggaa            \n",
       "2     SNV   c(-7, 8)  ...          ccttccttcctTcttcctgtttcttttagac             \n",
       "3     SNV  c(-15, 4)  ...      atggggcaagggggcAaggctgggaacaaggctgtggcc         \n",
       "4     SNV   c(-9, 2)  ...              gccccccGccccccccccccccc                 \n",
       "\n",
       "     pctRef    pctAlt   scoreRef   scoreAlt  Refpvalue  Altpvalue  altPos  \\\n",
       "0  0.858048  0.890445  10.935724  14.099791        NaN        NaN       1   \n",
       "1  0.903968  0.918037   9.979373  10.774472        NaN        NaN       1   \n",
       "2  0.869172  0.960194   6.350885  12.108658        NaN        NaN       1   \n",
       "3  0.847194  0.867085   8.574766   9.675056        NaN        NaN       1   \n",
       "4  0.958712  0.924593  10.942699   8.989977        NaN        NaN       1   \n",
       "\n",
       "   alleleDiff  effect  \n",
       "0    3.164068  strong  \n",
       "1    0.795099  strong  \n",
       "2    5.757773  strong  \n",
       "3    1.100290  strong  \n",
       "4   -1.952722  strong  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enhancer_SNPs_motifbreakr = pd.read_csv(\"%senhancer_rare_enriched_SNPs_motifbreakR-scores.csv\" % OUTPUT, sep = \"\\t\")\n",
    "enhancer_SNPs_motifbreakr.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Extract SNPs which were predicted to have \"strong\" effect on TF binding. The strength is defined as absolute difference of proportional frequencies of REF and ALT alleles at position motifPos in the motif. If this difference is > 0.7 then the effect is classified as \"strong\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "#select records with \"strong\" effect\n",
    "promoter_SNPs_motifbreakr_strong = promoter_SNPs_motifbreakr[promoter_SNPs_motifbreakr[\"effect\"] == \"strong\"]\n",
    "enhancer_SNPs_motifbreakr_strong = enhancer_SNPs_motifbreakr[enhancer_SNPs_motifbreakr[\"effect\"] == \"strong\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select only one record per SNP and keep information about motif with the highest pct score and biggest difference between alleles. These will be stored in motif_best_match and motif_highest_diff columns, in a format motif_id:score. I decide to keep both motifs because it is difficult to decide if good match is more or less important than high difference between alleles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-43-c5359b181366>:27: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  enhancer_SNPs_motifbreakr_strong[\"motif_best_match\"] =  enhancer_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"best_match\"])\n",
      "<ipython-input-43-c5359b181366>:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  enhancer_SNPs_motifbreakr_strong[\"motif_highest_diff\"] =  enhancer_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"highest_diff\"])\n",
      "<ipython-input-43-c5359b181366>:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  promoter_SNPs_motifbreakr_strong[\"motif_best_match\"] =  promoter_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"best_match\"])\n",
      "<ipython-input-43-c5359b181366>:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  promoter_SNPs_motifbreakr_strong[\"motif_highest_diff\"] =  promoter_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"highest_diff\"])\n"
     ]
    }
   ],
   "source": [
    "#add columns with info about best matches: motif with highest pct score and alleleDiff\n",
    "\n",
    "def find_best_matching_motif(group):\n",
    "    #find motif with highest pct score (either for REF or ALT)\n",
    "    best_pctRef_score = max(group[\"pctRef\"])\n",
    "    best_pctAlt_score = max(group[\"pctAlt\"])\n",
    "    if best_pctRef_score > best_pctAlt_score:\n",
    "        best_pct_score_motif = group[group[\"pctRef\"] == best_pctRef_score][\"providerId\"].values[0]\n",
    "    else:\n",
    "        best_pct_score_motif = group[group[\"pctAlt\"] == best_pctAlt_score][\"providerId\"].values[0]\n",
    "    \n",
    "    #find motif with highest abs(diff) between alleles\n",
    "    best_alleleDiff = max(group[\"alleleDiff\"].abs())\n",
    "    best_alleleDiff_motif = group[group[\"alleleDiff\"].abs() == best_alleleDiff][\"providerId\"].values[0]\n",
    "    \n",
    "    return best_pct_score_motif + \":\" + \"%.2f\" % max(best_pctRef_score, best_pctAlt_score), best_alleleDiff_motif + \":\" + \"%.2f\" % best_alleleDiff\n",
    "\n",
    "#information about best motifs for each SNP will be stored in a dict in which SNP_ids will be keys\n",
    "best_motifs_dict = {}\n",
    "\n",
    "for df in [enhancer_SNPs_motifbreakr_strong, promoter_SNPs_motifbreakr_strong]:\n",
    "    for snp_id, snp_records in df.groupby(\"SNP_id\"):\n",
    "        best_match, highest_diff = find_best_matching_motif(snp_records)\n",
    "        best_motifs_dict[snp_id] = {\"best_match\" : best_match, \"highest_diff\" : highest_diff}\n",
    "\n",
    "#extract information from the dict to fill appropriate columns in enhancer and promoter SNPs dataframes\n",
    "enhancer_SNPs_motifbreakr_strong[\"motif_best_match\"] =  enhancer_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"best_match\"]) \n",
    "enhancer_SNPs_motifbreakr_strong[\"motif_highest_diff\"] =  enhancer_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"highest_diff\"]) \n",
    "\n",
    "promoter_SNPs_motifbreakr_strong[\"motif_best_match\"] =  promoter_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"best_match\"]) \n",
    "promoter_SNPs_motifbreakr_strong[\"motif_highest_diff\"] =  promoter_SNPs_motifbreakr_strong.SNP_id.apply(lambda x: best_motifs_dict[x][\"highest_diff\"]) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally keep one record per SNP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#extract information about SNP location and best motifs, drop duplicates\n",
    "promoter_SNPs_motifbreakr_strong_snps_only = promoter_SNPs_motifbreakr_strong[[\"seqnames\", \"start\", \"REF\", \"ALT\", \"motif_best_match\", \"motif_highest_diff\"]].drop_duplicates()\n",
    "enhancer_SNPs_motifbreakr_strong_snps_only = enhancer_SNPs_motifbreakr_strong[[\"seqnames\", \"start\", \"REF\", \"ALT\", \"motif_best_match\", \"motif_highest_diff\"]].drop_duplicates()\n",
    "\n",
    "#change column names to keep the convention used in the whole notebook\n",
    "promoter_SNPs_motifbreakr_strong_snps_only = promoter_SNPs_motifbreakr_strong_snps_only.rename(columns = {\"seqnames\":\"CHROM\", \"start\":\"POS\"})\n",
    "enhancer_SNPs_motifbreakr_strong_snps_only = enhancer_SNPs_motifbreakr_strong_snps_only.rename(columns = {\"seqnames\":\"CHROM\", \"start\":\"POS\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "925 and 449 SNPs in promoters and enhancers have predicted strong effect of motif binding (out of 1313 and 784, respectively).\n"
     ]
    }
   ],
   "source": [
    "print(len(promoter_SNPs_motifbreakr_strong_snps_only), \"and\", len(enhancer_SNPs_motifbreakr_strong_snps_only),\n",
    "      \"SNPs in promoters and enhancers have predicted strong effect of motif binding (out of %s and %s, respectively).\" % (str(len(rare_enriched_promoter_snps)), str(len(rare_enriched_enhancer_snps))))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge information about allele counts and frequencies with selected SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "rare_enriched_promoter_snps_motif = pd.merge(rare_enriched_promoter_snps, promoter_SNPs_motifbreakr_strong_snps_only, how = \"right\", on = [\"CHROM\", \"POS\", \"REF\", \"ALT\"])\n",
    "rare_enriched_enhancer_snps_motif = pd.merge(rare_enriched_enhancer_snps, enhancer_SNPs_motifbreakr_strong_snps_only, how = \"right\", on = [\"CHROM\", \"POS\", \"REF\", \"ALT\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check how SNP data look like now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>45783029</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>KLF15_HUMAN.H11MO.0.A:0.93</td>\n",
       "      <td>ZN770_HUMAN.H11MO.0.C:3.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>222320241</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:0.98</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:3.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>6556629</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>True</td>\n",
       "      <td>0.004048</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:7.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>26692643</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:0.99</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:4.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>106247724</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.063</td>\n",
       "      <td>32</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000320</td>\n",
       "      <td>OSR2_HUMAN.H11MO.0.C:0.95</td>\n",
       "      <td>WT1_HUMAN.H11MO.0.C:3.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr19   45783029   T   G   2  0.043  46           0.000076   \n",
       "1   chr2  222320241   T   C   4  0.087  46           0.000200   \n",
       "2  chr17    6556629   G   C   2  0.091  22           0.001300   \n",
       "3   chr2   26692643   C   G   2  0.048  42           0.000046   \n",
       "4  chr12  106247724   G   T   2  0.063  32           0.000200   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000    0.000000           True              0.000000   \n",
       "1             0.0000    0.000000           True              0.000000   \n",
       "2             0.0017    0.000653           True              0.004048   \n",
       "3             0.0000    0.000000           True              0.000000   \n",
       "4             0.0003    0.000044           True              0.000320   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  \n",
       "0  KLF15_HUMAN.H11MO.0.A:0.93  ZN770_HUMAN.H11MO.0.C:3.80  \n",
       "1   E2F6_HUMAN.H11MO.0.A:0.98   E2F6_HUMAN.H11MO.0.A:3.84  \n",
       "2    MAZ_HUMAN.H11MO.1.A:0.96    MAZ_HUMAN.H11MO.1.A:7.56  \n",
       "3  ZN740_HUMAN.H11MO.0.D:0.99  ZN740_HUMAN.H11MO.0.D:4.41  \n",
       "4   OSR2_HUMAN.H11MO.0.C:0.95    WT1_HUMAN.H11MO.0.C:3.25  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_promoter_snps_motif.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>True</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "1             0.0005  6.410374e-27           True          8.314926e-26   \n",
       "2             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "3             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "4             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  \n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32  \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09  \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10  \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assign target genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Promoter SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "rare_enriched_promoter_snps_motif[\"genomic element\"] = \"promoter\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will intersect the promoter SNPs data with promoters data to be able to assign genes to SNPs. This will be done with pybedtools package - a python interface to bedtools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>45783029</td>\n",
       "      <td>ENSG00000104936/DMPK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>222320241</td>\n",
       "      <td>ENSG00000237732/AC010980.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>6556629</td>\n",
       "      <td>ENSG00000091622/PITPNM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>26692643</td>\n",
       "      <td>ENSG00000171303/KCNK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>106247724</td>\n",
       "      <td>ENSG00000136026/CKAP4,ENSG00000258355/RP11-651...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS                                               Gene\n",
       "0  chr19   45783029                               ENSG00000104936/DMPK\n",
       "1   chr2  222320241                         ENSG00000237732/AC010980.2\n",
       "2  chr17    6556629                            ENSG00000091622/PITPNM3\n",
       "3   chr2   26692643                              ENSG00000171303/KCNK3\n",
       "4  chr12  106247724  ENSG00000136026/CKAP4,ENSG00000258355/RP11-651..."
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create BedTool object from promoter regions bed\n",
    "promoters_info = pbt.BedTool(PROMOTER_REGIONS)\n",
    "\n",
    "#create BedTool object from dataframe with selected promoter SNPs\n",
    "rare_enriched_promoter_snps_motif[\"POS-1\"] = rare_enriched_promoter_snps_motif[\"POS\"] - 1\n",
    "rare_enriched_promoter_snps_motif_bedtool = pbt.BedTool.from_dataframe(rare_enriched_promoter_snps_motif[[\"CHROM\", \"POS-1\", \"POS\"]])\n",
    "rare_enriched_promoter_snps_motif = rare_enriched_promoter_snps_motif.drop(labels = [\"POS-1\"], axis=1)\n",
    "\n",
    "#intersect promoters and SNPs\n",
    "rare_enriched_promoter_snps_motif_intersection = rare_enriched_promoter_snps_motif_bedtool.intersect(promoters_info, wa=True, wb=True)\n",
    "\n",
    "#create a dataframe from the intersection results, keep only columns with SNP location and gene(s) name(s)\n",
    "rare_enriched_promoter_snps_motif_intersection_df = rare_enriched_promoter_snps_motif_intersection.to_dataframe(names = [\"CHROM\", \"POS\", \"Gene\"], usecols = [0, 2, 6]).drop_duplicates()\n",
    "rare_enriched_promoter_snps_motif_intersection_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>45783029</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>KLF15_HUMAN.H11MO.0.A:0.93</td>\n",
       "      <td>ZN770_HUMAN.H11MO.0.C:3.80</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000104936/DMPK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>222320241</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:0.98</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:3.84</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000237732/AC010980.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>6556629</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>True</td>\n",
       "      <td>0.004048</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:7.56</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000091622/PITPNM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>26692643</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:0.99</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:4.41</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000171303/KCNK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>106247724</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.063</td>\n",
       "      <td>32</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000320</td>\n",
       "      <td>OSR2_HUMAN.H11MO.0.C:0.95</td>\n",
       "      <td>WT1_HUMAN.H11MO.0.C:3.25</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000136026/CKAP4,ENSG00000258355/RP11-651...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr19   45783029   T   G   2  0.043  46           0.000076   \n",
       "1   chr2  222320241   T   C   4  0.087  46           0.000200   \n",
       "2  chr17    6556629   G   C   2  0.091  22           0.001300   \n",
       "3   chr2   26692643   C   G   2  0.048  42           0.000046   \n",
       "4  chr12  106247724   G   T   2  0.063  32           0.000200   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000    0.000000           True              0.000000   \n",
       "1             0.0000    0.000000           True              0.000000   \n",
       "2             0.0017    0.000653           True              0.004048   \n",
       "3             0.0000    0.000000           True              0.000000   \n",
       "4             0.0003    0.000044           True              0.000320   \n",
       "\n",
       "             motif_best_match          motif_highest_diff genomic element  \\\n",
       "0  KLF15_HUMAN.H11MO.0.A:0.93  ZN770_HUMAN.H11MO.0.C:3.80        promoter   \n",
       "1   E2F6_HUMAN.H11MO.0.A:0.98   E2F6_HUMAN.H11MO.0.A:3.84        promoter   \n",
       "2    MAZ_HUMAN.H11MO.1.A:0.96    MAZ_HUMAN.H11MO.1.A:7.56        promoter   \n",
       "3  ZN740_HUMAN.H11MO.0.D:0.99  ZN740_HUMAN.H11MO.0.D:4.41        promoter   \n",
       "4   OSR2_HUMAN.H11MO.0.C:0.95    WT1_HUMAN.H11MO.0.C:3.25        promoter   \n",
       "\n",
       "                                                Gene  \n",
       "0                               ENSG00000104936/DMPK  \n",
       "1                         ENSG00000237732/AC010980.2  \n",
       "2                            ENSG00000091622/PITPNM3  \n",
       "3                              ENSG00000171303/KCNK3  \n",
       "4  ENSG00000136026/CKAP4,ENSG00000258355/RP11-651...  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge the intersection dataframe with df containing frequency information\n",
    "rare_enriched_promoter_snps_motif_gene = pd.merge(rare_enriched_promoter_snps_motif, rare_enriched_promoter_snps_motif_intersection_df, how = \"left\", on = [\"CHROM\", \"POS\"])\n",
    "rare_enriched_promoter_snps_motif_gene.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enhancer SNPs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intronic enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For enhancers located inside genes the gene in which they reside is treated as candidate. Provided bed file with active enhancers contains information about intersecting gene in the last column. If enhancer is intergenic then the last column contains \".\" ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>ENSG00000227232/WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>ENSG00000279457/FO538757.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1134548</td>\n",
       "      <td>1135587</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1156466</td>\n",
       "      <td>1159847</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1504484</td>\n",
       "      <td>1506451</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1550199</td>\n",
       "      <td>1551447</td>\n",
       "      <td>ENSG00000160075/SSU72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1567417</td>\n",
       "      <td>1568787</td>\n",
       "      <td>ENSG00000160075/SSU72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1806062</td>\n",
       "      <td>1808150</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1846213</td>\n",
       "      <td>1848350</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1857785</td>\n",
       "      <td>1859590</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end                        Gene\n",
       "0  chr1    19482    22400      ENSG00000227232/WASH7P\n",
       "1  chr1   189803   193125  ENSG00000279457/FO538757.2\n",
       "2  chr1  1134548  1135587                           .\n",
       "3  chr1  1156466  1159847                           .\n",
       "4  chr1  1504484  1506451                           .\n",
       "5  chr1  1550199  1551447       ENSG00000160075/SSU72\n",
       "6  chr1  1567417  1568787       ENSG00000160075/SSU72\n",
       "7  chr1  1806062  1808150        ENSG00000078369/GNB1\n",
       "8  chr1  1846213  1848350        ENSG00000078369/GNB1\n",
       "9  chr1  1857785  1859590        ENSG00000078369/GNB1"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_genes = pd.read_csv(ENHANCER_REGIONS, sep='\\t', names = ['chr', 'start', 'end', 'Gene'])\n",
    "enh_genes[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>ENSG00000227232/WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>ENSG00000279457/FO538757.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1134548</td>\n",
       "      <td>1135587</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1156466</td>\n",
       "      <td>1159847</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1504484</td>\n",
       "      <td>1506451</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1550199</td>\n",
       "      <td>1551447</td>\n",
       "      <td>ENSG00000160075/SSU72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1567417</td>\n",
       "      <td>1568787</td>\n",
       "      <td>ENSG00000160075/SSU72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1806062</td>\n",
       "      <td>1808150</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1846213</td>\n",
       "      <td>1848350</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1857785</td>\n",
       "      <td>1859590</td>\n",
       "      <td>ENSG00000078369/GNB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr    start      end                        Gene\n",
       "0  chr1    19482    22400      ENSG00000227232/WASH7P\n",
       "1  chr1   189803   193125  ENSG00000279457/FO538757.2\n",
       "2  chr1  1134548  1135587                           .\n",
       "3  chr1  1156466  1159847                           .\n",
       "4  chr1  1504484  1506451                           .\n",
       "5  chr1  1550199  1551447       ENSG00000160075/SSU72\n",
       "6  chr1  1567417  1568787       ENSG00000160075/SSU72\n",
       "7  chr1  1806062  1808150        ENSG00000078369/GNB1\n",
       "8  chr1  1846213  1848350        ENSG00000078369/GNB1\n",
       "9  chr1  1857785  1859590        ENSG00000078369/GNB1"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reformat to have one gene ID in cell\n",
    "enh_gene = pd.DataFrame() \n",
    "for i, row in enh_genes.iterrows(): \n",
    "    genes = row['Gene'].split(',') \n",
    "    if len(genes) == 1: \n",
    "        enh_gene = enh_gene.append(row, ignore_index=True) \n",
    "    else: \n",
    "        for gene in genes: \n",
    "            new_row = row \n",
    "            new_row['Gene'] = gene \n",
    "            enh_gene = enh_gene.append(new_row, ignore_index=True) \n",
    "            enh_gene = enh_gene.reindex(enh_genes.columns, axis=1) \n",
    "            enh_gene[\"start\"] = enh_gene[\"start\"].astype(int)\n",
    "            enh_gene[\"end\"] = enh_gene[\"end\"].astype(int)\n",
    "enh_gene[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Intersect information about enhancers with SNPs to assign gene names to SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>True</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "      <td>116358414</td>\n",
       "      <td>116360334</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "      <td>26475763</td>\n",
       "      <td>26476965</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "      <td>4855091</td>\n",
       "      <td>4857297</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "1             0.0005  6.410374e-27           True          8.314926e-26   \n",
       "2             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "3             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "4             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32   90397237   \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09   90397237   \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  116358414   \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10   26475763   \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26    4855091   \n",
       "\n",
       "     enh_end Gene  \n",
       "0   90398831    .  \n",
       "1   90398831    .  \n",
       "2  116360334    .  \n",
       "3   26476965    .  \n",
       "4    4857297    .  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#prepare bedtool objects\n",
    "rare_enriched_enhancer_snps_motif[\"POS-1\"] = rare_enriched_enhancer_snps_motif[\"POS\"] - 1\n",
    "rare_enriched_enhancer_snps_motif_bedtool = pbt.BedTool.from_dataframe(rare_enriched_enhancer_snps_motif[[\"CHROM\", \"POS-1\", \"POS\", \"REF\", \"ALT\", \"AC\", \"AF\", \"AN\", \n",
    "                                                                                                          \"gnomAD_genome_ALL\", \"gnomAD_genome_NFE\", \"binom_pval\",\n",
    "                                                                                                          \"B-H_reject_H0\", \"corrected_binom_pval\", \"motif_best_match\", \n",
    "                                                                                                          \"motif_highest_diff\"]])\n",
    "enh_genes_bedtool = pbt.BedTool.from_dataframe(enh_gene)\n",
    "\n",
    "#intersect\n",
    "rare_enriched_enhancer_snps_motif_intersection = rare_enriched_enhancer_snps_motif_bedtool.intersect(enh_genes_bedtool, wa=True, wb=True, loj=True)\n",
    "\n",
    "#reformat intersection to dataframe, keep columns with enhancer coordinates - they will be usefull in the next step\n",
    "rare_enriched_enhancer_snps_motif_gene = rare_enriched_enhancer_snps_motif_intersection.to_dataframe(usecols = [0,2,3,4,5,6,7,8,9,10,11,12,13,14,16,17,18], names = [\"CHROM\", \"POS\", \"REF\", \"ALT\", \"AC\", \"AF\", \"AN\", \n",
    "                                                                                                          \"gnomAD_genome_ALL\", \"gnomAD_genome_NFE\", \"binom_pval\",\n",
    "                                                                                                          \"B-H_reject_H0\", \"corrected_binom_pval\", \"motif_best_match\", \n",
    "                                                                                                          \"motif_highest_diff\", \"enh_start\", \"enh_end\",\"Gene\"])\n",
    "rare_enriched_enhancer_snps_motif_gene.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Add column which will inform if enhancer is intronic or intergenic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>True</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "      <td>116358414</td>\n",
       "      <td>116360334</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "      <td>26475763</td>\n",
       "      <td>26476965</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "      <td>4855091</td>\n",
       "      <td>4857297</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "1             0.0005  6.410374e-27           True          8.314926e-26   \n",
       "2             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "3             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "4             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32   90397237   \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09   90397237   \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  116358414   \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10   26475763   \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26    4855091   \n",
       "\n",
       "     enh_end Gene      genomic element  \n",
       "0   90398831    .  enhancer intergenic  \n",
       "1   90398831    .  enhancer intergenic  \n",
       "2  116360334    .  enhancer intergenic  \n",
       "3   26476965    .  enhancer intergenic  \n",
       "4    4857297    .  enhancer intergenic  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_gene[\"genomic element\"] = rare_enriched_enhancer_snps_motif_gene.Gene.apply(lambda x: \"enhancer intergenic\" if x == \".\" else \"enhancer intronic\")\n",
    "rare_enriched_enhancer_snps_motif_gene.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intergenic enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For enhancers located in intergenic regions targets can be assigned based on distance - by selecting the closest genes or based on chromatin contacts information inferred from Hi-C data.  \n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Closest gene"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To find closest genes we will first obtain TSS locations from hg38 genome annotation from Ensembl (GRCh38.p5). The gtf file used here contains only \"gene\" records."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972/DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232/WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267/MIR6859-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485/RP11-34P13.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890/MIR1302-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000237613\"; gene_version \"2\"; g...</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>ENSG00000237613/FAM138A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>53312</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000268020\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "      <td>ENSG00000268020/OR4G4P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>62948</td>\n",
       "      <td>63887</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000240361\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000240361</td>\n",
       "      <td>ENSG00000240361/OR4G11P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>69091</td>\n",
       "      <td>70008</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000186092\"; gene_version \"4\"; g...</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENSG00000186092/OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>89295</td>\n",
       "      <td>133723</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000238009\"; gene_version \"6\"; g...</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009/RP11-34P13.7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  type  start     end strand  \\\n",
       "0  chr1  gene  11869   14409      +   \n",
       "1  chr1  gene  14404   29570      -   \n",
       "2  chr1  gene  17369   17436      -   \n",
       "3  chr1  gene  29554   31109      +   \n",
       "4  chr1  gene  30366   30503      +   \n",
       "5  chr1  gene  34554   36081      -   \n",
       "6  chr1  gene  52473   53312      +   \n",
       "7  chr1  gene  62948   63887      +   \n",
       "8  chr1  gene  69091   70008      +   \n",
       "9  chr1  gene  89295  133723      -   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "5  gene_id \"ENSG00000237613\"; gene_version \"2\"; g...  ENSG00000237613   \n",
       "6  gene_id \"ENSG00000268020\"; gene_version \"3\"; g...  ENSG00000268020   \n",
       "7  gene_id \"ENSG00000240361\"; gene_version \"1\"; g...  ENSG00000240361   \n",
       "8  gene_id \"ENSG00000186092\"; gene_version \"4\"; g...  ENSG00000186092   \n",
       "9  gene_id \"ENSG00000238009\"; gene_version \"6\"; g...  ENSG00000238009   \n",
       "\n",
       "                           Gene  \n",
       "0       ENSG00000223972/DDX11L1  \n",
       "1        ENSG00000227232/WASH7P  \n",
       "2     ENSG00000278267/MIR6859-1  \n",
       "3  ENSG00000243485/RP11-34P13.3  \n",
       "4     ENSG00000274890/MIR1302-2  \n",
       "5       ENSG00000237613/FAM138A  \n",
       "6        ENSG00000268020/OR4G4P  \n",
       "7       ENSG00000240361/OR4G11P  \n",
       "8         ENSG00000186092/OR4F5  \n",
       "9  ENSG00000238009/RP11-34P13.7  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_annot = pd.read_csv('data/hg38_full.genes.gtf', sep='\\t', usecols = [0, 2, 3, 4, 6, 8], skiprows = 5,\n",
    "                            names = [\"chr\", \"type\", \"start\", \"end\", \"strand\", 'info'])\n",
    "genes_info = full_annot[full_annot[\"type\"] == \"gene\"]\n",
    "genes_info['ID'] = genes_info['info'].str.split('\"').str[1]\n",
    "genes_info['Gene'] = genes_info['ID'] + \"/\" + genes_info['info'].str.split('\"').str[5]\n",
    "\n",
    "genes_info[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_tss(row):\n",
    "    if row['strand'] == '+':\n",
    "        return row['start']\n",
    "    else:\n",
    "        return row['end']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000223972\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>ENSG00000223972/DDX11L1</td>\n",
       "      <td>11869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000227232\"; gene_version \"5\"; g...</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>ENSG00000227232/WASH7P</td>\n",
       "      <td>29570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000278267\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>ENSG00000278267/MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000243485\"; gene_version \"3\"; g...</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>ENSG00000243485/RP11-34P13.3</td>\n",
       "      <td>29554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>gene</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000274890\"; gene_version \"1\"; g...</td>\n",
       "      <td>ENSG00000274890</td>\n",
       "      <td>ENSG00000274890/MIR1302-2</td>\n",
       "      <td>30366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  type  start    end strand  \\\n",
       "0  chr1  gene  11869  14409      +   \n",
       "1  chr1  gene  14404  29570      -   \n",
       "2  chr1  gene  17369  17436      -   \n",
       "3  chr1  gene  29554  31109      +   \n",
       "4  chr1  gene  30366  30503      +   \n",
       "\n",
       "                                                info               ID  \\\n",
       "0  gene_id \"ENSG00000223972\"; gene_version \"5\"; g...  ENSG00000223972   \n",
       "1  gene_id \"ENSG00000227232\"; gene_version \"5\"; g...  ENSG00000227232   \n",
       "2  gene_id \"ENSG00000278267\"; gene_version \"1\"; g...  ENSG00000278267   \n",
       "3  gene_id \"ENSG00000243485\"; gene_version \"3\"; g...  ENSG00000243485   \n",
       "4  gene_id \"ENSG00000274890\"; gene_version \"1\"; g...  ENSG00000274890   \n",
       "\n",
       "                           Gene    tss  \n",
       "0       ENSG00000223972/DDX11L1  11869  \n",
       "1        ENSG00000227232/WASH7P  29570  \n",
       "2     ENSG00000278267/MIR6859-1  17436  \n",
       "3  ENSG00000243485/RP11-34P13.3  29554  \n",
       "4     ENSG00000274890/MIR1302-2  30366  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info[\"tss\"] = genes_info.apply(find_tss, axis=1)\n",
    "genes_info.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will use bedtools closest tool to identify TSS closest to each enhancer containing analyzed SNPs. In cases when more than one TSS can be found at the shortest distance all results will be reported."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n",
      "/usr/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n",
      "/usr/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>closest gene</th>\n",
       "      <th>distance to closest gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868223</td>\n",
       "      <td>6870996</td>\n",
       "      <td>ENSG00000227950/RP11-312B8.2</td>\n",
       "      <td>33605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8888663</td>\n",
       "      <td>8891414</td>\n",
       "      <td>ENSG00000238249/HMGN2P17</td>\n",
       "      <td>1995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000201746/RNU6-828P</td>\n",
       "      <td>18494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21253526</td>\n",
       "      <td>21255567</td>\n",
       "      <td>ENSG00000236936/RP3-329E20.2</td>\n",
       "      <td>10515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24288566</td>\n",
       "      <td>24291017</td>\n",
       "      <td>ENSG00000266511/AL590683.2</td>\n",
       "      <td>8711</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM  enh_start   enh_end                  closest gene  \\\n",
       "0  chr1    6868223   6870996  ENSG00000227950/RP11-312B8.2   \n",
       "1  chr1    8888663   8891414      ENSG00000238249/HMGN2P17   \n",
       "2  chr1   10143598  10144774     ENSG00000201746/RNU6-828P   \n",
       "3  chr1   21253526  21255567  ENSG00000236936/RP3-329E20.2   \n",
       "4  chr1   24288566  24291017    ENSG00000266511/AL590683.2   \n",
       "\n",
       "   distance to closest gene  \n",
       "0                     33605  \n",
       "1                      1995  \n",
       "2                     18494  \n",
       "3                     10515  \n",
       "4                      8711  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_info_tss_bed = pbt.BedTool.from_dataframe(genes_info[['chr', 'tss', 'tss', 'Gene']])\n",
    "genes_info_tss_bed_sorted = genes_info_tss_bed.sort()\n",
    "\n",
    "enhancers_bed = pbt.BedTool.from_dataframe(rare_enriched_enhancer_snps_motif_gene[['CHROM', 'enh_start', 'enh_end']].drop_duplicates()).sort()\n",
    "\n",
    "tss_closest_to_enh = enhancers_bed.closest(genes_info_tss_bed_sorted, t='all', d=True)\n",
    "\n",
    "tss_closest_to_enh_df = tss_closest_to_enh.to_dataframe(names = ['CHROM', 'enh_start', 'enh_end', \"closest gene\", \"distance to closest gene\"],\n",
    "                                                       usecols = [0,1,2,6,7])\n",
    "tss_closest_to_enh_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge information about closest TSS with previously collected enhancer SNPs annotations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>closest gene</th>\n",
       "      <th>distance to closest gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>True</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "      <td>116358414</td>\n",
       "      <td>116360334</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000258346/RP11-148B3.2</td>\n",
       "      <td>8430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "      <td>26475763</td>\n",
       "      <td>26476965</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000132970/WASF3</td>\n",
       "      <td>80738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "      <td>4855091</td>\n",
       "      <td>4857297</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000163132/MSX1</td>\n",
       "      <td>2369</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "1             0.0005  6.410374e-27           True          8.314926e-26   \n",
       "2             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "3             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "4             0.0000  0.000000e+00           True          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32   90397237   \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09   90397237   \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  116358414   \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10   26475763   \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26    4855091   \n",
       "\n",
       "     enh_end Gene      genomic element                   closest gene  \\\n",
       "0   90398831    .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "1   90398831    .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "2  116360334    .  enhancer intergenic   ENSG00000258346/RP11-148B3.2   \n",
       "3   26476965    .  enhancer intergenic          ENSG00000132970/WASF3   \n",
       "4    4857297    .  enhancer intergenic           ENSG00000163132/MSX1   \n",
       "\n",
       "   distance to closest gene  \n",
       "0                     31500  \n",
       "1                     31500  \n",
       "2                      8430  \n",
       "3                     80738  \n",
       "4                      2369  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_gene_closest = pd.merge(rare_enriched_enhancer_snps_motif_gene, tss_closest_to_enh_df, how=\"left\", on = [\"CHROM\", \"enh_start\", \"enh_end\"])\n",
    "rare_enriched_enhancer_snps_motif_gene_closest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "477 477\n"
     ]
    }
   ],
   "source": [
    "print(len(rare_enriched_enhancer_snps_motif_gene_closest), len(rare_enriched_enhancer_snps_motif_gene))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Chromatin contacts predicted from Hi-C data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To predict target genes we will also use chromatin contacts predicted based on Hi-C data from developing human brain ([Won et al., 2016](https://pubmed.ncbi.nlm.nih.gov/27760116/)) using [HiCEnterprise software]( https://github.com/regulomics/HiCEnterprise)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>ENST</th>\n",
       "      <th>contacting gene</th>\n",
       "      <th>-log10(qval)</th>\n",
       "      <th>confirmed_both_ways</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000606034</td>\n",
       "      <td>ENSG00000272512</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>ENSG00000188290</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>ENSG00000187608</td>\n",
       "      <td>8.697366</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000330388</td>\n",
       "      <td>ENSG00000184163</td>\n",
       "      <td>2.648810</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>ENST00000478065</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>3.934685</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM  enh_start  enh_end             ENST  contacting gene  -log10(qval)  \\\n",
       "1   chr1    1024210  1025994  ENST00000606034  ENSG00000272512      8.697366   \n",
       "2   chr1    1024210  1025994  ENST00000484667  ENSG00000188290      8.697366   \n",
       "6   chr1    1024210  1025994  ENST00000624697  ENSG00000187608      8.697366   \n",
       "9   chr1    1024210  1025994  ENST00000330388  ENSG00000184163      2.648810   \n",
       "12  chr1    1024210  1025994  ENST00000478065  ENSG00000131584      3.934685   \n",
       "\n",
       "    confirmed_both_ways  \n",
       "1                     1  \n",
       "2                     1  \n",
       "6                     1  \n",
       "9                     1  \n",
       "12                    1  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read bed file with predicted contacts, select one record for each engancer-gene pair\n",
    "contacts = pd.read_csv(CHROMATIN_CONTACTS, sep=' ')\n",
    "contacts_to_genes = contacts[contacts[\"ENSG\"] != '-']\n",
    "contacts_to_genes = contacts_to_genes.drop_duplicates(subset = [\"chr\", \"start\", \"end\", \"ENSG\"])\n",
    "contacts_to_genes = contacts_to_genes.rename(columns = {\"chr\":\"CHROM\", \n",
    "                                                        \"start\":\"enh_start\", \n",
    "                                                        \"end\":\"enh_end\",\n",
    "                                                        \"ENSG\":\"contacting gene\"})\n",
    "contacts_to_genes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge enhancer variants with contacts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>...</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>closest gene</th>\n",
       "      <th>distance to closest gene</th>\n",
       "      <th>contacting gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>...</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "      <td>116358414</td>\n",
       "      <td>116360334</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000258346/RP11-148B3.2</td>\n",
       "      <td>8430</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "      <td>26475763</td>\n",
       "      <td>26476965</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000132970/WASF3</td>\n",
       "      <td>80738</td>\n",
       "      <td>ENSG00000132964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "      <td>4855091</td>\n",
       "      <td>4857297</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000163132/MSX1</td>\n",
       "      <td>2369</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  ...  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "1             0.0005  6.410374e-27  ...          8.314926e-26   \n",
       "2             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "3             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "4             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32   90397237   \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09   90397237   \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  116358414   \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10   26475763   \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26    4855091   \n",
       "\n",
       "     enh_end  Gene      genomic element                   closest gene  \\\n",
       "0   90398831     .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "1   90398831     .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "2  116360334     .  enhancer intergenic   ENSG00000258346/RP11-148B3.2   \n",
       "3   26476965     .  enhancer intergenic          ENSG00000132970/WASF3   \n",
       "4    4857297     .  enhancer intergenic           ENSG00000163132/MSX1   \n",
       "\n",
       "  distance to closest gene  contacting gene  \n",
       "0                    31500                .  \n",
       "1                    31500                .  \n",
       "2                     8430                .  \n",
       "3                    80738  ENSG00000132964  \n",
       "4                     2369                .  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_gene_closest_contacts = pd.merge(rare_enriched_enhancer_snps_motif_gene_closest, \n",
    "                                                                   contacts_to_genes[[\"CHROM\", \"enh_start\", \"enh_end\", \"contacting gene\"]], \n",
    "                                                                   on = [\"CHROM\", \"enh_start\", \"enh_end\"], how = \"left\").fillna('.')\n",
    "rare_enriched_enhancer_snps_motif_gene_closest_contacts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Replace gene ID from the \"contacting gene\" column with ID/name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_gene_name(gene_id):\n",
    "    if len(genes_info[genes_info[\"ID\"]==gene_id]) != 0:\n",
    "        return genes_info[genes_info[\"ID\"]==gene_id][\"Gene\"].values[0]\n",
    "    else:\n",
    "        return '.'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>...</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>closest gene</th>\n",
       "      <th>distance to closest gene</th>\n",
       "      <th>contacting gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>0.109</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:0.99</td>\n",
       "      <td>SMCA5_HUMAN.H11MO.0.C:4.32</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>90397732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>0.239</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>6.410374e-27</td>\n",
       "      <td>...</td>\n",
       "      <td>8.314926e-26</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:1.00</td>\n",
       "      <td>ZN394_HUMAN.H11MO.1.D:4.09</td>\n",
       "      <td>90397237</td>\n",
       "      <td>90398831</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000281904/CH17-132F21.5</td>\n",
       "      <td>31500</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>116359966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>ETS2_HUMAN.H11MO.0.B:0.96</td>\n",
       "      <td>ETV2_HUMAN.H11MO.0.B:5.76</td>\n",
       "      <td>116358414</td>\n",
       "      <td>116360334</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000258346/RP11-148B3.2</td>\n",
       "      <td>8430</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13</td>\n",
       "      <td>26475911</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:0.87</td>\n",
       "      <td>SP4_HUMAN.H11MO.0.A:1.10</td>\n",
       "      <td>26475763</td>\n",
       "      <td>26476965</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000132970/WASF3</td>\n",
       "      <td>80738</td>\n",
       "      <td>ENSG00000132964/CDK8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>4857069</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>SALL4_HUMAN.H11MO.0.B:1.00</td>\n",
       "      <td>ZN320_HUMAN.H11MO.0.C:4.26</td>\n",
       "      <td>4855091</td>\n",
       "      <td>4857297</td>\n",
       "      <td>.</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>ENSG00000163132/MSX1</td>\n",
       "      <td>2369</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0   chr2   90397298   T   G   5  0.109  46           0.000088   \n",
       "1   chr2   90397732   G   A  11  0.239  46           0.000400   \n",
       "2  chr12  116359966   T   C   1  0.022  46           0.000000   \n",
       "3  chr13   26475911   A   C   1  0.024  42           0.000065   \n",
       "4   chr4    4857069   C   A   2  0.043  46           0.001100   \n",
       "\n",
       "   gnomAD_genome_NFE    binom_pval  ...  corrected_binom_pval  \\\n",
       "0             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "1             0.0005  6.410374e-27  ...          8.314926e-26   \n",
       "2             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "3             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "4             0.0000  0.000000e+00  ...          0.000000e+00   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0  ZN394_HUMAN.H11MO.1.D:0.99  SMCA5_HUMAN.H11MO.0.C:4.32   90397237   \n",
       "1  ZN394_HUMAN.H11MO.1.D:1.00  ZN394_HUMAN.H11MO.1.D:4.09   90397237   \n",
       "2   ETS2_HUMAN.H11MO.0.B:0.96   ETV2_HUMAN.H11MO.0.B:5.76  116358414   \n",
       "3    SP4_HUMAN.H11MO.0.A:0.87    SP4_HUMAN.H11MO.0.A:1.10   26475763   \n",
       "4  SALL4_HUMAN.H11MO.0.B:1.00  ZN320_HUMAN.H11MO.0.C:4.26    4855091   \n",
       "\n",
       "     enh_end  Gene      genomic element                   closest gene  \\\n",
       "0   90398831     .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "1   90398831     .  enhancer intergenic  ENSG00000281904/CH17-132F21.5   \n",
       "2  116360334     .  enhancer intergenic   ENSG00000258346/RP11-148B3.2   \n",
       "3   26476965     .  enhancer intergenic          ENSG00000132970/WASF3   \n",
       "4    4857297     .  enhancer intergenic           ENSG00000163132/MSX1   \n",
       "\n",
       "  distance to closest gene       contacting gene  \n",
       "0                    31500                     .  \n",
       "1                    31500                     .  \n",
       "2                     8430                     .  \n",
       "3                    80738  ENSG00000132964/CDK8  \n",
       "4                     2369                     .  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_gene_closest_contacts[\"contacting gene\"] = rare_enriched_enhancer_snps_motif_gene_closest_contacts[\"contacting gene\"].apply(add_gene_name)\n",
    "rare_enriched_enhancer_snps_motif_gene_closest_contacts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reformat to have all target genes in one cell"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we have separate columns for different sources of target gene predictions (containing, closest, contacts) and multiple rows for each variant can be present if more than one closest or contacting gene was found for a particular enhancer. We will collect all predicted targets in the Gene column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "rare_enriched_enhancer_snps_motif_genes_collected = pd.DataFrame()\n",
    "\n",
    "for name, group in rare_enriched_enhancer_snps_motif_gene_closest_contacts.groupby([\"CHROM\", \"POS\", \"REF\", \"ALT\"]):\n",
    "    containing_genes = [gene + \"(containing)\" for gene in group[\"Gene\"].unique() if gene != \".\"]\n",
    "    closest_genes = [gene + \"(closest)\" for gene in group[\"closest gene\"].unique() if gene != \".\"]\n",
    "    contacting_genes = [gene + \"(contacting)\" for gene in group[\"contacting gene\"].unique() if gene != \".\"]\n",
    "\n",
    "    all_genes = []\n",
    "    all_genes.extend(containing_genes)\n",
    "    all_genes.extend(closest_genes)\n",
    "    all_genes.extend(contacting_genes)\n",
    "    \n",
    "    group[\"Gene\"] = \";\".join(all_genes)\n",
    "    \n",
    "    rare_enriched_enhancer_snps_motif_genes_collected = rare_enriched_enhancer_snps_motif_genes_collected.append(group[['CHROM', 'POS', 'REF', 'ALT', \n",
    "                                                                                                                        'AC', 'AF', 'AN', 'gnomAD_genome_ALL',\n",
    "                                                                                                                        'gnomAD_genome_NFE', 'binom_pval', 'B-H_reject_H0',\n",
    "                                                                                                                        'corrected_binom_pval', 'motif_best_match', 'motif_highest_diff',\n",
    "                                                                                                                        'enh_start', 'enh_end', 'Gene', 'genomic element']].drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868940</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>FLI1_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>ETS1_HUMAN.H11MO.0.A:1.60</td>\n",
       "      <td>6868223</td>\n",
       "      <td>6870996</td>\n",
       "      <td>ENSG00000171735/CAMTA1(containing);ENSG0000022...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8890584</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:0.96</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:3.86</td>\n",
       "      <td>8888663</td>\n",
       "      <td>8891414</td>\n",
       "      <td>ENSG00000238249/HMGN2P17(closest);ENSG00000116...</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:0.90</td>\n",
       "      <td>ZFP28_HUMAN.H11MO.0.C:1.89</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:1.00</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:3.01</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:0.89</td>\n",
       "      <td>IRF5_HUMAN.H11MO.0.D:2.20</td>\n",
       "      <td>21253526</td>\n",
       "      <td>21255567</td>\n",
       "      <td>ENSG00000117298/ECE1(containing);ENSG000002369...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "189  chr1   6868940   G   C   1  0.022  46           0.000200   \n",
       "82   chr1   8890584   T   C   1  0.022  46           0.000100   \n",
       "298  chr1  10144750   G   A   2  0.043  46           0.000100   \n",
       "326  chr1  10144754   G   A   2  0.043  46           0.000035   \n",
       "378  chr1  21254094   A   G   2  0.048  42           0.000037   \n",
       "\n",
       "     gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "189           0.000000    0.000000           True              0.000000   \n",
       "82            0.000000    0.000000           True              0.000000   \n",
       "298           0.000200    0.000041           True              0.000384   \n",
       "326           0.000071    0.000005           True              0.000055   \n",
       "378           0.000073    0.000005           True              0.000050   \n",
       "\n",
       "               motif_best_match          motif_highest_diff  enh_start  \\\n",
       "189   FLI1_HUMAN.H11MO.1.A:0.96   ETS1_HUMAN.H11MO.0.A:1.60    6868223   \n",
       "82   PRDM6_HUMAN.H11MO.0.C:0.96   IRF2_HUMAN.H11MO.0.A:3.86    8888663   \n",
       "298  CPEB1_HUMAN.H11MO.0.D:0.90  ZFP28_HUMAN.H11MO.0.C:1.89   10143598   \n",
       "326  PRDM6_HUMAN.H11MO.0.C:1.00  CPEB1_HUMAN.H11MO.0.D:3.01   10143598   \n",
       "378   IRF2_HUMAN.H11MO.0.A:0.89   IRF5_HUMAN.H11MO.0.D:2.20   21253526   \n",
       "\n",
       "      enh_end                                               Gene  \\\n",
       "189   6870996  ENSG00000171735/CAMTA1(containing);ENSG0000022...   \n",
       "82    8891414  ENSG00000238249/HMGN2P17(closest);ENSG00000116...   \n",
       "298  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "326  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "378  21255567  ENSG00000117298/ECE1(containing);ENSG000002369...   \n",
       "\n",
       "         genomic element  \n",
       "189    enhancer intronic  \n",
       "82   enhancer intergenic  \n",
       "298    enhancer intronic  \n",
       "326    enhancer intronic  \n",
       "378    enhancer intronic  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_genes_collected.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check correlation between H3K27ac in enhancer and gene expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We expect that enhancer activity should positively correlate with gene expression. Therefore we will use information about coverage from ChIP-seq on H3K27ac as enhancer activity measure and check if it correlates with expression of putative target genes. Putative targets with positive correlation and p-value < 0.15 will be reported. Spearman correlation will be calculated.\n",
    "\n",
    "The ChIP-seq and RNA-seq data used here are from Stepniak et al [ref]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>GB08</th>\n",
       "      <th>GB02</th>\n",
       "      <th>GB01</th>\n",
       "      <th>PA04</th>\n",
       "      <th>PA01</th>\n",
       "      <th>PA02</th>\n",
       "      <th>GB04</th>\n",
       "      <th>GB05</th>\n",
       "      <th>DA03</th>\n",
       "      <th>GB06</th>\n",
       "      <th>DA04</th>\n",
       "      <th>GB07</th>\n",
       "      <th>DA05</th>\n",
       "      <th>GB03</th>\n",
       "      <th>DA01</th>\n",
       "      <th>DA06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19482</td>\n",
       "      <td>22400</td>\n",
       "      <td>0.510363</td>\n",
       "      <td>0.834031</td>\n",
       "      <td>0.698638</td>\n",
       "      <td>0.988984</td>\n",
       "      <td>0.756236</td>\n",
       "      <td>1.132902</td>\n",
       "      <td>0.657006</td>\n",
       "      <td>0.566208</td>\n",
       "      <td>0.528208</td>\n",
       "      <td>0.451753</td>\n",
       "      <td>0.574673</td>\n",
       "      <td>0.526820</td>\n",
       "      <td>0.966401</td>\n",
       "      <td>0.426569</td>\n",
       "      <td>0.689541</td>\n",
       "      <td>0.638432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183942</td>\n",
       "      <td>184568</td>\n",
       "      <td>0.671342</td>\n",
       "      <td>0.697314</td>\n",
       "      <td>0.402064</td>\n",
       "      <td>0.463842</td>\n",
       "      <td>0.788018</td>\n",
       "      <td>1.099185</td>\n",
       "      <td>0.665276</td>\n",
       "      <td>0.743958</td>\n",
       "      <td>0.581258</td>\n",
       "      <td>0.533743</td>\n",
       "      <td>0.658219</td>\n",
       "      <td>0.651496</td>\n",
       "      <td>0.687957</td>\n",
       "      <td>0.541331</td>\n",
       "      <td>0.785030</td>\n",
       "      <td>0.634088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>189803</td>\n",
       "      <td>193125</td>\n",
       "      <td>0.877638</td>\n",
       "      <td>0.989761</td>\n",
       "      <td>0.888490</td>\n",
       "      <td>1.141639</td>\n",
       "      <td>1.091900</td>\n",
       "      <td>1.293182</td>\n",
       "      <td>0.825588</td>\n",
       "      <td>0.921340</td>\n",
       "      <td>0.935483</td>\n",
       "      <td>0.614575</td>\n",
       "      <td>0.967135</td>\n",
       "      <td>0.663150</td>\n",
       "      <td>1.189149</td>\n",
       "      <td>0.475793</td>\n",
       "      <td>1.106543</td>\n",
       "      <td>0.791204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1024210</td>\n",
       "      <td>1025994</td>\n",
       "      <td>0.793004</td>\n",
       "      <td>0.777391</td>\n",
       "      <td>0.688113</td>\n",
       "      <td>0.785313</td>\n",
       "      <td>0.587288</td>\n",
       "      <td>0.661786</td>\n",
       "      <td>0.584530</td>\n",
       "      <td>0.526820</td>\n",
       "      <td>0.855769</td>\n",
       "      <td>0.487969</td>\n",
       "      <td>0.661978</td>\n",
       "      <td>0.602755</td>\n",
       "      <td>0.899062</td>\n",
       "      <td>0.581647</td>\n",
       "      <td>0.603402</td>\n",
       "      <td>0.811515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1031391</td>\n",
       "      <td>1031884</td>\n",
       "      <td>0.905488</td>\n",
       "      <td>0.851251</td>\n",
       "      <td>0.894948</td>\n",
       "      <td>1.180019</td>\n",
       "      <td>0.951712</td>\n",
       "      <td>1.047810</td>\n",
       "      <td>0.822269</td>\n",
       "      <td>0.311449</td>\n",
       "      <td>0.755113</td>\n",
       "      <td>0.372133</td>\n",
       "      <td>0.701185</td>\n",
       "      <td>0.663920</td>\n",
       "      <td>0.967906</td>\n",
       "      <td>0.420525</td>\n",
       "      <td>0.440595</td>\n",
       "      <td>0.634359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM  enh_start  enh_end      GB08      GB02      GB01      PA04      PA01  \\\n",
       "0  chr1      19482    22400  0.510363  0.834031  0.698638  0.988984  0.756236   \n",
       "1  chr1     183942   184568  0.671342  0.697314  0.402064  0.463842  0.788018   \n",
       "2  chr1     189803   193125  0.877638  0.989761  0.888490  1.141639  1.091900   \n",
       "3  chr1    1024210  1025994  0.793004  0.777391  0.688113  0.785313  0.587288   \n",
       "4  chr1    1031391  1031884  0.905488  0.851251  0.894948  1.180019  0.951712   \n",
       "\n",
       "       PA02      GB04      GB05      DA03      GB06      DA04      GB07  \\\n",
       "0  1.132902  0.657006  0.566208  0.528208  0.451753  0.574673  0.526820   \n",
       "1  1.099185  0.665276  0.743958  0.581258  0.533743  0.658219  0.651496   \n",
       "2  1.293182  0.825588  0.921340  0.935483  0.614575  0.967135  0.663150   \n",
       "3  0.661786  0.584530  0.526820  0.855769  0.487969  0.661978  0.602755   \n",
       "4  1.047810  0.822269  0.311449  0.755113  0.372133  0.701185  0.663920   \n",
       "\n",
       "       DA05      GB03      DA01      DA06  \n",
       "0  0.966401  0.426569  0.689541  0.638432  \n",
       "1  0.687957  0.541331  0.785030  0.634088  \n",
       "2  1.189149  0.475793  1.106543  0.791204  \n",
       "3  0.899062  0.581647  0.603402  0.811515  \n",
       "4  0.967906  0.420525  0.440595  0.634359  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# H3K27ac coverage\n",
    "h3k27ac_cov = pd.read_csv(ENHANCER_ACTIVITY, sep = \"\\t\")\n",
    "h3k27ac_cov = h3k27ac_cov.rename(columns = {\"chr\":\"CHROM\", \n",
    "                                            \"start\":\"enh_start\", \n",
    "                                            \"end\":\"enh_end\"})\n",
    "h3k27ac_cov.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript</th>\n",
       "      <th>DA02</th>\n",
       "      <th>GB08</th>\n",
       "      <th>GB02</th>\n",
       "      <th>GB01</th>\n",
       "      <th>PA10</th>\n",
       "      <th>PA11</th>\n",
       "      <th>PA08</th>\n",
       "      <th>PA03</th>\n",
       "      <th>PA09</th>\n",
       "      <th>...</th>\n",
       "      <th>GB06</th>\n",
       "      <th>DA04</th>\n",
       "      <th>GB10</th>\n",
       "      <th>GB07</th>\n",
       "      <th>DA05</th>\n",
       "      <th>GB03</th>\n",
       "      <th>DA01</th>\n",
       "      <th>DA07</th>\n",
       "      <th>DA06</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000000233_ARF5</td>\n",
       "      <td>800.0</td>\n",
       "      <td>1820.0</td>\n",
       "      <td>551.0</td>\n",
       "      <td>1412.0</td>\n",
       "      <td>922.0</td>\n",
       "      <td>869.0</td>\n",
       "      <td>713.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>798.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1010.0</td>\n",
       "      <td>1195.0</td>\n",
       "      <td>802.0</td>\n",
       "      <td>901.0</td>\n",
       "      <td>664.0</td>\n",
       "      <td>1051.0</td>\n",
       "      <td>986.0</td>\n",
       "      <td>843.0</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>ARF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000000412_M6PR</td>\n",
       "      <td>1302.0</td>\n",
       "      <td>485.0</td>\n",
       "      <td>790.0</td>\n",
       "      <td>1131.0</td>\n",
       "      <td>952.0</td>\n",
       "      <td>840.0</td>\n",
       "      <td>831.0</td>\n",
       "      <td>1044.0</td>\n",
       "      <td>1316.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1209.0</td>\n",
       "      <td>772.0</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>1131.0</td>\n",
       "      <td>699.0</td>\n",
       "      <td>1250.0</td>\n",
       "      <td>707.0</td>\n",
       "      <td>849.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>M6PR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000000442_ESRRA</td>\n",
       "      <td>347.0</td>\n",
       "      <td>393.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>183.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>241.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>...</td>\n",
       "      <td>422.0</td>\n",
       "      <td>298.0</td>\n",
       "      <td>308.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>232.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>323.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>295.0</td>\n",
       "      <td>ESRRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000001008_FKBP4</td>\n",
       "      <td>556.0</td>\n",
       "      <td>682.0</td>\n",
       "      <td>704.0</td>\n",
       "      <td>2868.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>539.0</td>\n",
       "      <td>739.0</td>\n",
       "      <td>414.0</td>\n",
       "      <td>...</td>\n",
       "      <td>542.0</td>\n",
       "      <td>658.0</td>\n",
       "      <td>590.0</td>\n",
       "      <td>595.0</td>\n",
       "      <td>710.0</td>\n",
       "      <td>935.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>744.0</td>\n",
       "      <td>1076.0</td>\n",
       "      <td>FKBP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000001146_CYP26B1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>...</td>\n",
       "      <td>57.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Transcript    DA02    GB08   GB02    GB01   PA10   PA11  \\\n",
       "0     ENST00000000233_ARF5   800.0  1820.0  551.0  1412.0  922.0  869.0   \n",
       "1     ENST00000000412_M6PR  1302.0   485.0  790.0  1131.0  952.0  840.0   \n",
       "2    ENST00000000442_ESRRA   347.0   393.0  214.0   261.0  208.0  183.0   \n",
       "3    ENST00000001008_FKBP4   556.0   682.0  704.0  2868.0  300.0  462.0   \n",
       "4  ENST00000001146_CYP26B1    24.0    22.0   53.0    46.0  110.0  166.0   \n",
       "\n",
       "    PA08    PA03    PA09  ...    GB06    DA04    GB10    GB07   DA05    GB03  \\\n",
       "0  713.0   912.0   798.0  ...  1010.0  1195.0   802.0   901.0  664.0  1051.0   \n",
       "1  831.0  1044.0  1316.0  ...  1209.0   772.0  1287.0  1131.0  699.0  1250.0   \n",
       "2  159.0   241.0   220.0  ...   422.0   298.0   308.0   301.0  232.0   286.0   \n",
       "3  539.0   739.0   414.0  ...   542.0   658.0   590.0   595.0  710.0   935.0   \n",
       "4  126.0   188.0    74.0  ...    57.0    42.0    24.0    47.0   28.0    11.0   \n",
       "\n",
       "    DA01   DA07    DA06     Gene  \n",
       "0  986.0  843.0  1127.0     ARF5  \n",
       "1  707.0  849.0   912.0     M6PR  \n",
       "2  323.0  223.0   295.0    ESRRA  \n",
       "3  652.0  744.0  1076.0    FKBP4  \n",
       "4   13.0   31.0     9.0  CYP26B1  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genes/transcripts normalized counts\n",
    "counts = pd.read_csv(GENE_EXPRESSION, sep='\\t')\n",
    "counts['Gene'] = counts.Transcript.apply(lambda x: '_'.join(x.split('_')[1:]))\n",
    "counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>...</th>\n",
       "      <th>GB04</th>\n",
       "      <th>GB05</th>\n",
       "      <th>DA03</th>\n",
       "      <th>GB06</th>\n",
       "      <th>DA04</th>\n",
       "      <th>GB07</th>\n",
       "      <th>DA05</th>\n",
       "      <th>GB03</th>\n",
       "      <th>DA01</th>\n",
       "      <th>DA06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868940</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.766341</td>\n",
       "      <td>0.885556</td>\n",
       "      <td>0.899062</td>\n",
       "      <td>1.049015</td>\n",
       "      <td>0.979862</td>\n",
       "      <td>0.946164</td>\n",
       "      <td>0.878843</td>\n",
       "      <td>0.705809</td>\n",
       "      <td>0.757225</td>\n",
       "      <td>0.907536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8890584</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.718616</td>\n",
       "      <td>0.772834</td>\n",
       "      <td>0.924597</td>\n",
       "      <td>1.050524</td>\n",
       "      <td>0.808201</td>\n",
       "      <td>0.989958</td>\n",
       "      <td>0.997923</td>\n",
       "      <td>0.631530</td>\n",
       "      <td>1.012666</td>\n",
       "      <td>1.006235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>...</td>\n",
       "      <td>0.787857</td>\n",
       "      <td>0.409702</td>\n",
       "      <td>0.679355</td>\n",
       "      <td>0.392537</td>\n",
       "      <td>0.693585</td>\n",
       "      <td>0.384560</td>\n",
       "      <td>0.552377</td>\n",
       "      <td>0.351660</td>\n",
       "      <td>0.295737</td>\n",
       "      <td>0.423495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.787857</td>\n",
       "      <td>0.409702</td>\n",
       "      <td>0.679355</td>\n",
       "      <td>0.392537</td>\n",
       "      <td>0.693585</td>\n",
       "      <td>0.384560</td>\n",
       "      <td>0.552377</td>\n",
       "      <td>0.351660</td>\n",
       "      <td>0.295737</td>\n",
       "      <td>0.423495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.753438</td>\n",
       "      <td>0.799153</td>\n",
       "      <td>0.877427</td>\n",
       "      <td>0.955926</td>\n",
       "      <td>1.046168</td>\n",
       "      <td>1.056799</td>\n",
       "      <td>0.937268</td>\n",
       "      <td>0.765077</td>\n",
       "      <td>0.804443</td>\n",
       "      <td>0.834471</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr1   6868940   G   C   1  0.022  46           0.000200   \n",
       "1  chr1   8890584   T   C   1  0.022  46           0.000100   \n",
       "2  chr1  10144750   G   A   2  0.043  46           0.000100   \n",
       "3  chr1  10144754   G   A   2  0.043  46           0.000035   \n",
       "4  chr1  21254094   A   G   2  0.048  42           0.000037   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  ...      GB04      GB05      DA03      GB06  \\\n",
       "0           0.000000    0.000000  ...  0.766341  0.885556  0.899062  1.049015   \n",
       "1           0.000000    0.000000  ...  0.718616  0.772834  0.924597  1.050524   \n",
       "2           0.000200    0.000041  ...  0.787857  0.409702  0.679355  0.392537   \n",
       "3           0.000071    0.000005  ...  0.787857  0.409702  0.679355  0.392537   \n",
       "4           0.000073    0.000005  ...  0.753438  0.799153  0.877427  0.955926   \n",
       "\n",
       "       DA04      GB07      DA05      GB03      DA01      DA06  \n",
       "0  0.979862  0.946164  0.878843  0.705809  0.757225  0.907536  \n",
       "1  0.808201  0.989958  0.997923  0.631530  1.012666  1.006235  \n",
       "2  0.693585  0.384560  0.552377  0.351660  0.295737  0.423495  \n",
       "3  0.693585  0.384560  0.552377  0.351660  0.295737  0.423495  \n",
       "4  1.046168  1.056799  0.937268  0.765077  0.804443  0.834471  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge SNPs with H3K27ac coverage on enhancers\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_coverage = pd.merge(rare_enriched_enhancer_snps_motif_genes_collected,\n",
    "                                                                     h3k27ac_cov, on = [\"CHROM\", \"enh_start\", \"enh_end\"], how = \"left\")\n",
    "\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_coverage.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_correlation(row, sample_names):\n",
    "    correlations = \"\"\n",
    "    \n",
    "    enh_act_vector = row[sample_names].values\n",
    "    \n",
    "    genes = set([el.split(\"(\")[0] for el in row[\"Gene\"].split(';')])\n",
    "    \n",
    "    #iterate over all target genes assigned to this variant\n",
    "    for gene in genes:\n",
    "        gene_name = gene.split('/')[1]\n",
    "        gene_expr_rows = counts[counts[\"Gene\"] == gene_name]\n",
    "        if len(gene_expr_rows) != 0:\n",
    "            \n",
    "            #calculate correlations for each transcript of the analyzed gene\n",
    "            gene_correlations = {}\n",
    "            for j, expr_row in gene_expr_rows.iterrows():\n",
    "                expr_vector = expr_row[sample_names].values\n",
    "                rho, pval = spearmanr(enh_act_vector, expr_vector)\n",
    "                #collect pvalues for positive correlations\n",
    "                if str(rho) != 'nan' and rho > 0:\n",
    "                    gene_correlations[pval] = [rho, expr_row[\"Transcript\"]]\n",
    "                    \n",
    "            #find best correlating transcript\n",
    "            if len(gene_correlations.keys()) > 0:\n",
    "                min_pval = min(gene_correlations.keys())\n",
    "                if min_pval < 0.15:\n",
    "                    correlations += gene_name + \"/\" + gene_correlations[min_pval][1].split(\"_\")[0] + \"/\" + \"%.5f\" % min_pval + \";\"\n",
    "                    \n",
    "        else:\n",
    "            pass\n",
    "    if len(correlations) == 0:\n",
    "        return \".\"\n",
    "    else:\n",
    "        return correlations.rstrip(\";\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/researcher/.local/lib/python3.8/site-packages/scipy/stats/stats.py:4264: SpearmanRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(SpearmanRConstantInputWarning())\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>H3K27ac-expression correlation p-values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868940</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>FLI1_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>ETS1_HUMAN.H11MO.0.A:1.60</td>\n",
       "      <td>6868223</td>\n",
       "      <td>6870996</td>\n",
       "      <td>ENSG00000171735/CAMTA1(containing);ENSG0000022...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8890584</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:0.96</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:3.86</td>\n",
       "      <td>8888663</td>\n",
       "      <td>8891414</td>\n",
       "      <td>ENSG00000238249/HMGN2P17(closest);ENSG00000116...</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:0.90</td>\n",
       "      <td>ZFP28_HUMAN.H11MO.0.C:1.89</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:1.00</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:3.01</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:0.89</td>\n",
       "      <td>IRF5_HUMAN.H11MO.0.D:2.20</td>\n",
       "      <td>21253526</td>\n",
       "      <td>21255567</td>\n",
       "      <td>ENSG00000117298/ECE1(containing);ENSG000002369...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ECE1/ENST00000415912/0.06776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr1   6868940   G   C   1  0.022  46           0.000200   \n",
       "1  chr1   8890584   T   C   1  0.022  46           0.000100   \n",
       "2  chr1  10144750   G   A   2  0.043  46           0.000100   \n",
       "3  chr1  10144754   G   A   2  0.043  46           0.000035   \n",
       "4  chr1  21254094   A   G   2  0.048  42           0.000037   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0           0.000000    0.000000           True              0.000000   \n",
       "1           0.000000    0.000000           True              0.000000   \n",
       "2           0.000200    0.000041           True              0.000384   \n",
       "3           0.000071    0.000005           True              0.000055   \n",
       "4           0.000073    0.000005           True              0.000050   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0   FLI1_HUMAN.H11MO.1.A:0.96   ETS1_HUMAN.H11MO.0.A:1.60    6868223   \n",
       "1  PRDM6_HUMAN.H11MO.0.C:0.96   IRF2_HUMAN.H11MO.0.A:3.86    8888663   \n",
       "2  CPEB1_HUMAN.H11MO.0.D:0.90  ZFP28_HUMAN.H11MO.0.C:1.89   10143598   \n",
       "3  PRDM6_HUMAN.H11MO.0.C:1.00  CPEB1_HUMAN.H11MO.0.D:3.01   10143598   \n",
       "4   IRF2_HUMAN.H11MO.0.A:0.89   IRF5_HUMAN.H11MO.0.D:2.20   21253526   \n",
       "\n",
       "    enh_end                                               Gene  \\\n",
       "0   6870996  ENSG00000171735/CAMTA1(containing);ENSG0000022...   \n",
       "1   8891414  ENSG00000238249/HMGN2P17(closest);ENSG00000116...   \n",
       "2  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "3  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "4  21255567  ENSG00000117298/ECE1(containing);ENSG000002369...   \n",
       "\n",
       "       genomic element H3K27ac-expression correlation p-values  \n",
       "0    enhancer intronic                                       .  \n",
       "1  enhancer intergenic                                       .  \n",
       "2    enhancer intronic                                       .  \n",
       "3    enhancer intronic                                       .  \n",
       "4    enhancer intronic            ECE1/ENST00000415912/0.06776  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples = h3k27ac_cov.columns[3:]\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_coverage[\"H3K27ac-expression correlation p-values\"] = rare_enriched_enhancer_snps_motif_genes_collected_coverage.apply(calculate_correlation, args = (samples,), axis=1)\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations = rare_enriched_enhancer_snps_motif_genes_collected_coverage.drop(labels = samples, axis=1)\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_best_candidate_target(putative_targets):\n",
    "    if putative_targets != \".\":\n",
    "        putative_targets_list = putative_targets.split(';')\n",
    "        pvalues = [float(target.split('/')[2]) for target in putative_targets_list]\n",
    "        min_pval = min(pvalues)\n",
    "        best_candidate = putative_targets_list[pvalues.index(min_pval)]\n",
    "        return best_candidate\n",
    "    else:\n",
    "        return '.'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>H3K27ac-expression correlation p-values</th>\n",
       "      <th>Putative target with highest correlation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868940</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>FLI1_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>ETS1_HUMAN.H11MO.0.A:1.60</td>\n",
       "      <td>6868223</td>\n",
       "      <td>6870996</td>\n",
       "      <td>ENSG00000171735/CAMTA1(containing);ENSG0000022...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8890584</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:0.96</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:3.86</td>\n",
       "      <td>8888663</td>\n",
       "      <td>8891414</td>\n",
       "      <td>ENSG00000238249/HMGN2P17(closest);ENSG00000116...</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:0.90</td>\n",
       "      <td>ZFP28_HUMAN.H11MO.0.C:1.89</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:1.00</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:3.01</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:0.89</td>\n",
       "      <td>IRF5_HUMAN.H11MO.0.D:2.20</td>\n",
       "      <td>21253526</td>\n",
       "      <td>21255567</td>\n",
       "      <td>ENSG00000117298/ECE1(containing);ENSG000002369...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ECE1/ENST00000415912/0.06776</td>\n",
       "      <td>ECE1/ENST00000415912/0.06776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr1   6868940   G   C   1  0.022  46           0.000200   \n",
       "1  chr1   8890584   T   C   1  0.022  46           0.000100   \n",
       "2  chr1  10144750   G   A   2  0.043  46           0.000100   \n",
       "3  chr1  10144754   G   A   2  0.043  46           0.000035   \n",
       "4  chr1  21254094   A   G   2  0.048  42           0.000037   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0           0.000000    0.000000           True              0.000000   \n",
       "1           0.000000    0.000000           True              0.000000   \n",
       "2           0.000200    0.000041           True              0.000384   \n",
       "3           0.000071    0.000005           True              0.000055   \n",
       "4           0.000073    0.000005           True              0.000050   \n",
       "\n",
       "             motif_best_match          motif_highest_diff  enh_start  \\\n",
       "0   FLI1_HUMAN.H11MO.1.A:0.96   ETS1_HUMAN.H11MO.0.A:1.60    6868223   \n",
       "1  PRDM6_HUMAN.H11MO.0.C:0.96   IRF2_HUMAN.H11MO.0.A:3.86    8888663   \n",
       "2  CPEB1_HUMAN.H11MO.0.D:0.90  ZFP28_HUMAN.H11MO.0.C:1.89   10143598   \n",
       "3  PRDM6_HUMAN.H11MO.0.C:1.00  CPEB1_HUMAN.H11MO.0.D:3.01   10143598   \n",
       "4   IRF2_HUMAN.H11MO.0.A:0.89   IRF5_HUMAN.H11MO.0.D:2.20   21253526   \n",
       "\n",
       "    enh_end                                               Gene  \\\n",
       "0   6870996  ENSG00000171735/CAMTA1(containing);ENSG0000022...   \n",
       "1   8891414  ENSG00000238249/HMGN2P17(closest);ENSG00000116...   \n",
       "2  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "3  10144774  ENSG00000130939/UBE4B(containing);ENSG00000201...   \n",
       "4  21255567  ENSG00000117298/ECE1(containing);ENSG000002369...   \n",
       "\n",
       "       genomic element H3K27ac-expression correlation p-values  \\\n",
       "0    enhancer intronic                                       .   \n",
       "1  enhancer intergenic                                       .   \n",
       "2    enhancer intronic                                       .   \n",
       "3    enhancer intronic                                       .   \n",
       "4    enhancer intronic            ECE1/ENST00000415912/0.06776   \n",
       "\n",
       "  Putative target with highest correlation  \n",
       "0                                        .  \n",
       "1                                        .  \n",
       "2                                        .  \n",
       "3                                        .  \n",
       "4             ECE1/ENST00000415912/0.06776  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations[\"Putative target with highest correlation\"] = rare_enriched_enhancer_snps_motif_genes_collected_corelations[\"H3K27ac-expression correlation p-values\"].apply(find_best_candidate_target)\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check expression of TFs and target genes in brain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex = pd.read_csv(GTEX, sep='\\t', skiprows=[0,1])\n",
    "brain_columns = [col for col in list(gtex.columns) if \"Brain\" in col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_names(genes_string):\n",
    "    #promoters will have ENSG00000136026/CKAP4, comma separated\n",
    "    #enhancers will have \";\"-separated lists with the following format: ENSG00000171735/CAMTA1(containing)\n",
    "    if genes_string:\n",
    "        if \"(\" not in genes_string:\n",
    "            return [el.split('/')[1] for el in genes_string.split(',')]\n",
    "        else:    \n",
    "            return [el.split('/')[1].split('(')[0] for el in genes_string.split(';')]\n",
    "    else:\n",
    "        return \"\"\n",
    "\n",
    "def check_expression_in_brain(genes):\n",
    "    \n",
    "    gene_names_list = get_gene_names(genes)\n",
    "    expression_list = []\n",
    "    for gene in gene_names_list:\n",
    "        if gene != \"\" and gene != \".\":\n",
    "            try:    \n",
    "                mean_median_tpm = sum(gtex[gtex['Description'] == gene.strip()][brain_columns].values[0]) / float(len(brain_columns))\n",
    "                expression_list.append(gene + ':' + \"%.2f\" % mean_median_tpm)\n",
    "            except:\n",
    "                print(\"no gtex brain data for:\", gene)\n",
    "            \n",
    "    return \",\".join(expression_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no gtex brain data for: AL121992.1\n",
      "no gtex brain data for: RP13-582O9.6\n",
      "no gtex brain data for: AC138430.4\n",
      "no gtex brain data for: AC138028.1\n",
      "no gtex brain data for: RP11-460N20.3\n",
      "no gtex brain data for: FAM150B\n",
      "no gtex brain data for: AL590233.1\n",
      "no gtex brain data for: FAM150B\n",
      "no gtex brain data for: RP11-327F22.2\n",
      "no gtex brain data for: AL022326.1\n",
      "no gtex brain data for: CTD-3105H18.14\n",
      "no gtex brain data for: RP4-665J23.2\n",
      "no gtex brain data for: TMEM56-RWDD3\n",
      "no gtex brain data for: BORCS8-MEF2B\n",
      "no gtex brain data for: CTC-435M10.3\n",
      "no gtex brain data for: TCEB2\n",
      "no gtex brain data for: AC226119.5\n",
      "no gtex brain data for: FAM150B\n",
      "no gtex brain data for: RP4-665J23.2\n",
      "no gtex brain data for: C1orf95\n",
      "no gtex brain data for: AC138028.1\n",
      "no gtex brain data for: AC226119.5\n",
      "no gtex brain data for: TGIF2-C20orf24\n",
      "no gtex brain data for: AC138430.4\n",
      "no gtex brain data for: RP13-395E19.3\n",
      "no gtex brain data for: RP13-395E19.3\n",
      "no gtex brain data for: RP11-697E2.6\n",
      "no gtex brain data for: FLJ35934\n",
      "no gtex brain data for: XRCC6BP1\n",
      "no gtex brain data for: MIR3650\n",
      "no gtex brain data for: MIR1302-11\n",
      "no gtex brain data for: AL132780.1\n",
      "no gtex brain data for: CH17-232I21.1\n",
      "no gtex brain data for: MLLT4\n",
      "no gtex brain data for: MLLT4-AS1\n",
      "no gtex brain data for: MIR1184-3\n",
      "no gtex brain data for: GUSBP11\n",
      "no gtex brain data for: KB-1572G7.2\n",
      "no gtex brain data for: RN7SL671P\n",
      "no gtex brain data for: DHFRL1\n",
      "no gtex brain data for: RP13-539J13.1\n",
      "no gtex brain data for: EBLN3\n",
      "no gtex brain data for: RP11-396C23.2\n",
      "no gtex brain data for: MIR1292\n",
      "no gtex brain data for: NAMA_2\n",
      "no gtex brain data for: AL356020.1\n",
      "no gtex brain data for: STX16-NPEPL1\n",
      "no gtex brain data for: RP13-395E19.3\n",
      "no gtex brain data for: MIR1302-2\n",
      "no gtex brain data for: RP11-34P13.3\n",
      "no gtex brain data for: AC007128.1\n",
      "no gtex brain data for: AC007040.11\n",
      "no gtex brain data for: AL590683.2\n",
      "no gtex brain data for: AL391730.1\n",
      "no gtex brain data for: AL355795.1\n",
      "no gtex brain data for: RP11-25K21.6\n",
      "no gtex brain data for: RP11-222A5.1\n",
      "no gtex brain data for: AL590085.1\n",
      "no gtex brain data for: RP11-134G8.6\n",
      "no gtex brain data for: AL138925.1\n",
      "no gtex brain data for: RP11-420K10.1\n",
      "no gtex brain data for: RP11-548K23.11\n",
      "no gtex brain data for: OBFC1\n",
      "no gtex brain data for: OBFC1\n",
      "no gtex brain data for: RP11-179H18.5\n",
      "no gtex brain data for: AP002498.1\n",
      "no gtex brain data for: RP11-95F22.1\n",
      "no gtex brain data for: AL132988.1\n",
      "no gtex brain data for: AL358340.1\n",
      "no gtex brain data for: AF111168.4\n",
      "no gtex brain data for: AC104002.1\n",
      "no gtex brain data for: RP11-680G10.1\n",
      "no gtex brain data for: AC010311.1\n",
      "no gtex brain data for: AC010311.1\n",
      "no gtex brain data for: CTC-548K16.2\n",
      "no gtex brain data for: CTC-548K16.2\n",
      "no gtex brain data for: AC092580.4\n",
      "no gtex brain data for: AC096772.6\n",
      "no gtex brain data for: AC093865.1\n",
      "no gtex brain data for: AL035106.1\n",
      "no gtex brain data for: AP000320.7\n",
      "no gtex brain data for: AC121332.1\n",
      "no gtex brain data for: NPHP3-ACAD11\n",
      "no gtex brain data for: AC063932.1\n",
      "no gtex brain data for: RP11-215A19.2\n",
      "no gtex brain data for: CTD-3224K15.2\n",
      "no gtex brain data for: CTB-35F21.2\n",
      "no gtex brain data for: CTD-3224K15.2\n",
      "no gtex brain data for: CTB-35F21.2\n",
      "no gtex brain data for: CTD-3224K15.2\n",
      "no gtex brain data for: AC005592.1\n",
      "no gtex brain data for: AC005592.1\n",
      "no gtex brain data for: RP1-167F1.2\n",
      "no gtex brain data for: ATP6V1G2-DDX39B\n",
      "no gtex brain data for: XXbac-BPG32J3.22\n",
      "no gtex brain data for: MSH5-SAPCD1\n",
      "no gtex brain data for: AC006483.1\n",
      "no gtex brain data for: AC004520.1\n",
      "no gtex brain data for: AC004520.1\n",
      "no gtex brain data for: RP4-777O23.3\n",
      "no gtex brain data for: RP5-1165K10.2\n",
      "no gtex brain data for: RP5-1165K10.2\n",
      "no gtex brain data for: WBSCR16\n",
      "no gtex brain data for: WBSCR16\n",
      "no gtex brain data for: AC006014.7\n",
      "no gtex brain data for: RP11-514P8.8\n",
      "no gtex brain data for: MESTIT1_1\n",
      "no gtex brain data for: MESTIT1_2\n",
      "no gtex brain data for: MESTIT1_3\n",
      "no gtex brain data for: AC104133.1\n",
      "no gtex brain data for: AP003356.1\n",
      "no gtex brain data for: AC022909.1\n",
      "no gtex brain data for: KIAA0196\n",
      "no gtex brain data for: GS1-393G12.12\n",
      "no gtex brain data for: MIR6847\n",
      "no gtex brain data for: GS1-393G12.12\n",
      "no gtex brain data for: MIR6847\n",
      "no gtex brain data for: GS1-393G12.12\n",
      "no gtex brain data for: CDKN2B-AS_3\n",
      "no gtex brain data for: RP11-15J10.1\n",
      "no gtex brain data for: RP11-548B3.3\n",
      "no gtex brain data for: PPP2R4\n",
      "no gtex brain data for: RP11-247A12.8\n",
      "no gtex brain data for: AF241734.1\n"
     ]
    }
   ],
   "source": [
    "rare_enriched_promoter_snps_motif_gene[\"Median TPM in brain tissues in GTEx\"] = rare_enriched_promoter_snps_motif_gene.Gene.apply(check_expression_in_brain)\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations[\"Median TPM in brain tissues in GTEx\"] = rare_enriched_enhancer_snps_motif_genes_collected_corelations.Gene.apply(check_expression_in_brain)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>B-H_reject_H0</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Median TPM in brain tissues in GTEx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>45783029</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>KLF15_HUMAN.H11MO.0.A:0.93</td>\n",
       "      <td>ZN770_HUMAN.H11MO.0.C:3.80</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000104936/DMPK</td>\n",
       "      <td>DMPK:37.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>222320241</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>0.087</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:0.98</td>\n",
       "      <td>E2F6_HUMAN.H11MO.0.A:3.84</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000237732/AC010980.2</td>\n",
       "      <td>AC010980.2:2.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>6556629</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>0.091</td>\n",
       "      <td>22</td>\n",
       "      <td>0.001300</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>True</td>\n",
       "      <td>0.004048</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>MAZ_HUMAN.H11MO.1.A:7.56</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000091622/PITPNM3</td>\n",
       "      <td>PITPNM3:29.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>26692643</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:0.99</td>\n",
       "      <td>ZN740_HUMAN.H11MO.0.D:4.41</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000171303/KCNK3</td>\n",
       "      <td>KCNK3:6.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>106247724</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>0.063</td>\n",
       "      <td>32</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>True</td>\n",
       "      <td>0.000320</td>\n",
       "      <td>OSR2_HUMAN.H11MO.0.C:0.95</td>\n",
       "      <td>WT1_HUMAN.H11MO.0.C:3.25</td>\n",
       "      <td>promoter</td>\n",
       "      <td>ENSG00000136026/CKAP4,ENSG00000258355/RP11-651...</td>\n",
       "      <td>CKAP4:11.70,RP11-651L5.2:0.02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr19   45783029   T   G   2  0.043  46           0.000076   \n",
       "1   chr2  222320241   T   C   4  0.087  46           0.000200   \n",
       "2  chr17    6556629   G   C   2  0.091  22           0.001300   \n",
       "3   chr2   26692643   C   G   2  0.048  42           0.000046   \n",
       "4  chr12  106247724   G   T   2  0.063  32           0.000200   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  B-H_reject_H0  corrected_binom_pval  \\\n",
       "0             0.0000    0.000000           True              0.000000   \n",
       "1             0.0000    0.000000           True              0.000000   \n",
       "2             0.0017    0.000653           True              0.004048   \n",
       "3             0.0000    0.000000           True              0.000000   \n",
       "4             0.0003    0.000044           True              0.000320   \n",
       "\n",
       "             motif_best_match          motif_highest_diff genomic element  \\\n",
       "0  KLF15_HUMAN.H11MO.0.A:0.93  ZN770_HUMAN.H11MO.0.C:3.80        promoter   \n",
       "1   E2F6_HUMAN.H11MO.0.A:0.98   E2F6_HUMAN.H11MO.0.A:3.84        promoter   \n",
       "2    MAZ_HUMAN.H11MO.1.A:0.96    MAZ_HUMAN.H11MO.1.A:7.56        promoter   \n",
       "3  ZN740_HUMAN.H11MO.0.D:0.99  ZN740_HUMAN.H11MO.0.D:4.41        promoter   \n",
       "4   OSR2_HUMAN.H11MO.0.C:0.95    WT1_HUMAN.H11MO.0.C:3.25        promoter   \n",
       "\n",
       "                                                Gene  \\\n",
       "0                               ENSG00000104936/DMPK   \n",
       "1                         ENSG00000237732/AC010980.2   \n",
       "2                            ENSG00000091622/PITPNM3   \n",
       "3                              ENSG00000171303/KCNK3   \n",
       "4  ENSG00000136026/CKAP4,ENSG00000258355/RP11-651...   \n",
       "\n",
       "  Median TPM in brain tissues in GTEx  \n",
       "0                          DMPK:37.81  \n",
       "1                     AC010980.2:2.01  \n",
       "2                       PITPNM3:29.02  \n",
       "3                          KCNK3:6.22  \n",
       "4       CKAP4:11.70,RP11-651L5.2:0.02  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_promoter_snps_motif_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>gnomAD_genome_NFE</th>\n",
       "      <th>binom_pval</th>\n",
       "      <th>...</th>\n",
       "      <th>corrected_binom_pval</th>\n",
       "      <th>motif_best_match</th>\n",
       "      <th>motif_highest_diff</th>\n",
       "      <th>enh_start</th>\n",
       "      <th>enh_end</th>\n",
       "      <th>Gene</th>\n",
       "      <th>genomic element</th>\n",
       "      <th>H3K27ac-expression correlation p-values</th>\n",
       "      <th>Putative target with highest correlation</th>\n",
       "      <th>Median TPM in brain tissues in GTEx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6868940</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>FLI1_HUMAN.H11MO.1.A:0.96</td>\n",
       "      <td>ETS1_HUMAN.H11MO.0.A:1.60</td>\n",
       "      <td>6868223</td>\n",
       "      <td>6870996</td>\n",
       "      <td>ENSG00000171735/CAMTA1(containing);ENSG0000022...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CAMTA1:19.49,RP11-312B8.2:0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8890584</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:0.96</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:3.86</td>\n",
       "      <td>8888663</td>\n",
       "      <td>8891414</td>\n",
       "      <td>ENSG00000238249/HMGN2P17(closest);ENSG00000116...</td>\n",
       "      <td>enhancer intergenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>HMGN2P17:0.02,ERRFI1:16.11,RP11-431K24.1:0.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:0.90</td>\n",
       "      <td>ZFP28_HUMAN.H11MO.0.C:1.89</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UBE4B:18.16,RNU6-828P:0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10144754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.043</td>\n",
       "      <td>46</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>PRDM6_HUMAN.H11MO.0.C:1.00</td>\n",
       "      <td>CPEB1_HUMAN.H11MO.0.D:3.01</td>\n",
       "      <td>10143598</td>\n",
       "      <td>10144774</td>\n",
       "      <td>ENSG00000130939/UBE4B(containing);ENSG00000201...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UBE4B:18.16,RNU6-828P:0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21254094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>0.048</td>\n",
       "      <td>42</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>IRF2_HUMAN.H11MO.0.A:0.89</td>\n",
       "      <td>IRF5_HUMAN.H11MO.0.D:2.20</td>\n",
       "      <td>21253526</td>\n",
       "      <td>21255567</td>\n",
       "      <td>ENSG00000117298/ECE1(containing);ENSG000002369...</td>\n",
       "      <td>enhancer intronic</td>\n",
       "      <td>ECE1/ENST00000415912/0.06776</td>\n",
       "      <td>ECE1/ENST00000415912/0.06776</td>\n",
       "      <td>ECE1:13.67,RP3-329E20.2:0.01,RAP1GAP:103.43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  AC     AF  AN  gnomAD_genome_ALL  \\\n",
       "0  chr1   6868940   G   C   1  0.022  46           0.000200   \n",
       "1  chr1   8890584   T   C   1  0.022  46           0.000100   \n",
       "2  chr1  10144750   G   A   2  0.043  46           0.000100   \n",
       "3  chr1  10144754   G   A   2  0.043  46           0.000035   \n",
       "4  chr1  21254094   A   G   2  0.048  42           0.000037   \n",
       "\n",
       "   gnomAD_genome_NFE  binom_pval  ...  corrected_binom_pval  \\\n",
       "0           0.000000    0.000000  ...              0.000000   \n",
       "1           0.000000    0.000000  ...              0.000000   \n",
       "2           0.000200    0.000041  ...              0.000384   \n",
       "3           0.000071    0.000005  ...              0.000055   \n",
       "4           0.000073    0.000005  ...              0.000050   \n",
       "\n",
       "             motif_best_match          motif_highest_diff enh_start   enh_end  \\\n",
       "0   FLI1_HUMAN.H11MO.1.A:0.96   ETS1_HUMAN.H11MO.0.A:1.60   6868223   6870996   \n",
       "1  PRDM6_HUMAN.H11MO.0.C:0.96   IRF2_HUMAN.H11MO.0.A:3.86   8888663   8891414   \n",
       "2  CPEB1_HUMAN.H11MO.0.D:0.90  ZFP28_HUMAN.H11MO.0.C:1.89  10143598  10144774   \n",
       "3  PRDM6_HUMAN.H11MO.0.C:1.00  CPEB1_HUMAN.H11MO.0.D:3.01  10143598  10144774   \n",
       "4   IRF2_HUMAN.H11MO.0.A:0.89   IRF5_HUMAN.H11MO.0.D:2.20  21253526  21255567   \n",
       "\n",
       "                                                Gene      genomic element  \\\n",
       "0  ENSG00000171735/CAMTA1(containing);ENSG0000022...    enhancer intronic   \n",
       "1  ENSG00000238249/HMGN2P17(closest);ENSG00000116...  enhancer intergenic   \n",
       "2  ENSG00000130939/UBE4B(containing);ENSG00000201...    enhancer intronic   \n",
       "3  ENSG00000130939/UBE4B(containing);ENSG00000201...    enhancer intronic   \n",
       "4  ENSG00000117298/ECE1(containing);ENSG000002369...    enhancer intronic   \n",
       "\n",
       "  H3K27ac-expression correlation p-values  \\\n",
       "0                                       .   \n",
       "1                                       .   \n",
       "2                                       .   \n",
       "3                                       .   \n",
       "4            ECE1/ENST00000415912/0.06776   \n",
       "\n",
       "  Putative target with highest correlation  \\\n",
       "0                                        .   \n",
       "1                                        .   \n",
       "2                                        .   \n",
       "3                                        .   \n",
       "4             ECE1/ENST00000415912/0.06776   \n",
       "\n",
       "             Median TPM in brain tissues in GTEx  \n",
       "0                 CAMTA1:19.49,RP11-312B8.2:0.00  \n",
       "1  HMGN2P17:0.02,ERRFI1:16.11,RP11-431K24.1:0.16  \n",
       "2                     UBE4B:18.16,RNU6-828P:0.02  \n",
       "3                     UBE4B:18.16,RNU6-828P:0.02  \n",
       "4    ECE1:13.67,RP3-329E20.2:0.01,RAP1GAP:103.43  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Save output to files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "rare_enriched_promoter_snps_motif_gene.to_csv(ANNOTATED_PROMOTER_SNPs, sep = \"\\t\", index = False)\n",
    "rare_enriched_enhancer_snps_motif_genes_collected_corelations.to_csv(ANNOTATED_ENHANCER_SNPs, sep = \"\\t\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Timestamp('2021-03-15 12:55:41.025907')"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t2 = pd.to_datetime('today')\n",
    "t2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
